<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>BiomartGeneRegionTrack class and methods — BiomartGeneRegionTrack-class • Gviz</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="BiomartGeneRegionTrack class and methods — BiomartGeneRegionTrack-class" />
<meta property="og:description" content="

A class to hold gene model data for a genomic region fetched
  dynamically from EBI's Biomart Ensembl data source." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Gviz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.37.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Gviz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ivanek/Gviz/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>BiomartGeneRegionTrack class and methods</h1>
    
    <div class="hidden name"><code>BiomartGeneRegionTrack-class.Rd</code></div>
    </div>

    <div class="ref-description">
    

<p>A class to hold gene model data for a genomic region fetched
  dynamically from EBI's Biomart Ensembl data source.</p>
    </div>

    <pre class="usage"><span class='fu'>BiomartGeneRegionTrack</span><span class='op'>(</span><span class='va'>start</span>, <span class='va'>end</span>, <span class='va'>biomart</span>, <span class='va'>chromosome</span>, <span class='va'>strand</span>, <span class='va'>genome</span>,
  stacking<span class='op'>=</span><span class='st'>"squish"</span>, filters<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span>, featureMap<span class='op'>=</span><span class='cn'>NULL</span>,
  name<span class='op'>=</span><span class='st'>"BiomartGeneRegionTrack"</span>, symbol<span class='op'>=</span><span class='cn'>NULL</span>, gene<span class='op'>=</span><span class='cn'>NULL</span>, entrez<span class='op'>=</span><span class='cn'>NULL</span>,
  transcript<span class='op'>=</span><span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>start</th>
      <td><p>An integer scalar with the genomic start coordinates for
    the gene model range.</p></td>
    </tr>
    <tr>
      <th>end</th>
      <td><p>An integer scalar with the genomic end coordinates for
    the gene model range.</p></td>
    </tr>
    <tr>
      <th>biomart</th>
      <td><p>An optional <code>Mart</code> object
    providing access to the EBI Biomart webservice. As default the
    appropriate Ensembl data source is selected based on the provided
    genome and chromosome.</p></td>
    </tr>
    <tr>
      <th>strand</th>
      <td><p>Character scalar, the strand for which to fetch gene
    information from Biomart. One in <code><a href='https://rdrr.io/r/base/Arithmetic.html'>+</a></code>, <code><a href='https://rdrr.io/r/base/Arithmetic.html'>-</a></code>, or <code>+-</code>.</p></td>
    </tr>
    <tr>
      <th>chromosome</th>
      <td><p>The chromosome on which the track's genomic ranges
    are defined. A valid UCSC chromosome identifier. Please note that at
    this stage only syntactic checking takes place, i.e., the argument
    value needs to be a single integer, numeric character or a character
    of the form <code>chrx</code>, where <code>x</code> may be any possible
    string. The user has to make sure that the respective chromosome is
    indeed defined for the the track's genome.</p></td>
    </tr>
    <tr>
      <th>genome</th>
      <td><p>The genome on which the track's ranges are
    defined. Usually this is a valid UCSC genome identifier, however
    this is not being formally checked at this point. If no mapping from
    genome to Biomart Ensembl data source is possible, the
    <code>biomart</code> argument needs to be provided by the user.</p></td>
    </tr>
    <tr>
      <th>stacking</th>
      <td><p>The stacking type for overlapping items of the
    track. One in <code><a href='https://rdrr.io/r/base/c.html'>c(hide, dense, squish, pack,full)</a></code>. Currently,
    only hide (don't show the track items, squish (make best use of the
    available space) and dense (no stacking at all) are implemented.</p></td>
    </tr>
    <tr>
      <th>filters</th>
      <td><p>A list of additional filters to be applied in the
    Biomart query. See <code>getBM</code> for details.</p></td>
    </tr>
    <tr>
      <th>featureMap</th>
      <td><p>Named character vector or list to map between the fields in
    the Biomart data base and the features as they are used to construct
    the track. If multiple values are provided in a single list item,
    the package will use the first one that is defined in the selected Biomart.</p></td>
    </tr>
    <tr>
      <th>name</th>
      <td><p>Character scalar of the track's name used in the title
    panel when plotting.</p></td>
    </tr>
    <tr>
      <th>symbol,transcript,gene,entrez</th>
      <td><p>Character vector giving one or
    several gene symbols, Ensembl transcript identifiers, Ensembl gene
    identifiers, or ENTREZ gene identifiers, respectively. The genomic
    locus of their gene model will be fetch from Biomart instead of
    providing explicit start and end coordinates.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional items which will all be interpreted as further
    display parameters. See <code><a href='settings.html'>settings</a></code> and the "Display
    Parameters" section below for details.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="objects-from-the-class"><a class="anchor" href="#objects-from-the-class"></a>Objects from the class</h2>

    

<p>Objects can be created using the constructor function
  <code>BiomartGeneRegionTrack</code>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    

<p>A track containing all gene models in a particular region as fetched
  from EBI's Biomart service. Usually the user does not have to take
  care of the Biomart connection, which will be established
  automatically based on the provided genome and chromosome
  information. However, for full flexibility a valid
  <code>Mart</code> object may be passed on to the
  constructor. Please note that this assumes a connection to one of the
  Ensembl gene data sources, mapping the available query data back to
  the internal object slots.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    

<p>The return value of the constructor function is a new object of class
  <code>BiomartGeneRegionTrack</code>.</p>
    <h2 class="hasAnchor" id="slots"><a class="anchor" href="#slots"></a>Slots</h2>

    <p></p><dl>
    <dt><code>biomart</code>:</dt><dd><p>Object of class <code>"MartOrNULL"</code>, the
      connection to the Ensembl Biomart webservice.</p></dd>

    <dt><code>filter</code>:</dt><dd><p>Object of class <code>"list"</code>, additional
      filters for the data base query.</p></dd>

    <dt><code>start</code>:</dt><dd><p>Object of class <code>"numeric"</code>, inherited
      from class <code><a href='GeneRegionTrack-class.html'>GeneRegionTrack</a></code>. The start
      coordinates of the annotation range. The coorrdinates for the
      individual gene model items are stored in the <code>range</code> slot.</p></dd>

    <dt><code>end</code>:</dt><dd><p>Object of class <code>"numeric"</code>, inherited from
      class <code><a href='GeneRegionTrack-class.html'>GeneRegionTrack</a></code>. The end
      coordinates of the annotation range. The corrdinates for the
      individual gene model items are stored in the <code>range</code> slot.</p></dd>

     <dt><code>stacking</code>:</dt><dd><p>Object of class <code>"character"</code>,
      inherited from class <code><a href='StackedTrack-class.html'>StackedTrack</a></code></p></dd>

    <dt><code>stacks</code>:</dt><dd><p>Object of class <code>"environment"</code>,
      inherited from class <code><a href='StackedTrack-class.html'>StackedTrack</a></code></p></dd>

    <dt><code>range</code>:</dt><dd><p>Object of class <code>GRanges</code>,
      inherited from class <code><a href='RangeTrack-class.html'>RangeTrack</a></code></p></dd>

    <dt><code>chromosome</code>:</dt><dd><p>Object of class <code>"character"</code>,
      inherited from class <code><a href='RangeTrack-class.html'>RangeTrack</a></code></p></dd>

    <dt><code>genome</code>:</dt><dd><p>Object of class <code>"character"</code>, inherited
      from class <code><a href='RangeTrack-class.html'>RangeTrack</a></code></p></dd>

    <dt><code>dp</code>:</dt><dd><p>Object of class
      <code><a href='DisplayPars-class.html'>DisplayPars</a></code>, inherited from class
      <code><a href='GdObject-class.html'>GdObject</a></code></p></dd>

    <dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code>, inherited
      from class <code><a href='GdObject-class.html'>GdObject</a></code></p></dd>

     <dt><code>imageMap</code>:</dt><dd><p>Object of class
      <code><a href='ImageMap-class.html'>ImageMap</a></code>, inherited from class
      <code><a href='GdObject-class.html'>GdObject</a></code></p></dd>

  
</dl>

    <h2 class="hasAnchor" id="extends"><a class="anchor" href="#extends"></a>Extends</h2>

    

<p>Class <code>"<a href='GeneRegionTrack-class.html'>GeneRegionTrack</a>"</code>, directly.</p>
<p>Class <code>"<a href='AnnotationTrack-class.html'>AnnotationTrack</a>"</code>, by class "GeneRegionTrack", distance 2.</p>
<p>Class <code>"<a href='StackedTrack-class.html'>StackedTrack</a>"</code>, by class "GeneRegionTrack", distance 3.</p>
<p>Class <code>"<a href='RangeTrack-class.html'>RangeTrack</a>"</code>, by class "GeneRegionTrack", distance 4.</p>
<p>Class <code>"<a href='GdObject-class.html'>GdObject</a>"</code>, by class "GeneRegionTrack", distance 5.</p>
    <h2 class="hasAnchor" id="methods"><a class="anchor" href="#methods"></a>Methods</h2>

    

<p>In the following code chunks, <code>obj</code> is considered to be an
  object of class <code>BiomartGeneRegionTrack</code>.</p>
<p></p><dl>

    <em>Internal methods:</em>

    <dt>initialize</dt><dd><p><code>signature(.Object =
    "BiomartGeneRegionTrack")</code>: initialize the object.</p></dd>

  
</dl>

<p><b><em>Inherited methods:</em></b></p>
<p></p><dl>

     <dt>group</dt><dd><p><code>signature(gdObject="BiomartGeneRegionTrack")</code>: extract
      the group membership for all track items.      <em>Usage:</em>      <code><a href='AnnotationTrack-class.html'>group(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='AnnotationTrack-class.html'>group(obj)</a></code></p></dd></dl></p></dd>

    <dt>group&lt;-</dt><dd><p><code>signature(gdObject="BiomartGeneRegionTrack",
      value="character")</code>: replace the grouping information for track
      items. The replacement value must be a factor of appropriate
      length or another vector that can be coerced into such.      <em>Usage:</em>      <code>group&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>group(obj) &lt;- c("a", "a", "b", "c", "a")</code></p></dd></dl></p></dd>

    <dt>identifier</dt><dd><p><code>signature(gdObject="BiomartGeneRegionTrack")</code>:
      return track item identifiers. Depending on the setting of the
      optional argument <code>lowest</code>, these are either the group
      identifiers or the individual item identifiers.      <em>Usage:</em>      <code><a href='AnnotationTrack-class.html'>identifier(GdObject, lowest=FALSE)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>lowest</code>: return the lowest-level identifier, i.e.,
    the item IDs, or the higher level group IDs which do not have to
    be unqiue.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='AnnotationTrack-class.html'>identifier(obj, lowest=FALSE)</a></code></p></dd></dl></p></dd>

    <dt>identifier&lt;-</dt><dd><p><code>signature(gdObject="BiomartGeneRegionTrack",
    value="character")</code>: Set the track item identifiers. The
      replacement value has to be a character vector of appropriate
      length. This always replaces the group-level identifiers, so
      essentially it is similar to <code>groups&lt;-</code>.      <em>Usage:</em>      <code>identifier&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>identifier(obj) &lt;- c("foo", "bar")</code></p></dd></dl></p></dd>

    <dt>exon</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>: Extract
      the exon identifiers for all exons in the gene models.      <em>Usage:</em>      <code><a href='GeneRegionTrack-class.html'>exon(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='GeneRegionTrack-class.html'>exon(obj)</a></code></p></dd></dl></p></dd>

    <dt>exon&lt;-</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack",
      value="character")</code>: replace the exon identifiers for all exons in
      the gene model.  The replacement value must be a character of
      appropriate length or another vector that can be coerced into
      such.      <em>Usage:</em>      <code>exon&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>exon(obj) &lt;- paste("Exon", 1:5)</code></p></dd></dl></p></dd>

    <dt>gene</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>: Extract
      the gene identifiers for all gene models.      <em>Usage:</em>      <code><a href='GeneRegionTrack-class.html'>gene(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='GeneRegionTrack-class.html'>gene(obj)</a></code></p></dd></dl></p></dd>

    <dt>gene&lt;-</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack",
    value="character")</code>: replace the gene identifiers for all gene
      models.  The replacement value must be a character of
      appropriate length or another vector that can be coerced into
      such.      <em>Usage:</em>      <code>gene&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>gene(obj) &lt;- paste("Gene", LETTERS[1:5])</code></p></dd></dl></p></dd>

    <dt>symbol</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>: Extract
      the human-readble gene symbol for all gene models.      <em>Usage:</em>      <code><a href='GeneRegionTrack-class.html'>symbol(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='GeneRegionTrack-class.html'>symbol(obj)</a></code></p></dd></dl></p></dd>

    <dt>symbol&lt;-</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack",
    value="character")</code>: replace the human-readable gene symbol for
      all gene models.  The replacement value must be a character of
      appropriate length or another vector that can be coerced into
      such.      <em>Usage:</em>      <code>gene&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>symbol(obj) &lt;- letters[1:5]</code></p></dd></dl></p></dd>

    <dt>transcript</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>:
      Extract the transcript identifiers for all transcripts in the
      gene models.      <em>Usage:</em>      <code><a href='GeneRegionTrack-class.html'>transcript(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='GeneRegionTrack-class.html'>transcript(obj)</a></code></p></dd></dl></p></dd>

    <dt>transcript&lt;-</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack",
    value="character")</code>: replace the transcript identifiers for
      all transcripts in the gene model.  The replacement value must
      be a character of appropriate length or another vector that
      can be coerced into such.      <em>Usage:</em>      <code>transcript&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>transcript(obj) &lt;- paste("Exon", 1:5)</code></p></dd></dl></p></dd>
  
</dl>

<p><em>Internal methods:</em></p>
<p></p><dl>

    <dt>coerce</dt><dd><p><code>signature(from="BiomartGeneRegionTrack",
    to="UCSCData")</code>: coerce to a <code>UCSCData</code> object for export to
      the UCSC genome browser.       <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>as(obj, "UCSCData")</code></p></dd></dl></p></dd>

    <dt>collapseTrack</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>:
      preprocess the track before plotting. This will collapse
      overlapping track items based on the available resolution and
      increase the width and height of all track objects to a minimum
      value to avoid rendering issues. See <code><a href='collapsing.html'>collapsing</a></code> for
      details.      <em>Usage:</em>      <code>collapseTrack(GdObject, diff=.pxResolution(coord="x"))</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>diff</code>: the minimum pixel width to display,
      everything below that will be inflated to a width of
      <code>diff</code>.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>Gviz:::collapseTrack(obj)</code></p></dd></dl></p></dd>

    <dt>show</dt><dd><p><code>signature(object="BiomartGeneRegionTrack")</code>: show a
      human-readable summary of the object</p></dd>

    <dt>drawGD</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>: plot the
      object to a graphics device. The return value of this method is
      the input object, potentially updated during the plotting
      operation. Internally, there are two modes in which the method can
      be called. Either in 'prepare' mode, in which case no plotting is
      done but the object is preprocessed based on the
      available space, or in 'plotting' mode, in which case the actual
      graphical output is created. Since subsetting of the object can be
      potentially costly, this can be switched off in case subsetting
      has already been performed before or is not necessary.      <em>Usage:</em>      <code><a href='GdObject-class.html'>drawGD(GdObject, minBase, maxBase, prepare=FALSE,
    subset=TRUE, ...)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>minBase</code>, <code>maxBase</code>: the coordinate range to
      plot.</p></dd>    <dt></dt><dd><p><code>prepare</code>: run method in preparation or in
      production mode.</p></dd>    <dt></dt><dd><p><code>subset</code>: subset the object to the visible region
    or skip the potentially expensive subsetting operation.</p></dd>    <dt></dt><dd><p><code>...</code>: all further arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>Gviz:::drawGD(obj)</code></p></dd>    <dt></dt><dd><p><code>Gviz:::drawGD(obj, minBase=1, maxBase=100)</code></p></dd>    <dt></dt><dd><p><code>Gviz:::drawGD(obj, prepare=TRUE,
        subset=FALSE)</code></p></dd></dl></p></dd>

    <dt>drawGrid</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>: superpose a grid on top of a track.      <em>Usage:</em>      <code>drawGrid(GdObject, from, to)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>from</code>, <code>to</code>: integer scalars, draw grid
      within a certain coordinates range. This needs to be supplied
      for the plotting function to know the current genomic
      coordinates.</p></dd></dl>      <em>Examples:</em>      <dl>          <dt></dt><dd><p><code>Gviz:::drawGrid(obj, from=10, to=100)</code></p></dd></dl></p></dd>

    <dt>setStacks</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>:
      recompute the stacks based on the available space and on the
      object's track items and stacking settings.      <em>Usage:</em>      <code>setStacks(GdObject, from, to)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>from</code>, <code>to</code>: integer scalars, compute
      stacking within a certain coordinates range. This needs to be
      supplied for the plotting function to know the current genomic
      coordinates.</p></dd></dl>      <em>Examples:</em>      <dl>          <dt></dt><dd><p><code>Gviz:::setStacks(obj, from=1, to=100)</code></p></dd></dl></p></dd>

    <dt>stacking</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>: return
      the current stacking type.      <em>Usage:</em>      <code><a href='StackedTrack-class.html'>stacking(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='StackedTrack-class.html'>stacking(obj)</a></code></p></dd></dl></p></dd>

    <dt>stacking&lt;-</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack",
      value="character")</code>: set the object's stacking type to one in
      <code><a href='https://rdrr.io/r/base/c.html'>c(hide, dense, squish, pack,full)</a></code>.      <em>Usage:</em>      <code>stacking&lt;-(GdObject, value)</code>      <em>Additional Arguments:</em>    <dl>      <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>stacking(obj) &lt;- "squish" </code></p></dd></dl></p></dd>

    <dt>stacks</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>: return
      the stack indices for each track item.      <em>Usage:</em>      <code><a href='StackedTrack-class.html'>stacks(GdObject)</a></code>      <em>Examples:</em>      <dl>          <dt></dt><dd><p><code>Gviz:::stacks(obj)</code></p></dd></dl></p></dd>

    <dt>[</dt><dd><p><code>signature(x="BiomartGeneRegionTrack", i="ANY", j="ANY",
    drop="ANY")</code>: subset the items in the <code>BiomartGeneRegionTrack</code>
      object. This is essentially similar to subsetting of the
      <code>GRanges</code> object in the <code>range</code>
      slot. For most applications, the <code>subset</code> method may be
      more appropriate.      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>i</code>, <code>j</code>: subsetting indices, <code>j</code> is
      ignored.</p></dd>    <dt></dt><dd><p><code>drop</code>: argument is ignored.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>obj[1:5]</code></p></dd></dl></p></dd>


    <dt>chromosome</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>:
      return the chromosome for which the track is defined.      <em>Usage:</em>      <code><a href='RangeTrack-class.html'>chromosome(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='RangeTrack-class.html'>chromosome(obj)</a></code></p></dd></dl></p></dd>

    <dt>chromosome&lt;-</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>:
      replace the value of the track's chromosome. This has to be a
      valid UCSC chromosome identifier or an integer or character
      scalar that can be reasonably coerced into one.      <em>Usage:</em>      <code>chromosome&lt;-(GdObject, value)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>chromosome(obj) &lt;- "chr12"</code></p></dd></dl></p></dd>

    <dt>start, end, width</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: the
      start or end coordinates of the track items, or their width in
      genomic coordinates.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/stats/start.html'>start(x)</a></code>      <code><a href='https://rdrr.io/r/stats/start.html'>end(x)</a></code>      <code>width(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/stats/start.html'>start(obj)</a></code></p></dd>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/stats/start.html'>end(obj)</a></code></p></dd>    <dt></dt><dd><p><code>width(obj)</code></p></dd></dl></p></dd>

    <dt>start&lt;-, end&lt;-, width&lt;-</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>:
      replace the start or end coordinates of the track items, or their
      width.      <em>Usage:</em>      <code>start&lt;-(x, value)</code>      <code>end&lt;-(x, value)</code>      <code>width&lt;-(x, value)</code>      <em>Additional Arguments:</em>    <dl>      <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>start(obj) &lt;- 1:10</code></p></dd>    <dt></dt><dd><p><code>end(obj) &lt;- 20:30</code></p></dd>    <dt></dt><dd><p><code>width(obj) &lt;- 1</code></p></dd></dl></p></dd>

    <dt>position</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>: the
      arithmetic mean of the track item's coordionates, i.e.,
      <code>(end(obj)-start(obj))/2</code>.      <em>Usage:</em>      <code><a href='RangeTrack-class.html'>position(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='RangeTrack-class.html'>position(obj)</a></code></p></dd></dl></p></dd>

    <dt>feature</dt><dd><p><code>signature(GdObject="BiomartGeneRegionTrack")</code>: return the
      grouping information for track items. For certain sub-classes,
      groups may be indicated by different color schemes when
      plotting. See <code><a href='grouping.html'>grouping</a></code> for details.      <em>Usage:</em>      <code><a href='RangeTrack-class.html'>feature(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='RangeTrack-class.html'>feature(obj)</a></code></p></dd></dl></p></dd>

    <dt>feature&lt;-</dt><dd><p><code>signature(gdObject="BiomartGeneRegionTrack",
      value="character")</code>: set the grouping information for track
      items. This has to be a factor vector (or another type of vector
      that can be coerced into one) of the same length as the number of
      items in the <code>BiomartGeneRegionTrack</code>. See <code><a href='grouping.html'>grouping</a></code>
      for details.      <em>Usage:</em>      <code>feature&lt;-(GdObject, value)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>feature(obj) &lt;- c("a", "a", "b", "c", "a")</code></p></dd></dl></p></dd>

    <dt>genome</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: return the track's genome.      <em>Usage:</em>      <code>genome(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>genome(obj)</code></p></dd></dl></p></dd>

    <dt>genome&lt;-</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: set the track's
      genome. Usually this has to be a valid UCSC identifier, however
      this is not formally enforced here.      <em>Usage:</em>      <code>genome&lt;-(x, value)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>genome(obj) &lt;- "mm9"</code></p></dd></dl></p></dd>

    <dt>length</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: return the number
      of items in the track.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/length.html'>length(x)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/length.html'>length(obj)</a></code></p></dd></dl></p></dd>

    <dt>range</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: return the genomic
      coordinates for the track as an object of class
      <code>IRanges</code>.      <em>Usage:</em>      <code><a href='RangeTrack-class.html'>range(x)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='RangeTrack-class.html'>range(obj)</a></code></p></dd></dl></p></dd>

    <dt>ranges</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: return the genomic
      coordinates for the track along with all additional annotation
      information as an object of class <code>GRanges</code>.      <em>Usage:</em>      <code>ranges(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>ranges(obj)</code></p></dd></dl></p></dd>

    <dt>split</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: split a
      <code>BiomartGeneRegionTrack</code> object by an appropriate factor vector (or
      another vector that can be coerced into one). The output of this
      operation is a list of objects of the same class as the input
      object, all inheriting from class <code>BiomartGeneRegionTrack</code>.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/split.html'>split(x, f, ...)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>f</code>: the splitting factor.</p></dd>    <dt></dt><dd><p><code>...</code>: all further arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/split.html'>split(obj, c("a", "a", "b", "c", "a"))</a></code></p></dd></dl></p></dd>

    <dt>strand</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: return a vector of
      strand specifiers for all track items, in the form '+' for the
      Watson strand, '-' for the Crick strand or '*' for either of the
      two.      <em>Usage:</em>      <code>strand(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>strand(obj)</code></p></dd></dl></p></dd>

    <dt>strand&lt;-</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: replace the
      strand information for the track items. The replacement value
      needs to be an appropriate scalar or vector of strand values.      <em>Usage:</em>      <code>strand&lt;-(x, value)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>strand(obj) &lt;- "+"</code></p></dd></dl></p></dd>

    <dt>values</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: return all
      additional annotation information except for the genomic coordinates
      for the track items as a data.frame.      <em>Usage:</em>      <code>values(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>values(obj)</code></p></dd></dl></p></dd>

    <dt>coerce</dt><dd><p><code>signature(from="BiomartGeneRegionTrack",
    to="data.frame")</code>: coerce the <code>GRanges</code>
    object in the <code>range</code> slot into a regular data.frame.       <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>as(obj, "data.frame")</code></p></dd></dl></p></dd>

    <dt>subset</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: subset a
      <code>BiomartGeneRegionTrack</code> by coordinates and sort if necessary.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/subset.html'>subset(x, from, to, sort=FALSE, ...)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>from</code>, <code>to</code>: the coordinates range to subset
      to.</p></dd>    <dt></dt><dd><p><code>sort</code>: sort the object after subsetting. Usually
      not necessary.</p></dd>    <dt></dt><dd><p><code>...</code>: additional arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/subset.html'>subset(obj, from=10, to=20, sort=TRUE)</a></code></p></dd></dl></p></dd>

    <dt>displayPars</dt><dd><p><code>signature(x="BiomartGeneRegionTrack",
      name="character")</code>: list the value of the display parameter
      <code>name</code>. See <code><a href='settings.html'>settings</a></code> for details on display
      parameters and customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>displayPars(x, name)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>displayPars(obj, "col")</a></code></p></dd></dl></p></dd>


    <dt>displayPars</dt><dd><p><code>signature(x="BiomartGeneRegionTrack",
      name="missing")</code>: list the value of all available display
      parameters. See <code><a href='settings.html'>settings</a></code> for details on display
      parameters and customization.      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>displayPars(obj)</a></code></p></dd></dl></p></dd>

    <dt>getPar</dt><dd><p><code>signature(x="BiomartGeneRegionTrack", name="character")</code>:
      alias for the <code>displayPars</code> method. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>getPar(x, name)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>getPar(obj, "col")</a></code></p></dd></dl></p></dd>

    <dt>getPar</dt><dd><p><code>signature(x="BiomartGeneRegionTrack", name="missing")</code>:
      alias for the <code>displayPars</code> method. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>getPar(obj)</a></code></p></dd></dl></p></dd>

    <dt>displayPars&lt;-</dt><dd><p><code>signature(x="BiomartGeneRegionTrack",
      value="list")</code>: set display parameters using the values of the
      named list in <code>value</code>. See <code><a href='settings.html'>settings</a></code> for details
      on display parameters and customization.      <em>Usage:</em>      <code>displayPars&lt;-(x, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>displayPars(obj) &lt;- list(col="red", lwd=2)</code></p></dd></dl></p></dd>

    <dt>setPar</dt><dd><p><code>signature(x="BiomartGeneRegionTrack", value="character")</code>:
      set the single display parameter <code>name</code> to <code>value</code>. Note
      that display parameters in the <code>BiomartGeneRegionTrack</code> class are
      pass-by-reference, so no re-assignmnet to the symbol <code>obj</code> is
      necessary. See <code><a href='settings.html'>settings</a></code> for details on display
      parameters and customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>setPar(x, name, value)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>name</code>: the name of the display parameter to set.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>setPar(obj, "col", "red")</a></code></p></dd></dl></p></dd>

    <dt>setPar</dt><dd><p><code>signature(x="BiomartGeneRegionTrack", value="list")</code>: set
      display parameters by the values of the named list in
      <code>value</code>.  Note that display parameters in the
      <code>BiomartGeneRegionTrack</code> class are pass-by-reference, so no
      re-assignmnet to the symbol <code>obj</code> is necessary. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>setPar(obj, list(col="red", lwd=2))</a></code></p></dd></dl></p></dd>

    <dt>names</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: return the value of
      the <code>name</code> slot.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/names.html'>names(x)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/names.html'>names(obj)</a></code></p></dd></dl></p></dd>

    <dt>names&lt;-</dt><dd><p><code>signature(x="BiomartGeneRegionTrack", value="character")</code>:
      set the value of the <code>name</code> slot.      <em>Usage:</em>      <code>names&lt;-(x, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>names(obj) &lt;- "foo"</code></p></dd></dl></p></dd>

    <dt>coords</dt><dd><p><code>signature(ImageMap="BiomartGeneRegionTrack")</code>: return the
      coordinates from the internal image map.      <em>Usage:</em>      <code><a href='GdObject-class.html'>coords(ImageMap)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='GdObject-class.html'>coords(obj)</a></code></p></dd></dl></p></dd>

    <dt>tags</dt><dd><p><code>signature(x="BiomartGeneRegionTrack")</code>: return the tags from the
      internal image map.      <em>Usage:</em>      <code><a href='GdObject-class.html'>tags(x)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='GdObject-class.html'>tags(obj)</a></code></p></dd></dl></p></dd>
  
</dl>


    <h2 class="hasAnchor" id="display-parameters"><a class="anchor" href="#display-parameters"></a>Display Parameters</h2>

     

<p>The following display parameters are set for objects of class
  <code>BiomartGeneRegionTrack</code> upon instantiation, unless one or more
  of them have already been set by one of the optional sub-class
  initializers, which always get precedence over these global defaults.
  See <code><a href='settings.html'>settings</a></code> for details on setting graphical parameters
  for tracks.</p><dl>

    <dt></dt><dd><p><code>C_segment="burlywood4"</code>: Character or integer scalar.
      Fill color for annotation objects of type 'C_segment'.</p></dd> 

    <dt></dt><dd><p><code>D_segment="lightblue"</code>: Character or integer scalar.
      Fill color for annotation objects of type 'C_segment'.</p></dd> 

    <dt></dt><dd><p><code>J_segment="dodgerblue2"</code>: Character or integer scalar.
      Fill color for annotation objects of type 'C_segment'.</p></dd> 

    <dt></dt><dd><p><code>miRNA="cornflowerblue"</code>: Character or integer scalar.
      Fill color for annotation objects of type 'L_segment'.</p></dd> 

    <dt></dt><dd><p><code>miRNA_pseudogene="cornsilk"</code>: Character or integer
      scalar. Fill color for annotation objects of type 'miRNA_pseudogene'.</p></dd> 

    <dt></dt><dd><p><code>misc_RNA="cornsilk3"</code>: Character or integer scalar.
      Fill color for annotation objects of type 'misc_RNA'.</p></dd> 

    <dt></dt><dd><p><code>misc_RNA_pseudogene="cornsilk4"</code>: Character or integer
      scalar. Fill color for annotation objects of type 'misc_RNA_pseudogene'.</p></dd> 

    <dt></dt><dd><p><code>Mt_rRNA="yellow"</code>: Character or integer scalar. Fill
      color for annotation objects of type 'Mt_rRNA'.</p></dd> 

    <dt></dt><dd><p><code>Mt_tRNA="darkgoldenrod"</code>: Character or integer scalar.
      Fill color for annotation objects of type 'Mt_tRNA'.</p></dd> 

    <dt></dt><dd><p><code>Mt_tRNA_pseudogene="darkgoldenrod1"</code>: Character or
      integer scalar. Fill color for annotation objects of type
      'Mt_tRNA_pseudogene'.</p></dd> 

    <dt></dt><dd><p><code>protein_coding="#FFD58A"</code>: Character or integer
      scalar. Fill color for annotation objects of type 'protein_coding'.</p></dd> 

    <dt></dt><dd><p><code>pseudogene="brown1"</code>: Character or integer scalar.
      Fill color for annotation objects of type 'pseudogene'.</p></dd> 

    <dt></dt><dd><p><code>retrotransposed="blueviolet"</code>: Character or integer
      scalar. Fill color for annotation objects of type 'retrotransposed'.</p></dd> 

    <dt></dt><dd><p><code>rRNA="darkolivegreen1"</code>: Character or integer scalar.
      Fill color for annotation objects of type 'rRNA'.</p></dd> 

    <dt></dt><dd><p><code>rRNA_pseudogene="darkolivegreen"</code>: Character or
      integer scalar. Fill color for annotation objects of type
      'rRNA_pseudogene'.</p></dd> 

    <dt></dt><dd><p><code>scRNA="gold4"</code>: Character or integer scalar. Fill
      color for annotation objects of type 'scRNA'.</p></dd> 

    <dt></dt><dd><p><code>scRNA_pseudogene="darkorange2"</code>: Character or integer
      scalar. Fill color for annotation objects of type 'scRNA_pseudogene'.</p></dd> 

    <dt></dt><dd><p><code>snoRNA="cyan"</code>: Character or integer scalar. Fill
      color for annotation objects of type 'snoRNA'.</p></dd> 

    <dt></dt><dd><p><code>snoRNA_pseudogene="cyan2"</code>: Character or integer
      scalar. Fill color for annotation objects of type 'snoRNA_pseudogene'.</p></dd> 

    <dt></dt><dd><p><code>snRNA="coral"</code>: Character or integer scalar. Fill
      color for annotation objects of type 'snRNA'.</p></dd> 

    <dt></dt><dd><p><code>snRNA_pseudogene="coral3"</code>: Character or integer
      scalar. Fill color for annotation objects of type 'snRNA_pseudogene'.</p></dd> 

    <dt></dt><dd><p><code>tRNA_pseudogene="antiquewhite3"</code>: Character or integer
      scalar. Fill color for annotation objects of type 'tRNA_pseudogene'.</p></dd> 

    <dt></dt><dd><p><code>utr3="#FFD58A"</code>: Character or integer scalar. Fill
      color for annotation objects of type 'utr3'.</p></dd> 

    <dt></dt><dd><p><code>utr5="#FFD58A"</code>: Character or integer scalar. Fill
      color for annotation objects of type 'utr5'.</p></dd> 

    <dt></dt><dd><p><code>V_segment="aquamarine"</code>: Character or integer scalar.
      Fill color for annotation objects of type 'V_segment'.</p></dd> 

    <dt></dt><dd><p><code>verbose=FALSE</code>: Logical scalar. Report data loading
      events from Bioamart or retrieval from cache.</p></dd> 

  
</dl> 

<p>Additional display parameters are being inherited from the respective
  parent classes. Note that not all of them may have an effect on the
  plotting of <code>BiomartGeneRegionTrack</code> objects.</p>
<p></p><dl>

    <dt></dt><dd><p><code><a href='GeneRegionTrack-class.html'>GeneRegionTrack</a></code>:       <dl>        <dt></dt><dd><p><code>arrowHeadMaxWidth=20</code>: Numeric scalar. The maximum
          width of the arrow head in pixels if <code>shape</code> is <code>arrow</code>.</p></dd>         <dt></dt><dd><p><code>arrowHeadWidth=10</code>: Numeric scalar. The width of
          the arrow head in pixels if <code>shape</code> is <code>fixedArrow</code>.</p></dd>         <dt></dt><dd><p><code>col=NULL</code>: Character or integer scalar. The border
          color for all track items. Defaults to using the same color
          as in <code>fill</code>, also taking into account different track
          <code>features</code>.</p></dd>         <dt></dt><dd><p><code>collapseTranscripts=FALSE</code>: Logical or character
          scalar. Can be one in <code>gene</code>, <code>longest</code>, <code>shortest</code>
          or <code>meta</code>. Merge all transcripts of the same gene into
          one single gene model. In the case of <code>gene</code> (or
          <code>TRUE</code>), this will only keep the start location of the
          first exon and the end location of the last exon from all
          transcripts of the gene. For <code>shortest</code> and <code>longest</code>,
          only the longest or shortest transcript model is retained.
          For <code>meta</code>, a meta-transcript containing the union of
          all exons is formed (essentially identical to the operation
          <code>reduce(geneModel)</code>).</p></dd>         <dt></dt><dd><p><code>exonAnnotation=NULL</code>: Character scalar. Add
          annotation information to the individual exon models. This
          can be a value in <code>symbol</code>, <code>gene</code>, <code>transcript</code>,
          <code>exon</code> or <code>feature</code>. Defaults to <code>exon</code>. Only
          works if <code>showExonId</code> is not <code>FALSE</code>.</p></dd>         <dt></dt><dd><p><code>fill="orange"</code>: Character or integer scalar. The
          fill color for untyped items. This is also used to connect
          grouped items. See <code><a href='grouping.html'>grouping</a></code> for details.</p></dd>         <dt></dt><dd><p><code>min.distance=0</code>: Numeric scalar. The minimum
          pixel distance before collapsing range items, only if
          <code>collapse==TRUE</code>. See <code><a href='collapsing.html'>collapsing</a></code> for
          details. Note that a value larger than 0 may lead to UTR
          regions being merged to CDS regions, which in most cases is
          not particularly useful.</p></dd>         <dt></dt><dd><p><code>shape=c("smallArrow", "box")</code>: Character scalar.
          The shape in which to display the track items. Currently only
          <code>box</code>, <code>arrow</code>, <code>ellipse</code>, and <code>smallArrow</code>
          are implemented.</p></dd>         <dt></dt><dd><p><code>showExonId=NULL</code>: Logical scalar. Control whether
          to plot the individual exon identifiers.</p></dd>         <dt></dt><dd><p><code>thinBoxFeature=c("utr", "ncRNA", "utr3", "utr5",
          "3UTR", "5UTR", "miRNA", "lincRNA", "three_prime_UTR",
          "five_prime_UTR")</code>: Character vector. A listing of feature
          types that should be drawn with thin boxes. Typically those
          are non-coding elements.</p></dd>         <dt></dt><dd><p><code>transcriptAnnotation=NULL</code> <code>(Aliases:
          transcriptAnnotation)</code>: Character scalar. Add annotation
          information as transcript labels. This can be a value in
          <code>symbol</code>, <code>gene</code>, <code>transcript</code>, <code>exon</code>
          or <code>feature</code>. Defaults to <code>symbol</code>. Only works if
          <code>showId</code> is not <code>FALSE</code>.</p></dd> </dl></p></dd> 

    <dt></dt><dd><p><code><a href='AnnotationTrack-class.html'>AnnotationTrack</a></code>:       <dl>        <dt></dt><dd><p><code>cex=1</code>: Numeric scalar. The font expansion factor
          for item identifiers.</p></dd>         <dt></dt><dd><p><code>cex.group=0.6</code>: Numeric scalar. The font expansion
          factor for the group-level annotation.</p></dd>         <dt></dt><dd><p><code>col.line="darkgray"</code>: Character scalar. The color
          used for connecting lines between grouped items. Defaults to
          a light gray, but if set to <code>NULL</code> the same color as
          for the first item in the group is used.</p></dd>         <dt></dt><dd><p><code>featureAnnotation=NULL</code>: Character scalar. Add
          annotation information to the individual track elements. This
          can be a value in <code>id</code>, <code>group</code> or <code>feature</code>.
          Defaults to <code>id</code>. Only works if <code>showFeatureId</code> is
          not <code>FALSE</code>.</p></dd>         <dt></dt><dd><p><code>fontcolor.group="#808080"</code> <code>(Aliases:
          fontcolor.group)</code>: Character or integer scalar. The font
          color for the group-level annotation.</p></dd>         <dt></dt><dd><p><code>fontcolor.item="white"</code> <code>(Aliases:
          fontcolor.item)</code>: Character or integer scalar. The font color
          for item identifiers.</p></dd>         <dt></dt><dd><p><code>fontface.group=2</code>: Numeric scalar. The font face
          for the group-level annotation.</p></dd>         <dt></dt><dd><p><code>fontfamily.group="sans"</code>: Character scalar. The
          font family for the group-level annotation.</p></dd>         <dt></dt><dd><p><code>fontsize.group=12</code>: Numeric scalar. The font size
          for the group-level annotation.</p></dd>         <dt></dt><dd><p><code>groupAnnotation=NULL</code>: Character scalar. Add
          annotation information as group labels. This can be a value
          in <code>id</code>, <code>group</code> or <code>feature</code>. Defaults to
          <code>group</code>. Only works if <code>showId</code> is not <code>FALSE</code>.</p></dd>         <dt></dt><dd><p><code>just.group="left"</code> <code>(Aliases: just.group)</code>:
          Character scalar. the justification of group labels. Either
          <code>left</code>, <code>right</code>, <code>above</code> or <code>below</code>.</p></dd>         <dt></dt><dd><p><code>lex=1</code>: Numeric scalar. The line expansion factor
          for all track items. This is also used to connect grouped
          items. See <code><a href='grouping.html'>grouping</a></code> for details.</p></dd>         <dt></dt><dd><p><code>lineheight=1</code>: Numeric scalar. The font line
          height for item identifiers.</p></dd>         <dt></dt><dd><p><code>lty="solid"</code>: Character or integer scalar. The
          line type for all track items. This is also used to connect
          grouped items. See <code><a href='grouping.html'>grouping</a></code> for details.</p></dd>         <dt></dt><dd><p><code>lwd=1</code>: Integer scalar. The line width for all
          track items. This is also used to connect grouped items. See
          <code><a href='grouping.html'>grouping</a></code> for details.</p></dd>         <dt></dt><dd><p><code>mergeGroups=FALSE</code>: Logical scalar. Merge fully
          overlapping groups if <code>collapse==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>min.height=3</code>: Numeric scalar. The minimum range
          height in pixels to display. All ranges are expanded to this
          size in order to avoid rendering issues. See <code><a href='collapsing.html'>collapsing</a></code>
          for details. For feathered bars indicating the strandedness
          of grouped items this also controls the height of the arrow
          feathers.</p></dd>         <dt></dt><dd><p><code>min.width=1</code>: Numeric scalar. The minimum range
          width in pixels to display. All ranges are expanded to this
          size in order to avoid rendering issues. See <code><a href='collapsing.html'>collapsing</a></code>
          for details.</p></dd>         <dt></dt><dd><p><code>rotation=0</code>: Numeric scalar. The degree of text
          rotation for item identifiers.</p></dd>         <dt></dt><dd><p><code>rotation.group=0</code>: Numeric scalar. The degree of
          text rotation for group labels.</p></dd>         <dt></dt><dd><p><code>rotation.item=0</code>: Numeric scalar. The degree of
          text rotation for item identifiers.</p></dd>         <dt></dt><dd><p><code>showFeatureId=FALSE</code>: Logical scalar. Control
          whether to plot the individual track item identifiers.</p></dd>         <dt></dt><dd><p><code>showId=FALSE</code>: Logical scalar. Control whether
          to annotate individual groups.</p></dd>         <dt></dt><dd><p><code>showOverplotting=FALSE</code>: Logical scalar. Use a
          color gradient to show the amount of overplotting for collapsed
          items. This implies that <code>collapse==TRUE</code></p></dd>         <dt></dt><dd><p><code>size=1</code>: Numeric scalar. The relative size of
          the track. Can be overridden in the <code><a href='plotTracks.html'>plotTracks</a></code>
          function.</p></dd> </dl></p></dd> 

    <dt></dt><dd><p><code><a href='StackedTrack-class.html'>StackedTrack</a></code>:       <dl>        <dt></dt><dd><p><code>reverseStacking=FALSE</code>: Logical flag. Reverse
          the y-ordering of stacked items. I.e., features that are
          plotted on the bottom-most stacks will be moved to the top-most
          stack and vice versa.</p></dd>         <dt></dt><dd><p><code>stackHeight=0.75</code>: Numeric between 0 and 1.
          Controls the vertical size and spacing between stacked
          elements. The number defines the proportion of the total
          available space for the stack that is used to draw the glyphs.
          E.g., a value of 0.5 means that half of the available vertical
          drawing space (for each stacking line) is used for the glyphs,
          and thus one quarter of the available space each is used for
          spacing above and below the glyph. Defaults to 0.75.</p></dd> </dl></p></dd> 

    <dt></dt><dd><p><code><a href='GdObject-class.html'>GdObject</a></code>:       <dl>        <dt></dt><dd><p><code>alpha=1</code>: Numeric scalar. The transparency for
          all track items.</p></dd>         <dt></dt><dd><p><code>alpha.title=NULL</code>: Numeric scalar. The transparency
          for the title panel.</p></dd>         <dt></dt><dd><p><code>background.legend="transparent"</code>: Integer or
          character scalar. The background color for the legend.</p></dd>         <dt></dt><dd><p><code>background.panel="transparent"</code>: Integer or
          character scalar. The background color of the content panel.</p></dd>         <dt></dt><dd><p><code>background.title="lightgray"</code>: Integer or character
          scalar. The background color for the title panel.</p></dd>         <dt></dt><dd><p><code>cex.axis=NULL</code>: Numeric scalar. The expansion
          factor for the axis annotation. Defaults to <code>NULL</code>, in
          which case it is automatically determined based on the
          available space.</p></dd>         <dt></dt><dd><p><code>cex.title=NULL</code>: Numeric scalar. The expansion
          factor for the title panel. This effects the fontsize of both
          the title and the axis, if any. Defaults to <code>NULL</code>,
          which means that the text size is automatically adjusted to
          the available space.</p></dd>         <dt></dt><dd><p><code>col.axis="white"</code>: Integer or character scalar.
          The font and line color for the y axis, if any.</p></dd>         <dt></dt><dd><p><code>col.border.title="white"</code>: Integer or character
          scalar. The border color for the title panels.</p></dd>         <dt></dt><dd><p><code>col.frame="lightgray"</code>: Integer or character
          scalar. The line color used for the panel frame, if
          <code>frame==TRUE</code></p></dd>         <dt></dt><dd><p><code>col.grid="#808080"</code>: Integer or character scalar.
          Default line color for grid lines, both when <code>type=="g"</code>
          in <code><a href='DataTrack-class.html'>DataTrack</a></code>s and when display parameter
          <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>col.symbol=NULL</code>: Integer or character scalar.
          Default colors for plot symbols. Usually the same as the
          global <code>col</code> parameter.</p></dd>         <dt></dt><dd><p><code>col.title="white"</code> <code>(Aliases: fontcolor.title)</code>:
          Integer or character scalar. The border color for the title panels</p></dd>         <dt></dt><dd><p><code>collapse=TRUE</code>: Boolean controlling whether to
          collapse the content of the track to accomodate the minimum
          current device resolution. See <code><a href='collapsing.html'>collapsing</a></code> for
          details.</p></dd>         <dt></dt><dd><p><code>fontcolor="black"</code>: Integer or character scalar.
          The font color for all text, unless a more specific definition
          exists.</p></dd>         <dt></dt><dd><p><code>fontface=1</code>: Integer or character scalar. The
          font face for all text, unless a more specific definition exists.</p></dd>         <dt></dt><dd><p><code>fontface.title=2</code>: Integer or character scalar.
          The font face for the title panels.</p></dd>         <dt></dt><dd><p><code>fontfamily="sans"</code>: Integer or character scalar.
          The font family for all text, unless a more specific definition
          exists.</p></dd>         <dt></dt><dd><p><code>fontfamily.title="sans"</code>: Integer or character
          scalar. The font family for the title panels.</p></dd>         <dt></dt><dd><p><code>fontsize=12</code>: Numeric scalar. The font size for
          all text, unless a more specific definition exists.</p></dd>         <dt></dt><dd><p><code>frame=FALSE</code>: Boolean. Draw a frame around the
          track when plotting.</p></dd>         <dt></dt><dd><p><code>grid=FALSE</code>: Boolean, switching on/off the plotting
          of a grid.</p></dd>         <dt></dt><dd><p><code>h=-1</code>: Integer scalar. Parameter controlling the
          number of horizontal grid lines, see <code><a href='https://rdrr.io/pkg/lattice/man/panel.functions.html'>panel.grid</a></code>
          for details.</p></dd>         <dt></dt><dd><p><code>lty.grid="solid"</code>: Integer or character scalar.
          Default line type for grid lines, both when <code>type=="g"</code>
          in <code><a href='DataTrack-class.html'>DataTrack</a></code>s and when display parameter
          <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>lwd.border.title=1</code>: Integer scalar. The border
          width for the title panels.</p></dd>         <dt></dt><dd><p><code>lwd.grid=1</code>: Numeric scalar. Default line width
          for grid lines, both when <code>type=="g"</code> in <code><a href='DataTrack-class.html'>DataTrack</a></code>s
          and when display parameter <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>lwd.title=1</code>: Integer scalar. The border width
          for the title panels</p></dd>         <dt></dt><dd><p><code>reverseStrand=FALSE</code>: Logical scalar. Set up the
          plotting coordinates in 3' -&gt; 5' direction if <code>TRUE</code>.
          This will effectively mirror the plot on the vertical axis.</p></dd>         <dt></dt><dd><p><code>rotation.title=90</code> <code>(Aliases: rotation.title)</code>:
          The rotation angle for the text in the title panel. Even
          though this can be adjusted, the automatic resizing of the
          title panel will currently not work, so use at own risk.</p></dd>         <dt></dt><dd><p><code>showAxis=TRUE</code>: Boolean controlling whether to
          plot a y axis (only applies to track types where axes are
          implemented).</p></dd>         <dt></dt><dd><p><code>showTitle=TRUE</code>: Boolean controlling whether to
          plot a title panel. Although this can be set individually
          for each track, in multi-track plots as created by
          <code><a href='plotTracks.html'>plotTracks</a></code> there will still be an empty
          placeholder in case any of the other tracks include a title.
          The same holds true for axes. Note that the the title panel
          background color could be set to transparent in order to
          completely hide it.</p></dd>         <dt></dt><dd><p><code>v=-1</code>: Integer scalar. Parameter controlling the
          number of vertical grid lines, see <code><a href='https://rdrr.io/pkg/lattice/man/panel.functions.html'>panel.grid</a></code>
          for details.</p></dd> </dl></p></dd> 

  
</dl> 


    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    

<p>EBI Biomart webservice at <a href='http://www.biomart.org'>http://www.biomart.org</a>.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Florian Hahne</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'>

<p><code><a href='AnnotationTrack-class.html'>AnnotationTrack</a></code></p>
<p><code><a href='DisplayPars-class.html'>DisplayPars</a></code></p>
<p><code><a href='GdObject-class.html'>GdObject</a></code></p>
<p><code><a href='GeneRegionTrack-class.html'>GeneRegionTrack</a></code></p>
<p><code>GRanges</code></p>
<p><code><a href='ImageMap-class.html'>ImageMap</a></code></p>
<p><code>IRanges</code></p>
<p><code>Mart</code></p>
<p><code><a href='RangeTrack-class.html'>RangeTrack</a></code></p>
<p><code><a href='StackedTrack-class.html'>StackedTrack</a></code></p>
<p><code><a href='collapsing.html'>collapsing</a></code></p>
<p><code><a href='DataTrack-class.html'>DataTrack</a></code></p>
<p><code>getBM</code></p>
<p><code><a href='grouping.html'>grouping</a></code></p>
<p><code><a href='https://rdrr.io/pkg/lattice/man/panel.functions.html'>panel.grid</a></code></p>
<p><code><a href='plotTracks.html'>plotTracks</a></code></p>
<p><code><a href='settings.html'>settings</a></code></p>
<p><code>useMart</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'># \dontshow{</span>
<span class='co'>## Load some sample data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
<span class='co'># }</span>

<span class='co'>## Construct the object</span>
<span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='va'>bmTrack</span> <span class='op'>&lt;-</span> <span class='fu'>BiomartGeneRegionTrack</span><span class='op'>(</span>start<span class='op'>=</span><span class='fl'>26682683</span>, end<span class='op'>=</span><span class='fl'>26711643</span>,
chromosome<span class='op'>=</span><span class='fl'>7</span>, genome<span class='op'>=</span><span class='st'>"mm9"</span><span class='op'>)</span>
<span class='op'>}</span>

<span class='co'># \dontshow{</span>
<span class='co'>## For some annoying reason the postscript device does not know about</span>
<span class='co'>## the sans font</span>
<span class='kw'>if</span><span class='op'>(</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/interactive.html'>interactive</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>
<span class='op'>{</span>
<span class='va'>font</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/ps.options.html'>ps.options</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>$</span><span class='va'>family</span>
<span class='fu'><a href='DisplayPars-class.html'>displayPars</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>fontfamily<span class='op'>=</span><span class='va'>font</span>, fontfamily.title<span class='op'>=</span><span class='va'>font</span><span class='op'>)</span>
<span class='op'>}</span>
<span class='co'># }</span>

<span class='co'>## Plotting</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='img'><img src='BiomartGeneRegionTrack-class-1.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Track names</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "BiomartGeneRegionTrack"</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"foo"</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='img'><img src='BiomartGeneRegionTrack-class-2.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Subsetting and splitting</span>
<span class='va'>subTrack</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/subset.html'>subset</a></span><span class='op'>(</span><span class='va'>bmTrack</span>, from<span class='op'>=</span><span class='fl'>26700000</span>, to<span class='op'>=</span><span class='fl'>26705000</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] 21</div><div class='input'><span class='va'>subTrack</span> <span class='op'>&lt;-</span> <span class='va'>bmTrack</span><span class='op'>[</span><span class='fu'><a href='GeneRegionTrack-class.html'>transcript</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span><span class='op'>==</span><span class='st'>"ENSMUST00000144140"</span><span class='op'>]</span>
<span class='fu'><a href='https://rdrr.io/r/base/split.html'>split</a></span><span class='op'>(</span><span class='va'>bmTrack</span>, <span class='fu'><a href='GeneRegionTrack-class.html'>transcript</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; $ENSMUST00000005477
#&gt; GeneRegionTrack 'foo'
#&gt; | genome: mm9
#&gt; | active chromosome: chr7
#&gt; | annotation features: 10
#&gt; 
#&gt; $ENSMUST00000072438
#&gt; GeneRegionTrack 'foo'
#&gt; | genome: mm9
#&gt; | active chromosome: chr7
#&gt; | annotation features: 9
#&gt; 
#&gt; $ENSMUST00000144140
#&gt; GeneRegionTrack 'foo'
#&gt; | genome: mm9
#&gt; | active chromosome: chr7
#&gt; | annotation features: 2
#&gt; </div><div class='input'>
<span class='co'>## Accessors</span>
<span class='fu'><a href='https://rdrr.io/r/stats/start.html'>start</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] 26682639 26696257 26696573 26698884 26699767 26700411 26701953 26702307
#&gt;  [9] 26711087 26682695 26696257 26696573 26698884 26699767 26700411 26701953
#&gt; [17] 26702307 26710392 26711087 26701785 26702307</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/stats/start.html'>end</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] 26682874 26696419 26696722 26699044 26699943 26700552 26702140 26702448
#&gt;  [9] 26711643 26682874 26696419 26696722 26699044 26699943 26700552 26702140
#&gt; [17] 26702448 26710418 26711578 26702140 26702449</div><div class='input'><span class='fu'>width</span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] 236 163 150 161 177 142 188 142 557 180 163 150 161 177 142 188 142  27 492
#&gt; [20] 356 143</div><div class='input'><span class='fu'><a href='RangeTrack-class.html'>position</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] 26682756 26696338 26696648 26698964 26699855 26700482 26702046 26702378
#&gt;  [9] 26711365 26682784 26696338 26696648 26698964 26699855 26700482 26702046
#&gt; [17] 26702378 26710405 26711332 26701962 26702378</div><div class='input'><span class='fu'>width</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>width</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span><span class='op'>+</span><span class='fl'>100</span>

<span class='fu'>strand</span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+"
#&gt; [20] "+" "+"</div><div class='input'><span class='fu'>strand</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"-"</span>

<span class='fu'><a href='RangeTrack-class.html'>chromosome</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;   chr7 
#&gt; "chr7" </div><div class='input'><span class='fu'><a href='RangeTrack-class.html'>chromosome</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"chrX"</span>

<span class='fu'>genome</span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "mm9"</div><div class='input'><span class='fu'>genome</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"hg19"</span>

<span class='fu'><a href='RangeTrack-class.html'>range</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; IRanges object with 21 ranges and 0 metadata columns:
#&gt;            start       end     width
#&gt;        &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
#&gt;    [1]  26682639  26682874       236
#&gt;    [2]  26696257  26696419       163
#&gt;    [3]  26696573  26696722       150
#&gt;    [4]  26698884  26699044       161
#&gt;    [5]  26699767  26699943       177
#&gt;    ...       ...       ...       ...
#&gt;   [17]  26702307  26702448       142
#&gt;   [18]  26710392  26710418        27
#&gt;   [19]  26711087  26711578       492
#&gt;   [20]  26701785  26702140       356
#&gt;   [21]  26702307  26702449       143</div><div class='input'><span class='fu'>ranges</span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; GRanges object with 21 ranges and 6 metadata columns:
#&gt;        seqnames            ranges strand |        feature               gene
#&gt;           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;character&gt;        &lt;character&gt;
#&gt;    [1]     chr7 26682639-26682874      + | protein_coding ENSMUSG00000030483
#&gt;    [2]     chr7 26696257-26696419      + | protein_coding ENSMUSG00000030483
#&gt;    [3]     chr7 26696573-26696722      + | protein_coding ENSMUSG00000030483
#&gt;    [4]     chr7 26698884-26699044      + | protein_coding ENSMUSG00000030483
#&gt;    [5]     chr7 26699767-26699943      + | protein_coding ENSMUSG00000030483
#&gt;    ...      ...               ...    ... .            ...                ...
#&gt;   [17]     chr7 26702307-26702448      + | protein_coding ENSMUSG00000030483
#&gt;   [18]     chr7 26710392-26710418      + | protein_coding ENSMUSG00000030483
#&gt;   [19]     chr7 26711087-26711578      + | protein_coding ENSMUSG00000030483
#&gt;   [20]     chr7 26701785-26702140      + | protein_coding ENSMUSG00000030483
#&gt;   [21]     chr7 26702307-26702449      + | protein_coding ENSMUSG00000030483
#&gt;                      exon         transcript      symbol      rank
#&gt;               &lt;character&gt;        &lt;character&gt; &lt;character&gt; &lt;numeric&gt;
#&gt;    [1] ENSMUSE00000742021 ENSMUST00000072438     Cyp2b10         1
#&gt;    [2] ENSMUSE00000449901 ENSMUST00000072438     Cyp2b10         2
#&gt;    [3] ENSMUSE00000199425 ENSMUST00000072438     Cyp2b10         3
#&gt;    [4] ENSMUSE00000511070 ENSMUST00000072438     Cyp2b10         4
#&gt;    [5] ENSMUSE00000514929 ENSMUST00000072438     Cyp2b10         5
#&gt;    ...                ...                ...         ...       ...
#&gt;   [17] ENSMUSE00000507557 ENSMUST00000005477     Cyp2b10         8
#&gt;   [18] ENSMUSE00000496705 ENSMUST00000005477     Cyp2b10         9
#&gt;   [19] ENSMUSE00000750625 ENSMUST00000005477     Cyp2b10        10
#&gt;   [20] ENSMUSE00000736922 ENSMUST00000144140     Cyp2b10         1
#&gt;   [21] ENSMUSE00000748299 ENSMUST00000144140     Cyp2b10         2
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## Annotation</span>
<span class='fu'><a href='AnnotationTrack-class.html'>identifier</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt;  [8] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt; [15] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</div><div class='input'><span class='fu'><a href='AnnotationTrack-class.html'>identifier</a></span><span class='op'>(</span><span class='va'>bmTrack</span>, <span class='st'>"lowest"</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "ENSMUSE00000742021" "ENSMUSE00000449901" "ENSMUSE00000199425"
#&gt;  [4] "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423"
#&gt;  [7] "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000498616"
#&gt; [10] "ENSMUSE00000489385" "ENSMUSE00000449901" "ENSMUSE00000199425"
#&gt; [13] "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423"
#&gt; [16] "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000496705"
#&gt; [19] "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299"</div><div class='input'><span class='fu'><a href='AnnotationTrack-class.html'>identifier</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"bar"</span>

<span class='fu'><a href='RangeTrack-class.html'>feature</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "protein_coding" "protein_coding" "protein_coding" "protein_coding"
#&gt;  [5] "protein_coding" "protein_coding" "protein_coding" "protein_coding"
#&gt;  [9] "protein_coding" "protein_coding" "protein_coding" "protein_coding"
#&gt; [13] "protein_coding" "protein_coding" "protein_coding" "protein_coding"
#&gt; [17] "protein_coding" "protein_coding" "protein_coding" "protein_coding"
#&gt; [21] "protein_coding"</div><div class='input'><span class='fu'><a href='RangeTrack-class.html'>feature</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"foo"</span>

<span class='fu'><a href='GeneRegionTrack-class.html'>exon</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "ENSMUSE00000742021" "ENSMUSE00000449901" "ENSMUSE00000199425"
#&gt;  [4] "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423"
#&gt;  [7] "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000498616"
#&gt; [10] "ENSMUSE00000489385" "ENSMUSE00000449901" "ENSMUSE00000199425"
#&gt; [13] "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423"
#&gt; [16] "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000496705"
#&gt; [19] "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299"</div><div class='input'><span class='fu'><a href='GeneRegionTrack-class.html'>exon</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>letters</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span><span class='op'>]</span>

<span class='fu'><a href='GeneRegionTrack-class.html'>gene</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt;  [4] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt;  [7] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt; [10] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt; [13] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt; [16] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt; [19] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"</div><div class='input'><span class='fu'><a href='GeneRegionTrack-class.html'>gene</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"bar"</span>

<span class='fu'><a href='GeneRegionTrack-class.html'>symbol</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt;  [8] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt; [15] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</div><div class='input'><span class='fu'><a href='GeneRegionTrack-class.html'>symbol</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"foo"</span>

<span class='fu'><a href='GeneRegionTrack-class.html'>transcript</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438"
#&gt;  [4] "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438"
#&gt;  [7] "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438"
#&gt; [10] "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477"
#&gt; [13] "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477"
#&gt; [16] "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477"
#&gt; [19] "ENSMUST00000005477" "ENSMUST00000144140" "ENSMUST00000144140"</div><div class='input'><span class='fu'><a href='GeneRegionTrack-class.html'>transcript</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"foo"</span>, <span class='st'>"bar"</span><span class='op'>)</span>
<span class='fu'><a href='RangeTrack-class.html'>chromosome</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"chr7"</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span>
</div><div class='img'><img src='BiomartGeneRegionTrack-class-3.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='fu'>values</span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;           feature               gene               exon         transcript
#&gt; 1  protein_coding ENSMUSG00000030483 ENSMUSE00000742021 ENSMUST00000072438
#&gt; 2  protein_coding ENSMUSG00000030483 ENSMUSE00000449901 ENSMUST00000072438
#&gt; 3  protein_coding ENSMUSG00000030483 ENSMUSE00000199425 ENSMUST00000072438
#&gt; 4  protein_coding ENSMUSG00000030483 ENSMUSE00000511070 ENSMUST00000072438
#&gt; 5  protein_coding ENSMUSG00000030483 ENSMUSE00000514929 ENSMUST00000072438
#&gt; 6  protein_coding ENSMUSG00000030483 ENSMUSE00000199423 ENSMUST00000072438
#&gt; 7  protein_coding ENSMUSG00000030483 ENSMUSE00000535814 ENSMUST00000072438
#&gt; 8  protein_coding ENSMUSG00000030483 ENSMUSE00000507557 ENSMUST00000072438
#&gt; 9  protein_coding ENSMUSG00000030483 ENSMUSE00000498616 ENSMUST00000072438
#&gt; 10 protein_coding ENSMUSG00000030483 ENSMUSE00000489385 ENSMUST00000005477
#&gt; 11 protein_coding ENSMUSG00000030483 ENSMUSE00000449901 ENSMUST00000005477
#&gt; 12 protein_coding ENSMUSG00000030483 ENSMUSE00000199425 ENSMUST00000005477
#&gt; 13 protein_coding ENSMUSG00000030483 ENSMUSE00000511070 ENSMUST00000005477
#&gt; 14 protein_coding ENSMUSG00000030483 ENSMUSE00000514929 ENSMUST00000005477
#&gt; 15 protein_coding ENSMUSG00000030483 ENSMUSE00000199423 ENSMUST00000005477
#&gt; 16 protein_coding ENSMUSG00000030483 ENSMUSE00000535814 ENSMUST00000005477
#&gt; 17 protein_coding ENSMUSG00000030483 ENSMUSE00000507557 ENSMUST00000005477
#&gt; 18 protein_coding ENSMUSG00000030483 ENSMUSE00000496705 ENSMUST00000005477
#&gt; 19 protein_coding ENSMUSG00000030483 ENSMUSE00000750625 ENSMUST00000005477
#&gt; 20 protein_coding ENSMUSG00000030483 ENSMUSE00000736922 ENSMUST00000144140
#&gt; 21 protein_coding ENSMUSG00000030483 ENSMUSE00000748299 ENSMUST00000144140
#&gt;     symbol rank
#&gt; 1  Cyp2b10    1
#&gt; 2  Cyp2b10    2
#&gt; 3  Cyp2b10    3
#&gt; 4  Cyp2b10    4
#&gt; 5  Cyp2b10    5
#&gt; 6  Cyp2b10    6
#&gt; 7  Cyp2b10    7
#&gt; 8  Cyp2b10    8
#&gt; 9  Cyp2b10    9
#&gt; 10 Cyp2b10    1
#&gt; 11 Cyp2b10    2
#&gt; 12 Cyp2b10    3
#&gt; 13 Cyp2b10    4
#&gt; 14 Cyp2b10    5
#&gt; 15 Cyp2b10    6
#&gt; 16 Cyp2b10    7
#&gt; 17 Cyp2b10    8
#&gt; 18 Cyp2b10    9
#&gt; 19 Cyp2b10   10
#&gt; 20 Cyp2b10    1
#&gt; 21 Cyp2b10    2</div><div class='input'>
<span class='co'>## Grouping</span>
<span class='fu'><a href='AnnotationTrack-class.html'>group</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438"
#&gt;  [4] "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438"
#&gt;  [7] "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438"
#&gt; [10] "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477"
#&gt; [13] "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477"
#&gt; [16] "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477"
#&gt; [19] "ENSMUST00000005477" "ENSMUST00000144140" "ENSMUST00000144140"</div><div class='input'><span class='fu'><a href='AnnotationTrack-class.html'>group</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"Group 1"</span>
<span class='fu'><a href='GeneRegionTrack-class.html'>transcript</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "Group 1" "Group 1"</div><div class='input'><span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span>
</div><div class='img'><img src='BiomartGeneRegionTrack-class-4.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Stacking</span>
<span class='fu'><a href='StackedTrack-class.html'>stacking</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "squish"</div><div class='input'><span class='fu'><a href='StackedTrack-class.html'>stacking</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"dense"</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='img'><img src='BiomartGeneRegionTrack-class-5.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## coercion</span>
<span class='fu'>as</span><span class='op'>(</span><span class='va'>bmTrack</span>, <span class='st'>"data.frame"</span><span class='op'>)</span>
</div><div class='output co'>#&gt;    X.seqnames  X.start    X.end X.width X.strand      X.feature
#&gt; 1        chr7 26682639 26682874     236        + protein_coding
#&gt; 2        chr7 26696257 26696419     163        + protein_coding
#&gt; 3        chr7 26696573 26696722     150        + protein_coding
#&gt; 4        chr7 26698884 26699044     161        + protein_coding
#&gt; 5        chr7 26699767 26699943     177        + protein_coding
#&gt; 6        chr7 26700411 26700552     142        + protein_coding
#&gt; 7        chr7 26701953 26702140     188        + protein_coding
#&gt; 8        chr7 26702307 26702448     142        + protein_coding
#&gt; 9        chr7 26711087 26711643     557        + protein_coding
#&gt; 10       chr7 26682695 26682874     180        + protein_coding
#&gt; 11       chr7 26696257 26696419     163        + protein_coding
#&gt; 12       chr7 26696573 26696722     150        + protein_coding
#&gt; 13       chr7 26698884 26699044     161        + protein_coding
#&gt; 14       chr7 26699767 26699943     177        + protein_coding
#&gt; 15       chr7 26700411 26700552     142        + protein_coding
#&gt; 16       chr7 26701953 26702140     188        + protein_coding
#&gt; 17       chr7 26702307 26702448     142        + protein_coding
#&gt; 18       chr7 26710392 26710418      27        + protein_coding
#&gt; 19       chr7 26711087 26711578     492        + protein_coding
#&gt; 20       chr7 26701785 26702140     356        + protein_coding
#&gt; 21       chr7 26702307 26702449     143        + protein_coding
#&gt;                X.gene             X.exon       X.transcript X.symbol X.rank
#&gt; 1  ENSMUSG00000030483 ENSMUSE00000742021 ENSMUST00000072438  Cyp2b10      1
#&gt; 2  ENSMUSG00000030483 ENSMUSE00000449901 ENSMUST00000072438  Cyp2b10      2
#&gt; 3  ENSMUSG00000030483 ENSMUSE00000199425 ENSMUST00000072438  Cyp2b10      3
#&gt; 4  ENSMUSG00000030483 ENSMUSE00000511070 ENSMUST00000072438  Cyp2b10      4
#&gt; 5  ENSMUSG00000030483 ENSMUSE00000514929 ENSMUST00000072438  Cyp2b10      5
#&gt; 6  ENSMUSG00000030483 ENSMUSE00000199423 ENSMUST00000072438  Cyp2b10      6
#&gt; 7  ENSMUSG00000030483 ENSMUSE00000535814 ENSMUST00000072438  Cyp2b10      7
#&gt; 8  ENSMUSG00000030483 ENSMUSE00000507557 ENSMUST00000072438  Cyp2b10      8
#&gt; 9  ENSMUSG00000030483 ENSMUSE00000498616 ENSMUST00000072438  Cyp2b10      9
#&gt; 10 ENSMUSG00000030483 ENSMUSE00000489385 ENSMUST00000005477  Cyp2b10      1
#&gt; 11 ENSMUSG00000030483 ENSMUSE00000449901 ENSMUST00000005477  Cyp2b10      2
#&gt; 12 ENSMUSG00000030483 ENSMUSE00000199425 ENSMUST00000005477  Cyp2b10      3
#&gt; 13 ENSMUSG00000030483 ENSMUSE00000511070 ENSMUST00000005477  Cyp2b10      4
#&gt; 14 ENSMUSG00000030483 ENSMUSE00000514929 ENSMUST00000005477  Cyp2b10      5
#&gt; 15 ENSMUSG00000030483 ENSMUSE00000199423 ENSMUST00000005477  Cyp2b10      6
#&gt; 16 ENSMUSG00000030483 ENSMUSE00000535814 ENSMUST00000005477  Cyp2b10      7
#&gt; 17 ENSMUSG00000030483 ENSMUSE00000507557 ENSMUST00000005477  Cyp2b10      8
#&gt; 18 ENSMUSG00000030483 ENSMUSE00000496705 ENSMUST00000005477  Cyp2b10      9
#&gt; 19 ENSMUSG00000030483 ENSMUSE00000750625 ENSMUST00000005477  Cyp2b10     10
#&gt; 20 ENSMUSG00000030483 ENSMUSE00000736922 ENSMUST00000144140  Cyp2b10      1
#&gt; 21 ENSMUSG00000030483 ENSMUSE00000748299 ENSMUST00000144140  Cyp2b10      2
#&gt;           feature               gene               exon         transcript
#&gt; 1  protein_coding ENSMUSG00000030483 ENSMUSE00000742021 ENSMUST00000072438
#&gt; 2  protein_coding ENSMUSG00000030483 ENSMUSE00000449901 ENSMUST00000072438
#&gt; 3  protein_coding ENSMUSG00000030483 ENSMUSE00000199425 ENSMUST00000072438
#&gt; 4  protein_coding ENSMUSG00000030483 ENSMUSE00000511070 ENSMUST00000072438
#&gt; 5  protein_coding ENSMUSG00000030483 ENSMUSE00000514929 ENSMUST00000072438
#&gt; 6  protein_coding ENSMUSG00000030483 ENSMUSE00000199423 ENSMUST00000072438
#&gt; 7  protein_coding ENSMUSG00000030483 ENSMUSE00000535814 ENSMUST00000072438
#&gt; 8  protein_coding ENSMUSG00000030483 ENSMUSE00000507557 ENSMUST00000072438
#&gt; 9  protein_coding ENSMUSG00000030483 ENSMUSE00000498616 ENSMUST00000072438
#&gt; 10 protein_coding ENSMUSG00000030483 ENSMUSE00000489385 ENSMUST00000005477
#&gt; 11 protein_coding ENSMUSG00000030483 ENSMUSE00000449901 ENSMUST00000005477
#&gt; 12 protein_coding ENSMUSG00000030483 ENSMUSE00000199425 ENSMUST00000005477
#&gt; 13 protein_coding ENSMUSG00000030483 ENSMUSE00000511070 ENSMUST00000005477
#&gt; 14 protein_coding ENSMUSG00000030483 ENSMUSE00000514929 ENSMUST00000005477
#&gt; 15 protein_coding ENSMUSG00000030483 ENSMUSE00000199423 ENSMUST00000005477
#&gt; 16 protein_coding ENSMUSG00000030483 ENSMUSE00000535814 ENSMUST00000005477
#&gt; 17 protein_coding ENSMUSG00000030483 ENSMUSE00000507557 ENSMUST00000005477
#&gt; 18 protein_coding ENSMUSG00000030483 ENSMUSE00000496705 ENSMUST00000005477
#&gt; 19 protein_coding ENSMUSG00000030483 ENSMUSE00000750625 ENSMUST00000005477
#&gt; 20 protein_coding ENSMUSG00000030483 ENSMUSE00000736922 ENSMUST00000144140
#&gt; 21 protein_coding ENSMUSG00000030483 ENSMUSE00000748299 ENSMUST00000144140
#&gt;     symbol rank
#&gt; 1  Cyp2b10    1
#&gt; 2  Cyp2b10    2
#&gt; 3  Cyp2b10    3
#&gt; 4  Cyp2b10    4
#&gt; 5  Cyp2b10    5
#&gt; 6  Cyp2b10    6
#&gt; 7  Cyp2b10    7
#&gt; 8  Cyp2b10    8
#&gt; 9  Cyp2b10    9
#&gt; 10 Cyp2b10    1
#&gt; 11 Cyp2b10    2
#&gt; 12 Cyp2b10    3
#&gt; 13 Cyp2b10    4
#&gt; 14 Cyp2b10    5
#&gt; 15 Cyp2b10    6
#&gt; 16 Cyp2b10    7
#&gt; 17 Cyp2b10    8
#&gt; 18 Cyp2b10    9
#&gt; 19 Cyp2b10   10
#&gt; 20 Cyp2b10    1
#&gt; 21 Cyp2b10    2</div><div class='input'><span class='fu'>as</span><span class='op'>(</span><span class='va'>bmTrack</span>, <span class='st'>"UCSCData"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; UCSC track 'foo'
#&gt; UCSCData object with 3 ranges and 6 metadata columns:
#&gt;       seqnames            ranges strand |                 id        name
#&gt;          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt; &lt;character&gt;
#&gt;   [1]     chr7 26682695-26711578      + | ENSMUST00000005477     Cyp2b10
#&gt;   [2]     chr7 26682639-26711643      + | ENSMUST00000072438     Cyp2b10
#&gt;   [3]     chr7 26701785-26702449      + | ENSMUST00000144140     Cyp2b10
#&gt;           itemRgb blockCount             blockSizes            blockStarts
#&gt;       &lt;character&gt;  &lt;numeric&gt;            &lt;character&gt;            &lt;character&gt;
#&gt;   [1]       gold4         10 180,163,150,161,177,.. 0,13562,13878,16189,..
#&gt;   [2]       gold4          9 236,163,150,161,177,.. 0,13618,13934,16245,..
#&gt;   [3]       gold4          2                356,143                  0,522
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## HTML image map</span>
<span class='fu'><a href='GdObject-class.html'>coords</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; NULL</div><div class='input'><span class='fu'><a href='GdObject-class.html'>tags</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; NULL</div><div class='input'><span class='va'>bmTrack</span> <span class='op'>&lt;-</span> <span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span><span class='op'>$</span><span class='va'>foo</span>
</div><div class='img'><img src='BiomartGeneRegionTrack-class-6.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'><a href='GdObject-class.html'>coords</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                             x1   y1        x2    y2
#&gt; ENSMUSE00000742021    38.32764 64.5  41.79997 415.5
#&gt; ENSMUSE00000449901   239.54551 64.5 241.93921 415.5
#&gt; ENSMUSE00000199425   244.21469 64.5 246.41630 415.5
#&gt; ENSMUSE00000511070   278.36174 64.5 280.72588 415.5
#&gt; ENSMUSE00000514929   291.40884 64.5 294.00939 415.5
#&gt; ENSMUSE00000199423   300.92450 64.5 303.00790 415.5
#&gt; ENSMUSE00000535814   323.70890 64.5 326.47199 415.5
#&gt; ENSMUSE00000507557   328.93956 64.5 331.02296 415.5
#&gt; ENSMUSE00000498616   458.67175 64.5 466.88714 415.5
#&gt; ENSMUSE00000489385    39.15509 64.5  41.79997 415.5
#&gt; ENSMUSE00000449901.1 239.54551 64.5 241.93921 415.5
#&gt; ENSMUSE00000199425.1 244.21469 64.5 246.41630 415.5
#&gt; ENSMUSE00000511070.1 278.36174 64.5 280.72588 415.5
#&gt; ENSMUSE00000514929.1 291.40884 64.5 294.00939 415.5
#&gt; ENSMUSE00000199423.1 300.92450 64.5 303.00790 415.5
#&gt; ENSMUSE00000535814.1 323.70890 64.5 326.47199 415.5
#&gt; ENSMUSE00000507557.1 328.93956 64.5 331.02296 415.5
#&gt; ENSMUSE00000496705   448.09222 64.5 449.08221 415.5
#&gt; ENSMUSE00000750625   458.67175 64.5 465.92671 415.5
#&gt; ENSMUSE00000736922   321.22656 64.5 326.47199 415.5
#&gt; ENSMUSE00000748299   328.93956 64.5 331.03774 415.5</div><div class='input'><span class='fu'><a href='GdObject-class.html'>tags</a></span><span class='op'>(</span><span class='va'>bmTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; $fill
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt;              "gold4"              "gold4"              "gold4" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt;              "gold4"              "gold4"              "gold4" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt;              "gold4"              "gold4"              "gold4" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt;              "gold4"              "gold4"              "gold4" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt;              "gold4"              "gold4"              "gold4" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt;              "gold4"              "gold4"              "gold4" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;              "gold4"              "gold4"              "gold4" 
#&gt; 
#&gt; $strand
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt;                  "+"                  "+"                  "+" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt;                  "+"                  "+"                  "+" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt;                  "+"                  "+"                  "+" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt;                  "+"                  "+"                  "+" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt;                  "+"                  "+"                  "+" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt;                  "+"                  "+"                  "+" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;                  "+"                  "+"                  "+" 
#&gt; 
#&gt; $text
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt; "ENSMUSE00000742021" "ENSMUSE00000449901" "ENSMUSE00000199425" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt; "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt; "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000498616" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt; "ENSMUSE00000489385" "ENSMUSE00000449901" "ENSMUSE00000199425" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt; "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt; "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000496705" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" 
#&gt; 
#&gt; $start
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt;           "26682639"           "26696257"           "26696573" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt;           "26698884"           "26699767"           "26700411" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt;           "26701953"           "26702307"           "26711087" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt;           "26682695"           "26696257"           "26696573" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt;           "26698884"           "26699767"           "26700411" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt;           "26701953"           "26702307"           "26710371" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;           "26711087"           "26701785"           "26702307" 
#&gt; 
#&gt; $end
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt;           "26682874"           "26696419"           "26696722" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt;           "26699044"           "26699943"           "26700552" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt;           "26702140"           "26702448"           "26711643" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt;           "26682874"           "26696419"           "26696722" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt;           "26699044"           "26699943"           "26700552" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt;           "26702140"           "26702448"           "26710438" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;           "26711578"           "26702140"           "26702449" 
#&gt; 
#&gt; $feature
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt; 
#&gt; $gene
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt; 
#&gt; $exon
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt; "ENSMUSE00000742021" "ENSMUSE00000449901" "ENSMUSE00000199425" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt; "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt; "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000498616" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt; "ENSMUSE00000489385" "ENSMUSE00000449901" "ENSMUSE00000199425" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt; "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt; "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000496705" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" 
#&gt; 
#&gt; $transcript
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt; "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt; "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt; "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt; "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt; "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt; "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUST00000005477" "ENSMUST00000144140" "ENSMUST00000144140" 
#&gt; 
#&gt; $symbol
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt; 
#&gt; $rank
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt;                  "1"                  "2"                  "3" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt;                  "4"                  "5"                  "6" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt;                  "7"                  "8"                  "9" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt;                  "1"                  "2"                  "3" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt;                  "4"                  "5"                  "6" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt;                  "7"                  "8"                  "9" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;                 "10"                  "1"                  "2" 
#&gt; 
#&gt; $exonId
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt; "ENSMUSE00000742021" "ENSMUSE00000449901" "ENSMUSE00000199425" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt; "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt; "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000498616" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt; "ENSMUSE00000489385" "ENSMUSE00000449901" "ENSMUSE00000199425" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt; "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt; "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000496705" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" 
#&gt; 
#&gt; $origExonId
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt; "ENSMUSE00000742021" "ENSMUSE00000449901" "ENSMUSE00000199425" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt; "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt; "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000498616" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt; "ENSMUSE00000489385" "ENSMUSE00000449901" "ENSMUSE00000199425" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt; "ENSMUSE00000511070" "ENSMUSE00000514929" "ENSMUSE00000199423" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt; "ENSMUSE00000535814" "ENSMUSE00000507557" "ENSMUSE00000496705" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" 
#&gt; 
#&gt; $col
#&gt;   ENSMUSE00000742021   ENSMUSE00000449901   ENSMUSE00000199425 
#&gt;        "transparent"        "transparent"        "transparent" 
#&gt;   ENSMUSE00000511070   ENSMUSE00000514929   ENSMUSE00000199423 
#&gt;        "transparent"        "transparent"        "transparent" 
#&gt;   ENSMUSE00000535814   ENSMUSE00000507557   ENSMUSE00000498616 
#&gt;        "transparent"        "transparent"        "transparent" 
#&gt;   ENSMUSE00000489385 ENSMUSE00000449901.1 ENSMUSE00000199425.1 
#&gt;        "transparent"        "transparent"        "transparent" 
#&gt; ENSMUSE00000511070.1 ENSMUSE00000514929.1 ENSMUSE00000199423.1 
#&gt;        "transparent"        "transparent"        "transparent" 
#&gt; ENSMUSE00000535814.1 ENSMUSE00000507557.1   ENSMUSE00000496705 
#&gt;        "transparent"        "transparent"        "transparent" 
#&gt;   ENSMUSE00000750625   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;        "transparent"        "transparent"        "transparent" 
#&gt; 
#&gt; $title
#&gt;  [1] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt;  [8] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt; [15] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt; </div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Florian Hahne, Steffen Durinck, Robert Ivanek, Arne Mueller, Steve Lianoglou, Ge Tan, Lance Parsons, Shraddha Pai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


