<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>DataTrack class and methods — DataTrack-class • Gviz</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="DataTrack class and methods — DataTrack-class"><meta property="og:description" content="

A class to store numeric data values along genomic
  coordinates. Multiple samples as well as sample groupings are
  supported, with the restriction of equal genomic coordinates for a
  single observation across samples."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Gviz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.39.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/Gviz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ivanek/Gviz/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>DataTrack class and methods</h1>
    
    <div class="hidden name"><code>DataTrack-class.Rd</code></div>
    </div>

    <div class="ref-description">
    

<p>A class to store numeric data values along genomic
  coordinates. Multiple samples as well as sample groupings are
  supported, with the restriction of equal genomic coordinates for a
  single observation across samples.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">DataTrack</span><span class="op">(</span>range<span class="op">=</span><span class="cn">NULL</span>, start<span class="op">=</span><span class="cn">NULL</span>, end<span class="op">=</span><span class="cn">NULL</span>, width<span class="op">=</span><span class="cn">NULL</span>, <span class="va">data</span>, <span class="va">chromosome</span>, <span class="va">strand</span>, <span class="va">genome</span>,
name<span class="op">=</span><span class="st">"DataTrack"</span>, <span class="va">importFunction</span>, stream<span class="op">=</span><span class="cn">FALSE</span>, <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="objects-from-the-class">
    <h2>Objects from the class</h2>
    

<p>Objects can be created using the constructor function
  <code>DataTrack</code>.</p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <p>We tried to keep instantiation of <code>DataTrack</code> objects as flexible
  as possible to accomodate different use cases. For instance, one
  natural way to create a <code>DataTrack</code> is from an existing
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code> object. In other cases it might be more
  appropriate to build the object using individual function arguments.</p>
<dl><dt>range</dt>
<dd>

<p>An optional meta argument to handle the different input types. If
    the <code>range</code> argument is missing, all the relevant information
    to create the object has to be provided as individual function
    arguments (see below).</p>
<p>The different input options for <code>range</code> are:</p>
<p></p><dl><dt></dt>
<dd><p>A <code>GRanges</code> object: essentially all the necessary
    information to create a <code>DataTrack</code> can be contained in a
    single <code>GRanges</code> object. The track's coordinates are taken
    from the <code>start</code>, <code>end</code> and <code>seqnames</code> slots, the
    genome information from the genome slot, and the numeric data
    values can be extracted from additional metadata columns
    columns (please note that non-numeric columns are being ignored
    with a warning). As a matter of fact, calling the constructor on
    a <code>GRanges</code> object without further arguments, e.g.
    <code>DataTrack(range=obj)</code> is equivalent to calling the coerce
    method <code>as(obj, "DataTrack")</code>. Alternatively, the
    <code>GRanges</code> object may only contain the coordinate
    information, in which case the numeric data part is expected to
    be present in the separate <code>data</code> argument, and the ranges
    have to match the dimensions of the data matrix. If <code>data</code>
    is not <code>NULL</code>, this will always take precedence over
    anything defined in the <code>range</code> argument. See below for
    details.</p></dd>


      <dt></dt>
<dd><p>An <code><a href="https://rdrr.io/pkg/IRanges/man/IRanges-class.html" class="external-link">IRanges</a></code> object: this is very
    similar to the above case, except that the numeric data part now
    always has to be provided in the separate <code>data</code>
    argument. Also the chromosome information must be provided in
    the <code>chromosome</code> argument, because neither of the two can
    be directly encoded in an <code>IRange</code> object.</p></dd>


      <dt></dt>
<dd><p>A <code>data.frame</code> object: the <code>data.frame</code> needs to
    contain at least the two mandatory columns <code>start</code> and
    <code>end</code> with the range coordinates. It may also contain a
    <code>chromosome</code> column with the chromosome information for
    each range. If missing it will be drawn from the separate
    <code>chromosome</code> argument. All additional numeric columns will
    be interpreted as data columns, unless the <code>data</code> argument
    is explicitely provided.</p></dd>


       <dt></dt>
<dd><p>A <code>character</code> scalar: in this case the value of the
    <code>range</code> argument is considered to be a file path to an
    annotation file on disk. A range of file types are supported by
    the <code>Gviz</code> package as identified by the file extension. See
    the <code>importFunction</code> documentation below for further
    details.</p></dd>


    
</dl><p></p></dd>
<dt>start, end, width</dt>
<dd><p>Integer vectors, giving the start and the end
    end coordinates for the individual track items, or their width. Two
    of the three need to be specified, and have to be of equal length or
    of length one, in which case the single value will be recycled
    accordingly. Otherwise, the usual R recycling rules for vectors do
    not apply and the function will cast an error.</p></dd>
<dt>data</dt>
<dd><p>A numeric matrix of data points with the number of columns
    equal to the number of coordinates in <code>range</code>, or a numeric
    vector of appropriate length that will be coerced into such a
    one-row matrix. Each individual row is supposed to contain data for
    a given sample, where the coordinates for each single observation
    are constant across samples. Depending on the plotting type of the
    data (see 'Details' and 'Display Parameters' sections), sample
    grouping or data aggregation may be available. Alternatively, this
    can be a character vector of column names that point into the
    element metadata of the <code>range</code> object for
    subsetting. Naturally, this is only supported when the <code>range</code>
    argument is of class <code>GRanges</code>.</p></dd>
<dt>strand</dt>
<dd><p>Character vector, the strand information for the
    individual track items. Currently this has to be unique for the
    whole track and doesn't really have any visible consequences, but we
    might decide to make <code>DataTracks</code> strand-specific at a later
    stage.</p></dd>
<dt>chromosome</dt>
<dd><p>The chromosome on which the track's genomic ranges
    are defined. A valid UCSC chromosome identifier if
    <code>options(ucscChromosomeNames=TRUE)</code>. Please note that in this
    case only syntactic checking takes place, i.e., the argument value
    needs to be an integer, numeric character or a character of the form
    <code>chrx</code>, where <code>x</code> may be any possible string. The user has
    to make sure that the respective chromosome is indeed defined for
    the the track's genome. If not provided here, the constructor will
    try to construct the chromosome information based on the available
    inputs, and as a last resort will fall back to the value
    <code>chrNA</code>. Please note that by definition all objects in the
    <code>Gviz</code> package can only have a single active chromosome at a
    time (although internally the information for more than one
    chromosome may be present), and the user has to call the
    <code>chromosome&lt;-</code> replacement method in order to change to a
    different active chromosome.</p></dd>
<dt>genome</dt>
<dd><p>The genome on which the track's ranges are
    defined. Usually this is a valid UCSC genome identifier, however
    this is not being formally checked at this point. If not provided
    here the constructor will try to extract this information from the
    provided input, and eventually will fall back to the default value
    of <code>NA</code>.</p></dd>
<dt>name</dt>
<dd><p>Character scalar of the track's name used in the title
    panel when plotting.</p></dd>
<dt>importFunction</dt>
<dd><p>A user-defined function to be used to import the
    data from a file. This only applies when the <code>range</code> argument
    is a character string with the path to the input data file. The
    function needs to accept an argument <code>file</code> containing the file
    path and has to return a proper <code>GRanges</code> object with the data
    part attached as numeric metadata columns. Essentially
    the process is equivalent to constructing a <code>DataTrack</code>
    directly from a <code>GRanges</code> object in that non-numeric columns
    will be dropped, and further subsetting can be archived by means of
    the <code>data</code> argument. A set of default import functions is
    already implemented in the package for a number of different file
    types, and one of these defaults will be picked automatically based
    on the extension of the input file name. If the extension can not be
    mapped to any of the existing import function, an error is raised
    asking for a user-defined import function. Currently the following
    file types can be imported with the default functions: <code>wig</code>,
    <code>bigWig/bw</code>, <code>bedGraph</code> and <code>bam</code>.</p>
<p>Some file types support indexing by genomic coordinates (e.g.,
    <code>bigWig</code> and <code>bam</code>), and it makes sense to only load the
    part of the file that is needed for plotting. To this end, the
    <code>Gviz</code> package defines the derived <code>ReferenceDataTrack</code>
    class, which supports streaming data from the file system. The user
    typically does not have to deal with this distinction but may rely
    on the constructor function to make the right choice as long as the
    default import functions are used. However, once a user-defined import
    function has been provided and if this function adds support for
    indexed files, you will have to make the constructor aware of this
    fact by setting the <code>stream</code> argument to <code>TRUE</code>. Please
    note that in this case the import function needs to accept a second
    mandatory argument <code>selection</code> which is a <code>GRanges</code> object
    containing the dimensions of the plotted genomic range. As before,
    the function has to return an appropriate <code>GRanges</code> object.</p></dd>
<dt>stream</dt>
<dd><p>A logical flag indicating that the user-provided import
    function can deal with indexed files and knows how to process the
    additional <code>selection</code> argument when accessing the data on
    disk. This causes the constructor to return a
    <code>ReferenceDataTrack</code> object which will grab the necessary data
    on the fly during each plotting operation.</p></dd>
<dt>...</dt>
<dd><p>Additional items which will all be interpreted as further
    display parameters.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>The return value of the constructor function is a new object of class
<code>DataTrack</code> or <code>ReferenceDataTrack</code>.</p>
    </div>
    <div id="slots">
    <h2>Slots</h2>
    

<p></p><dl><dt><code>data</code>:</dt>
<dd><p>Object of class <code>"matrix"</code>, containing the
      data values to be plotted. Individual rows of the matrix
      correspond to individual samples, and the number of columns has to
      be identical to the feature number of the <code>GRanges</code> object in
      the <code>range</code> slot.</p></dd>


    <dt><code>strand</code>:</dt>
<dd><p>Object of class <code>"character"</code>, the strand
      information for the track, in the form '+' for the Watson strand,
      '-' for the Crick strand or '*' for either of the two.</p></dd>


    <dt><code>range</code>:</dt>
<dd><p>Object of class <code><a href="https://rdrr.io/pkg/IRanges/man/IRanges-class.html" class="external-link">IRanges</a></code>,
      inherited from class <code><a href="RangeTrack-class.html">RangeTrack</a></code>. The genomic
      coordinates for the data values. The length of the object needs to
      be identical to the number of columns of the data matrix in the
      <code>data</code> slot.</p></dd>


    <dt><code>chromosome</code>:</dt>
<dd><p>Object of class <code>"character"</code>,
      inherited from class <code><a href="RangeTrack-class.html">RangeTrack</a></code></p></dd>


    <dt><code>genome</code>:</dt>
<dd><p>Object of class <code>"character"</code>, inherited
      from class <code><a href="RangeTrack-class.html">RangeTrack</a></code></p></dd>


    <dt><code>dp</code>:</dt>
<dd><p>Object of class
      <code><a href="DisplayPars-class.html">DisplayPars</a></code>, inherited from class
      <code><a href="GdObject-class.html">GdObject</a></code></p></dd>


    <dt><code>name</code>:</dt>
<dd><p>Object of class <code>"character"</code>, inherited
      from class <code><a href="GdObject-class.html">GdObject</a></code></p></dd>


    <dt><code>imageMap</code>:</dt>
<dd><p>Object of class
      <code><a href="ImageMap-class.html">ImageMap</a></code>, inherited from class
      <code><a href="GdObject-class.html">GdObject</a></code></p></dd>


  
</dl></div>
    <div id="extends">
    <h2>Extends</h2>
    

<p>Class <code>"<a href="NumericTrack-class.html">NumericTrack</a>"</code>, directly.</p>
<p>Class <code>"<a href="RangeTrack-class.html">RangeTrack</a>"</code>, by class "NumericTrack", distance 2.</p>
<p>Class <code>"<a href="GdObject-class.html">GdObject</a>"</code>, by class "NumericTrack", distance 3.</p>
    </div>
    <div id="methods">
    <h2>Methods</h2>
    

<p>In the following code chunks, <code>obj</code> is considered to be an
  object of class <code>DataTrack</code>.</p>
<p><b><em>Exported in the name space:</em></b></p>
<p></p><dl><dt>[</dt>
<dd><p><code>signature(x="DataTrack")</code>: subsetting of the
      object, either to a subet of coordinates, or to a subset of
      samples.</p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>i</code>, <code>j</code>: subsetting indices for coordinates
      (<code>i</code>) or samples (<code>j</code>).</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>obj[1:3,]</code></p></dd>


    <dt></dt>
<dd><p><code>obj[,2:4]</code></p></dd>


      
</dl><p></p></dd>


    <dt>values</dt>
<dd><p><code>signature(x="DataTrack")</code>: return the raw data
      values of the object, i.e., the data matrix in the <code>data</code>
      slot.</p>
<p><em>Usage:</em></p>
<p><code>values(x)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>values(obj)</code></p></dd>


      
</dl><p></p></dd>


     <dt>values&lt;-</dt>
<dd><p><code>signature(x="DataTrack")</code>: replace the data
      matrix in the <code>data</code> slot.</p>
<p><em>Usage:</em></p>
<p><code>values&lt;-(x, value)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>value</code>: replacement value.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>values(obj) &lt;- matrix(1:10, ncol=2)</code></p></dd>


      
</dl><p></p></dd>


    <dt>score</dt>
<dd><p><code>signature(x="DataTrack")</code>: return processed data
      values of the object exactly like they would be plotted to the
      device (modulo any potential aggregration or collapsing), i.e.,
      the raw data with optional transformations applied.</p>
<p><em>Usage:</em></p>
<p><code>score(x, from=NULL, to=NULL, sort=FALSE, transformation=TRUE)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>from</code>, <code>to</code>: restrict to data
      within a certain coordinates range.</p></dd>


    <dt></dt>
<dd><p><code>sort</code>: sort the return values by
      coordinates. This is usually not necessary since the data
      should already be ordererd, however this is not formaly
      checked anywhere and some operations strictly depend on
      ordered data.</p></dd>


    <dt></dt>
<dd><p><code>transformation</code>: apply a data transformation
      in case one is defined as the <code>transformation</code> display
      parameter.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>score(obj)</code></p></dd>


    <dt></dt>
<dd><p><code>score(obj, from=100, to=10000)</code></p></dd>


    <dt></dt>
<dd><p><code>score(obj, sort=TRUE, transformation=FALSE)</code></p></dd>


      
</dl><p></p></dd>


    <dt>split</dt>
<dd><p><code>signature(x="DataTrack")</code>: split a
      <code>DataTrack</code> object by an appropriate factor vector (or
      another vector that can be coerced into one). The output of this
      operation is a list of <code>DataTrack</code> objects.</p>
<p><em>Usage:</em></p>
<p><code>split(x, f, ...)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>f</code>: the splitting factor.</p></dd>


    <dt></dt>
<dd><p><code>...</code>: all further arguments are ignored.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>split(obj, c("a", "a", "b", "c", "a"))</code></p></dd>


      
</dl><p></p></dd>


    <dt>range, ranges</dt>
<dd><p><code>signature(x="DataTrack")</code>: return the genomic
      coordinates for the track as an object of class
      <code><a href="https://rdrr.io/pkg/IRanges/man/IRanges-class.html" class="external-link">IRanges</a></code>.</p>
<p><em>Usage:</em></p>
<p><code>range(x)</code></p>
<p><code>ranges(x)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>range(obj)</code></p></dd>


    <dt></dt>
<dd><p><code>ranges(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>strand</dt>
<dd><p><code>signature(x="DataTrack")</code>: return a vector of
      strand specifiers for all track items, in the form '+' for the
      Watson strand, '-' for the Crick strand or '*' for either of the
      two.</p>
<p><em>Usage:</em></p>
<p><code>strand(x)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>strand(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>strand&lt;-</dt>
<dd><p><code>signature(x="DataTrack")</code>: replace the
      strand information for the track items. The replacement value
      needs to be an appropriate scalar or vector of strand values.</p>
<p><em>Usage:</em></p>
<p><code>strand&lt;-(x, value)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>value</code>: replacement value.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>strand(obj) &lt;- "+"</code></p></dd>


      
</dl><p></p></dd>


    <dt>feature</dt>
<dd><p><code>signature(GdObject="DataTrack")</code>: returns NULL
      since there is no grouping information for the ranges in a
      <code>DataTrack</code>.</p>
<p><em>Usage:</em></p>
<p><code>feature(GdObject)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>feature(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>feature&lt;-</dt>
<dd><p><code>signature(gdObject="DataTrack",
      value="character")</code>: this return the unaltered input object since
      there  is no grouping information for the ranges in a
      <code>DataTrack</code>.</p>
<p><em>Usage:</em></p>
<p><code>feature&lt;-(GdObject, value)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>value</code>: replacement value.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>feature(obj) &lt;- c("a", "a", "b", "c", "a")</code></p></dd>


      
</dl><p></p></dd>

  
</dl><p><b><em>Internal methods:</em></b></p>
<p></p><dl><dt>collapseTrack</dt>
<dd><p><code>signature(gdObject="DataTrack")</code>:
      preprocess the track before plotting. This will collapse
      overlapping track items based on the available resolution and
      increase the width and height of all track objects to a minimum
      value to avoid rendering issues. See <code><a href="collapsing.html">collapsing</a></code> for
      details.</p>
<p><em>Usage:</em></p>
<p><code>collapseTrack(GdObject, diff=.pxResolution(coord="x"))</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>diff</code>: the minimum pixel width to display,
      everything below that will be inflated to a width of
      <code>diff</code>.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>Gviz:::collapseTrack(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>drawGD</dt>
<dd><p><code>signature(GdObject="DataTrack")</code>: plot the
      object to a graphics device. The return value of this method is
      the input object, potentially updated during the plotting
      operation. Internally, there are two modes in which the method can
      be called. Either in 'prepare' mode, in which case no plotting is
      done but the object is preprocessed based on the
      available space, or in 'plotting' mode, in which case the actual
      graphical output is created. Since subsetting of the object can be
      potentially costly, this can be switched off in case subsetting
      has already been performed before or is not necessary.</p>
<p><em>Usage:</em></p>
<p><code>drawGD(GdObject, minBase, maxBase, prepare=FALSE,
    subset=TRUE, ...)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>minBase</code>, <code>maxBase</code>: the coordinate range to
      plot.</p></dd>


    <dt></dt>
<dd><p><code>prepare</code>: run method in preparation or in
      production mode.</p></dd>


    <dt></dt>
<dd><p><code>subset</code>: subset the object to the visible region
    or skip the potentially expensive subsetting operation.</p></dd>


    <dt></dt>
<dd><p><code>...</code>: all further arguments are ignored.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>Gviz:::drawGD(obj)</code></p></dd>


    <dt></dt>
<dd><p><code>Gviz:::drawGD(obj, minBase=1, maxBase=100)</code></p></dd>


    <dt></dt>
<dd><p><code>Gviz:::drawGD(obj, prepare=TRUE,
        subset=FALSE)</code></p></dd>


      
</dl><p></p></dd>


    <dt>drawAxis</dt>
<dd><p><code>signature(GdObject="DataTrack")</code>: add a y-axis
      to the title panel of a track.</p>
<p><em>Usage:</em></p>
<p><code>drawAxis(GdObject, from, to, ...)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>from</code>, <code>to</code>: compute axis range from the data
      within a certain coordinates range only.</p></dd>


    <dt></dt>
<dd><p><code>...</code>: all further arguments are ignored.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>Gviz:::drawAxis(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>initialize</dt>
<dd><p><code>signature(.Object="DataTrack")</code>:
      initialize the object</p></dd>


    <dt>show</dt>
<dd><p><code>signature(object="DataTrack")</code>: show a
      human-readable summary of the object</p></dd>


  
</dl><p><b><em>Inherited methods:</em></b></p>
<p></p><dl><dt>drawGrid</dt>
<dd><p><code>signature(GdObject="DataTrack")</code>:
      superpose a grid on top of a track.</p>
<p><em>Usage:</em></p>
<p><code>drawGrid(GdObject, from, to, ...)</code></p>

<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>from</code>, <code>to</code>: integer scalars, restrict to
      coordinate range before computing the grid lines.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>Gviz:::drawGrid(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>chromosome</dt>
<dd><p><code>signature(GdObject="DataTrack")</code>: return the
      currently active chromosome for which the track is defined. For
      consistancy with other Bioconductor packages, the
      <code>isActiveSeq</code> alias is also provided.</p>
<p><em>Usage:</em></p>
<p><code>chromosome(GdObject)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>chromosome(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>chromosome&lt;-</dt>
<dd><p><code>signature(GdObject="DataTrack")</code>: replace
      the value of the track's active chromosome. This has to be a valid
      UCSC chromosome identifier or an integer or character scalar that
      can be reasonably coerced into one, unless
      <code>options(ucscChromosomeNames=FALSE)</code>. For consistancy with
      other Bioconductor packages, the <code>isActiveSeq&lt;-</code> alias is
      also provided.</p>
<p><em>Usage:</em></p>
<p><code>chromosome&lt;-(GdObject, value)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>value</code>: replacement value.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>chromosome(obj) &lt;- "chr12"</code></p></dd>


      
</dl><p></p></dd>


    <dt>start, end, width</dt>
<dd><p><code>signature(x="DataTrack")</code>: the
      start or end coordinates of the track items, or their width in
      genomic coordinates.</p>
<p><em>Usage:</em></p>
<p><code>start(x)</code></p>
<p><code>end(x)</code></p>
<p><code>width(x)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>start(obj)</code></p></dd>


    <dt></dt>
<dd><p><code>end(obj)</code></p></dd>


    <dt></dt>
<dd><p><code>width(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>start&lt;-, end&lt;-, width&lt;-</dt>
<dd><p><code>signature(x="DataTrack")</code>:
      replace the start or end coordinates of the track items, or their
      width.</p>
<p><em>Usage:</em></p>
<p><code>start&lt;-(x, value)</code></p>
<p><code>end&lt;-(x, value)</code></p>
<p><code>width&lt;-(x, value)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>value</code>: replacement value.</p></dd>


    
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>start(obj) &lt;- 1:10</code></p></dd>


    <dt></dt>
<dd><p><code>end(obj) &lt;- 20:30</code></p></dd>


    <dt></dt>
<dd><p><code>width(obj) &lt;- 1</code></p></dd>


      
</dl><p></p></dd>


    <dt>position</dt>
<dd><p><code>signature(GdObject="DataTrack")</code>: the
      arithmetic mean of the track item's coordionates, i.e.,
      <code>(end(obj)-start(obj))/2</code>.</p>
<p><em>Usage:</em></p>
<p><code>position(GdObject)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>position(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>genome</dt>
<dd><p><code>signature(x="DataTrack")</code>: return the track's genome.</p>
<p><em>Usage:</em></p>
<p><code>genome(x)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>genome(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>genome&lt;-</dt>
<dd><p><code>signature(x="DataTrack")</code>: set the track's
      genome. Usually this has to be a valid UCSC identifier, however
      this is not formally enforced here.</p>
<p><em>Usage:</em></p>
<p><code>genome&lt;-(x, value)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>value</code>: replacement value.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>genome(obj) &lt;- "mm9"</code></p></dd>


      
</dl><p></p></dd>


    <dt>length</dt>
<dd><p><code>signature(x="DataTrack")</code>: return the number
      of items in the track.</p>
<p><em>Usage:</em></p>
<p><code>length(x)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>length(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>coerce</dt>
<dd><p><code>signature(from="DataTrack",
    to="data.frame")</code>: coerce the <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code>
    object in the <code>range</code> slot into a regular data.frame.</p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>as(obj, "data.frame")</code></p></dd>


      
</dl><p></p></dd>


    <dt>subset</dt>
<dd><p><code>signature(x="DataTrack")</code>: subset a
      <code>NumericTrack</code> by coordinates and sort if necessary.</p>
<p><em>Usage:</em></p>
<p><code>subset(x, from, to, sort=FALSE, drop=TRUE, ...)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>from</code>, <code>to</code>: the coordinates range to subset
      to.</p></dd>


    <dt></dt>
<dd><p><code>sort</code>: sort the object after subsetting. Usually
      not necessary.</p></dd>


    <dt></dt>
<dd><p><code>drop</code>: drop unused regions on the other,
      non-active chromosomes.w</p></dd>


    <dt></dt>
<dd><p><code>...</code>: additional arguments are ignored.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>subset(obj, from=10, to=20, sort=TRUE)</code></p></dd>


      
</dl><p></p></dd>


    <dt>displayPars</dt>
<dd><p><code>signature(x="DataTrack",
      name="character")</code>: list the value of the display parameter
      <code>name</code>. See <code><a href="settings.html">settings</a></code> for details on display
      parameters and customization.</p>
<p><em>Usage:</em></p>
<p><code>displayPars(x, name)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>displayPars(obj, "col")</code></p></dd>


      
</dl><p></p></dd>



    <dt>displayPars</dt>
<dd><p><code>signature(x="DataTrack",
      name="missing")</code>: list the value of all available display
      parameters. See <code><a href="settings.html">settings</a></code> for details on display
      parameters and customization.</p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>displayPars(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>getPar</dt>
<dd><p><code>signature(x="DataTrack", name="character")</code>:
      alias for the <code>displayPars</code> method. See
      <code><a href="settings.html">settings</a></code> for details on display parameters and
      customization.</p>
<p><em>Usage:</em></p>
<p><code>getPar(x, name)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>getPar(obj, "col")</code></p></dd>


      
</dl><p></p></dd>


    <dt>getPar</dt>
<dd><p><code>signature(x="DataTrack", name="missing")</code>:
      alias for the <code>displayPars</code> method. See
      <code><a href="settings.html">settings</a></code> for details on display parameters and
      customization.</p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>getPar(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>displayPars&lt;-</dt>
<dd><p><code>signature(x="DataTrack",
      value="list")</code>: set display parameters using the values of the
      named list in <code>value</code>. See <code><a href="settings.html">settings</a></code> for details
      on display parameters and customization.</p>
<p><em>Usage:</em></p>
<p><code>displayPars&lt;-(x, value)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>displayPars(obj) &lt;- list(col="red", lwd=2)</code></p></dd>


      
</dl><p></p></dd>


    <dt>setPar</dt>
<dd><p><code>signature(x="DataTrack", value="character")</code>:
      set the single display parameter <code>name</code> to <code>value</code>. Note
      that display parameters in the <code>DataTrack</code> class are
      pass-by-reference, so no re-assignmnet to the symbol <code>obj</code> is
      necessary. See <code><a href="settings.html">settings</a></code> for details on display
      parameters and customization.</p>
<p><em>Usage:</em></p>
<p><code>setPar(x, name, value)</code></p>
<p><em>Additional Arguments:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>name</code>: the name of the display parameter to set.</p></dd>


      
</dl><p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>setPar(obj, "col", "red")</code></p></dd>


      
</dl><p></p></dd>


    <dt>setPar</dt>
<dd><p><code>signature(x="DataTrack", value="list")</code>: set
      display parameters by the values of the named list in
      <code>value</code>.  Note that display parameters in the
      <code>DataTrack</code> class are pass-by-reference, so no
      re-assignmnet to the symbol <code>obj</code> is necessary. See
      <code><a href="settings.html">settings</a></code> for details on display parameters and
      customization.</p>

<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>setPar(obj, list(col="red", lwd=2))</code></p></dd>


      
</dl><p></p></dd>


    <dt>group</dt>
<dd><p><code>signature(GdObject="DataTrack")</code>: return
      grouping information for the individual items in the track. Unless
      overwritten in one of the sub-classes, this usualy returns
      <code>NULL</code>.</p>
<p><em>Usage:</em></p>
<p><code>group(GdObject)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>group(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>names</dt>
<dd><p><code>signature(x="DataTrack")</code>: return the value of
      the <code>name</code> slot.</p>
<p><em>Usage:</em></p>
<p><code>names(x)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>names(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>names&lt;-</dt>
<dd><p><code>signature(x="DataTrack", value="character")</code>:
      set the value of the <code>name</code> slot.</p>
<p><em>Usage:</em></p>
<p><code>names&lt;-(x, value)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>names(obj) &lt;- "foo"</code></p></dd>


      
</dl><p></p></dd>


    <dt>coords</dt>
<dd><p><code>signature(ImageMap="DataTrack")</code>: return the
      coordinates from the internal image map.</p>
<p><em>Usage:</em></p>
<p><code>coords(ImageMap)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>coords(obj)</code></p></dd>


      
</dl><p></p></dd>


    <dt>tags</dt>
<dd><p><code>signature(x="DataTrack")</code>: return the tags from the
      internal image map.</p>
<p><em>Usage:</em></p>
<p><code>tags(x)</code></p>
<p><em>Examples:</em></p>
<p></p><dl><dt></dt>
<dd><p><code>tags(obj)</code></p></dd>


      
</dl><p></p></dd>

  
</dl></div>
    <div id="display-parameters">
    <h2>Display Parameters</h2>
     

<p>The following display parameters are set for objects of class
  <code>DataTrack</code> upon instantiation, unless one or more of them have
  already been set by one of the optional sub-class initializers, which
  always get precedence over these global defaults. See <code><a href="settings.html">settings</a></code>
  for details on setting graphical parameters for tracks.</p><dl><dt></dt>
<dd><p><code>aggregateGroups=FALSE</code>: Logical scalar. Aggregate
      the values within a sample group using the aggregation funnction
      specified in the <code>aggregation</code> parameter.</p></dd>
 

    <dt></dt>
<dd><p><code>aggregation="mean"</code>: Function or character scalar.
      Used to aggregate values in windows or for collapsing overlapping
      items. The function has to accept a numeric vector as a single
      input parameter and has to return a numeric scalar with the
      aggregated value. Alternatively, one of the predefined options
      <code>mean</code>, <code>median</code> <code>sum</code>, <code>min</code>, <code>max</code> or
      <code>extreme</code> can be supplied as a character scalar. Defaults
      to <code>mean</code>.</p></dd>
 

    <dt></dt>
<dd><p><code>alpha.confint=0.3</code>: Numeric scalar. The transparency
      for the confidence intervalls in confint-type plots.</p></dd>
 

    <dt></dt>
<dd><p><code>amount=NULL</code>: Numeric scalar. Amount of jittering in
      xy-type plots. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.xyplot.html" class="external-link">panel.xyplot</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>baseline=NULL</code>: Numeric scalar. Y-axis position of
      an optional baseline. This parameter has a special meaning for
      mountain-type and polygon-type plots, see the 'Details' section
      in <code>DataTrack</code> for more information.</p></dd>
 

    <dt></dt>
<dd><p><code>box.legend=FALSE</code>: Logical scalar. Draw a box around
      a legend.</p></dd>
 

    <dt></dt>
<dd><p><code>box.ratio=1</code>: Numeric scalar. Parameter controlling
      the boxplot appearance. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.bwplot.html" class="external-link">panel.bwplot</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>box.width=NULL</code>: Numeric scalar. Parameter controlling
      the boxplot appearance. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.bwplot.html" class="external-link">panel.bwplot</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>cex=0.7</code>: Numeric scalar. The default pixel size for
      plotting symbols.</p></dd>
 

    <dt></dt>
<dd><p><code>cex.legend=0.8</code>: Numeric scalar. The size factor for
      the legend text.</p></dd>
 

    <dt></dt>
<dd><p><code>cex.sampleNames=NULL</code>: Numeric scalar. The size factor
      for the sample names text in heatmap or horizon plots. Defaults
      to an automatic setting.</p></dd>
 

    <dt></dt>
<dd><p><code>coef=1.5</code>: Numeric scalar. Parameter controlling the
      boxplot appearance. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.bwplot.html" class="external-link">panel.bwplot</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>col=c("#0080ff", "#ff00ff", "darkgreen", "#ff0000",
      "orange", "#00ff00", "brown")</code>: Character or integer vector. The
      color used for all line and symbol elements, unless there is a
      more specific control defined elsewhere. Unless <code>groups</code>
      are specified, only the first color in the vector is usually
      regarded.</p></dd>
 

    <dt></dt>
<dd><p><code>col.baseline=NULL</code>: Character scalar. Color for the
      optional baseline, defaults to the setting of <code>col</code>.</p></dd>
 

    <dt></dt>
<dd><p><code>col.boxplotFrame="#808080"</code>: Character scalar. Line
      color of the frame around grouped boxplots.</p></dd>
 

    <dt></dt>
<dd><p><code>col.confint</code>: Character vector. Border colors for
      the confidence intervals for confint-type plots.</p></dd>
 

    <dt></dt>
<dd><p><code>col.histogram="#808080"</code>: Character scalar. Line
      color in histogram-type plots.</p></dd>
 

    <dt></dt>
<dd><p><code>col.horizon</code>: The line color for the segments in the
      <code>horizon</code>-type plot. See <code>horizonplot</code> for
      details.</p></dd>
 

    <dt></dt>
<dd><p><code>col.mountain=NULL</code>: Character scalar. Line color in
      mountain-type and polygon-type plots, defaults to the setting of
      <code>col</code>.</p></dd>
 

    <dt></dt>
<dd><p><code>col.sampleNames="white"</code>: Character or integer scalar.
      The color used for the sample names in heatmap plots.</p></dd>
 

    <dt></dt>
<dd><p><code>collapse=FALSE</code>: Logical scalar. Collapse overlapping
      ranges and aggregate the underlying data.</p></dd>
 

    <dt></dt>
<dd><p><code>degree=1</code>: Numeric scalar. Parameter controlling the
      loess calculation for smooth and mountain-type plots. See
      <code><a href="https://rdrr.io/pkg/lattice/man/panel.loess.html" class="external-link">panel.loess</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>do.out=TRUE</code>: Logical scalar. Parameter controlling
      the boxplot appearance. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.bwplot.html" class="external-link">panel.bwplot</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>evaluation=50</code>: Numeric scalar. Parameter controlling
      the loess calculation for smooth and mountain-type plots. See
      <code><a href="https://rdrr.io/pkg/lattice/man/panel.loess.html" class="external-link">panel.loess</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>factor=0.5</code>: Numeric scalar. Factor to control amount
      of jittering in xy-type plots. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.xyplot.html" class="external-link">panel.xyplot</a></code>
      for details.</p></dd>
 

    <dt></dt>
<dd><p><code>family="symmetric"</code>: Character scalar. Parameter
      controlling the loess calculation for smooth and mountain-type
      plots. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.loess.html" class="external-link">panel.loess</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>fill.confint=NULL</code>: Character vector. Fill colors
      for the confidence intervals for confint-type plots.</p></dd>
 

    <dt></dt>
<dd><p><code>fill.histogram=NULL</code>: Character scalar. Fill color
      in histogram-type plots, defaults to the setting of <code>fill</code>.</p></dd>
 

    <dt></dt>
<dd><p><code>fill.horizon=c("#B41414", "#E03231", "#F7A99C",
      "#9FC8DC", "#468CC8", "#0165B3")</code>: The fill colors for the segments
      in the <code>horizon</code>-type plot. This should be a vector of length
      six, where the first three entries are the colors for positive
      changes, and the latter three entries are the colors for negative
      changes. Defaults to a red-blue color scheme. See <code>horizonplot</code>
      for details.</p></dd>
 

    <dt></dt>
<dd><p><code>fill.mountain=c("#CCFFFF", "#FFCCFF")</code>: Character
      vector of length 2. Fill color in mountain-type and polygon-type
      plots.</p></dd>
 

    <dt></dt>
<dd><p><code>fontcolor.legend="#808080"</code>: Integer or character
      scalar. The font color for the legend text.</p></dd>
 

    <dt></dt>
<dd><p><code>fontface.legend=NULL</code>: Integer or character scalar.
      The font face for the legend text.</p></dd>
 

    <dt></dt>
<dd><p><code>fontfamily.legend=NULL</code>: Integer or character scalar.
      The font family for the legend text.</p></dd>
 

    <dt></dt>
<dd><p><code>fontsize.legend=NULL</code>: Numeric scalar. The pixel size
      for the legend text.</p></dd>
 

    <dt></dt>
<dd><p><code>gradient=c("#F7FBFF", "#DEEBF7", "#C6DBEF", "#9ECAE1",
      "#6BAED6", "#4292C6", "#2171B5", "#08519C", "#08306B")</code>: Character
      vector. The base colors for the <code>gradient</code> plotting type or
      the <code>heatmap</code> type with a single group. When plotting heatmaps
      with more than one group, the <code>col</code> parameter can be used
      to control the group color scheme, however the gradient will
      always be from white to 'col' and thus does not offer as much
      flexibility as this <code>gradient</code> parameter.</p></dd>
 

    <dt></dt>
<dd><p><code>grid=FALSE</code>: Logical vector. Draw a line grid under
      the track content.</p></dd>
 

    <dt></dt>
<dd><p><code>groups=NULL</code>: Vector coercable to a factor. Optional
      sample grouping. See 'Details' section in <code>DataTrack</code>
      for further information.</p></dd>
 

    <dt></dt>
<dd><p><code>horizon.origin=0</code>: The baseline relative to which
      changes are indicated on the <code>horizon</code>-type plot. See
      <code>horizonplot</code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>horizon.scale=NULL</code>: The scale for each of the segments
      in the <code>horizon</code>-type plot. Defaults to 1/3 of the absolute
      data range. See <code>horizonplot</code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>jitter.x=FALSE</code>: Logical scalar. Toggle on jittering
      on the x axis in xy-type plots. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.xyplot.html" class="external-link">panel.xyplot</a></code>
      for details.</p></dd>
 

    <dt></dt>
<dd><p><code>jitter.y=FALSE</code>: Logical scalar. Toggle off jittering
      on the y axis in xy-type plots. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.xyplot.html" class="external-link">panel.xyplot</a></code>
      for details.</p></dd>
 

    <dt></dt>
<dd><p><code>legend=TRUE</code>: Boolean triggering the addition of a
      legend to the track to indicate groups. This only has an effect
      if at least two groups are present.</p></dd>
 

    <dt></dt>
<dd><p><code>levels.fos=NULL</code>: Numeric scalar. Parameter controlling
      the boxplot appearance. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.bwplot.html" class="external-link">panel.bwplot</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>lineheight.legend=NULL</code>: Numeric scalar. The line
      height for the legend text.</p></dd>
 

    <dt></dt>
<dd><p><code>lty.baseline=NULL</code>: Character or numeric scalar. Line
      type of the optional baseline, defaults to the setting of
      <code>lty</code>.</p></dd>
 

    <dt></dt>
<dd><p><code>lty.mountain=NULL</code>: Character or numeric scalar. Line
      type in mountain-type and polygon-type plots, defaults to the
      setting of <code>lty</code>.</p></dd>
 

    <dt></dt>
<dd><p><code>lwd.baseline=NULL</code>: Numeric scalar. Line width of
      the optional baseline, defaults to the setting of <code>lwd</code>.</p></dd>
 

    <dt></dt>
<dd><p><code>lwd.mountain=NULL</code>: Numeric scalar. Line width in
      mountain-type and polygon-type plots, defaults to the setting of
      <code>lwd</code>.</p></dd>
 

    <dt></dt>
<dd><p><code>min.distance=0</code>: Numeric scalar. The mimimum distance
      in pixel below which to collapse ranges.</p></dd>
 

    <dt></dt>
<dd><p><code>missingAsZero=TRUE</code>: Logical scalar. Defines how the
      missing values are treated in the aggregation procedure with
      running window. Setting it to <code>TRUE</code> fills empty positions
      with zeros, which is default. <code>FALSE</code> fills empty positions
      with <code>NA</code>.</p></dd>
 

    <dt></dt>
<dd><p><code>na.rm=FALSE</code>: Boolean controlling whether to discard
      all NA values when plotting or to keep empty spaces for NAs</p></dd>
 

    <dt></dt>
<dd><p><code>ncolor=100</code>: Integer scalar. The number of colors
      for the 'gradient' plotting type</p></dd>
 

    <dt></dt>
<dd><p><code>notch=FALSE</code>: Logical scalar. Parameter controlling
      the boxplot appearance. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.bwplot.html" class="external-link">panel.bwplot</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>notch.frac=0.5</code>: Numeric scalar. Parameter controlling
      the boxplot appearance. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.bwplot.html" class="external-link">panel.bwplot</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>pch=20</code>: Integer scalar. The type of glyph used for
      plotting symbols.</p></dd>
 

    <dt></dt>
<dd><p><code>separator=0</code>: Numeric scalar. Number of pixels used
      to separate individual samples in heatmap- and horizon-type plots.</p></dd>
 

    <dt></dt>
<dd><p><code>showColorBar=TRUE</code>: Boolean. Indicate the data range
      color mapping in the axis for 'heatmap' or 'gradient' types.</p></dd>
 

    <dt></dt>
<dd><p><code>showSampleNames=FALSE</code>: Boolean. Display the names
      of the individual samples in a heatmap or a horizon plot.</p></dd>
 

    <dt></dt>
<dd><p><code>size=NULL</code>: Numeric scalar. The relative size of the
      track. Can be overridden in the <code><a href="plotTracks.html">plotTracks</a></code> function.
      By default the size will be set automatically based on the selected
      plotting type.</p></dd>
 

    <dt></dt>
<dd><p><code>span=0.2</code>: Numeric scalar. Parameter controlling the
      loess calculation for smooth and mountain-type plots. See
      <code><a href="https://rdrr.io/pkg/lattice/man/panel.loess.html" class="external-link">panel.loess</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>stackedBars=TRUE</code>: Logical scalar. When there are
      several data groups, draw the histogram-type plots as stacked
      barplots or grouped side by side.</p></dd>
 

    <dt></dt>
<dd><p><code>stats=function</code>: Function. Parameter controlling the
      boxplot appearance. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.bwplot.html" class="external-link">panel.bwplot</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>transformation=NULL</code>: Function. Applied to the data
      matrix prior to plotting or when calling the <code>score</code> method.
      The function should accept exactly one input argument and its
      return value needs to be a numeric vector which can be coerced
      back into a data matrix of identical dimensionality as the input
      data.</p></dd>
 

    <dt></dt>
<dd><p><code>type="p"</code>: Character vector. The plot type, one or
      several in <code>p</code>,<code>l</code>, <code>b</code>, <code>a</code>, <code>a_confint</code>,
      <code>s</code>, <code>g</code>, <code>r</code>, <code>S</code>, <code>confint</code>, <code>smooth</code>,
      <code>histogram</code>, <code>mountain</code>, <code>polygon</code>, <code>h</code>,
      <code>boxplot</code>, <code>gradient</code>, <code>heatmap</code>, <code>horizon</code>.
      See 'Details' section in <code>DataTrack</code> for more
      information on the individual plotting types.</p></dd>
 

    <dt></dt>
<dd><p><code>varwidth=FALSE</code>: Logical scalar. Parameter controlling
      the boxplot appearance. See <code><a href="https://rdrr.io/pkg/lattice/man/panel.bwplot.html" class="external-link">panel.bwplot</a></code> for details.</p></dd>
 

    <dt></dt>
<dd><p><code>window=NULL</code>: Numeric or character scalar. Aggregate
      the rows values of the data matrix to <code>window</code> equally sized
      slices on the data range using the method defined in <code>aggregation</code>.
      If negative, apply a running window of size <code>windowSize</code>
      using the same aggregation method. Alternatively, the special
      value <code>auto</code> causes the function to determine the optimal
      window size to avoid overplotting, and <code>fixed</code> uses fixed-size
      windows of size <code>windowSize</code>.</p></dd>
 

    <dt></dt>
<dd><p><code>windowSize=NULL</code>: Numeric scalar. The size of the
      running window when the value of <code>window</code> is negative.</p></dd>
 

    <dt></dt>
<dd><p><code>ylim=NULL</code>: Numeric vector of length 2. The range of
      the y-axis scale.</p></dd>
 

    <dt></dt>
<dd><p><code>yTicksAt=NULL</code>: Numeric vector. The points at which
      y-axis tick-marks are to be drawn. By default, when <code>NULL</code>,
      tickmark locations are computed.</p></dd>
 

  
</dl><p>Additional display parameters are being inherited from the respective
  parent classes. Note that not all of them may have an effect on the
  plotting of <code>DataTrack</code> objects.</p>
<p></p><dl><dt></dt>
<dd><p><code><a href="GdObject-class.html">GdObject</a></code>:</p>
<p></p><dl><dt></dt>
<dd><p><code>alpha=1</code>: Numeric scalar. The transparency for
          all track items.</p></dd>
 

        <dt></dt>
<dd><p><code>alpha.title=NULL</code>: Numeric scalar. The transparency
          for the title panel.</p></dd>
 

        <dt></dt>
<dd><p><code>background.legend="transparent"</code>: Integer or
          character scalar. The background color for the legend.</p></dd>
 

        <dt></dt>
<dd><p><code>background.panel="transparent"</code>: Integer or
          character scalar. The background color of the content panel.</p></dd>
 

        <dt></dt>
<dd><p><code>background.title="lightgray"</code>: Integer or character
          scalar. The background color for the title panel.</p></dd>
 

        <dt></dt>
<dd><p><code>cex.axis=NULL</code>: Numeric scalar. The expansion
          factor for the axis annotation. Defaults to <code>NULL</code>, in
          which case it is automatically determined based on the
          available space.</p></dd>
 

        <dt></dt>
<dd><p><code>cex.title=NULL</code>: Numeric scalar. The expansion
          factor for the title panel. This effects the fontsize of both
          the title and the axis, if any. Defaults to <code>NULL</code>,
          which means that the text size is automatically adjusted to
          the available space.</p></dd>
 

        <dt></dt>
<dd><p><code>col.axis="white"</code>: Integer or character scalar.
          The font and line color for the y axis, if any.</p></dd>
 

        <dt></dt>
<dd><p><code>col.border.title="white"</code>: Integer or character
          scalar. The border color for the title panels.</p></dd>
 

        <dt></dt>
<dd><p><code>col.frame="lightgray"</code>: Integer or character
          scalar. The line color used for the panel frame, if
          <code>frame==TRUE</code></p></dd>
 

        <dt></dt>
<dd><p><code>col.grid="#808080"</code>: Integer or character scalar.
          Default line color for grid lines, both when <code>type=="g"</code>
          in <code>DataTrack</code>s and when display parameter
          <code>grid==TRUE</code>.</p></dd>
 

        <dt></dt>
<dd><p><code>col.line=NULL</code>: Integer or character scalar.
          Default colors for plot lines. Usually the same as the global
          <code>col</code> parameter.</p></dd>
 

        <dt></dt>
<dd><p><code>col.symbol=NULL</code>: Integer or character scalar.
          Default colors for plot symbols. Usually the same as the
          global <code>col</code> parameter.</p></dd>
 

        <dt></dt>
<dd><p><code>col.title="white"</code> <code>(Aliases: fontcolor.title)</code>:
          Integer or character scalar. The border color for the title panels</p></dd>
 

        <dt></dt>
<dd><p><code>fill="lightgray"</code>: Integer or character scalar.
          Default fill color setting for all plotting elements, unless
          there is a more specific control defined elsewhere.</p></dd>
 

        <dt></dt>
<dd><p><code>fontcolor="black"</code>: Integer or character scalar.
          The font color for all text, unless a more specific definition
          exists.</p></dd>
 

        <dt></dt>
<dd><p><code>fontface=1</code>: Integer or character scalar. The
          font face for all text, unless a more specific definition exists.</p></dd>
 

        <dt></dt>
<dd><p><code>fontface.title=2</code>: Integer or character scalar.
          The font face for the title panels.</p></dd>
 

        <dt></dt>
<dd><p><code>fontfamily="sans"</code>: Integer or character scalar.
          The font family for all text, unless a more specific definition
          exists.</p></dd>
 

        <dt></dt>
<dd><p><code>fontfamily.title="sans"</code>: Integer or character
          scalar. The font family for the title panels.</p></dd>
 

        <dt></dt>
<dd><p><code>fontsize=12</code>: Numeric scalar. The font size for
          all text, unless a more specific definition exists.</p></dd>
 

        <dt></dt>
<dd><p><code>frame=FALSE</code>: Boolean. Draw a frame around the
          track when plotting.</p></dd>
 

        <dt></dt>
<dd><p><code>h=-1</code>: Integer scalar. Parameter controlling the
          number of horizontal grid lines, see <code><a href="https://rdrr.io/pkg/lattice/man/panel.functions.html" class="external-link">panel.grid</a></code>
          for details.</p></dd>
 

        <dt></dt>
<dd><p><code>lineheight=1</code>: Numeric scalar. The font line
          height for all text, unless a more specific definition exists.</p></dd>
 

        <dt></dt>
<dd><p><code>lty="solid"</code>: Numeric scalar. Default line type
          setting for all plotting elements, unless there is a more
          specific control defined elsewhere.</p></dd>
 

        <dt></dt>
<dd><p><code>lty.grid="solid"</code>: Integer or character scalar.
          Default line type for grid lines, both when <code>type=="g"</code>
          in <code>DataTrack</code>s and when display parameter
          <code>grid==TRUE</code>.</p></dd>
 

        <dt></dt>
<dd><p><code>lwd=1</code>: Numeric scalar. Default line width setting
          for all plotting elements, unless there is a more specific
          control defined elsewhere.</p></dd>
 

        <dt></dt>
<dd><p><code>lwd.border.title=1</code>: Integer scalar. The border
          width for the title panels.</p></dd>
 

        <dt></dt>
<dd><p><code>lwd.grid=1</code>: Numeric scalar. Default line width
          for grid lines, both when <code>type=="g"</code> in <code>DataTrack</code>s
          and when display parameter <code>grid==TRUE</code>.</p></dd>
 

        <dt></dt>
<dd><p><code>lwd.title=1</code>: Integer scalar. The border width
          for the title panels</p></dd>
 

        <dt></dt>
<dd><p><code>min.height=3</code>: Numeric scalar. The minimum range
          height in pixels to display. All ranges are expanded to this
          size in order to avoid rendering issues. See <code><a href="collapsing.html">collapsing</a></code>
          for details.</p></dd>
 

        <dt></dt>
<dd><p><code>min.width=1</code>: Numeric scalar. The minimum range
          width in pixels to display. All ranges are expanded to this
          size in order to avoid rendering issues. See <code><a href="collapsing.html">collapsing</a></code>
          for details.</p></dd>
 

        <dt></dt>
<dd><p><code>reverseStrand=FALSE</code>: Logical scalar. Set up the
          plotting coordinates in 3' -&gt; 5' direction if <code>TRUE</code>.
          This will effectively mirror the plot on the vertical axis.</p></dd>
 

        <dt></dt>
<dd><p><code>rotation=0</code>: The rotation angle for all text
          unless a more specific definiton exists.</p></dd>
 

        <dt></dt>
<dd><p><code>rotation.title=90</code> <code>(Aliases: rotation.title)</code>:
          The rotation angle for the text in the title panel. Even
          though this can be adjusted, the automatic resizing of the
          title panel will currently not work, so use at own risk.</p></dd>
 

        <dt></dt>
<dd><p><code>showAxis=TRUE</code>: Boolean controlling whether to
          plot a y axis (only applies to track types where axes are
          implemented).</p></dd>
 

        <dt></dt>
<dd><p><code>showTitle=TRUE</code>: Boolean controlling whether to
          plot a title panel. Although this can be set individually
          for each track, in multi-track plots as created by
          <code><a href="plotTracks.html">plotTracks</a></code> there will still be an empty
          placeholder in case any of the other tracks include a title.
          The same holds true for axes. Note that the the title panel
          background color could be set to transparent in order to
          completely hide it.</p></dd>
 

        <dt></dt>
<dd><p><code>v=-1</code>: Integer scalar. Parameter controlling the
          number of vertical grid lines, see <code><a href="https://rdrr.io/pkg/lattice/man/panel.functions.html" class="external-link">panel.grid</a></code>
          for details.</p></dd>
 

      
</dl><p></p></dd>
 

  
</dl></div>
    <div id="details">
    <h2>Details</h2>
    

<p>Depending on the setting of the <code>type</code> display parameter, the
  data can be plotted in various different forms as well as combinations
  thereof. Supported plotting types are:</p>
<p></p><dl><dt></dt>
<dd><p><code>p</code>: simple xy-plot.</p></dd>


    <dt></dt>
<dd><p><code>l</code>: lines plot. In the case of multiple samples this
      plotting type is not overly usefull since the points in the data
      matrix are connected in column-wise order. Type <code>a</code> might be
      more appropriate in these situations.</p></dd>


    <dt></dt>
<dd><p><code>b</code>: combination of xy-plot and lines plot.</p></dd>


    <dt></dt>
<dd><p><code>a</code>: lines plot of the column-wise average values.</p></dd>


    <dt></dt>
<dd><p><code>s</code>: sort and connect data points along the x-axis</p></dd>


    <dt></dt>
<dd><p><code>S</code>: sort and connect data points along the y-axis</p></dd>


    <dt></dt>
<dd><p><code>g</code>: add grid lines. To ensure a consitant look and
      feel across multiple tracks, grid lines should preferentially be
      added by using the <code>grid</code> display parameter.</p></dd>


    <dt></dt>
<dd><p><code>r</code>: add a regression line to the plot.</p></dd>


    <dt></dt>
<dd><p><code>h</code>: histogram-like vertical lines centered in the
      middle of the coordinate ranges.</p></dd>


    <dt></dt>
<dd><p><code>smooth</code>: add a loess fit to the plot. The following
      display parameters can be used to control the loess calculation:
      <code>span, degree, family, evaluation</code>. See
      <code><a href="https://rdrr.io/pkg/lattice/man/panel.loess.html" class="external-link">panel.loess</a></code> for details.</p></dd>


    <dt></dt>
<dd><p><code>histogram</code>: plot data as a histogram, where the width
      of the histogram bars reflects the width of the genomic ranges in
      the <code>range</code> slot.</p></dd>


    <dt></dt>
<dd><p><code>mountain</code>: plot a smoothed version of the data
      relative to a baseline, as defined by the <code>baseline</code> display
      parameter. The following display parameters can be used to control
      the smoothing: <code>span, degree, family, evaluation</code>. See
      <code><a href="https://rdrr.io/pkg/lattice/man/panel.loess.html" class="external-link">panel.loess</a></code> for details. The layout of the plot can
      be further customized via the following display parameters:
      <code>col.mountain, lwd.mountain, lty.mountain, fill.mountain</code>.</p></dd>


    <dt></dt>
<dd><p><code>polygon</code>: plot data as a polygon (similar to
      <code>mountain</code>-type but without smoothing). Data are plotted relative
      to a baseline, as defined by the <code>baseline</code> display
      parameter. The layout of the plot can be further customized via
      the following display parameters: <code>col.mountain,
      lwd.mountain, lty.mountain, fill.mountain</code>.</p></dd>



    <dt></dt>
<dd><p><code>boxplot</code>: plot the data as box-and-whisker
      plots. The layout of the plot can be further customized via the
      following display parameters: <code>box.ratio, box.width, varwidt,
      notch, notch.frac, levels.fos, stats, coef, do.out</code>. See
      <code><a href="https://rdrr.io/pkg/lattice/man/panel.bwplot.html" class="external-link">panel.bwplot</a></code> for details.</p></dd>



    <dt></dt>
<dd><p><code>gradient</code>: collapse the data across samples and plot this
      average value as a color-coded gradient. Essenitally this is
      similar to the heatmap-type plot of a single sample. The layout of
      the plot can be further customized via the display parameters
      <code>ncolor</code> and <code>gradient</code> which control the number of
      gradient colors as well as the gradient base colors,
      respectively.</p></dd>


    <dt></dt>
<dd><p><code>heatmap</code>: plot the color-coded values for all samples
      in the form of a heatmap. The data for individual samples can be
      visually separated by setting the <code>separator</code> display
      parameter. It's value is taken as the amount of spacing in pixels
      in between two heatmap rows. The layout of the plot can be further
      customized via the display parameters <code>ncolor</code> and
      <code>gradient</code> which control the number of gradient colors as
      well as the gradient base colors, respectively.</p></dd>


    <dt></dt>
<dd><p><code>horizon</code>: plot continuous data by cutting the y range into
      segments and overplotting them with color representing the
      magnitude and direction of deviation. This is particularly useful
      when comparing multiple samples, in which case the horizon strips
      are stacked. See <code>horizonplot</code> for details. Please
      note that the <code>origin</code> and <code>horizonscale</code> arguments of
      the Lattice <code>horizonplot</code> function are available as display
      parameters <code>horizon.origin</code> and <code>horizon.scale</code>.</p></dd>


  
</dl><p>For some of the above plotting-types the <code>groups</code> display
  parameter can be used to indicate sample sub-groupings. Its value is
  supposed to be a factor vector of similar length as the number of
  samples. In most cases, the groups are shown in different plotting
  colors and data aggregation operations are done in a stratified
  fashion.</p>
<p>The <code>window</code> display parameter can be used to aggregate the data
  prior to plotting. Its value is taken as the number of equal-sized
  windows along the genomic coordinates of the track for which to
  compute average values. The special value <code>auto</code> can be used to
  automatically determine a reasonable number of windows which can be
  particularly useful when plotting very large genomic regions with many
  data points.</p>
<p>The <code>aggregation</code> parameter can be set to define the aggregation
  function to be used when averaging in windows or across collapsed
  items. It takes the form of either a function which should condense a
  numeric vector into a single number, or one of the predefined options
  as character scalars <code>"mean"</code>, <code>"median"</code> or <code>"sum"</code>
  for mean, median or summation, respectively. Defaults to computing
  mean values for each sample. Note that the predefined options can be
  much faster because they are optimized to work on large numeric
  tables.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Florian Hahne</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index">

<p><code>DataTrack</code></p>
<p><code><a href="DisplayPars-class.html">DisplayPars</a></code></p>
<p><code><a href="GdObject-class.html">GdObject</a></code></p>
<p><code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code></p>
<p><code><a href="ImageMap-class.html">ImageMap</a></code></p>
<p><code><a href="https://rdrr.io/pkg/IRanges/man/IRanges-class.html" class="external-link">IRanges</a></code></p>
<p><code><a href="NumericTrack-class.html">NumericTrack</a></code></p>
<p><code><a href="RangeTrack-class.html">RangeTrack</a></code></p>
<p><code><a href="collapsing.html">collapsing</a></code></p>
<p><code><a href="grouping.html">grouping</a></code></p>
<p><code>horizonplot</code></p>
<p><code><a href="https://rdrr.io/pkg/lattice/man/panel.bwplot.html" class="external-link">panel.bwplot</a></code></p>
<p><code><a href="https://rdrr.io/pkg/lattice/man/panel.functions.html" class="external-link">panel.grid</a></code></p>
<p><code><a href="https://rdrr.io/pkg/lattice/man/panel.loess.html" class="external-link">panel.loess</a></code></p>
<p><code><a href="https://rdrr.io/pkg/lattice/man/panel.xyplot.html" class="external-link">panel.xyplot</a></code></p>
<p><code><a href="plotTracks.html">plotTracks</a></code></p>
<p><code><a href="settings.html">settings</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="co">## Object construction:</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## An empty object</span></span>
<span class="r-in"><span class="fu">DataTrack</span><span class="op">(</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataTrack 'DataTrack'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | genome: NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | active chromosome: chrNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | positions: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | samples:0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | strand: NA </span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## from individual arguments</span></span>
<span class="r-in"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fl">400</span><span class="op">)</span>, nrow<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dtTrack</span> <span class="op">&lt;-</span> <span class="fu">DataTrack</span><span class="op">(</span>start<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1000</span>, len<span class="op">=</span><span class="fl">100</span><span class="op">)</span>, width<span class="op">=</span><span class="fl">10</span>, data<span class="op">=</span><span class="va">dat</span>,</span>
<span class="r-in">chromosome<span class="op">=</span><span class="fl">1</span>, genome<span class="op">=</span><span class="st">"mm9"</span>, name<span class="op">=</span><span class="st">"random data"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## from GRanges</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span class="r-in"><span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="st">"chr1"</span>, ranges<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1000</span>, len<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,</span>
<span class="r-in">width<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">values</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dtTrack</span> <span class="op">&lt;-</span> <span class="fu">DataTrack</span><span class="op">(</span>range<span class="op">=</span><span class="va">gr</span>, genome<span class="op">=</span><span class="st">"mm9"</span>, name<span class="op">=</span><span class="st">"random data"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## from IRanges</span></span>
<span class="r-in"><span class="va">dtTrack</span> <span class="op">&lt;-</span> <span class="fu">DataTrack</span><span class="op">(</span>range<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/range-squeezers.html" class="external-link">ranges</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span>, data<span class="op">=</span><span class="va">dat</span>, genome<span class="op">=</span><span class="st">"mm9"</span>,</span>
<span class="r-in">name<span class="op">=</span><span class="st">"random data"</span>, chromosome<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## from a data.frame</span></span>
<span class="r-in"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"chromosome"</span></span>
<span class="r-in"><span class="va">dtTrack</span> <span class="op">&lt;-</span> <span class="fu">DataTrack</span><span class="op">(</span>range<span class="op">=</span><span class="va">df</span>,  genome<span class="op">=</span><span class="st">"mm9"</span>, name<span class="op">=</span><span class="st">"random data"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># \dontshow{</span></span>
<span class="r-in"><span class="co">## For some annoying reason the postscript device does not know about</span></span>
<span class="r-in"><span class="co">## the sans font</span></span>
<span class="r-in"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="op">{</span></span>
<span class="r-in"><span class="va">font</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/ps.options.html" class="external-link">ps.options</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">family</span></span>
<span class="r-in"><span class="fu"><a href="DisplayPars-class.html">displayPars</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>fontfamily<span class="op">=</span><span class="va">font</span>, fontfamily.title<span class="op">=</span><span class="va">font</span><span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-in"><span class="co"># }</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Plotting</span></span>
<span class="r-in"><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-plt img"><img src="DataTrack-class-1.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Track names</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "random data"</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"foo"</span></span>
<span class="r-in"><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-plt img"><img src="DataTrack-class-2.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Subsetting and splitting</span></span>
<span class="r-in"><span class="va">subTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">dtTrack</span>, from<span class="op">=</span><span class="fl">100</span>, to<span class="op">=</span><span class="fl">300</span><span class="op">)</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 19</span>
<span class="r-in"><span class="va">subTrack</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,<span class="op">]</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataTrack 'foo'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | genome: mm9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | active chromosome: chr1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | positions: 19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | samples:2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | strand: * </span>
<span class="r-in"><span class="va">subTrack</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataTrack 'foo'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | genome: mm9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | active chromosome: chr1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | positions: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | samples:4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | strand: * </span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">dtTrack</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, each<span class="op">=</span><span class="fl">50</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $`1`</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataTrack 'foo'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | genome: mm9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | active chromosome: chr1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | positions: 50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | samples:4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | strand: * </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $`2`</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataTrack 'foo'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | genome: mm9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | active chromosome: chr1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | positions: 50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | samples:4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | strand: * </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Accessors</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]    1   11   21   31   41   51   61   71   81   91  101  112  122  132  142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [16]  152  162  172  182  192  202  212  223  233  243  253  263  273  283  293</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [31]  303  313  323  334  344  354  364  374  384  394  404  414  424  434  445</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [46]  455  465  475  485  495  505  515  525  535  545  556  566  576  586  596</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [61]  606  616  626  636  646  656  667  677  687  697  707  717  727  737  747</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [76]  757  767  778  788  798  808  818  828  838  848  858  868  878  889  899</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [91]  909  919  929  939  949  959  969  979  989 1000</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]   10   20   30   40   50   60   70   80   90  100  110  121  131  141  151</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [16]  161  171  181  191  201  211  221  232  242  252  262  272  282  292  302</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [31]  312  322  332  343  353  363  373  383  393  403  413  423  433  443  454</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [46]  464  474  484  494  504  514  524  534  544  554  565  575  585  595  605</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [61]  615  625  635  645  655  665  676  686  696  706  716  726  736  746  756</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [76]  766  776  787  797  807  817  827  837  847  857  867  877  887  898  908</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [91]  918  928  938  948  958  968  978  988  998 1009</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1] 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [26] 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [51] 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [76] 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10 10</span>
<span class="r-in"><span class="fu"><a href="RangeTrack-class.html">position</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]    5.5   15.5   25.5   35.5   45.5   55.5   65.5   75.5   85.5   95.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [11]  105.5  116.5  126.5  136.5  146.5  156.5  166.5  176.5  186.5  196.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [21]  206.5  216.5  227.5  237.5  247.5  257.5  267.5  277.5  287.5  297.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [31]  307.5  317.5  327.5  338.5  348.5  358.5  368.5  378.5  388.5  398.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [41]  408.5  418.5  428.5  438.5  449.5  459.5  469.5  479.5  489.5  499.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [51]  509.5  519.5  529.5  539.5  549.5  560.5  570.5  580.5  590.5  600.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [61]  610.5  620.5  630.5  640.5  650.5  660.5  671.5  681.5  691.5  701.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [71]  711.5  721.5  731.5  741.5  751.5  761.5  771.5  782.5  792.5  802.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [81]  812.5  822.5  832.5  842.5  852.5  862.5  872.5  882.5  893.5  903.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [91]  913.5  923.5  933.5  943.5  953.5  963.5  973.5  983.5  993.5 1004.5</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span><span class="op">-</span><span class="fl">5</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "*"</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"-"</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="RangeTrack-class.html">chromosome</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "chr1"</span>
<span class="r-in"><span class="fu"><a href="RangeTrack-class.html">chromosome</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"chrX"</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  chr1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "mm9" </span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"mm9"</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="RangeTrack-class.html">range</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> IRanges object with 100 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             start       end     width</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]         1        10        10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]        11        20        10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]        21        30        10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]        31        40        10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]        41        50        10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...       ...       ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [96]       959       968        10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [97]       969       978        10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [98]       979       988        10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [99]       989       998        10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [100]      1000      1009        10</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/range-squeezers.html" class="external-link">ranges</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 100 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr1      1-10      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr1     11-20      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr1     21-30      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr1     31-40      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr1     41-50      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...       ...    ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [96]     chr1   959-968      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [97]     chr1   969-978      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [98]     chr1   979-988      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [99]     chr1   989-998      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [100]     chr1 1000-1009      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from mm9 genome; no seqlengths</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Data</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">values</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          [,1]      [,2]      [,3]      [,4]      [,5]       [,6]      [,7]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.57358702 0.6223585 0.9672963 0.5130287 0.6620476 0.93234592 0.3387958</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.18869879 0.4159132 0.2159033 0.7749373 0.9077198 0.19993327 0.2756013</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.75826533 0.7765109 0.8670089 0.1325433 0.3444916 0.05401831 0.6634905</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.09268309 0.8078028 0.4169180 0.4119076 0.1052431 0.42584008 0.7838961</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [,8]      [,9]     [,10]     [,11]     [,12]     [,13]      [,14]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.4396974 0.3802893 0.8146524 0.9885015 0.6132155 0.9982450 0.24688645</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.9303239 0.9353908 0.8824088 0.5552681 0.0336093 0.9601184 0.60656298</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.2722072 0.7355009 0.7454413 0.8797018 0.2547007 0.7509288 0.57397056</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.6590290 0.5900905 0.5017916 0.6304393 0.7759688 0.6805039 0.04959491</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,15]     [,16]     [,17]        [,18]     [,19]     [,20]      [,21]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.3728201 0.1752801 0.6251975 0.0006121558 0.5117742 0.5235778 0.13615199</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.8947680 0.1926055 0.5722048 0.9553637172 0.4671612 0.5604211 0.07388093</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.3917717 0.5465087 0.1406191 0.3994099500 0.7238355 0.6603428 0.76151218</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.5169453 0.3931208 0.2892717 0.9770535454 0.1420736 0.3722657 0.70279491</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,22]     [,23]      [,24]      [,25]     [,26]     [,27]     [,28]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.7638953 0.9126530 0.60425038 0.66891915 0.6744341 0.1279001 0.9301581</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.1570563 0.3080208 0.08395502 0.80104330 0.8458933 0.7941084 0.5924445</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.5993446 0.6570659 0.60566724 0.85545589 0.7463983 0.3555059 0.1104943</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.7633418 0.3378046 0.59591103 0.05013991 0.8295644 0.9480817 0.7838355</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [,29]     [,30]      [,31]     [,32]     [,33]     [,34]     [,35]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.60182294 0.8354190 0.56301801 0.6325430 0.4056898 0.6498573 0.3803301</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.08827442 0.2477733 0.80686536 0.2390868 0.7984731 0.6556109 0.6113083</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.65726228 0.3635909 0.03892874 0.2525250 0.1174629 0.7474796 0.9660003</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.33208094 0.9246486 0.03982544 0.6246245 0.9944973 0.5142732 0.2158587</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,36]     [,37]     [,38]      [,39]     [,40]     [,41]      [,42]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.7487494 0.9797886 0.7906329 0.49753823 0.1558116 0.2416608 0.20855917</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.4323756 0.9140007 0.1650948 0.03980805 0.9902038 0.8625791 0.05941213</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.8285962 0.2574866 0.8380759 0.62515660 0.6808823 0.7920867 0.62528744</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.8833405 0.2234478 0.9719204 0.83038591 0.8646979 0.6123248 0.80393136</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,43]      [,44]      [,45]     [,46]     [,47]     [,48]     [,49]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.1986306 0.06022841 0.83779684 0.0386787 0.9511434 0.1199900 0.6381902</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.6656501 0.42028041 0.01209806 0.3583329 0.6160606 0.5962579 0.6968670</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.4029328 0.55109725 0.57269810 0.2674532 0.9648370 0.3569202 0.6251960</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.3869627 0.81607572 0.30164086 0.9740237 0.8780345 0.6748865 0.1222221</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,50]     [,51]     [,52]      [,53]     [,54]      [,55]     [,56]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.2809718 0.4288006 0.5094248 0.30699922 0.3274520 0.92945157 0.3366140</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.1741882 0.3891868 0.3949653 0.07813034 0.6648269 0.04988784 0.9315206</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.1702244 0.8437064 0.2581601 0.94677874 0.7773210 0.67135756 0.4840459</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.5605436 0.2087907 0.8029614 0.37866848 0.4841216 0.82551193 0.9460470</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,57]     [,58]     [,59]     [,60]      [,61]     [,62]     [,63]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.3112607 0.4405584 0.2871900 0.5569071 0.30697206 0.5119930 0.8605094</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.5990519 0.5600596 0.8435601 0.1878760 0.35628694 0.8047119 0.4501529</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.8902234 0.9394272 0.2587780 0.5096489 0.05689367 0.2368177 0.1703747</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.6073635 0.8345116 0.3393378 0.9270733 0.88473156 0.6868353 0.5086963</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [,64]      [,65]     [,66]     [,67]     [,68]     [,69]     [,70]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.87001938 0.61503638 0.3019702 0.4649488 0.3935495 0.4152224 0.4784585</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.03330662 0.94951899 0.5120426 0.7293890 0.2491241 0.8916960 0.1102770</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.25815528 0.81559449 0.7103606 0.1281431 0.9473613 0.2067407 0.6293535</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.93708628 0.04931545 0.2846531 0.9342953 0.8416525 0.9501987 0.2934964</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [,71]     [,72]       [,73]     [,74]      [,75]     [,76]      [,77]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.83029322 0.5400352 0.222356177 0.3185764 0.01027815 0.2165860 0.05624466</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.32796644 0.4789401 0.509144905 0.5453064 0.59863873 0.1164273 0.56831566</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.87367987 0.8779045 0.009525969 0.3029777 0.71310268 0.4105797 0.70799067</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.02763326 0.4868288 0.088490856 0.7398839 0.46599317 0.6391856 0.97579369</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,78]      [,79]       [,80]     [,81]      [,82]      [,83]     [,84]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.9707030 0.45519014 0.824800678 0.4979119 0.30515477 0.09924971 0.4815282</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.1169157 0.20389975 0.101026231 0.9856622 0.98665275 0.54111919 0.8071745</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.1899509 0.97433515 0.972129777 0.4394909 0.92429570 0.62069789 0.9878339</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.1869202 0.09940844 0.005362803 0.3435958 0.01221103 0.79361237 0.9275750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [,85]        [,86]     [,87]     [,88]     [,89]       [,90]     [,91]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.27158255 0.5919763285 0.2006996 0.3914566 0.6917567 0.033254308 0.2807610</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.09879319 0.0008630857 0.6645027 0.8356536 0.9165655 0.647654844 0.6105293</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.06561258 0.4841866000 0.5526230 0.2486454 0.3700721 0.004003891 0.9270527</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.18732520 0.6553977022 0.7441226 0.8860824 0.6150685 0.577345077 0.5732150</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,92]     [,93]     [,94]        [,95]      [,96]     [,97]     [,98]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.7990210 0.8370746 0.0658973 0.9448925692 0.21729766 0.7015088 0.6762002</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.3004978 0.0721004 0.2241470 0.0565309483 0.68857418 0.1667424 0.5202724</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.8878272 0.6239092 0.5814365 0.0003418126 0.86487015 0.7850592 0.7049990</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.8550553 0.5107359 0.3622224 0.2899808860 0.02983773 0.1458532 0.2645380</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,99]    [,100]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.5058416 0.1099460</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.8223393 0.8714368</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.1797052 0.5144198</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.8646465 0.2856988</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/score.html" class="external-link">score</a></span><span class="op">(</span><span class="va">dtTrack</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          [,1]      [,2]      [,3]      [,4]      [,5]       [,6]      [,7]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.57358702 0.6223585 0.9672963 0.5130287 0.6620476 0.93234592 0.3387958</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.18869879 0.4159132 0.2159033 0.7749373 0.9077198 0.19993327 0.2756013</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.75826533 0.7765109 0.8670089 0.1325433 0.3444916 0.05401831 0.6634905</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.09268309 0.8078028 0.4169180 0.4119076 0.1052431 0.42584008 0.7838961</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [,8]      [,9]     [,10]     [,11]     [,12]     [,13]      [,14]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.4396974 0.3802893 0.8146524 0.9885015 0.6132155 0.9982450 0.24688645</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.9303239 0.9353908 0.8824088 0.5552681 0.0336093 0.9601184 0.60656298</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.2722072 0.7355009 0.7454413 0.8797018 0.2547007 0.7509288 0.57397056</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.6590290 0.5900905 0.5017916 0.6304393 0.7759688 0.6805039 0.04959491</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,15]     [,16]     [,17]        [,18]     [,19]     [,20]      [,21]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.3728201 0.1752801 0.6251975 0.0006121558 0.5117742 0.5235778 0.13615199</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.8947680 0.1926055 0.5722048 0.9553637172 0.4671612 0.5604211 0.07388093</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.3917717 0.5465087 0.1406191 0.3994099500 0.7238355 0.6603428 0.76151218</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.5169453 0.3931208 0.2892717 0.9770535454 0.1420736 0.3722657 0.70279491</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,22]     [,23]      [,24]      [,25]     [,26]     [,27]     [,28]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.7638953 0.9126530 0.60425038 0.66891915 0.6744341 0.1279001 0.9301581</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.1570563 0.3080208 0.08395502 0.80104330 0.8458933 0.7941084 0.5924445</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.5993446 0.6570659 0.60566724 0.85545589 0.7463983 0.3555059 0.1104943</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.7633418 0.3378046 0.59591103 0.05013991 0.8295644 0.9480817 0.7838355</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [,29]     [,30]      [,31]     [,32]     [,33]     [,34]     [,35]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.60182294 0.8354190 0.56301801 0.6325430 0.4056898 0.6498573 0.3803301</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.08827442 0.2477733 0.80686536 0.2390868 0.7984731 0.6556109 0.6113083</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.65726228 0.3635909 0.03892874 0.2525250 0.1174629 0.7474796 0.9660003</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.33208094 0.9246486 0.03982544 0.6246245 0.9944973 0.5142732 0.2158587</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,36]     [,37]     [,38]      [,39]     [,40]     [,41]      [,42]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.7487494 0.9797886 0.7906329 0.49753823 0.1558116 0.2416608 0.20855917</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.4323756 0.9140007 0.1650948 0.03980805 0.9902038 0.8625791 0.05941213</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.8285962 0.2574866 0.8380759 0.62515660 0.6808823 0.7920867 0.62528744</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.8833405 0.2234478 0.9719204 0.83038591 0.8646979 0.6123248 0.80393136</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,43]      [,44]      [,45]     [,46]     [,47]     [,48]     [,49]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.1986306 0.06022841 0.83779684 0.0386787 0.9511434 0.1199900 0.6381902</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.6656501 0.42028041 0.01209806 0.3583329 0.6160606 0.5962579 0.6968670</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.4029328 0.55109725 0.57269810 0.2674532 0.9648370 0.3569202 0.6251960</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.3869627 0.81607572 0.30164086 0.9740237 0.8780345 0.6748865 0.1222221</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,50]     [,51]     [,52]      [,53]     [,54]      [,55]     [,56]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.2809718 0.4288006 0.5094248 0.30699922 0.3274520 0.92945157 0.3366140</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.1741882 0.3891868 0.3949653 0.07813034 0.6648269 0.04988784 0.9315206</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.1702244 0.8437064 0.2581601 0.94677874 0.7773210 0.67135756 0.4840459</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.5605436 0.2087907 0.8029614 0.37866848 0.4841216 0.82551193 0.9460470</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,57]     [,58]     [,59]     [,60]      [,61]     [,62]     [,63]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.3112607 0.4405584 0.2871900 0.5569071 0.30697206 0.5119930 0.8605094</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.5990519 0.5600596 0.8435601 0.1878760 0.35628694 0.8047119 0.4501529</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.8902234 0.9394272 0.2587780 0.5096489 0.05689367 0.2368177 0.1703747</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.6073635 0.8345116 0.3393378 0.9270733 0.88473156 0.6868353 0.5086963</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [,64]      [,65]     [,66]     [,67]     [,68]     [,69]     [,70]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.87001938 0.61503638 0.3019702 0.4649488 0.3935495 0.4152224 0.4784585</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.03330662 0.94951899 0.5120426 0.7293890 0.2491241 0.8916960 0.1102770</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.25815528 0.81559449 0.7103606 0.1281431 0.9473613 0.2067407 0.6293535</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.93708628 0.04931545 0.2846531 0.9342953 0.8416525 0.9501987 0.2934964</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [,71]     [,72]       [,73]     [,74]      [,75]     [,76]      [,77]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.83029322 0.5400352 0.222356177 0.3185764 0.01027815 0.2165860 0.05624466</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.32796644 0.4789401 0.509144905 0.5453064 0.59863873 0.1164273 0.56831566</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.87367987 0.8779045 0.009525969 0.3029777 0.71310268 0.4105797 0.70799067</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.02763326 0.4868288 0.088490856 0.7398839 0.46599317 0.6391856 0.97579369</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,78]      [,79]       [,80]     [,81]      [,82]      [,83]     [,84]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.9707030 0.45519014 0.824800678 0.4979119 0.30515477 0.09924971 0.4815282</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.1169157 0.20389975 0.101026231 0.9856622 0.98665275 0.54111919 0.8071745</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.1899509 0.97433515 0.972129777 0.4394909 0.92429570 0.62069789 0.9878339</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.1869202 0.09940844 0.005362803 0.3435958 0.01221103 0.79361237 0.9275750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [,85]        [,86]     [,87]     [,88]     [,89]       [,90]     [,91]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.27158255 0.5919763285 0.2006996 0.3914566 0.6917567 0.033254308 0.2807610</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.09879319 0.0008630857 0.6645027 0.8356536 0.9165655 0.647654844 0.6105293</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.06561258 0.4841866000 0.5526230 0.2486454 0.3700721 0.004003891 0.9270527</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.18732520 0.6553977022 0.7441226 0.8860824 0.6150685 0.577345077 0.5732150</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,92]     [,93]     [,94]        [,95]      [,96]     [,97]     [,98]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.7990210 0.8370746 0.0658973 0.9448925692 0.21729766 0.7015088 0.6762002</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.3004978 0.0721004 0.2241470 0.0565309483 0.68857418 0.1667424 0.5202724</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.8878272 0.6239092 0.5814365 0.0003418126 0.86487015 0.7850592 0.7049990</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.8550553 0.5107359 0.3622224 0.2899808860 0.02983773 0.1458532 0.2645380</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        [,99]    [,100]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V1 0.5058416 0.1099460</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V2 0.8223393 0.8714368</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V3 0.1797052 0.5144198</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V4 0.8646465 0.2856988</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## coercion</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">dtTrack</span>, <span class="st">"data.frame"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     chromosome start  end width strand           V1           V2           V3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1         chr1     1   10    10      * 0.5735870199 0.1886987926 0.7582653339</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2         chr1    11   20    10      * 0.6223585266 0.4159132016 0.7765108705</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3         chr1    21   30    10      * 0.9672963202 0.2159032512 0.8670089110</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4         chr1    31   40    10      * 0.5130287160 0.7749373205 0.1325433177</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5         chr1    41   50    10      * 0.6620475734 0.9077197670 0.3444916373</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6         chr1    51   60    10      * 0.9323459219 0.1999332728 0.0540183072</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7         chr1    61   70    10      * 0.3387958198 0.2756013027 0.6634905422</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8         chr1    71   80    10      * 0.4396973667 0.9303238676 0.2722072056</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9         chr1    81   90    10      * 0.3802892941 0.9353907774 0.7355009299</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10        chr1    91  100    10      * 0.8146524148 0.8824088296 0.7454413087</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11        chr1   101  110    10      * 0.9885014733 0.5552680509 0.8797018330</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12        chr1   112  121    10      * 0.6132155319 0.0336092997 0.2547006856</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13        chr1   122  131    10      * 0.9982450008 0.9601184016 0.7509288136</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14        chr1   132  141    10      * 0.2468864541 0.6065629770 0.5739705635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15        chr1   142  151    10      * 0.3728201450 0.8947680488 0.3917717456</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16        chr1   152  161    10      * 0.1752800765 0.1926054636 0.5465086957</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17        chr1   162  171    10      * 0.6251974967 0.5722047726 0.1406190705</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18        chr1   172  181    10      * 0.0006121558 0.9553637172 0.3994099500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19        chr1   182  191    10      * 0.5117741989 0.4671611886 0.7238354937</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20        chr1   192  201    10      * 0.5235777793 0.5604211201 0.6603428058</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21        chr1   202  211    10      * 0.1361519932 0.0738809260 0.7615121801</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22        chr1   212  221    10      * 0.7638952618 0.1570562583 0.5993446065</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23        chr1   223  232    10      * 0.9126529882 0.3080208497 0.6570658754</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24        chr1   233  242    10      * 0.6042503819 0.0839550188 0.6056672356</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25        chr1   243  252    10      * 0.6689191493 0.8010432960 0.8554558854</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26        chr1   253  262    10      * 0.6744340854 0.8458932983 0.7463983176</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27        chr1   263  272    10      * 0.1279001259 0.7941084381 0.3555059133</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28        chr1   273  282    10      * 0.9301581059 0.5924445491 0.1104942642</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29        chr1   283  292    10      * 0.6018229409 0.0882744200 0.6572622757</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30        chr1   293  302    10      * 0.8354190320 0.2477733262 0.3635909229</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31        chr1   303  312    10      * 0.5630180079 0.8068653576 0.0389287404</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32        chr1   313  322    10      * 0.6325429690 0.2390867730 0.2525250022</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33        chr1   323  332    10      * 0.4056898011 0.7984730732 0.1174629412</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34        chr1   334  343    10      * 0.6498573283 0.6556109313 0.7474796236</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35        chr1   344  353    10      * 0.3803301165 0.6113082704 0.9660003211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36        chr1   354  363    10      * 0.7487493630 0.4323756348 0.8285962390</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37        chr1   364  373    10      * 0.9797886356 0.9140007265 0.2574865802</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38        chr1   374  383    10      * 0.7906329180 0.1650948089 0.8380758595</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39        chr1   384  393    10      * 0.4975382260 0.0398080489 0.6251565958</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40        chr1   394  403    10      * 0.1558116069 0.9902037513 0.6808823370</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41        chr1   404  413    10      * 0.2416607619 0.8625790973 0.7920866730</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42        chr1   414  423    10      * 0.2085591650 0.0594121274 0.6252874383</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43        chr1   424  433    10      * 0.1986306231 0.6656501235 0.4029327796</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44        chr1   434  443    10      * 0.0602284134 0.4202804118 0.5510972508</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45        chr1   445  454    10      * 0.8377968394 0.0120980635 0.5726980967</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46        chr1   455  464    10      * 0.0386787029 0.3583329110 0.2674532351</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47        chr1   465  474    10      * 0.9511434464 0.6160606141 0.9648370075</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48        chr1   475  484    10      * 0.1199900017 0.5962578633 0.3569201881</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49        chr1   485  494    10      * 0.6381902022 0.6968670192 0.6251959745</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50        chr1   495  504    10      * 0.2809718144 0.1741881554 0.1702244373</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51        chr1   505  514    10      * 0.4288005915 0.3891867667 0.8437064125</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52        chr1   515  524    10      * 0.5094247796 0.3949653304 0.2581600642</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53        chr1   525  534    10      * 0.3069992187 0.0781303388 0.9467787365</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54        chr1   535  544    10      * 0.3274520368 0.6648268595 0.7773210085</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55        chr1   545  554    10      * 0.9294515708 0.0498878444 0.6713575637</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56        chr1   556  565    10      * 0.3366140190 0.9315205917 0.4840459090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57        chr1   566  575    10      * 0.3112607158 0.5990518702 0.8902234128</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58        chr1   576  585    10      * 0.4405584221 0.5600596257 0.9394272438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59        chr1   586  595    10      * 0.2871900161 0.8435600812 0.2587780301</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60        chr1   596  605    10      * 0.5569070971 0.1878760487 0.5096489282</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61        chr1   606  615    10      * 0.3069720590 0.3562869385 0.0568936686</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62        chr1   616  625    10      * 0.5119929586 0.8047118725 0.2368176600</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63        chr1   626  635    10      * 0.8605094308 0.4501529348 0.1703747334</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64        chr1   636  645    10      * 0.8700193819 0.0333066215 0.2581552805</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65        chr1   646  655    10      * 0.6150363816 0.9495189879 0.8155944902</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66        chr1   656  665    10      * 0.3019702085 0.5120426093 0.7103606241</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67        chr1   667  676    10      * 0.4649487969 0.7293889807 0.1281430512</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68        chr1   677  686    10      * 0.3935495033 0.2491241221 0.9473612858</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69        chr1   687  696    10      * 0.4152223791 0.8916959967 0.2067407048</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70        chr1   697  706    10      * 0.4784584530 0.1102770171 0.6293535226</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71        chr1   707  716    10      * 0.8302932212 0.3279664356 0.8736798677</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72        chr1   717  726    10      * 0.5400351961 0.4789401426 0.8779045155</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73        chr1   727  736    10      * 0.2223561769 0.5091449048 0.0095259694</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74        chr1   737  746    10      * 0.3185763857 0.5453064234 0.3029776684</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75        chr1   747  756    10      * 0.0102781467 0.5986387285 0.7131026797</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76        chr1   757  766    10      * 0.2165860315 0.1164272521 0.4105797180</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77        chr1   767  776    10      * 0.0562446604 0.5683156576 0.7079906715</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78        chr1   778  787    10      * 0.9707029760 0.1169157147 0.1899509421</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79        chr1   788  797    10      * 0.4551901366 0.2038997530 0.9743351496</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80        chr1   798  807    10      * 0.8248006778 0.1010262312 0.9721297773</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81        chr1   808  817    10      * 0.4979119201 0.9856622049 0.4394909204</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82        chr1   818  827    10      * 0.3051547669 0.9866527512 0.9242956992</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83        chr1   828  837    10      * 0.0992497059 0.5411191899 0.6206978902</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84        chr1   838  847    10      * 0.4815281834 0.8071745096 0.9878338575</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85        chr1   848  857    10      * 0.2715825546 0.0987931858 0.0656125834</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86        chr1   858  867    10      * 0.5919763285 0.0008630857 0.4841866000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87        chr1   868  877    10      * 0.2006995643 0.6645026628 0.5526229665</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88        chr1   878  887    10      * 0.3914565546 0.8356535558 0.2486454002</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89        chr1   889  898    10      * 0.6917566981 0.9165655458 0.3700721294</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90        chr1   899  908    10      * 0.0332543079 0.6476548435 0.0040038908</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91        chr1   909  918    10      * 0.2807610431 0.6105293497 0.9270527456</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92        chr1   919  928    10      * 0.7990209830 0.3004978194 0.8878271687</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93        chr1   929  938    10      * 0.8370745510 0.0721003956 0.6239091882</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94        chr1   939  948    10      * 0.0658973011 0.2241470253 0.5814365137</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95        chr1   949  958    10      * 0.9448925692 0.0565309483 0.0003418126</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96        chr1   959  968    10      * 0.2172976583 0.6885741784 0.8648701455</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97        chr1   969  978    10      * 0.7015088277 0.1667423935 0.7850592399</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98        chr1   979  988    10      * 0.6762002339 0.5202723837 0.7049989968</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 99        chr1   989  998    10      * 0.5058415511 0.8223392977 0.1797051979</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 100       chr1  1000 1009    10      * 0.1099460442 0.8714367896 0.5144197687</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              V4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   0.092683093</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   0.807802802</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   0.416918028</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   0.411907642</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   0.105243064</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   0.425840076</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7   0.783896097</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8   0.659029029</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9   0.590090453</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10  0.501791561</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11  0.630439264</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12  0.775968801</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13  0.680503861</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14  0.049594908</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15  0.516945271</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16  0.393120839</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17  0.289271669</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18  0.977053545</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19  0.142073553</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20  0.372265702</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21  0.702794913</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22  0.763341787</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23  0.337804629</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24  0.595911026</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25  0.050139910</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26  0.829564404</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27  0.948081671</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28  0.783835468</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29  0.332080937</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30  0.924648634</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31  0.039825438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32  0.624624500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33  0.994497281</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34  0.514273213</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35  0.215858667</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36  0.883340520</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37  0.223447818</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38  0.971920439</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39  0.830385908</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40  0.864697936</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41  0.612324767</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42  0.803931365</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43  0.386962687</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44  0.816075719</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45  0.301640862</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46  0.974023738</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47  0.878034516</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48  0.674886464</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49  0.122222089</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50  0.560543634</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51  0.208790706</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52  0.802961438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53  0.378668481</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54  0.484121621</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55  0.825511929</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56  0.946046997</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57  0.607363547</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58  0.834511601</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59  0.339337832</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60  0.927073283</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61  0.884731559</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62  0.686835337</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63  0.508696325</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64  0.937086280</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65  0.049315451</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66  0.284653093</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67  0.934295350</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68  0.841652545</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69  0.950198708</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70  0.293496415</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71  0.027633261</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72  0.486828848</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73  0.088490856</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74  0.739883858</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75  0.465993169</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76  0.639185569</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77  0.975793693</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78  0.186920193</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79  0.099408438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80  0.005362803</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81  0.343595788</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82  0.012211026</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83  0.793612365</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84  0.927575003</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85  0.187325199</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86  0.655397702</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87  0.744122574</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88  0.886082397</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89  0.615068463</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90  0.577345077</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91  0.573214963</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92  0.855055271</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93  0.510735895</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94  0.362222408</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95  0.289980886</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96  0.029837726</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97  0.145853204</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98  0.264538009</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 99  0.864646499</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 100 0.285698822</span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Florian Hahne, Steffen Durinck, Robert Ivanek, Arne Mueller, Steve Lianoglou, Ge Tan, Lance Parsons, Shraddha Pai.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer></div>

  


  

  </body></html>

