<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>GeneRegionTrack class and methods — GeneRegionTrack-class • Gviz</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="GeneRegionTrack class and methods — GeneRegionTrack-class"><meta property="og:description" content="A class to hold gene model data for a genomic region."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Gviz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.45.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/Gviz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ivanek/Gviz/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>GeneRegionTrack class and methods</h1>
    <small class="dont-index">Source: <a href="https://github.com/ivanek/Gviz/blob/HEAD/R/GeneRegionTrack-class.R" class="external-link"><code>R/GeneRegionTrack-class.R</code></a></small>
    <div class="hidden name"><code>GeneRegionTrack-class.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A class to hold gene model data for a genomic region.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="co"># S4 method for GeneRegionTrack</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/new.html" class="external-link">initialize</a></span><span class="op">(</span><span class="va">.Object</span>, <span class="va">start</span>, <span class="va">end</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for ReferenceGeneRegionTrack</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/new.html" class="external-link">initialize</a></span><span class="op">(</span></span>
<span>  <span class="va">.Object</span>,</span>
<span>  <span class="va">stream</span>,</span>
<span>  <span class="va">reference</span>,</span>
<span>  mapping <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  defaults <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">GeneRegionTrack</span><span class="op">(</span></span>
<span>  range <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rstarts <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rends <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rwidths <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">strand</span>,</span>
<span>  <span class="va">feature</span>,</span>
<span>  <span class="va">exon</span>,</span>
<span>  <span class="va">transcript</span>,</span>
<span>  <span class="va">gene</span>,</span>
<span>  <span class="va">symbol</span>,</span>
<span>  <span class="va">chromosome</span>,</span>
<span>  <span class="va">genome</span>,</span>
<span>  stacking <span class="op">=</span> <span class="st">"squish"</span>,</span>
<span>  name <span class="op">=</span> <span class="st">"GeneRegionTrack"</span>,</span>
<span>  start <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  end <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">importFunction</span>,</span>
<span>  stream <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack</span></span>
<span><span class="fu"><a href="GdObject-class.html">gene</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack,character</span></span>
<span><span class="fu"><a href="GdObject-class.html">gene</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">value</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack</span></span>
<span><span class="fu"><a href="GdObject-class.html">symbol</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack,character</span></span>
<span><span class="fu"><a href="GdObject-class.html">symbol</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">value</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack</span></span>
<span><span class="fu"><a href="GdObject-class.html">transcript</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack,character</span></span>
<span><span class="fu"><a href="GdObject-class.html">transcript</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">value</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack</span></span>
<span><span class="fu"><a href="GdObject-class.html">exon</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack,character</span></span>
<span><span class="fu"><a href="GdObject-class.html">exon</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">value</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack</span></span>
<span><span class="fu"><a href="GdObject-class.html">group</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack,character</span></span>
<span><span class="fu"><a href="GdObject-class.html">group</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">value</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack</span></span>
<span><span class="fu"><a href="GdObject-class.html">identifier</a></span><span class="op">(</span></span>
<span>  <span class="va">GdObject</span>,</span>
<span>  type <span class="op">=</span> <span class="fu">.dpOrDefault</span><span class="op">(</span><span class="va">GdObject</span>, <span class="st">"transcriptAnnotation"</span>, <span class="st">"symbol"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack,character</span></span>
<span><span class="fu"><a href="GdObject-class.html">identifier</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">value</span></span>
<span></span>
<span><span class="co"># S4 method for ReferenceGeneRegionTrack</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack</span></span>
<span><span class="fu"><a href="GdObject-class.html">drawGD</a></span><span class="op">(</span><span class="va">GdObject</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for GeneRegionTrack</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for ReferenceGeneRegionTrack</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>.Object</dt>
<dd><p>.Object</p></dd>


<dt>start, end</dt>
<dd><p>An integer scalar with the genomic start or end coordinate
for the gene model range. If those are missing, the default value will
automatically be the smallest (or largest) value, respectively in
<code>rstarts</code> and <code>rends</code> for the currently active chromosome. When
building a <code>GeneRegionTrack</code> from a <code>TxDb</code> object, these arguments
can be used to subset the desired annotation data by genomic coordinates.
Please note this in that case the <code>chromosome</code> parameter must also be
set.</p></dd>


<dt>...</dt>
<dd><p>Additional items which will all be interpreted as further
display parameters. See <code><a href="settings.html">settings</a></code> and the "Display Parameters"
section below for details.</p></dd>


<dt>stream</dt>
<dd><p>A logical flag indicating that the user-provided import
function can deal with indexed files and knows how to process the additional
<code>selection</code> argument when accessing the data on disk. This causes the
constructor to return a <code>ReferenceGeneRegionTrack</code> object which will
grab the necessary data on the fly during each plotting operation.</p></dd>


<dt>reference</dt>
<dd><p>reference file</p></dd>


<dt>mapping</dt>
<dd><p>mapping</p></dd>


<dt>args</dt>
<dd><p>args</p></dd>


<dt>defaults</dt>
<dd><p><code>logical</code></p></dd>


<dt>range</dt>
<dd><p>An optional meta argument to handle the different input types. If the
<code>range</code> argument is missing, all the relevant information to create the
object has to be provided as individual function arguments (see below).</p>
<p>The different input options for <code>range</code> are:</p>
<dl><dt>A <code>TxDb</code> object:</dt>
<dd><p>all the necessary gene model information
including exon locations, transcript groupings and associated gene ids are
contained in <code>TxDb</code> objects, and the coercion between the two is almost
completely automated. If desired, the data to be fetched from the
<code>TxDb</code> object can be restricted using the constructor's
<code>chromosome</code>, <code>start</code> and <code>end</code> arguments. See below for
details. A direct coercion method <code>as(obj, "GeneRegionTrack")</code> is also
available. A nice added benefit of this input option is that the UTR and
coding region information that is part of the original <code>TxDb</code> object is
retained in the <code>GeneRegionTrack</code>.</p></dd>


<dt>A <code>GRanges</code> object:</dt>
<dd><p>the genomic ranges for the
<code>GeneRegion</code> track as well as the optional additional metadata columns
<code>feature</code>, <code>transcript</code>, <code>gene</code>, <code>exon</code> and
<code>symbol</code> (see description of the individual function parameters below
for details). Calling the constructor on a <code>GRanges</code> object without
further arguments, e.g.  <code>GeneRegionTrack(range=obj)</code> is equivalent to
calling the coerce method <code>as(obj, "GeneRegionTrack")</code>.</p></dd>


<dt>A <code>GRangesList</code> object:</dt>
<dd><p>this is very similar to the previous
case, except that the grouping information that is part of the list
structure is preserved in the <code>GeneRegionTrack</code>. I.e., all the elements
within one list item receive the same group id. For consistancy, there is
also a coercion method from <code>GRangesLists</code> <code>as(obj,
"GeneRegionTrack")</code>. Please note that unless the necessary information about
gene ids, symbols, etc. is present in the individual <code>GRanges</code> meta
data slots, the object will not be particularly useful, because all the
identifiers will be set to a common default value.</p></dd>


<dt>An <code><a href="https://rdrr.io/pkg/IRanges/man/IRanges-class.html" class="external-link">IRanges</a></code> object:</dt>
<dd><p>almost identical to the
<code>GRanges</code> case, except that the chromosome and strand information as
well as all additional data has to be provided in the separate
<code>chromosome</code>, <code>strand</code>, <code>feature</code>, <code>transcript</code>,
<code>symbol</code>, <code>exon</code> or <code>gene</code> arguments, because it can not be
directly encoded in an <code>IRanges</code> object. Note that only the former two
are mandatory (if not provided explicitely the more or less reasonable
default values <code>chromosome=NA</code> and <code>strand=*</code> are used, but not
providing information about the gene-to-transcript relationship or the
human-readble symbols renders a lot of the class' functionality useles.</p></dd>


<dt>A <code>data.frame</code> object:</dt>
<dd><p>the <code>data.frame</code> needs to contain
at least the two mandatory columns <code>start</code> and <code>end</code> with the
range coordinates. It may also contain a <code>chromosome</code> and a
<code>strand</code> column with the chromosome and strand information for each
range. If missing, this information will be drawn from the constructor's
<code>chromosome</code> or <code>strand</code> arguments. In addition, the
<code>feature</code>, <code>exon</code>, <code>transcript</code>, <code>gene</code> and
<code>symbol</code> data can be provided as columns in the <code>data.frame</code>. The
above comments about potential default values also apply here.</p></dd>


<dt>A <code>character</code> scalar:</dt>
<dd><p>in this case the value of the
<code>range</code> argument is considered to be a file path to an annotation file
on disk. A range of file types are supported by the <code>Gviz</code> package as
identified by the file extension. See the <code>importFunction</code>
documentation below for further details.</p></dd>



</dl></dd>


<dt>rstarts</dt>
<dd><p>An integer vector of the start coordinates for the actual
gene model items, i.e., for the individual exons. The relationship between
exons is handled via the <code>gene</code> and <code>transcript</code> factors.
Alternatively, this can be a vector of comma-separated lists of integer
coordinates, one vector item for each transcript, and each comma-separated
element being the start location of a single exon within that transcript.
Those lists will be exploded upon object instantiation and all other
annotation arguments will be recycled accordingly to regenerate the
exon/transcript/gene relationship structure. This implies the approriate
number of items in all annotation and coordinates arguments.</p></dd>


<dt>rends</dt>
<dd><p>An integer vector of the end coordinates for the actual gene
model items. Both <code>rstarts</code> and <code>rends</code> have to be of equal
length.</p></dd>


<dt>rwidths</dt>
<dd><p>An integer vector of widths for the actual gene model items.
This can be used instead of either <code>rstarts</code> or <code>rends</code> to specify
the range coordinates.</p></dd>


<dt>strand</dt>
<dd><p>Character vector, the strand information for the individual
track exons. It may be provided in the form <code>+</code> for the Watson strand,
<code>-</code> for the Crick strand or <code>*</code> for either one of the two. Please
note that all items within a single gene or transcript model need to be on
the same strand, and erroneous entries will result in casting of an error.</p></dd>


<dt>feature</dt>
<dd><p>Factor (or other vector that can be coerced into one), giving
the feature types for the individual track exons.  When plotting the track
to the device, if a display parameter with the same name as the value of
<code>feature</code> is set, this will be used as the track item's fill color.
Additionally, the feature type defines whether an element in the
<code>GeneRegionTrack</code> is considered to be coding or non-coding. The details
section as well as the section about the <code>thinBoxFeature</code> display
parameter further below has more information on this. See also
<code><a href="grouping.html">grouping</a></code> for details.</p></dd>


<dt>exon</dt>
<dd><p>Character vector of exon identifiers. It's values will be used
as the identifier tag when plotting to the device if the display parameter
<code>showExonId=TRUE</code>.</p></dd>


<dt>transcript</dt>
<dd><p>Factor (or other vector that can be coerced into one),
giving the transcript memberships for the individual track exons. All items
with the same transcript identifier will be visually connected when plotting
to the device.  See <code><a href="grouping.html">grouping</a></code> for details. Will be used as
labels when <code>showId=TRUE</code>, and <code>geneSymbol=FALSE</code>.</p></dd>


<dt>gene</dt>
<dd><p>Factor (or other vector that can be coerced into one), giving
the gene memberships for the individual track exons.</p></dd>


<dt>symbol</dt>
<dd><p>A factor with human-readable gene name aliases which will be
used as labels when <code>showId=TRUE</code>, and <code>geneSymbol=TRUE</code>.</p></dd>


<dt>chromosome</dt>
<dd><p>The chromosome on which the track's genomic ranges are
defined. A valid UCSC chromosome identifier if
<code>options(ucscChromosomeNames=TRUE)</code>. Please note that in this case only
syntactic checking takes place, i.e., the argument value needs to be an
integer, numeric character or a character of the form <code>chrx</code>, where
<code>x</code> may be any possible string. The user has to make sure that the
respective chromosome is indeed defined for the the track's genome. If not
provided here, the constructor will try to build the chromosome information
based on the available inputs, and as a last resort will fall back to the
value <code>chrNA</code>. Please note that by definition all objects in the
<code>Gviz</code> package can only have a single active chromosome at a time
(although internally the information for more than one chromosome may be
present), and the user has to call the <code>chromosome&lt;-</code> replacement
method in order to change to a different active chromosome. When creating a
<code>GeneRegionTrack</code> from a <code>TxDb</code> object, the value of this
parameter can be used to subset the data to fetch only transcripts from a
single chromosome.</p></dd>


<dt>genome</dt>
<dd><p>The genome on which the track's ranges are defined. Usually
this is a valid UCSC genome identifier, however this is not being formally
checked at this point. If not provided here the constructor will try to
extract this information from the provided inputs, and eventually will fall
back to the default value of <code>NA</code>.</p></dd>


<dt>stacking</dt>
<dd><p>The stacking type for overlapping items of the track. One in
<code>c(hide, dense, squish, pack,full)</code>. Currently, only hide (don't show
the track items, squish (make best use of the available space) and dense (no
stacking at all) are implemented.</p></dd>


<dt>name</dt>
<dd><p>Character scalar of the track's name used in the title panel
when plotting.</p></dd>


<dt>importFunction</dt>
<dd><p>A user-defined function to be used to import the data
from a file. This only applies when the <code>range</code> argument is a character
string with the path to the input data file. The function needs to accept an
argument <code>x</code> containing the file path and has to return a proper
<code>GRanges</code> object with all the necessary metadata columns set. A set of
default import functions is already implemented in the package for a number
of different file types, and one of these defaults will be picked
automatically based on the extension of the input file name. If the
extension can not be mapped to any of the existing import function, an error
is raised asking for a user-defined import function via this argument.
Currently the following file types can be imported with the default
functions: <code>gff</code>, <code>gff1</code>, <code>gff2</code>, <code>gff3</code>, <code>gtf</code>.</p></dd>


<dt>GdObject</dt>
<dd><p>Object of <code>GdObject-class</code>.</p></dd>


<dt>value</dt>
<dd><p>Value to be set.</p></dd>


<dt>type</dt>
<dd><p>type</p></dd>


<dt>x</dt>
<dd><p>A valid track object class name, or the object itself, in which
case the class is derived directly from it.</p></dd>


<dt>object</dt>
<dd><p>object</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>The return value of the constructor function is a new object of class
<code>GeneRegionTrack</code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>A track containing all gene models in a particular region. The data are
usually fetched dynamially from an online data store, but it is also
possible to manully construct objects from local data. Connections to
particular online data sources should be implemented as sub-classes, and
<code>GeneRegionTrack</code> is just the commone denominator that is being used
for plotting later on. There are several levels of data associated to a
<code>GeneRegionTrack</code>:</p>
<dl><dt>exon level:</dt>
<dd><p>identifiers are stored in the exon column of the
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code> object in the <code>range</code> slot. Data may be
extracted using the <code>exon</code> method.</p></dd>


<dt>transcript level:</dt>
<dd><p>identifiers are stored in the transcript column of
the <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code> object. Data may be extracted using the
<code>transcript</code> method.</p></dd>


<dt>gene level:</dt>
<dd><p>identifiers are stored in the gene column of the
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code> object, more human-readable versions in the
symbol column. Data may be extracted using the <code>gene</code> or the
<code>symbol</code> methods.</p></dd>


<dt>transcript-type level:</dt>
<dd><p>information is stored in the feature column of
the <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code> object. If a display parameter of the same
name is specified, the software will use its value for the coloring.</p></dd>



</dl><p><code>GeneRegionTrack</code> objects also know about coding regions and non-coding
regions (e.g., UTRs) in a transcript, and will indicate those by using
different shapes (wide boxes for all coding regions, thinner boxes for
non-coding regions). This is archived by setting the <code>feature</code> values
of the object for non-coding elements to one of the options that are
provided in the <code>thinBoxFeature</code> display parameters. All other elements
are considered to be coding elements.</p>
    </div>
    <div id="functions">
    <h2>Functions</h2>
    
<ul><li><p><code>initialize(GeneRegionTrack)</code>: Initialize.</p></li>
<li><p><code>ReferenceGeneRegionTrack-class</code>: The file-based version of the <code>GeneRegionTrack-class</code>.</p></li>
<li><p><code>initialize(ReferenceGeneRegionTrack)</code>: Initialize.</p></li>
<li><p><code>GeneRegionTrack()</code>: Constructor function for <code>GeneRegionTrack-class</code>.</p></li>
<li><p><code>gene(GeneRegionTrack)</code>: Extract the gene identifiers for all
gene models.</p></li>
<li><p><code>gene(GdObject = GeneRegionTrack) &lt;- value</code>: Replace the gene identifiers for all
gene models.
The replacement value must be a character of appropriate length or another
vector that can be coerced into such.</p></li>
<li><p><code>symbol(GeneRegionTrack)</code>: Extract the human-readble gene symbol
for all gene models.</p></li>
<li><p><code>symbol(GdObject = GeneRegionTrack) &lt;- value</code>: Replace the human-readable gene symbol
for all gene models.
The replacement value must be a character of appropriate length or another
vector that can be coerced into such.</p></li>
<li><p><code>transcript(GeneRegionTrack)</code>: Extract the transcript identifiers for all
transcripts in the gene models.</p></li>
<li><p><code>transcript(GdObject = GeneRegionTrack) &lt;- value</code>: Replace the transcript identifiers for all
transcripts in the gene model. The replacement value must be a character of
appropriate length or another vector that can be coerced into such.</p></li>
<li><p><code>exon(GeneRegionTrack)</code>: Extract the exon identifiers for all exons
in the gene models.</p></li>
<li><p><code>exon(GdObject = GeneRegionTrack) &lt;- value</code>: replace the exon identifiers for all exons
in the gene model. The replacement value must be a character of appropriate
length or another vector that can be coerced into such.</p></li>
<li><p><code>group(GeneRegionTrack)</code>: extract the group membership for all track items.</p></li>
<li><p><code>group(GdObject = GeneRegionTrack) &lt;- value</code>: replace the grouping information for track items.
The replacement value must be a factor of appropriate length or another
vector that can be coerced into such.</p></li>
<li><p><code>identifier(GeneRegionTrack)</code>: return track item identifiers.
Depending on the setting of the optional argument lowest, these are either
the group identifiers or the individual item identifiers.
export</p></li>
<li><p><code>identifier(GdObject = GeneRegionTrack) &lt;- value</code>: Set the track item identifiers.
The replacement value has to be a character vector of appropriate length.
This always replaces the group-level identifiers, so essentially it is
similar to <code>groups&lt;-</code>.</p></li>
<li><p><code>subset(ReferenceGeneRegionTrack)</code>: Subset a GeneRegionTrack by coordinates
and sort if necessary.</p></li>
<li><p><code>drawGD(GeneRegionTrack)</code>: plot the object to a graphics device.
The return value of this method is the input object, potentially updated
during the plotting operation. Internally, there are two modes in which the
method can be called. Either in 'prepare' mode, in which case no plotting is
done but the object is preprocessed based on the available space, or in
'plotting' mode, in which case the actual graphical output is created.
Since subsetting of the object can be potentially costly, this can be
switched off in case subsetting has already been performed before or
is not necessary.</p></li>
<li><p><code>show(GeneRegionTrack)</code>: Show method.</p></li>
<li><p><code>show(ReferenceGeneRegionTrack)</code>: Show method.</p></li>
</ul></div>
    <div id="objects-from-the-class">
    <h2>Objects from the class</h2>
    


<p>Objects can be created using the constructor function
<code>GeneRegionTrack</code>.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="DisplayPars-class.html">DisplayPars</a></code></p>
<p><code><a href="GdObject-class.html">GdObject</a></code></p>
<p><code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code></p>
<p><code><a href="HighlightTrack-class.html">HighlightTrack</a></code></p>
<p><code><a href="ImageMap-class.html">ImageMap</a></code></p>
<p><code><a href="https://rdrr.io/pkg/IRanges/man/IRanges-class.html" class="external-link">IRanges</a></code></p>
<p><code><a href="RangeTrack-class.html">RangeTrack</a></code></p>
<p><code><a href="DataTrack-class.html">DataTrack</a></code></p>
<p><code><a href="collapsing.html">collapsing</a></code></p>
<p><code><a href="grouping.html">grouping</a></code></p>
<p><code><a href="https://rdrr.io/pkg/lattice/man/panel.functions.html" class="external-link">panel.grid</a></code></p>
<p><code><a href="plotTracks.html">plotTracks</a></code></p>
<p><code><a href="settings.html">settings</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Florian Hahne, Steve Lianoglou</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The empty object</span></span></span>
<span class="r-in"><span><span class="fu">GeneRegionTrack</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GeneRegionTrack 'GeneRegionTrack'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | genome: NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | active chromosome: chrNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | annotation features: 0</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Load some sample data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cyp2b10</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Construct the object</span></span></span>
<span class="r-in"><span><span class="va">grTrack</span> <span class="op">&lt;-</span> <span class="fu">GeneRegionTrack</span><span class="op">(</span></span></span>
<span class="r-in"><span>    start <span class="op">=</span> <span class="fl">26682683</span>, end <span class="op">=</span> <span class="fl">26711643</span>,</span></span>
<span class="r-in"><span>    rstart <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">start</span>, rends <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">end</span>, chromosome <span class="op">=</span> <span class="fl">7</span>, genome <span class="op">=</span> <span class="st">"mm9"</span>,</span></span>
<span class="r-in"><span>    transcript <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">transcript</span>, gene <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">gene</span>, symbol <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">symbol</span>,</span></span>
<span class="r-in"><span>    feature <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">feature</span>, exon <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">exon</span>,</span></span>
<span class="r-in"><span>    name <span class="op">=</span> <span class="st">"Cyp2b10"</span>, strand <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">strand</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Directly from the data.frame</span></span></span>
<span class="r-in"><span><span class="va">grTrack</span> <span class="op">&lt;-</span> <span class="fu">GeneRegionTrack</span><span class="op">(</span><span class="va">cyp2b10</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## From a TxDb object</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicFeatures" class="external-link">GenomicFeatures</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">samplefile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span></span>
<span class="r-in"><span>                              <span class="st">"hg19_knownGene_sample.sqlite"</span>,</span></span>
<span class="r-in"><span>                              package <span class="op">=</span> <span class="st">"GenomicFeatures"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">txdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">loadDb</a></span><span class="op">(</span><span class="va">samplefile</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu">GeneRegionTrack</span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu">GeneRegionTrack</span><span class="op">(</span><span class="va">txdb</span>, chromosome <span class="op">=</span> <span class="st">"chr6"</span>, start <span class="op">=</span> <span class="fl">35000000</span>, end <span class="op">=</span> <span class="fl">40000000</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicFeatures</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: AnnotationDbi</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biobase</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to Bioconductor</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Vignettes contain introductory material; view with</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GeneRegionTrack 'GeneRegionTrack'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | genome: hg19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | active chromosome: chr6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | annotation features: 71</span>
<span class="r-in"><span><span class="co"># \dontshow{</span></span></span>
<span class="r-in"><span><span class="co">## For some annoying reason the postscript device does not know about</span></span></span>
<span class="r-in"><span><span class="co">## the sans font</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">font</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/ps.options.html" class="external-link">ps.options</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">family</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="DisplayPars-class.html">displayPars</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>fontfamily <span class="op">=</span> <span class="va">font</span>, fontfamily.title <span class="op">=</span> <span class="va">font</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Plotting</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="GeneRegionTrack-class-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Track names</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "GeneRegionTrack"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"foo"</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="GeneRegionTrack-class-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Subsetting and splitting</span></span></span>
<span class="r-in"><span><span class="va">subTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">grTrack</span>, from <span class="op">=</span> <span class="fl">26700000</span>, to <span class="op">=</span> <span class="fl">26705000</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0</span>
<span class="r-in"><span><span class="va">subTrack</span> <span class="op">&lt;-</span> <span class="va">grTrack</span><span class="op">[</span><span class="fu"><a href="GdObject-class.html">transcript</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span> <span class="op">==</span> <span class="st">"ENSMUST00000144140"</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">grTrack</span>, <span class="fu"><a href="GdObject-class.html">transcript</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GeneRegionTrack 'foo'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | genome: NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | active chromosome: chr7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | annotation features: 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GeneRegionTrack 'foo'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | genome: NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | active chromosome: chr7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | annotation features: 11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $ENSMUST00000144140</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GeneRegionTrack 'foo'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | genome: NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | active chromosome: chr7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | annotation features: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Accessors</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 25897620 25897676 25897685 25897685 25911238 25911238 25911554 25911554</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] 25913865 25913865 25914748 25914748 25915392 25915392 25916766 25916934</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [17] 25916934 25917288 25917288 25917288 25925373 25926068 25926068 25926250</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] 25926250</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 25897684 25897684 25897855 25897855 25911400 25911400 25911703 25911703</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] 25914025 25914025 25914924 25914924 25915533 25915533 25917121 25917121</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [17] 25917121 25917429 25917429 25917430 25925399 25926249 25926249 25926559</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] 25926624</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1]  65   9 171 171 163 163 150 150 161 161 177 177 142 142 356 188 188 142 142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [20] 143  27 182 182 310 375</span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">position</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 25897652 25897680 25897770 25897770 25911319 25911319 25911628 25911628</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] 25913945 25913945 25914836 25914836 25915462 25915462 25916944 25917028</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [17] 25917028 25917358 25917358 25917359 25925386 25926158 25926158 25926404</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] 25926437</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">+</span> <span class="fl">100</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [20] "+" "+" "+" "+" "+" "+"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"-"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">chromosome</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "chr7"</span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">chromosome</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"chrX"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] NA</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"hg19"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> IRanges object with 25 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            start       end     width</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]  25897620  25897684        65</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]  25897676  25897684         9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]  25897685  25897855       171</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]  25897685  25897855       171</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]  25911238  25911400       163</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...       ...       ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [21]  25925373  25925399        27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [22]  25926068  25926249       182</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [23]  25926068  25926249       182</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [24]  25926250  25926559       310</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [25]  25926250  25926624       375</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/range-squeezers.html" class="external-link">ranges</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 25 ranges and 7 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames            ranges strand |        feature          id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |       &lt;factor&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]     chr7 25897620-25897684      + | utr5               unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]     chr7 25897676-25897684      + | utr5               unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]     chr7 25897685-25897855      + | protein_coding     unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]     chr7 25897685-25897855      + | protein_coding     unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]     chr7 25911238-25911400      + | protein_coding     unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...               ...    ... .            ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [21]     chr7 25925373-25925399      + | protein_coding     unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [22]     chr7 25926068-25926249      + | protein_coding     unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [23]     chr7 25926068-25926249      + | protein_coding     unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [24]     chr7 25926250-25926559      + | utr3               unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [25]     chr7 25926250-25926624      + | utr3               unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      exon         transcript               gene   symbol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;factor&gt;           &lt;factor&gt;           &lt;factor&gt; &lt;factor&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1] ENSMUSE00000742021 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2] ENSMUSE00000489385 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3] ENSMUSE00000742021 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4] ENSMUSE00000489385 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5] ENSMUSE00000973560 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...                ...                ...                ...      ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [21] ENSMUSE00000496705 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [22] ENSMUSE00000498616 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [23] ENSMUSE00000750625 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [24] ENSMUSE00000750625 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [25] ENSMUSE00000498616 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          density</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [21]         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [22]         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [23]         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [24]         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [25]         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Annotation</span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">identifier</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [8] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [15] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [22] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">identifier</a></span><span class="op">(</span><span class="va">grTrack</span>, <span class="st">"lowest"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "ENSMUSE00000742021" "ENSMUSE00000489385" "ENSMUSE00000742021"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] "ENSMUSE00000489385" "ENSMUSE00000973560" "ENSMUSE00000973560"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "ENSMUSE00001043622" "ENSMUSE00001043622" "ENSMUSE00001013991"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [10] "ENSMUSE00001013991" "ENSMUSE00000996531" "ENSMUSE00000996531"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "ENSMUSE00000999790" "ENSMUSE00000999790" "ENSMUSE00000736922"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16] "ENSMUSE00000963884" "ENSMUSE00000963884" "ENSMUSE00001063063"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "ENSMUSE00001063063" "ENSMUSE00000748299" "ENSMUSE00000496705"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [22] "ENSMUSE00000498616" "ENSMUSE00000750625" "ENSMUSE00000750625"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "ENSMUSE00000498616"</span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">identifier</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"bar"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">feature</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "utr5"           "utr5"           "protein_coding" "protein_coding"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] "protein_coding" "protein_coding" "protein_coding" "protein_coding"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] "protein_coding" "protein_coding" "protein_coding" "protein_coding"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "protein_coding" "protein_coding" "protein_coding" "protein_coding"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [17] "protein_coding" "protein_coding" "protein_coding" "protein_coding"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [21] "protein_coding" "protein_coding" "protein_coding" "utr3"          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "utr3"          </span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">feature</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"foo"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">exon</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "ENSMUSE00000742021" "ENSMUSE00000489385" "ENSMUSE00000742021"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] "ENSMUSE00000489385" "ENSMUSE00000973560" "ENSMUSE00000973560"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "ENSMUSE00001043622" "ENSMUSE00001043622" "ENSMUSE00001013991"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [10] "ENSMUSE00001013991" "ENSMUSE00000996531" "ENSMUSE00000996531"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "ENSMUSE00000999790" "ENSMUSE00000999790" "ENSMUSE00000736922"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16] "ENSMUSE00000963884" "ENSMUSE00000963884" "ENSMUSE00001063063"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "ENSMUSE00001063063" "ENSMUSE00000748299" "ENSMUSE00000496705"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [22] "ENSMUSE00000498616" "ENSMUSE00000750625" "ENSMUSE00000750625"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "ENSMUSE00000498616"</span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">exon</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">letters</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">gene</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [10] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [22] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "ENSMUSG00000030483"</span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">gene</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"bar"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">symbol</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [8] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [15] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [22] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">symbol</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"foo"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">transcript</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] "ENSMUST00000005477" "ENSMUST00000072438" "ENSMUST00000005477"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [10] "ENSMUST00000005477" "ENSMUST00000072438" "ENSMUST00000005477"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000144140"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "ENSMUST00000005477" "ENSMUST00000144140" "ENSMUST00000005477"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [22] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000005477"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "ENSMUST00000072438"</span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">transcript</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"foo"</span>, <span class="st">"bar"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">chromosome</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"chr7"</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="GeneRegionTrack-class-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">values</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           feature      id               exon         transcript</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            utr5 unknown ENSMUSE00000742021 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            utr5 unknown ENSMUSE00000489385 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  protein_coding unknown ENSMUSE00000742021 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  protein_coding unknown ENSMUSE00000489385 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  protein_coding unknown ENSMUSE00000973560 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  protein_coding unknown ENSMUSE00000973560 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  protein_coding unknown ENSMUSE00001043622 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8  protein_coding unknown ENSMUSE00001043622 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  protein_coding unknown ENSMUSE00001013991 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 protein_coding unknown ENSMUSE00001013991 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11 protein_coding unknown ENSMUSE00000996531 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12 protein_coding unknown ENSMUSE00000996531 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13 protein_coding unknown ENSMUSE00000999790 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14 protein_coding unknown ENSMUSE00000999790 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15 protein_coding unknown ENSMUSE00000736922 ENSMUST00000144140</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16 protein_coding unknown ENSMUSE00000963884 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17 protein_coding unknown ENSMUSE00000963884 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18 protein_coding unknown ENSMUSE00001063063 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19 protein_coding unknown ENSMUSE00001063063 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20 protein_coding unknown ENSMUSE00000748299 ENSMUST00000144140</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21 protein_coding unknown ENSMUSE00000496705 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22 protein_coding unknown ENSMUSE00000498616 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23 protein_coding unknown ENSMUSE00000750625 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24           utr3 unknown ENSMUSE00000750625 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25           utr3 unknown ENSMUSE00000498616 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  gene  symbol density</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Grouping</span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">group</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] "ENSMUST00000005477" "ENSMUST00000072438" "ENSMUST00000005477"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [10] "ENSMUST00000005477" "ENSMUST00000072438" "ENSMUST00000005477"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000144140"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "ENSMUST00000005477" "ENSMUST00000144140" "ENSMUST00000005477"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [22] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000005477"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "ENSMUST00000072438"</span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">group</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Group 1"</span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">transcript</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Group 1" "Group 1"</span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="GeneRegionTrack-class-4.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Collapsing transcripts</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grTrack</span>,</span></span>
<span class="r-in"><span>    collapseTranscripts <span class="op">=</span> <span class="cn">TRUE</span>, showId <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    extend.left <span class="op">=</span> <span class="fl">10000</span>, shape <span class="op">=</span> <span class="st">"arrow"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="GeneRegionTrack-class-5.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Stacking</span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">stacking</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "squish"</span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">stacking</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"dense"</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="GeneRegionTrack-class-6.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## coercion</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">grTrack</span>, <span class="st">"data.frame"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    X.seqnames  X.start    X.end X.width X.strand      X.feature    X.id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        chr7 25897620 25897684      65        +           utr5 unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2        chr7 25897676 25897684       9        +           utr5 unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3        chr7 25897685 25897855     171        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4        chr7 25897685 25897855     171        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5        chr7 25911238 25911400     163        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6        chr7 25911238 25911400     163        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7        chr7 25911554 25911703     150        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8        chr7 25911554 25911703     150        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9        chr7 25913865 25914025     161        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10       chr7 25913865 25914025     161        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11       chr7 25914748 25914924     177        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12       chr7 25914748 25914924     177        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13       chr7 25915392 25915533     142        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14       chr7 25915392 25915533     142        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15       chr7 25916766 25917121     356        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16       chr7 25916934 25917121     188        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17       chr7 25916934 25917121     188        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18       chr7 25917288 25917429     142        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19       chr7 25917288 25917429     142        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20       chr7 25917288 25917430     143        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21       chr7 25925373 25925399      27        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22       chr7 25926068 25926249     182        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23       chr7 25926068 25926249     182        + protein_coding unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24       chr7 25926250 25926559     310        +           utr3 unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25       chr7 25926250 25926624     375        +           utr3 unknown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                X.exon       X.transcript             X.gene X.symbol X.density</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  ENSMUSE00000742021 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  ENSMUSE00000489385 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  ENSMUSE00000742021 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  ENSMUSE00000489385 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  ENSMUSE00000973560 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  ENSMUSE00000973560 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  ENSMUSE00001043622 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8  ENSMUSE00001043622 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  ENSMUSE00001013991 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 ENSMUSE00001013991 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11 ENSMUSE00000996531 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12 ENSMUSE00000996531 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13 ENSMUSE00000999790 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14 ENSMUSE00000999790 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15 ENSMUSE00000736922 ENSMUST00000144140 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16 ENSMUSE00000963884 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17 ENSMUSE00000963884 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18 ENSMUSE00001063063 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19 ENSMUSE00001063063 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20 ENSMUSE00000748299 ENSMUST00000144140 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21 ENSMUSE00000496705 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22 ENSMUSE00000498616 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23 ENSMUSE00000750625 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24 ENSMUSE00000750625 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25 ENSMUSE00000498616 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           feature      id               exon         transcript</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            utr5 unknown ENSMUSE00000742021 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            utr5 unknown ENSMUSE00000489385 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  protein_coding unknown ENSMUSE00000742021 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  protein_coding unknown ENSMUSE00000489385 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  protein_coding unknown ENSMUSE00000973560 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  protein_coding unknown ENSMUSE00000973560 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  protein_coding unknown ENSMUSE00001043622 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8  protein_coding unknown ENSMUSE00001043622 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  protein_coding unknown ENSMUSE00001013991 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 protein_coding unknown ENSMUSE00001013991 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11 protein_coding unknown ENSMUSE00000996531 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12 protein_coding unknown ENSMUSE00000996531 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13 protein_coding unknown ENSMUSE00000999790 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14 protein_coding unknown ENSMUSE00000999790 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15 protein_coding unknown ENSMUSE00000736922 ENSMUST00000144140</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16 protein_coding unknown ENSMUSE00000963884 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17 protein_coding unknown ENSMUSE00000963884 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18 protein_coding unknown ENSMUSE00001063063 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19 protein_coding unknown ENSMUSE00001063063 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20 protein_coding unknown ENSMUSE00000748299 ENSMUST00000144140</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21 protein_coding unknown ENSMUSE00000496705 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22 protein_coding unknown ENSMUSE00000498616 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23 protein_coding unknown ENSMUSE00000750625 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24           utr3 unknown ENSMUSE00000750625 ENSMUST00000005477</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25           utr3 unknown ENSMUSE00000498616 ENSMUST00000072438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  gene  symbol density</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25 ENSMUSG00000030483 Cyp2b10       1</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">grTrack</span>, <span class="st">"UCSCData"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> UCSC track 'foo'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> UCSCData object with 3 ranges and 6 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames            ranges strand |                 id        name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr7 25897676-25926559      + | ENSMUST00000005477     Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr7 25897620-25926624      + | ENSMUST00000072438     Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr7 25916766-25917430      + | ENSMUST00000144140     Cyp2b10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           itemRgb blockCount             blockSizes            blockStarts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;character&gt;  &lt;numeric&gt;            &lt;character&gt;            &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     #FFD58A         12 9,171,163,150,161,17.. 0,9,13562,13878,1618..</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     #FFD58A         11 65,171,163,150,161,1.. 0,65,13618,13934,162..</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     #FFD58A          2                356,143                  0,522</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## HTML image map</span></span></span>
<span class="r-in"><span><span class="fu"><a href="ImageMap-class.html">coords</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span><span class="fu"><a href="ImageMap-class.html">tags</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span><span class="va">grTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span><span class="op">$</span><span class="va">foo</span></span></span>
<span class="r-plt img"><img src="GeneRegionTrack-class-7.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="ImageMap-class.html">coords</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             x1     y1        x2     y2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560   239.54551  64.50 241.93921 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001043622   244.21469  64.50 246.41630 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001013991   278.36174  64.50 280.72588 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531   291.40884  64.50 294.00939 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000999790   300.92450  64.50 303.00790 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000963884   323.70890  64.50 326.47199 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063   328.93956  64.50 331.02296 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000498616   458.67175  64.50 461.34619 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000498616.1 461.36096 152.25 466.88714 327.75</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 239.54551  64.50 241.93921 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001043622.1 244.21469  64.50 246.41630 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001013991.1 278.36174  64.50 280.72588 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 291.40884  64.50 294.00939 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000999790.1 300.92450  64.50 303.00790 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000963884.1 323.70890  64.50 326.47199 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1 328.93956  64.50 331.02296 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000496705   448.09222  64.50 449.08221 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625   458.67175  64.50 461.34619 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1 461.36096 152.25 465.92671 327.75</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000736922   321.22656  64.50 326.47199 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000748299   328.93956  64.50 331.03774 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000742021    38.29809  64.50  41.79997 415.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000489385    38.71181  64.50  41.79997 415.50</span>
<span class="r-in"><span><span class="fu"><a href="ImageMap-class.html">tags</a></span><span class="op">(</span><span class="va">grTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $fill</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "#FFD58A"            "#FFD58A"            "#FFD58A" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "#FFD58A"            "#FFD58A"            "#FFD58A" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "#FFD58A"            "#FFD58A"            "#FFD58A" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "#FFD58A"            "#FFD58A"            "#FFD58A" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "#FFD58A"            "#FFD58A"            "#FFD58A" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "#FFD58A"            "#FFD58A"            "#FFD58A" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "#FFD58A"            "#FFD58A"            "#FFD58A" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "#FFD58A"            "#FFD58A" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "+"                  "+"                  "+" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "+"                  "+"                  "+" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "+"                  "+"                  "+" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "+"                  "+"                  "+" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "+"                  "+"                  "+" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "+"                  "+"                  "+" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "+"                  "+"                  "+" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "+"                  "+" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $text</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00001063063" "ENSMUSE00000498616" "ENSMUSE00000498616" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00001063063" "ENSMUSE00000496705" "ENSMUSE00000750625" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000742021" "ENSMUSE00000489385" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $start</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25911238"           "25911554"           "25913865" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25914748"           "25915392"           "25916934" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25917288"           "25926068"           "25926250" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25911238"           "25911554"           "25913865" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25914748"           "25915392"           "25916934" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25917288"           "25925352"           "25926068" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25926250"           "25916766"           "25917288" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25897618"           "25897646" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $end</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25911400"           "25911703"           "25914025" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25914924"           "25915533"           "25917121" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25917429"           "25926249"           "25926624" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25911400"           "25911703"           "25914025" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25914924"           "25915533"           "25917121" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25917429"           "25925419"           "25926249" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25926559"           "25917121"           "25917430" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "25897855"           "25897855" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $feature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     "protein_coding"     "protein_coding"     "protein_coding" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     "protein_coding"     "protein_coding"     "protein_coding" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     "protein_coding"     "protein_coding"               "utr3" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     "protein_coding"     "protein_coding"     "protein_coding" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     "protein_coding"     "protein_coding"     "protein_coding" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     "protein_coding"     "protein_coding"     "protein_coding" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               "utr3"     "protein_coding"     "protein_coding" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          "composite"          "composite" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "unknown"            "unknown"            "unknown" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "unknown"            "unknown"            "unknown" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "unknown"            "unknown"            "unknown" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "unknown"            "unknown"            "unknown" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "unknown"            "unknown"            "unknown" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "unknown"            "unknown"            "unknown" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "unknown"            "unknown"            "unknown" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "unknown"            "unknown" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $exon</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00001063063" "ENSMUSE00000498616" "ENSMUSE00000498616" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00001063063" "ENSMUSE00000496705" "ENSMUSE00000750625" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000742021" "ENSMUSE00000489385" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $transcript</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUST00000005477" "ENSMUST00000144140" "ENSMUST00000144140" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUST00000072438" "ENSMUST00000005477" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSG00000030483" "ENSMUSG00000030483" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $symbol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            "Cyp2b10"            "Cyp2b10" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $density</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "1"                  "1"                  "1" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "1"                  "1"                  "1" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "1"                  "1"                  "1" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "1"                  "1"                  "1" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "1"                  "1"                  "1" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "1"                  "1"                  "1" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "1"                  "1"                  "1" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "2"                  "2" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $gdensity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   NA                   NA                   NA </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   NA                   NA                   NA </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   NA                   NA                   NA </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   NA                   NA                   NA </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   NA                   NA                   NA </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   NA                   NA                   NA </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   NA                   NA                   NA </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  "1"                  "1" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $exonId</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00001063063" "ENSMUSE00000498616" "ENSMUSE00000498616" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00001063063" "ENSMUSE00000496705" "ENSMUSE00000750625" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000742021" "ENSMUSE00000489385" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $origExonId</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00001063063" "ENSMUSE00000498616" "ENSMUSE00000498616" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00001063063" "ENSMUSE00000496705" "ENSMUSE00000750625" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "ENSMUSE00000742021" "ENSMUSE00000489385" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $col</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "darkgray"           "darkgray"           "darkgray" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "darkgray"           "darkgray"           "darkgray" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "darkgray"           "darkgray"           "darkgray" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "darkgray"           "darkgray"           "darkgray" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "darkgray"           "darkgray"           "darkgray" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "darkgray"           "darkgray"           "darkgray" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "darkgray"           "darkgray"           "darkgray" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSMUSE00000742021   ENSMUSE00000489385 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           "darkgray"           "darkgray" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $title</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [8] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [15] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [22] "Cyp2b10" "Cyp2b10"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Florian Hahne, Steffen Durinck, Robert Ivanek, Arne Mueller, Steve Lianoglou, Ge Tan, Lance Parsons, Shraddha Pai.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

