<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>GeneRegionTrack class and methods — GeneRegionTrack-class • Gviz</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="GeneRegionTrack class and methods — GeneRegionTrack-class" />
<meta property="og:description" content="

A class to hold gene model data for a genomic region." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Gviz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.35.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/Gviz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ivanek/Gviz/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>GeneRegionTrack class and methods</h1>
    
    <div class="hidden name"><code>GeneRegionTrack-class.Rd</code></div>
    </div>

    <div class="ref-description">
    

<p>A class to hold gene model data for a genomic region.</p>
    </div>

    <pre class="usage"><span class='fu'>GeneRegionTrack</span><span class='op'>(</span>range<span class='op'>=</span><span class='cn'>NULL</span>, rstarts<span class='op'>=</span><span class='cn'>NULL</span>, rends<span class='op'>=</span><span class='cn'>NULL</span>, rwidths<span class='op'>=</span><span class='cn'>NULL</span>,
               <span class='va'>strand</span>, <span class='va'>feature</span>, <span class='va'>exon</span>, <span class='va'>transcript</span>, <span class='va'>gene</span>, <span class='va'>symbol</span>,
               <span class='va'>chromosome</span>, <span class='va'>genome</span>, stacking<span class='op'>=</span><span class='st'>"squish"</span>,
               name<span class='op'>=</span><span class='st'>"GeneRegionTrack"</span>, start<span class='op'>=</span><span class='cn'>NULL</span>, end<span class='op'>=</span><span class='cn'>NULL</span>,
               <span class='va'>importFunction</span>, stream<span class='op'>=</span><span class='cn'>FALSE</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>range</th>
      <td>

<p>An optional meta argument to handle the different input types. If
    the <code>range</code> argument is missing, all the relevant information
    to create the object has to be provided as individual function
    arguments (see below).</p>
<p>The different input options for <code>range</code> are:</p>
<p></p><dl>

      <dt></dt><dd><p>A <code>TxDb</code> object: all the necessary gene model
    information including exon locations, transcript groupings and
    associated gene ids are contained in <code>TxDb</code>
    objects, and the coercion between the two is almost completely
    automated. If desired, the data to be fetched from the
    <code>TxDb</code> object can be restricted using the
    constructor's <code>chromosome</code>, <code>start</code> and <code>end</code>
    arguments. See below for details. A direct coercion method
    <code>as(obj, "GeneRegionTrack")</code> is also available. A nice
    added benefit of this input option is that the UTR and coding
    region information that is part of the original
    <code>TxDb</code> object is retained in the
    <code>GeneRegionTrack</code>.</p></dd>

      <dt></dt><dd><p>A <code>GRanges</code> object: the genomic ranges for the
    <code>GeneRegion</code> track as well as the optional additional
    metadata columns <code>feature</code>,
    <code>transcript</code>, <code>gene</code>, <code>exon</code> and <code>symbol</code>
    (see description of the individual function parameters below for
    details). Calling the constructor on a <code>GRanges</code> object
    without further arguments, e.g.
    <code>GeneRegionTrack(range=obj)</code> is equivalent to calling the
    coerce method <code>as(obj, "GeneRegionTrack")</code>.</p></dd>

      <dt></dt><dd><p>A <code>GRangesList</code> object: this is very similar to the
    previous case, except that the grouping information that is part
    of the list structure is preserved in the
    <code>GeneRegionTrack</code>. I.e., all the elements within one list
    item receive the same group id. For consistancy, there is also a
    coercion method from <code>GRangesLists</code> <code>as(obj,
    "GeneRegionTrack")</code>. Please note that unless the necessary
    information about gene ids, symbols, etc. is present in the
    individual <code>GRanges</code> meta data slots, the object will not
    be particularly useful, because all the identifiers will be set
    to a common default value.</p></dd>

      <dt></dt><dd><p>An <code>IRanges</code> object: almost identical
    to the <code>GRanges</code> case, except that the chromosome and
    strand information as well as all additional data has to be
    provided in the separate <code>chromosome</code>, <code>strand</code>,
    <code>feature</code>, <code>transcript</code>, <code>symbol</code>, <code>exon</code> or
    <code>gene</code> arguments, because it can not be directly encoded in
    an <code>IRanges</code> object. Note that only the former two are
    mandatory (if not provided explicitely the more or less
    reasonable default values <code>chromosome=NA</code> and
    <code>strand=*</code> are used, but not providing information about
    the gene-to-transcript relationship or the human-readble symbols
    renders a lot of the class' functionality useles.</p></dd>

      <dt></dt><dd><p>A <code>data.frame</code> object: the <code>data.frame</code> needs to
    contain at least the two mandatory columns <code>start</code> and
    <code>end</code> with the range coordinates. It may also contain a
    <code>chromosome</code> and a <code>strand</code> column with the chromosome
    and strand information for each range. If missing, this
    information will be drawn from the constructor's
    <code>chromosome</code> or <code>strand</code> arguments. In addition, the
    <code>feature</code>, <code>exon</code>, <code>transcript</code>, <code>gene</code> and
    <code>symbol</code> data can be provided as columns in the
    <code>data.frame</code>. The above comments about potential default
    values also apply here.</p></dd>

      <dt></dt><dd><p>A <code>character</code> scalar: in this case the value of the
    <code>range</code> argument is considered to be a file path to an
    annotation file on disk. A range of file types are supported by
    the <code>Gviz</code> package as identified by the file extension. See
    the <code>importFunction</code> documentation below for further
    details.</p></dd>

    
</dl><p></p></td>
    </tr>
    <tr>
      <th>start, end</th>
      <td><p>An integer scalar with the genomic start or end
    coordinate for the gene model range. If those are missing, the
    default value will automatically be the smallest (or largest) value,
    respectively in <code>rstarts</code> and <code>rends</code> for the currently
    active chromosome. When building a <code>GeneRegionTrack</code> from a
    <code>TxDb</code> object, these arguments can be used to subset
    the desired annotation data by genomic coordinates. Please note this
    in that case the <code>chromosome</code> parameter must also be set.</p></td>
    </tr>
    <tr>
      <th>rstarts</th>
      <td><p>An integer vector of the start coordinates for the
    actual gene model items, i.e., for the individual exons. The
    relationship between exons is handled via the <code>gene</code> and
    <code>transcript</code> factors. Alternatively, this can be a vector of
    comma-separated lists of integer coordinates, one vector item for
    each transcript, and each comma-separated element being the start
    location of a single exon within that transcript. Those lists will
    be exploded upon object instantiation and all other annotation
    arguments will be recycled accordingly to regenerate the
    exon/transcript/gene relationship structure. This implies the
    approriate number of items in all annotation and coordinates
    arguments.</p></td>
    </tr>
    <tr>
      <th>rends</th>
      <td><p>An integer vector of the end coordinates for the actual
    gene model items. Both <code>rstarts</code> and <code>rends</code> have to be of
    equal length.</p></td>
    </tr>
    <tr>
      <th>rwidths</th>
      <td><p>An integer vector of widths for the actual gene model
    items. This can be used instead of either <code>rstarts</code> or
    <code>rends</code> to specify the range coordinates.</p></td>
    </tr>
    <tr>
      <th>feature</th>
      <td><p>Factor (or other vector that can be coerced into one),
    giving the feature types for the individual track exons.  When
    plotting the track to the device, if a display parameter with the
    same name as the value of <code>feature</code> is set, this will be used
    as the track item's fill color. Additionally, the feature type
    defines whether an element in the <code>GeneRegionTrack</code> is
    considered to be coding or non-coding. The details section as well
    as the section about the <code>thinBoxFeature</code> display parameter
    further below has more information on this. See also
    <code><a href='grouping.html'>grouping</a></code> for details.</p></td>
    </tr>
    <tr>
      <th>exon</th>
      <td><p>Character vector of exon identifiers. It's values will be
    used as the identifier tag when plotting to the device if the
    display parameter <code>showExonId=TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>strand</th>
      <td><p>Character vector, the strand information for the
    individual track exons. It may be provided in the form <code><a href='https://rdrr.io/r/base/Arithmetic.html'>+</a></code> for
    the Watson strand, <code><a href='https://rdrr.io/r/base/Arithmetic.html'>-</a></code> for the Crick strand or <code><a href='https://rdrr.io/r/base/Arithmetic.html'>*</a></code> for
    either one of the two. Please note that all items within a single
    gene or transcript model need to be on the same strand, and
    erroneous entries will result in casting of an error.</p></td>
    </tr>
    <tr>
      <th>transcript</th>
      <td><p>Factor (or other vector that can be coerced into
    one), giving the transcript memberships for the individual track
    exons. All items with the same transcript identifier will be
    visually connected when plotting to the device.  See
    <code><a href='grouping.html'>grouping</a></code> for details. Will be used as labels when
    <code>showId=TRUE</code>, and <code>geneSymbol=FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>gene</th>
      <td><p>Factor (or other vector that can be coerced into one),
    giving the gene memberships for the individual track exons.</p></td>
    </tr>
    <tr>
      <th>symbol</th>
      <td><p>A factor with human-readable gene name aliases which
    will be used as labels when <code>showId=TRUE</code>, and
    <code>geneSymbol=TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>chromosome</th>
      <td><p>The chromosome on which the track's genomic ranges
    are defined. A valid UCSC chromosome identifier if
    <code><a href='https://rdrr.io/r/base/options.html'>options(ucscChromosomeNames=TRUE)</a></code>. Please note that in this
    case only syntactic checking takes place, i.e., the argument value
    needs to be an integer, numeric character or a character of the form
    <code>chrx</code>, where <code>x</code> may be any possible string. The user has
    to make sure that the respective chromosome is indeed defined for
    the the track's genome. If not provided here, the constructor will
    try to build the chromosome information based on the available
    inputs, and as a last resort will fall back to the value
    <code>chrNA</code>. Please note that by definition all objects in the
    <code>Gviz</code> package can only have a single active chromosome at a
    time (although internally the information for more than one
    chromosome may be present), and the user has to call the
    <code>chromosome&lt;-</code> replacement method in order to change to a
    different active chromosome. When creating a <code>GeneRegionTrack</code>
    from a <code>TxDb</code> object, the value of this parameter can
    be used to subset the data to fetch only transcripts from a single
    chromosome.</p></td>
    </tr>
    <tr>
      <th>genome</th>
      <td><p>The genome on which the track's ranges are
    defined. Usually this is a valid UCSC genome identifier, however
    this is not being formally checked at this point. If not provided
    here the constructor will try to extract this information from the
    provided inputs, and eventually will fall back to the default value
    of <code>NA</code>.</p></td>
    </tr>
    <tr>
      <th>stacking</th>
      <td><p>The stacking type for overlapping items of the
    track. One in <code><a href='https://rdrr.io/r/base/c.html'>c(hide, dense, squish, pack,full)</a></code>. Currently,
    only hide (don't show the track items, squish (make best use of the
    available space) and dense (no stacking at all) are implemented.</p></td>
    </tr>
    <tr>
      <th>name</th>
      <td><p>Character scalar of the track's name used in the title
    panel when plotting.</p></td>
    </tr>
    <tr>
      <th>importFunction</th>
      <td><p>A user-defined function to be used to import the
    data from a file. This only applies when the <code>range</code> argument
    is a character string with the path to the input data file. The
    function needs to accept an argument <code>x</code> containing the file
    path and has to return a proper <code>GRanges</code> object with all the
    necessary metadata columns set. A set of default
    import functions is already implemented in the package for a number
    of different file types, and one of these defaults will be picked
    automatically based on the extension of the input file name. If the
    extension can not be mapped to any of the existing import function,
    an error is raised asking for a user-defined import function via
    this argument. Currently the following file types can be imported
    with the default functions: <code>gff</code>, <code>gff1</code>, <code>gff2</code>,
    <code>gff3</code>, <code>gtf</code>.</p></td>
    </tr>
    <tr>
      <th>stream</th>
      <td><p>A logical flag indicating that the user-provided import
    function can deal with indexed files and knows how to process the
    additional <code>selection</code> argument when accessing the data on
    disk. This causes the constructor to return a
    <code>ReferenceGeneRegionTrack</code> object which will grab the necessary data
    on the fly during each plotting operation.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional items which will all be interpreted as further
    display parameters. See <code><a href='settings.html'>settings</a></code> and the "Display
    Parameters" section below for details.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="objects-from-the-class"><a class="anchor" href="#objects-from-the-class"></a>Objects from the class</h2>

    

<p>Objects can be created using the constructor function
  <code>GeneRegionTrack</code>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    

<p>A track containing all gene models in a particular region. The data
  are usually fetched dynamially from an online data store, but it is
  also possible to manully construct objects from local
  data. Connections to particular online data sources should be
  implemented as sub-classes, and <code>GeneRegionTrack</code> is just the
  commone denominator that is being used for plotting later on. There
  are several levels of data associated to a <code>GeneRegionTrack</code>:</p>
<p></p><dl>

   <dt>exon level:</dt><dd><p>identifiers are stored in the exon column of the
     <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code> object in the <code>range</code> slot. Data
     may be extracted using the <code>exon</code> method.</p></dd>

    <dt>transcript level:</dt><dd><p>identifiers are stored in the transcript
      column of the <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code> object. Data may be
      extracted using the <code>transcript</code> method.</p></dd>

    <dt>gene level:</dt><dd><p>identifiers are stored in the gene column of the
      <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code> object, more human-readable versions
      in the symbol column. Data may be extracted using the <code>gene</code>
      or the <code>symbol</code> methods.</p></dd>

    <dt>transcript-type level:</dt><dd><p>information is stored in the feature
      column of the <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code> object. If a display
      parameter of the same name is specified, the software will use its
      value for the coloring.</p></dd>

  
</dl>

<p><code>GeneRegionTrack</code> objects also know about coding regions and
  non-coding regions (e.g., UTRs) in a transcript, and will indicate
  those by using different shapes (wide boxes for all coding regions,
  thinner boxes for non-coding regions). This is archived by setting the
  <code>feature</code> values of the object for non-coding elements to one of
  the options that are provided in the <code>thinBoxFeature</code> display
  parameters. All other elements are considered to be coding elements.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    

<p>The return value of the constructor function is a new object of class
  <code>GeneRegionTrack</code>.</p>
    <h2 class="hasAnchor" id="slots"><a class="anchor" href="#slots"></a>Slots</h2>

    

<p></p><dl>

    <dt><code>start</code>:</dt><dd><p>Object of class <code>"numeric"</code>, the start
      coordinates of the annotation range. The coorrdinates for the
      individual gene model items are stored in the <code>range</code> slot.</p></dd>

    <dt><code>end</code>:</dt><dd><p>Object of class <code>"numeric"</code>, the end
      coordinates of the annotation range. The corrdinates for the
      individual gene model items are stored in the <code>range</code> slot.</p></dd>

    <dt><code>stacking</code>:</dt><dd><p>Object of class <code>"character"</code>,
      inherited from class <code><a href='StackedTrack-class.html'>StackedTrack</a></code></p></dd>

    <dt><code>stacks</code>:</dt><dd><p>Object of class <code>"numeric"</code>,
      inherited from class <code><a href='StackedTrack-class.html'>StackedTrack</a></code></p></dd>

    <dt><code>range</code>:</dt><dd><p>Object of class <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code>,
      inherited from class <code><a href='RangeTrack-class.html'>RangeTrack</a></code></p></dd>

    <dt><code>chromosome</code>:</dt><dd><p>Object of class <code>"character"</code>,
      inherited from class <code><a href='RangeTrack-class.html'>RangeTrack</a></code></p></dd>

    <dt><code>genome</code>:</dt><dd><p>Object of class <code>"character"</code>, inherited
      from class <code><a href='RangeTrack-class.html'>RangeTrack</a></code></p></dd>

    <dt><code>dp</code>:</dt><dd><p>Object of class
      <code><a href='DisplayPars-class.html'>DisplayPars</a></code>, inherited from class
      <code><a href='GdObject-class.html'>GdObject</a></code></p></dd>

    <dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code>, inherited
      from class <code><a href='GdObject-class.html'>GdObject</a></code></p></dd>

     <dt><code>imageMap</code>:</dt><dd><p>Object of class
      <code><a href='ImageMap-class.html'>ImageMap</a></code>, inherited from class
      <code><a href='GdObject-class.html'>GdObject</a></code></p></dd>

  
</dl>

    <h2 class="hasAnchor" id="extends"><a class="anchor" href="#extends"></a>Extends</h2>

    

<p>Class <code>"<a href='AnnotationTrack-class.html'>AnnotationTrack</a>"</code>, directly.</p>
<p>Class <code>"<a href='StackedTrack-class.html'>StackedTrack</a>"</code>, by class "AnnotationTrack",
  distance2.</p>
<p>Class <code>"<a href='RangeTrack-class.html'>RangeTrack</a>"</code>, by class "AnnotationTrack",
  distance3.</p>
<p>Class <code>"<a href='GdObject-class.html'>GdObject</a>"</code>, by class "AnnotationTrack",
  distance4.</p>
    <h2 class="hasAnchor" id="methods"><a class="anchor" href="#methods"></a>Methods</h2>

    

<p>In the following code chunks, <code>obj</code> is considered to be an object of class <code>GeneRegionTrack</code>.</p>
<p><b><em>Exported in the name space:</em></b></p>
<p></p><dl>

     <dt>group</dt><dd><p><code>signature(gdObject="GeneRegionTrack")</code>: extract
      the group membership for all track items.      <em>Usage:</em>      <code><a href='AnnotationTrack-class.html'>group(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='AnnotationTrack-class.html'>group(obj)</a></code></p></dd></dl></p></dd>

    <dt>group&lt;-</dt><dd><p><code>signature(gdObject="GeneRegionTrack",
      value="character")</code>: replace the grouping information for track
      items. The replacement value must be a factor of appropriate
      length or another vector that can be coerced into such.      <em>Usage:</em>      <code>group&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>group(obj) &lt;- c("a", "a", "b", "c", "a")</code></p></dd></dl></p></dd>

    <dt>identifier</dt><dd><p><code>signature(gdObject="GeneRegionTrack")</code>:
      return track item identifiers. Depending on the setting of the
      optional argument <code>lowest</code>, these are either the group
      identifiers or the individual item identifiers.      <em>Usage:</em>      <code><a href='AnnotationTrack-class.html'>identifier(GdObject, lowest=FALSE)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>lowest</code>: return the lowest-level identifier, i.e.,
    the item IDs, or the higher level group IDs which do not have to
    be unqiue.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='AnnotationTrack-class.html'>identifier(obj, lowest=FALSE)</a></code></p></dd></dl></p></dd>

    <dt>identifier&lt;-</dt><dd><p><code>signature(gdObject="GeneRegionTrack",
    value="character")</code>: Set the track item identifiers. The
      replacement value has to be a character vector of appropriate
      length. This always replaces the group-level identifiers, so
      essentially it is similar to <code>groups&lt;-</code>.      <em>Usage:</em>      <code>identifier&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>identifier(obj) &lt;- c("foo", "bar")</code></p></dd></dl></p></dd>

    <dt>exon</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>: Extract
      the exon identifiers for all exons in the gene models.      <em>Usage:</em>      <code>exon(GdObject)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>exon(obj)</code></p></dd></dl></p></dd>

    <dt>exon&lt;-</dt><dd><p><code>signature(GdObject="GeneRegionTrack",
      value="character")</code>: replace the exon identifiers for all exons in
      the gene model.  The replacement value must be a character of
      appropriate length or another vector that can be coerced into
      such.      <em>Usage:</em>      <code>exon&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>exon(obj) &lt;- paste("Exon", 1:5)</code></p></dd></dl></p></dd>

    <dt>gene</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>: Extract
      the gene identifiers for all gene models.      <em>Usage:</em>      <code>gene(GdObject)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>gene(obj)</code></p></dd></dl></p></dd>

    <dt>gene&lt;-</dt><dd><p><code>signature(GdObject="GeneRegionTrack",
    value="character")</code>: replace the gene identifiers for all gene
      models.  The replacement value must be a character of
      appropriate length or another vector that can be coerced into
      such.      <em>Usage:</em>      <code>gene&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>gene(obj) &lt;- paste("Gene", LETTERS[1:5])</code></p></dd></dl></p></dd>

    <dt>symbol</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>: Extract
      the human-readble gene symbol for all gene models.      <em>Usage:</em>      <code>symbol(GdObject)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>symbol(obj)</code></p></dd></dl></p></dd>

    <dt>symbol&lt;-</dt><dd><p><code>signature(GdObject="GeneRegionTrack",
    value="character")</code>: replace the human-readable gene symbol for
      all gene models.  The replacement value must be a character of
      appropriate length or another vector that can be coerced into
      such.      <em>Usage:</em>      <code>gene&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>symbol(obj) &lt;- letters[1:5]</code></p></dd></dl></p></dd>

    <dt>transcript</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>:
      Extract the transcript identifiers for all transcripts in the
      gene models.      <em>Usage:</em>      <code>transcript(GdObject)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>transcript(obj)</code></p></dd></dl></p></dd>

    <dt>transcript&lt;-</dt><dd><p><code>signature(GdObject="GeneRegionTrack",
    value="character")</code>: replace the transcript identifiers for
      all transcripts in the gene model.  The replacement value must
      be a character of appropriate length or another vector that
      can be coerced into such.      <em>Usage:</em>      <code>transcript&lt;-(GdObject, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>transcript(obj) &lt;- paste("Exon", 1:5)</code></p></dd></dl></p></dd>
  
</dl>

<p><em>Internal methods:</em></p>
<p></p><dl>

    <dt>coerce</dt><dd><p><code>signature(from="GeneRegionTrack",
    to="UCSCData")</code>: coerce to a <code>UCSCData</code> object for export to
      the UCSC genome browser.       <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>as(obj, "UCSCData")</code></p></dd></dl></p></dd>

    <dt>collapseTrack</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>:
      preprocess the track before plotting. This will collapse
      overlapping track items based on the available resolution and
      increase the width and height of all track objects to a minimum
      value to avoid rendering issues. See <code><a href='collapsing.html'>collapsing</a></code> for
      details.      <em>Usage:</em>      <code>collapseTrack(GdObject, diff=.pxResolution(coord="x"))</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>diff</code>: the minimum pixel width to display,
      everything below that will be inflated to a width of
      <code>diff</code>.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>Gviz:::collapseTrack(obj)</code></p></dd></dl></p></dd>

    <dt>initialize</dt><dd><p><code>signature(.Object="GeneRegionTrack")</code>:
      initialize the object</p></dd>

    <dt>show</dt><dd><p><code>signature(object="GeneRegionTrack")</code>: show a
      human-readable summary of the object</p></dd>

  
</dl>

<p><b><em>Inherited methods:</em></b></p>
<p></p><dl>

     <dt>drawGD</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>: plot the
      object to a graphics device. The return value of this method is
      the input object, potentially updated during the plotting
      operation. Internally, there are two modes in which the method can
      be called. Either in 'prepare' mode, in which case no plotting is
      done but the object is preprocessed based on the
      available space, or in 'plotting' mode, in which case the actual
      graphical output is created. Since subsetting of the object can be
      potentially costly, this can be switched off in case subsetting
      has already been performed before or is not necessary.      <em>Usage:</em>      <code><a href='GdObject-class.html'>drawGD(GdObject, minBase, maxBase, prepare=FALSE,
    subset=TRUE, ...)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>minBase</code>, <code>maxBase</code>: the coordinate range to
      plot.</p></dd>    <dt></dt><dd><p><code>prepare</code>: run method in preparation or in
      production mode.</p></dd>    <dt></dt><dd><p><code>subset</code>: subset the object to the visible region
    or skip the potentially expensive subsetting operation.</p></dd>    <dt></dt><dd><p><code>...</code>: all further arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>Gviz:::drawGD(obj)</code></p></dd>    <dt></dt><dd><p><code>Gviz:::drawGD(obj, minBase=1, maxBase=100)</code></p></dd>    <dt></dt><dd><p><code>Gviz:::drawGD(obj, prepare=TRUE,
        subset=FALSE)</code></p></dd></dl></p></dd>

    <dt>drawGrid</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>: superpose a grid on top of a track.      <em>Usage:</em>      <code>drawGrid(GdObject, from, to)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>from</code>, <code>to</code>: integer scalars, draw grid
      within a certain coordinates range. This needs to be supplied
      for the plotting function to know the current genomic
      coordinates.</p></dd></dl>      <em>Examples:</em>      <dl>          <dt></dt><dd><p><code>Gviz:::drawGrid(obj, from=10, to=100)</code></p></dd></dl></p></dd>

    <dt>setStacks</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>:
      recompute the stacks based on the available space and on the
      object's track items and stacking settings.      <em>Usage:</em>      <code>setStacks(GdObject, from, to)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>from</code>, <code>to</code>: integer scalars, compute
      stacking within a certain coordinates range. This needs to be
      supplied for the plotting function to know the current genomic
      coordinates.</p></dd></dl>      <em>Examples:</em>      <dl>          <dt></dt><dd><p><code>Gviz:::setStacks(obj, from=1, to=100)</code></p></dd></dl></p></dd>

    <dt>stacking</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>: return
      the current stacking type.      <em>Usage:</em>      <code><a href='StackedTrack-class.html'>stacking(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='StackedTrack-class.html'>stacking(obj)</a></code></p></dd></dl></p></dd>

    <dt>stacking&lt;-</dt><dd><p><code>signature(GdObject="GeneRegionTrack",
      value="character")</code>: set the object's stacking type to one in
      <code><a href='https://rdrr.io/r/base/c.html'>c(hide, dense, squish, pack,full)</a></code>.      <em>Usage:</em>      <code>stacking&lt;-(GdObject, value)</code>      <em>Additional Arguments:</em>    <dl>      <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>stacking(obj) &lt;- "squish" </code></p></dd></dl></p></dd>

    <dt>stacks</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>: return
      the stack indices for each track item.      <em>Usage:</em>      <code><a href='StackedTrack-class.html'>stacks(GdObject)</a></code>      <em>Examples:</em>      <dl>          <dt></dt><dd><p><code>Gviz:::stacks(obj)</code></p></dd></dl></p></dd>

    <dt>[</dt><dd><p><code>signature(x="GeneRegionTrack", i="ANY", j="ANY",
    drop="ANY")</code>: subset the items in the <code>GeneRegionTrack</code>
      object. This is essentially similar to subsetting of the
      <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code> object in the <code>range</code>
      slot. For most applications, the <code>subset</code> method may be
      more appropriate.      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>i</code>, <code>j</code>: subsetting indices, <code>j</code> is
      ignored.</p></dd>    <dt></dt><dd><p><code>drop</code>: argument is ignored.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>obj[1:5]</code></p></dd></dl></p></dd>


    <dt>chromosome</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>:
      return the currently active chromosome for which the track is
       defined. For consistancy with other Bioconductor packages, the
       <code>isActiveSeq</code> alias is also provided.      <em>Usage:</em>      <code><a href='RangeTrack-class.html'>chromosome(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='RangeTrack-class.html'>chromosome(obj)</a></code></p></dd></dl></p></dd>

    <dt>chromosome&lt;-</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>:
      replace the value of the track's active chromosome. This has to
       be a valid UCSC chromosome identifier or an integer or character
       scalar that can be reasonably coerced into one, unless
       <code><a href='https://rdrr.io/r/base/options.html'>options(ucscChromosomeNames=FALSE)</a></code>. For consistancy with
       other Bioconductor packages, the <code>isActiveSeq&lt;-</code> alias is
       also provided.      <em>Usage:</em>      <code>chromosome&lt;-(GdObject, value)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>chromosome(obj) &lt;- "chr12"</code></p></dd></dl></p></dd>

    <dt>start, end, width</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: the
      start or end coordinates of the track items, or their width in
      genomic coordinates.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/stats/start.html'>start(x)</a></code>      <code><a href='https://rdrr.io/r/stats/start.html'>end(x)</a></code>      <code>width(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/stats/start.html'>start(obj)</a></code></p></dd>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/stats/start.html'>end(obj)</a></code></p></dd>    <dt></dt><dd><p><code>width(obj)</code></p></dd></dl></p></dd>

    <dt>start&lt;-, end&lt;-, width&lt;-</dt><dd><p><code>signature(x="GeneRegionTrack")</code>:
      replace the start or end coordinates of the track items, or their
      width.      <em>Usage:</em>      <code>start&lt;-(x, value)</code>      <code>end&lt;-(x, value)</code>      <code>width&lt;-(x, value)</code>      <em>Additional Arguments:</em>    <dl>      <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>start(obj) &lt;- 1:10</code></p></dd>    <dt></dt><dd><p><code>end(obj) &lt;- 20:30</code></p></dd>    <dt></dt><dd><p><code>width(obj) &lt;- 1</code></p></dd></dl></p></dd>

    <dt>position</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>: the
      arithmetic mean of the track item's coordionates, i.e.,
      <code>(end(obj)-start(obj))/2</code>.      <em>Usage:</em>      <code><a href='RangeTrack-class.html'>position(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='RangeTrack-class.html'>position(obj)</a></code></p></dd></dl></p></dd>

    <dt>feature</dt><dd><p><code>signature(GdObject="GeneRegionTrack")</code>: return the
      grouping information for track items. For certain sub-classes,
      groups may be indicated by different color schemes when
      plotting. See <code><a href='grouping.html'>grouping</a></code> for details.      <em>Usage:</em>      <code><a href='RangeTrack-class.html'>feature(GdObject)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='RangeTrack-class.html'>feature(obj)</a></code></p></dd></dl></p></dd>

    <dt>feature&lt;-</dt><dd><p><code>signature(gdObject="GeneRegionTrack",
      value="character")</code>: set the grouping information for track
      items. This has to be a factor vector (or another type of vector
      that can be coerced into one) of the same length as the number of
      items in the <code>GeneRegionTrack</code>. See <code><a href='grouping.html'>grouping</a></code>
      for details.      <em>Usage:</em>      <code>feature&lt;-(GdObject, value)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>feature(obj) &lt;- c("a", "a", "b", "c", "a")</code></p></dd></dl></p></dd>

    <dt>genome</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: return the track's genome.      <em>Usage:</em>      <code>genome(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>genome(obj)</code></p></dd></dl></p></dd>

    <dt>genome&lt;-</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: set the track's
      genome. Usually this has to be a valid UCSC identifier, however
      this is not formally enforced here.      <em>Usage:</em>      <code>genome&lt;-(x, value)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>genome(obj) &lt;- "mm9"</code></p></dd></dl></p></dd>

    <dt>length</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: return the number
      of items in the track.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/length.html'>length(x)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/length.html'>length(obj)</a></code></p></dd></dl></p></dd>

    <dt>range</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: return the genomic
      coordinates for the track as an object of class
      <code>IRanges</code>.      <em>Usage:</em>      <code><a href='RangeTrack-class.html'>range(x)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='RangeTrack-class.html'>range(obj)</a></code></p></dd></dl></p></dd>

    <dt>ranges</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: return the genomic
      coordinates for the track along with all additional annotation
      information as an object of class <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code>.      <em>Usage:</em>      <code>ranges(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>ranges(obj)</code></p></dd></dl></p></dd>

    <dt>split</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: split a
      <code>GeneRegionTrack</code> object by an appropriate factor vector (or
      another vector that can be coerced into one). The output of this
      operation is a list of objects of the same class as the input
      object, all inheriting from class <code>GeneRegionTrack</code>.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/split.html'>split(x, f, ...)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>f</code>: the splitting factor.</p></dd>    <dt></dt><dd><p><code>...</code>: all further arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/split.html'>split(obj, c("a", "a", "b", "c", "a"))</a></code></p></dd></dl></p></dd>

    <dt>strand</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: return a vector of
      strand specifiers for all track items, in the form '+' for the
      Watson strand, '-' for the Crick strand or '*' for either of the
      two.      <em>Usage:</em>      <code>strand(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>strand(obj)</code></p></dd></dl></p></dd>

    <dt>strand&lt;-</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: replace the
      strand information for the track items. The replacement value
      needs to be an appropriate scalar or vector of strand values.      <em>Usage:</em>      <code>strand&lt;-(x, value)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>strand(obj) &lt;- "+"</code></p></dd></dl></p></dd>

    <dt>values</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: return all
      additional annotation information except for the genomic coordinates
      for the track items as a data.frame.      <em>Usage:</em>      <code>values(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>values(obj)</code></p></dd></dl></p></dd>

    <dt>coerce</dt><dd><p><code>signature(from="GeneRegionTrack",
    to="data.frame")</code>: coerce the <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code>
    object in the <code>range</code> slot into a regular data.frame.       <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>as(obj, "data.frame")</code></p></dd></dl></p></dd>

    <dt>subset</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: subset a
      <code>GeneRegionTrack</code> by coordinates and sort if necessary.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/subset.html'>subset(x, from, to, sort=FALSE, ...)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>from</code>, <code>to</code>: the coordinates range to subset
      to.</p></dd>    <dt></dt><dd><p><code>sort</code>: sort the object after subsetting. Usually
      not necessary.</p></dd>    <dt></dt><dd><p><code>...</code>: additional arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/subset.html'>subset(obj, from=10, to=20, sort=TRUE)</a></code></p></dd></dl></p></dd>

    <dt>displayPars</dt><dd><p><code>signature(x="GeneRegionTrack",
      name="character")</code>: list the value of the display parameter
      <code>name</code>. See <code><a href='settings.html'>settings</a></code> for details on display
      parameters and customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>displayPars(x, name)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>displayPars(obj, "col")</a></code></p></dd></dl></p></dd>


    <dt>displayPars</dt><dd><p><code>signature(x="GeneRegionTrack",
      name="missing")</code>: list the value of all available display
      parameters. See <code><a href='settings.html'>settings</a></code> for details on display
      parameters and customization.      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>displayPars(obj)</a></code></p></dd></dl></p></dd>

    <dt>getPar</dt><dd><p><code>signature(x="GeneRegionTrack", name="character")</code>:
      alias for the <code>displayPars</code> method. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>getPar(x, name)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>getPar(obj, "col")</a></code></p></dd></dl></p></dd>

    <dt>getPar</dt><dd><p><code>signature(x="GeneRegionTrack", name="missing")</code>:
      alias for the <code>displayPars</code> method. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>getPar(obj)</a></code></p></dd></dl></p></dd>

    <dt>displayPars&lt;-</dt><dd><p><code>signature(x="GeneRegionTrack",
      value="list")</code>: set display parameters using the values of the
      named list in <code>value</code>. See <code><a href='settings.html'>settings</a></code> for details
      on display parameters and customization.      <em>Usage:</em>      <code>displayPars&lt;-(x, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>displayPars(obj) &lt;- list(col="red", lwd=2)</code></p></dd></dl></p></dd>

    <dt>setPar</dt><dd><p><code>signature(x="GeneRegionTrack", value="character")</code>:
      set the single display parameter <code>name</code> to <code>value</code>. Note
      that display parameters in the <code>GeneRegionTrack</code> class are
      pass-by-reference, so no re-assignmnet to the symbol <code>obj</code> is
      necessary. See <code><a href='settings.html'>settings</a></code> for details on display
      parameters and customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>setPar(x, name, value)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>name</code>: the name of the display parameter to set.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>setPar(obj, "col", "red")</a></code></p></dd></dl></p></dd>

    <dt>setPar</dt><dd><p><code>signature(x="GeneRegionTrack", value="list")</code>: set
      display parameters by the values of the named list in
      <code>value</code>.  Note that display parameters in the
      <code>GeneRegionTrack</code> class are pass-by-reference, so no
      re-assignmnet to the symbol <code>obj</code> is necessary. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>setPar(obj, list(col="red", lwd=2))</a></code></p></dd></dl></p></dd>

    <dt>names</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: return the value of
      the <code>name</code> slot.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/names.html'>names(x)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/names.html'>names(obj)</a></code></p></dd></dl></p></dd>

    <dt>names&lt;-</dt><dd><p><code>signature(x="GeneRegionTrack", value="character")</code>:
      set the value of the <code>name</code> slot.      <em>Usage:</em>      <code>names&lt;-(x, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>names(obj) &lt;- "foo"</code></p></dd></dl></p></dd>

    <dt>coords</dt><dd><p><code>signature(ImageMap="GeneRegionTrack")</code>: return the
      coordinates from the internal image map.      <em>Usage:</em>      <code><a href='GdObject-class.html'>coords(ImageMap)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='GdObject-class.html'>coords(obj)</a></code></p></dd></dl></p></dd>

    <dt>tags</dt><dd><p><code>signature(x="GeneRegionTrack")</code>: return the tags from the
      internal image map.      <em>Usage:</em>      <code><a href='GdObject-class.html'>tags(x)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='GdObject-class.html'>tags(obj)</a></code></p></dd></dl></p></dd>
  
</dl>

    <h2 class="hasAnchor" id="display-parameters"><a class="anchor" href="#display-parameters"></a>Display Parameters</h2>

     

<p>The following display parameters are set for objects of class
  <code>GeneRegionTrack</code> upon instantiation, unless one or more of them
  have already been set by one of the optional sub-class initializers,
  which always get precedence over these global defaults. See
  <code><a href='settings.html'>settings</a></code> for details on setting graphical parameters
  for tracks.</p><dl>

    <dt></dt><dd><p><code>arrowHeadMaxWidth=20</code>: Numeric scalar. The maximum
      width of the arrow head in pixels if <code>shape</code> is <code>arrow</code>.</p></dd> 

    <dt></dt><dd><p><code>arrowHeadWidth=10</code>: Numeric scalar. The width of the
      arrow head in pixels if <code>shape</code> is <code>fixedArrow</code>.</p></dd> 

    <dt></dt><dd><p><code>col=NULL</code>: Character or integer scalar. The border
      color for all track items. Defaults to using the same color as
      in <code>fill</code>, also taking into account different track
      <code>features</code>.</p></dd> 

    <dt></dt><dd><p><code>collapseTranscripts=FALSE</code>: Logical or character
      scalar. Can be one in <code>gene</code>, <code>longest</code>, <code>shortest</code>
      or <code>meta</code>. Merge all transcripts of the same gene into one
      single gene model. In the case of <code>gene</code> (or <code>TRUE</code>),
      this will only keep the start location of the first exon and the
      end location of the last exon from all transcripts of the gene.
      For <code>shortest</code> and <code>longest</code>, only the longest or
      shortest transcript model is retained. For <code>meta</code>, a
      meta-transcript containing the union of all exons is formed
      (essentially identical to the operation <code><a href='https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html'>reduce(geneModel)</a></code>).</p></dd> 

    <dt></dt><dd><p><code>exonAnnotation=NULL</code>: Character scalar. Add annotation
      information to the individual exon models. This can be a value
      in <code>symbol</code>, <code>gene</code>, <code>transcript</code>, <code>exon</code> or
      <code>feature</code>. Defaults to <code>exon</code>. Only works if
      <code>showExonId</code> is not <code>FALSE</code>.</p></dd> 

    <dt></dt><dd><p><code>fill="orange"</code>: Character or integer scalar. The fill
      color for untyped items. This is also used to connect grouped
      items. See <code><a href='grouping.html'>grouping</a></code> for details.</p></dd> 

    <dt></dt><dd><p><code>min.distance=0</code>: Numeric scalar. The minimum pixel
      distance before collapsing range items, only if <code>collapse==TRUE</code>.
      See <code><a href='collapsing.html'>collapsing</a></code> for details. Note that a value larger
      than 0 may lead to UTR regions being merged to CDS regions, which
      in most cases is not particularly useful.</p></dd> 

    <dt></dt><dd><p><code>shape=c("smallArrow", "box")</code>: Character scalar. The
      shape in which to display the track items. Currently only
      <code>box</code>, <code>arrow</code>, <code>ellipse</code>, and <code>smallArrow</code>
      are implemented.</p></dd> 

    <dt></dt><dd><p><code>showExonId=NULL</code>: Logical scalar. Control whether to
      plot the individual exon identifiers.</p></dd> 

    <dt></dt><dd><p><code>thinBoxFeature=c("utr", "ncRNA", "utr3", "utr5",
      "3UTR", "5UTR", "miRNA", "lincRNA", "three_prime_UTR",
      "five_prime_UTR")</code>: Character vector. A listing of feature types
      that should be drawn with thin boxes. Typically those are non-coding
      elements.</p></dd> 

    <dt></dt><dd><p><code>transcriptAnnotation=NULL</code> <code>(Aliases:
      transcriptAnnotation)</code>: Character scalar. Add annotation information
      as transcript labels. This can be a value in <code>symbol</code>,
      <code>gene</code>, <code>transcript</code>, <code>exon</code> or <code>feature</code>.
      Defaults to <code>symbol</code>. Only works if <code>showId</code> is not
      <code>FALSE</code>.</p></dd> 

  
</dl> 

<p>Additional display parameters are being inherited from the respective
  parent classes. Note that not all of them may have an effect on the
  plotting of <code>GeneRegionTrack</code> objects.</p>
<p></p><dl>

    <dt></dt><dd><p><code><a href='AnnotationTrack-class.html'>AnnotationTrack</a></code>:       <dl>        <dt></dt><dd><p><code>cex=1</code>: Numeric scalar. The font expansion factor
          for item identifiers.</p></dd>         <dt></dt><dd><p><code>cex.group=0.6</code>: Numeric scalar. The font expansion
          factor for the group-level annotation.</p></dd>         <dt></dt><dd><p><code>col.line="darkgray"</code>: Character scalar. The color
          used for connecting lines between grouped items. Defaults to
          a light gray, but if set to <code>NULL</code> the same color as
          for the first item in the group is used.</p></dd>         <dt></dt><dd><p><code>featureAnnotation=NULL</code>: Character scalar. Add
          annotation information to the individual track elements. This
          can be a value in <code>id</code>, <code>group</code> or <code>feature</code>.
          Defaults to <code>id</code>. Only works if <code>showFeatureId</code> is
          not <code>FALSE</code>.</p></dd>         <dt></dt><dd><p><code>fontcolor.group="#808080"</code> <code>(Aliases:
          fontcolor.group)</code>: Character or integer scalar. The font
          color for the group-level annotation.</p></dd>         <dt></dt><dd><p><code>fontcolor.item="white"</code> <code>(Aliases:
          fontcolor.item)</code>: Character or integer scalar. The font color
          for item identifiers.</p></dd>         <dt></dt><dd><p><code>fontface.group=2</code>: Numeric scalar. The font face
          for the group-level annotation.</p></dd>         <dt></dt><dd><p><code>fontfamily.group="sans"</code>: Character scalar. The
          font family for the group-level annotation.</p></dd>         <dt></dt><dd><p><code>fontsize.group=12</code>: Numeric scalar. The font size
          for the group-level annotation.</p></dd>         <dt></dt><dd><p><code>groupAnnotation=NULL</code>: Character scalar. Add
          annotation information as group labels. This can be a value
          in <code>id</code>, <code>group</code> or <code>feature</code>. Defaults to
          <code>group</code>. Only works if <code>showId</code> is not <code>FALSE</code>.</p></dd>         <dt></dt><dd><p><code>just.group="left"</code> <code>(Aliases: just.group)</code>:
          Character scalar. the justification of group labels. Either
          <code>left</code>, <code>right</code>, <code>above</code> or <code>below</code>.</p></dd>         <dt></dt><dd><p><code>lex=1</code>: Numeric scalar. The line expansion factor
          for all track items. This is also used to connect grouped
          items. See <code><a href='grouping.html'>grouping</a></code> for details.</p></dd>         <dt></dt><dd><p><code>lineheight=1</code>: Numeric scalar. The font line
          height for item identifiers.</p></dd>         <dt></dt><dd><p><code>lty="solid"</code>: Character or integer scalar. The
          line type for all track items. This is also used to connect
          grouped items. See <code><a href='grouping.html'>grouping</a></code> for details.</p></dd>         <dt></dt><dd><p><code>lwd=1</code>: Integer scalar. The line width for all
          track items. This is also used to connect grouped items. See
          <code><a href='grouping.html'>grouping</a></code> for details.</p></dd>         <dt></dt><dd><p><code>mergeGroups=FALSE</code>: Logical scalar. Merge fully
          overlapping groups if <code>collapse==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>min.height=3</code>: Numeric scalar. The minimum range
          height in pixels to display. All ranges are expanded to this
          size in order to avoid rendering issues. See <code><a href='collapsing.html'>collapsing</a></code>
          for details. For feathered bars indicating the strandedness
          of grouped items this also controls the height of the arrow
          feathers.</p></dd>         <dt></dt><dd><p><code>min.width=1</code>: Numeric scalar. The minimum range
          width in pixels to display. All ranges are expanded to this
          size in order to avoid rendering issues. See <code><a href='collapsing.html'>collapsing</a></code>
          for details.</p></dd>         <dt></dt><dd><p><code>rotation=0</code>: Numeric scalar. The degree of text
          rotation for item identifiers.</p></dd>         <dt></dt><dd><p><code>rotation.group=0</code>: Numeric scalar. The degree of
          text rotation for group labels.</p></dd>         <dt></dt><dd><p><code>rotation.item=0</code>: Numeric scalar. The degree of
          text rotation for item identifiers.</p></dd>         <dt></dt><dd><p><code>showFeatureId=FALSE</code>: Logical scalar. Control
          whether to plot the individual track item identifiers.</p></dd>         <dt></dt><dd><p><code>showId=FALSE</code>: Logical scalar. Control whether
          to annotate individual groups.</p></dd>         <dt></dt><dd><p><code>showOverplotting=FALSE</code>: Logical scalar. Use a
          color gradient to show the amount of overplotting for collapsed
          items. This implies that <code>collapse==TRUE</code></p></dd>         <dt></dt><dd><p><code>size=1</code>: Numeric scalar. The relative size of
          the track. Can be overridden in the <code><a href='plotTracks.html'>plotTracks</a></code>
          function.</p></dd> </dl></p></dd> 

    <dt></dt><dd><p><code><a href='StackedTrack-class.html'>StackedTrack</a></code>:       <dl>        <dt></dt><dd><p><code>reverseStacking=FALSE</code>: Logical flag. Reverse
          the y-ordering of stacked items. I.e., features that are
          plotted on the bottom-most stacks will be moved to the top-most
          stack and vice versa.</p></dd>         <dt></dt><dd><p><code>stackHeight=0.75</code>: Numeric between 0 and 1.
          Controls the vertical size and spacing between stacked
          elements. The number defines the proportion of the total
          available space for the stack that is used to draw the glyphs.
          E.g., a value of 0.5 means that half of the available vertical
          drawing space (for each stacking line) is used for the glyphs,
          and thus one quarter of the available space each is used for
          spacing above and below the glyph. Defaults to 0.75.</p></dd> </dl></p></dd> 

    <dt></dt><dd><p><code><a href='GdObject-class.html'>GdObject</a></code>:       <dl>        <dt></dt><dd><p><code>alpha=1</code>: Numeric scalar. The transparency for
          all track items.</p></dd>         <dt></dt><dd><p><code>alpha.title=NULL</code>: Numeric scalar. The transparency
          for the title panel.</p></dd>         <dt></dt><dd><p><code>background.legend="transparent"</code>: Integer or
          character scalar. The background color for the legend.</p></dd>         <dt></dt><dd><p><code>background.panel="transparent"</code>: Integer or
          character scalar. The background color of the content panel.</p></dd>         <dt></dt><dd><p><code>background.title="lightgray"</code>: Integer or character
          scalar. The background color for the title panel.</p></dd>         <dt></dt><dd><p><code>cex.axis=NULL</code>: Numeric scalar. The expansion
          factor for the axis annotation. Defaults to <code>NULL</code>, in
          which case it is automatically determined based on the
          available space.</p></dd>         <dt></dt><dd><p><code>cex.title=NULL</code>: Numeric scalar. The expansion
          factor for the title panel. This effects the fontsize of both
          the title and the axis, if any. Defaults to <code>NULL</code>,
          which means that the text size is automatically adjusted to
          the available space.</p></dd>         <dt></dt><dd><p><code>col.axis="white"</code>: Integer or character scalar.
          The font and line color for the y axis, if any.</p></dd>         <dt></dt><dd><p><code>col.border.title="white"</code>: Integer or character
          scalar. The border color for the title panels.</p></dd>         <dt></dt><dd><p><code>col.frame="lightgray"</code>: Integer or character
          scalar. The line color used for the panel frame, if
          <code>frame==TRUE</code></p></dd>         <dt></dt><dd><p><code>col.grid="#808080"</code>: Integer or character scalar.
          Default line color for grid lines, both when <code>type=="g"</code>
          in <code><a href='DataTrack-class.html'>DataTrack</a></code>s and when display parameter
          <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>col.symbol=NULL</code>: Integer or character scalar.
          Default colors for plot symbols. Usually the same as the
          global <code>col</code> parameter.</p></dd>         <dt></dt><dd><p><code>col.title="white"</code> <code>(Aliases: fontcolor.title)</code>:
          Integer or character scalar. The border color for the title panels</p></dd>         <dt></dt><dd><p><code>collapse=TRUE</code>: Boolean controlling whether to
          collapse the content of the track to accomodate the minimum
          current device resolution. See <code><a href='collapsing.html'>collapsing</a></code> for
          details.</p></dd>         <dt></dt><dd><p><code>fontcolor="black"</code>: Integer or character scalar.
          The font color for all text, unless a more specific definition
          exists.</p></dd>         <dt></dt><dd><p><code>fontface=1</code>: Integer or character scalar. The
          font face for all text, unless a more specific definition exists.</p></dd>         <dt></dt><dd><p><code>fontface.title=2</code>: Integer or character scalar.
          The font face for the title panels.</p></dd>         <dt></dt><dd><p><code>fontfamily="sans"</code>: Integer or character scalar.
          The font family for all text, unless a more specific definition
          exists.</p></dd>         <dt></dt><dd><p><code>fontfamily.title="sans"</code>: Integer or character
          scalar. The font family for the title panels.</p></dd>         <dt></dt><dd><p><code>fontsize=12</code>: Numeric scalar. The font size for
          all text, unless a more specific definition exists.</p></dd>         <dt></dt><dd><p><code>frame=FALSE</code>: Boolean. Draw a frame around the
          track when plotting.</p></dd>         <dt></dt><dd><p><code>grid=FALSE</code>: Boolean, switching on/off the plotting
          of a grid.</p></dd>         <dt></dt><dd><p><code>h=-1</code>: Integer scalar. Parameter controlling the
          number of horizontal grid lines, see <code><a href='https://rdrr.io/pkg/lattice/man/panel.functions.html'>panel.grid</a></code>
          for details.</p></dd>         <dt></dt><dd><p><code>lty.grid="solid"</code>: Integer or character scalar.
          Default line type for grid lines, both when <code>type=="g"</code>
          in <code><a href='DataTrack-class.html'>DataTrack</a></code>s and when display parameter
          <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>lwd.border.title=1</code>: Integer scalar. The border
          width for the title panels.</p></dd>         <dt></dt><dd><p><code>lwd.grid=1</code>: Numeric scalar. Default line width
          for grid lines, both when <code>type=="g"</code> in <code><a href='DataTrack-class.html'>DataTrack</a></code>s
          and when display parameter <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>lwd.title=1</code>: Integer scalar. The border width
          for the title panels</p></dd>         <dt></dt><dd><p><code>reverseStrand=FALSE</code>: Logical scalar. Set up the
          plotting coordinates in 3' -&gt; 5' direction if <code>TRUE</code>.
          This will effectively mirror the plot on the vertical axis.</p></dd>         <dt></dt><dd><p><code>rotation.title=90</code> <code>(Aliases: rotation.title)</code>:
          The rotation angle for the text in the title panel. Even
          though this can be adjusted, the automatic resizing of the
          title panel will currently not work, so use at own risk.</p></dd>         <dt></dt><dd><p><code>showAxis=TRUE</code>: Boolean controlling whether to
          plot a y axis (only applies to track types where axes are
          implemented).</p></dd>         <dt></dt><dd><p><code>showTitle=TRUE</code>: Boolean controlling whether to
          plot a title panel. Although this can be set individually
          for each track, in multi-track plots as created by
          <code><a href='plotTracks.html'>plotTracks</a></code> there will still be an empty
          placeholder in case any of the other tracks include a title.
          The same holds true for axes. Note that the the title panel
          background color could be set to transparent in order to
          completely hide it.</p></dd>         <dt></dt><dd><p><code>v=-1</code>: Integer scalar. Parameter controlling the
          number of vertical grid lines, see <code><a href='https://rdrr.io/pkg/lattice/man/panel.functions.html'>panel.grid</a></code>
          for details.</p></dd> </dl></p></dd> 

  
</dl> 


    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Florian Hahne, Steve Lianoglou</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'>

<p><code><a href='AnnotationTrack-class.html'>AnnotationTrack</a></code></p>
<p><code><a href='DisplayPars-class.html'>DisplayPars</a></code></p>
<p><code><a href='GdObject-class.html'>GdObject</a></code></p>
<p><code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code></p>
<p><code><a href='ImageMap-class.html'>ImageMap</a></code></p>
<p><code>IRanges</code></p>
<p><code><a href='RangeTrack-class.html'>RangeTrack</a></code></p>
<p><code><a href='StackedTrack-class.html'>StackedTrack</a></code></p>
<p><code><a href='https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html'>TxDb</a></code></p>
<p><code><a href='collapsing.html'>collapsing</a></code></p>
<p><code><a href='DataTrack-class.html'>DataTrack</a></code></p>
<p><code><a href='grouping.html'>grouping</a></code></p>
<p><code><a href='https://rdrr.io/pkg/lattice/man/panel.functions.html'>panel.grid</a></code></p>
<p><code><a href='plotTracks.html'>plotTracks</a></code></p>
<p><code><a href='settings.html'>settings</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>## The empty object</span>
<span class='fu'>GeneRegionTrack</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; GeneRegionTrack 'GeneRegionTrack'
#&gt; | genome: NA
#&gt; | active chromosome: chrNA
#&gt; | annotation features: 0</div><div class='input'>
<span class='co'>## Load some sample data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>cyp2b10</span><span class='op'>)</span>

<span class='co'>## Construct the object</span>
<span class='va'>grTrack</span> <span class='op'>&lt;-</span> <span class='fu'>GeneRegionTrack</span><span class='op'>(</span>start<span class='op'>=</span><span class='fl'>26682683</span>, end<span class='op'>=</span><span class='fl'>26711643</span>,
rstart<span class='op'>=</span><span class='va'>cyp2b10</span><span class='op'>$</span><span class='va'>start</span>, rends<span class='op'>=</span><span class='va'>cyp2b10</span><span class='op'>$</span><span class='va'>end</span>, chromosome<span class='op'>=</span><span class='fl'>7</span>, genome<span class='op'>=</span><span class='st'>"mm9"</span>,
transcript<span class='op'>=</span><span class='va'>cyp2b10</span><span class='op'>$</span><span class='va'>transcript</span>, gene<span class='op'>=</span><span class='va'>cyp2b10</span><span class='op'>$</span><span class='va'>gene</span>, symbol<span class='op'>=</span><span class='va'>cyp2b10</span><span class='op'>$</span><span class='va'>symbol</span>,
feature<span class='op'>=</span><span class='va'>cyp2b10</span><span class='op'>$</span><span class='va'>feature</span>, exon<span class='op'>=</span><span class='va'>cyp2b10</span><span class='op'>$</span><span class='va'>exon</span>,
name<span class='op'>=</span><span class='st'>"Cyp2b10"</span>, strand<span class='op'>=</span><span class='va'>cyp2b10</span><span class='op'>$</span><span class='va'>strand</span><span class='op'>)</span>

<span class='co'>## Directly from the data.frame</span>
<span class='va'>grTrack</span> <span class='op'>&lt;-</span> <span class='fu'>GeneRegionTrack</span><span class='op'>(</span><span class='va'>cyp2b10</span><span class='op'>)</span>

<span class='co'>## From a TxDb object</span>
<span class='kw'>if</span><span class='op'>(</span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>require</a></span><span class='op'>(</span><span class='va'><a href='https://bioconductor.org/packages/GenomicFeatures'>GenomicFeatures</a></span><span class='op'>)</span><span class='op'>)</span><span class='op'>{</span>
<span class='va'>samplefile</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"hg19_knownGene_sample.sqlite"</span>, package<span class='op'>=</span><span class='st'>"GenomicFeatures"</span><span class='op'>)</span>
<span class='va'>txdb</span> <span class='op'>&lt;-</span> <span class='fu'>loadDb</span><span class='op'>(</span><span class='va'>samplefile</span><span class='op'>)</span>
<span class='fu'>GeneRegionTrack</span><span class='op'>(</span><span class='va'>txdb</span><span class='op'>)</span>
<span class='fu'>GeneRegionTrack</span><span class='op'>(</span><span class='va'>txdb</span>, chromosome<span class='op'>=</span><span class='st'>"chr6"</span>, start<span class='op'>=</span><span class='fl'>35000000</span>, end<span class='op'>=</span><span class='fl'>40000000</span><span class='op'>)</span>
<span class='op'>}</span>
</div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicFeatures</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: AnnotationDbi</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biobase</span></div><div class='output co'>#&gt; <span class='message'>Welcome to Bioconductor</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    Vignettes contain introductory material; view with</span>
#&gt; <span class='message'>    'browseVignettes()'. To cite Bioconductor, see</span>
#&gt; <span class='message'>    'citation("Biobase")', and for packages 'citation("pkgname")'.</span></div><div class='output co'>#&gt; GeneRegionTrack 'GeneRegionTrack'
#&gt; | genome: hg19
#&gt; | active chromosome: chr6
#&gt; | annotation features: 71</div><div class='input'>


<span class='co'># \dontshow{</span>
<span class='co'>## For some annoying reason the postscript device does not know about</span>
<span class='co'>## the sans font</span>
<span class='kw'>if</span><span class='op'>(</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/interactive.html'>interactive</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>
<span class='op'>{</span>
<span class='va'>font</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/ps.options.html'>ps.options</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>$</span><span class='va'>family</span>
<span class='fu'><a href='DisplayPars-class.html'>displayPars</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>fontfamily<span class='op'>=</span><span class='va'>font</span>, fontfamily.title<span class='op'>=</span><span class='va'>font</span><span class='op'>)</span>
<span class='op'>}</span>
<span class='co'># }</span>

<span class='co'>## Plotting</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='img'><img src='GeneRegionTrack-class-1.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Track names</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "GeneRegionTrack"</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"foo"</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='img'><img src='GeneRegionTrack-class-2.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Subsetting and splitting</span>
<span class='va'>subTrack</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/subset.html'>subset</a></span><span class='op'>(</span><span class='va'>grTrack</span>, from<span class='op'>=</span><span class='fl'>26700000</span>, to<span class='op'>=</span><span class='fl'>26705000</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] 0</div><div class='input'><span class='va'>subTrack</span> <span class='op'>&lt;-</span> <span class='va'>grTrack</span><span class='op'>[</span><span class='fu'>transcript</span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span><span class='op'>==</span><span class='st'>"ENSMUST00000144140"</span><span class='op'>]</span>
<span class='fu'><a href='https://rdrr.io/r/base/split.html'>split</a></span><span class='op'>(</span><span class='va'>grTrack</span>, <span class='fu'>transcript</span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; $ENSMUST00000005477
#&gt; GeneRegionTrack 'foo'
#&gt; | genome: NA
#&gt; | active chromosome: chr7
#&gt; | annotation features: 12
#&gt; 
#&gt; $ENSMUST00000072438
#&gt; GeneRegionTrack 'foo'
#&gt; | genome: NA
#&gt; | active chromosome: chr7
#&gt; | annotation features: 11
#&gt; 
#&gt; $ENSMUST00000144140
#&gt; GeneRegionTrack 'foo'
#&gt; | genome: NA
#&gt; | active chromosome: chr7
#&gt; | annotation features: 2
#&gt; </div><div class='input'>
<span class='co'>## Accessors</span>
<span class='fu'><a href='https://rdrr.io/r/stats/start.html'>start</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] 25897620 25897676 25897685 25897685 25911238 25911238 25911554 25911554
#&gt;  [9] 25913865 25913865 25914748 25914748 25915392 25915392 25916766 25916934
#&gt; [17] 25916934 25917288 25917288 25917288 25925373 25926068 25926068 25926250
#&gt; [25] 25926250</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/stats/start.html'>end</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] 25897684 25897684 25897855 25897855 25911400 25911400 25911703 25911703
#&gt;  [9] 25914025 25914025 25914924 25914924 25915533 25915533 25917121 25917121
#&gt; [17] 25917121 25917429 25917429 25917430 25925399 25926249 25926249 25926559
#&gt; [25] 25926624</div><div class='input'><span class='fu'>width</span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1]  65   9 171 171 163 163 150 150 161 161 177 177 142 142 356 188 188 142 142
#&gt; [20] 143  27 182 182 310 375</div><div class='input'><span class='fu'><a href='RangeTrack-class.html'>position</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] 25897652 25897680 25897770 25897770 25911319 25911319 25911628 25911628
#&gt;  [9] 25913945 25913945 25914836 25914836 25915462 25915462 25916944 25917028
#&gt; [17] 25917028 25917358 25917358 25917359 25925386 25926158 25926158 25926404
#&gt; [25] 25926437</div><div class='input'><span class='fu'>width</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>width</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span><span class='op'>+</span><span class='fl'>100</span>

<span class='fu'>strand</span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+" "+"
#&gt; [20] "+" "+" "+" "+" "+" "+"</div><div class='input'><span class='fu'>strand</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"-"</span>

<span class='fu'><a href='RangeTrack-class.html'>chromosome</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "chr7"</div><div class='input'><span class='fu'><a href='RangeTrack-class.html'>chromosome</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"chrX"</span>

<span class='fu'>genome</span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] NA</div><div class='input'><span class='fu'>genome</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"hg19"</span>

<span class='fu'><a href='RangeTrack-class.html'>range</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; IRanges object with 25 ranges and 0 metadata columns:
#&gt;            start       end     width
#&gt;        &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
#&gt;    [1]  25897620  25897684        65
#&gt;    [2]  25897676  25897684         9
#&gt;    [3]  25897685  25897855       171
#&gt;    [4]  25897685  25897855       171
#&gt;    [5]  25911238  25911400       163
#&gt;    ...       ...       ...       ...
#&gt;   [21]  25925373  25925399        27
#&gt;   [22]  25926068  25926249       182
#&gt;   [23]  25926068  25926249       182
#&gt;   [24]  25926250  25926559       310
#&gt;   [25]  25926250  25926624       375</div><div class='input'><span class='fu'>ranges</span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; GRanges object with 25 ranges and 7 metadata columns:
#&gt;        seqnames            ranges strand |        feature          id
#&gt;           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |       &lt;factor&gt; &lt;character&gt;
#&gt;    [1]     chr7 25897620-25897684      + | utr5               unknown
#&gt;    [2]     chr7 25897676-25897684      + | utr5               unknown
#&gt;    [3]     chr7 25897685-25897855      + | protein_coding     unknown
#&gt;    [4]     chr7 25897685-25897855      + | protein_coding     unknown
#&gt;    [5]     chr7 25911238-25911400      + | protein_coding     unknown
#&gt;    ...      ...               ...    ... .            ...         ...
#&gt;   [21]     chr7 25925373-25925399      + | protein_coding     unknown
#&gt;   [22]     chr7 25926068-25926249      + | protein_coding     unknown
#&gt;   [23]     chr7 25926068-25926249      + | protein_coding     unknown
#&gt;   [24]     chr7 25926250-25926559      + | utr3               unknown
#&gt;   [25]     chr7 25926250-25926624      + | utr3               unknown
#&gt;                      exon         transcript               gene   symbol
#&gt;                  &lt;factor&gt;           &lt;factor&gt;           &lt;factor&gt; &lt;factor&gt;
#&gt;    [1] ENSMUSE00000742021 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10
#&gt;    [2] ENSMUSE00000489385 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10
#&gt;    [3] ENSMUSE00000742021 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10
#&gt;    [4] ENSMUSE00000489385 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10
#&gt;    [5] ENSMUSE00000973560 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10
#&gt;    ...                ...                ...                ...      ...
#&gt;   [21] ENSMUSE00000496705 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10
#&gt;   [22] ENSMUSE00000498616 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10
#&gt;   [23] ENSMUSE00000750625 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10
#&gt;   [24] ENSMUSE00000750625 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10
#&gt;   [25] ENSMUSE00000498616 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10
#&gt;          density
#&gt;        &lt;numeric&gt;
#&gt;    [1]         1
#&gt;    [2]         1
#&gt;    [3]         1
#&gt;    [4]         1
#&gt;    [5]         1
#&gt;    ...       ...
#&gt;   [21]         1
#&gt;   [22]         1
#&gt;   [23]         1
#&gt;   [24]         1
#&gt;   [25]         1
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## Annotation</span>
<span class='fu'><a href='AnnotationTrack-class.html'>identifier</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt;  [8] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt; [15] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt; [22] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</div><div class='input'><span class='fu'><a href='AnnotationTrack-class.html'>identifier</a></span><span class='op'>(</span><span class='va'>grTrack</span>, <span class='st'>"lowest"</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "ENSMUSE00000742021" "ENSMUSE00000489385" "ENSMUSE00000742021"
#&gt;  [4] "ENSMUSE00000489385" "ENSMUSE00000973560" "ENSMUSE00000973560"
#&gt;  [7] "ENSMUSE00001043622" "ENSMUSE00001043622" "ENSMUSE00001013991"
#&gt; [10] "ENSMUSE00001013991" "ENSMUSE00000996531" "ENSMUSE00000996531"
#&gt; [13] "ENSMUSE00000999790" "ENSMUSE00000999790" "ENSMUSE00000736922"
#&gt; [16] "ENSMUSE00000963884" "ENSMUSE00000963884" "ENSMUSE00001063063"
#&gt; [19] "ENSMUSE00001063063" "ENSMUSE00000748299" "ENSMUSE00000496705"
#&gt; [22] "ENSMUSE00000498616" "ENSMUSE00000750625" "ENSMUSE00000750625"
#&gt; [25] "ENSMUSE00000498616"</div><div class='input'><span class='fu'><a href='AnnotationTrack-class.html'>identifier</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"bar"</span>

<span class='fu'><a href='RangeTrack-class.html'>feature</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "utr5"           "utr5"           "protein_coding" "protein_coding"
#&gt;  [5] "protein_coding" "protein_coding" "protein_coding" "protein_coding"
#&gt;  [9] "protein_coding" "protein_coding" "protein_coding" "protein_coding"
#&gt; [13] "protein_coding" "protein_coding" "protein_coding" "protein_coding"
#&gt; [17] "protein_coding" "protein_coding" "protein_coding" "protein_coding"
#&gt; [21] "protein_coding" "protein_coding" "protein_coding" "utr3"          
#&gt; [25] "utr3"          </div><div class='input'><span class='fu'><a href='RangeTrack-class.html'>feature</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"foo"</span>

<span class='fu'>exon</span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "ENSMUSE00000742021" "ENSMUSE00000489385" "ENSMUSE00000742021"
#&gt;  [4] "ENSMUSE00000489385" "ENSMUSE00000973560" "ENSMUSE00000973560"
#&gt;  [7] "ENSMUSE00001043622" "ENSMUSE00001043622" "ENSMUSE00001013991"
#&gt; [10] "ENSMUSE00001013991" "ENSMUSE00000996531" "ENSMUSE00000996531"
#&gt; [13] "ENSMUSE00000999790" "ENSMUSE00000999790" "ENSMUSE00000736922"
#&gt; [16] "ENSMUSE00000963884" "ENSMUSE00000963884" "ENSMUSE00001063063"
#&gt; [19] "ENSMUSE00001063063" "ENSMUSE00000748299" "ENSMUSE00000496705"
#&gt; [22] "ENSMUSE00000498616" "ENSMUSE00000750625" "ENSMUSE00000750625"
#&gt; [25] "ENSMUSE00000498616"</div><div class='input'><span class='fu'>exon</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>letters</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span><span class='op'>]</span>

<span class='fu'>gene</span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt;  [4] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt;  [7] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt; [10] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt; [13] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt; [16] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt; [19] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt; [22] "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483"
#&gt; [25] "ENSMUSG00000030483"</div><div class='input'><span class='fu'>gene</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"bar"</span>

<span class='fu'>symbol</span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt;  [8] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt; [15] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt; [22] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"</div><div class='input'><span class='fu'>symbol</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"foo"</span>

<span class='fu'>transcript</span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"
#&gt;  [4] "ENSMUST00000005477" "ENSMUST00000072438" "ENSMUST00000005477"
#&gt;  [7] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"
#&gt; [10] "ENSMUST00000005477" "ENSMUST00000072438" "ENSMUST00000005477"
#&gt; [13] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000144140"
#&gt; [16] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"
#&gt; [19] "ENSMUST00000005477" "ENSMUST00000144140" "ENSMUST00000005477"
#&gt; [22] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000005477"
#&gt; [25] "ENSMUST00000072438"</div><div class='input'><span class='fu'>transcript</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"foo"</span>, <span class='st'>"bar"</span><span class='op'>)</span>
<span class='fu'><a href='RangeTrack-class.html'>chromosome</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"chr7"</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span>
</div><div class='img'><img src='GeneRegionTrack-class-3.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='fu'>values</span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;           feature      id               exon         transcript
#&gt; 1            utr5 unknown ENSMUSE00000742021 ENSMUST00000072438
#&gt; 2            utr5 unknown ENSMUSE00000489385 ENSMUST00000005477
#&gt; 3  protein_coding unknown ENSMUSE00000742021 ENSMUST00000072438
#&gt; 4  protein_coding unknown ENSMUSE00000489385 ENSMUST00000005477
#&gt; 5  protein_coding unknown ENSMUSE00000973560 ENSMUST00000072438
#&gt; 6  protein_coding unknown ENSMUSE00000973560 ENSMUST00000005477
#&gt; 7  protein_coding unknown ENSMUSE00001043622 ENSMUST00000072438
#&gt; 8  protein_coding unknown ENSMUSE00001043622 ENSMUST00000005477
#&gt; 9  protein_coding unknown ENSMUSE00001013991 ENSMUST00000072438
#&gt; 10 protein_coding unknown ENSMUSE00001013991 ENSMUST00000005477
#&gt; 11 protein_coding unknown ENSMUSE00000996531 ENSMUST00000072438
#&gt; 12 protein_coding unknown ENSMUSE00000996531 ENSMUST00000005477
#&gt; 13 protein_coding unknown ENSMUSE00000999790 ENSMUST00000072438
#&gt; 14 protein_coding unknown ENSMUSE00000999790 ENSMUST00000005477
#&gt; 15 protein_coding unknown ENSMUSE00000736922 ENSMUST00000144140
#&gt; 16 protein_coding unknown ENSMUSE00000963884 ENSMUST00000072438
#&gt; 17 protein_coding unknown ENSMUSE00000963884 ENSMUST00000005477
#&gt; 18 protein_coding unknown ENSMUSE00001063063 ENSMUST00000072438
#&gt; 19 protein_coding unknown ENSMUSE00001063063 ENSMUST00000005477
#&gt; 20 protein_coding unknown ENSMUSE00000748299 ENSMUST00000144140
#&gt; 21 protein_coding unknown ENSMUSE00000496705 ENSMUST00000005477
#&gt; 22 protein_coding unknown ENSMUSE00000498616 ENSMUST00000072438
#&gt; 23 protein_coding unknown ENSMUSE00000750625 ENSMUST00000005477
#&gt; 24           utr3 unknown ENSMUSE00000750625 ENSMUST00000005477
#&gt; 25           utr3 unknown ENSMUSE00000498616 ENSMUST00000072438
#&gt;                  gene  symbol density
#&gt; 1  ENSMUSG00000030483 Cyp2b10       1
#&gt; 2  ENSMUSG00000030483 Cyp2b10       1
#&gt; 3  ENSMUSG00000030483 Cyp2b10       1
#&gt; 4  ENSMUSG00000030483 Cyp2b10       1
#&gt; 5  ENSMUSG00000030483 Cyp2b10       1
#&gt; 6  ENSMUSG00000030483 Cyp2b10       1
#&gt; 7  ENSMUSG00000030483 Cyp2b10       1
#&gt; 8  ENSMUSG00000030483 Cyp2b10       1
#&gt; 9  ENSMUSG00000030483 Cyp2b10       1
#&gt; 10 ENSMUSG00000030483 Cyp2b10       1
#&gt; 11 ENSMUSG00000030483 Cyp2b10       1
#&gt; 12 ENSMUSG00000030483 Cyp2b10       1
#&gt; 13 ENSMUSG00000030483 Cyp2b10       1
#&gt; 14 ENSMUSG00000030483 Cyp2b10       1
#&gt; 15 ENSMUSG00000030483 Cyp2b10       1
#&gt; 16 ENSMUSG00000030483 Cyp2b10       1
#&gt; 17 ENSMUSG00000030483 Cyp2b10       1
#&gt; 18 ENSMUSG00000030483 Cyp2b10       1
#&gt; 19 ENSMUSG00000030483 Cyp2b10       1
#&gt; 20 ENSMUSG00000030483 Cyp2b10       1
#&gt; 21 ENSMUSG00000030483 Cyp2b10       1
#&gt; 22 ENSMUSG00000030483 Cyp2b10       1
#&gt; 23 ENSMUSG00000030483 Cyp2b10       1
#&gt; 24 ENSMUSG00000030483 Cyp2b10       1
#&gt; 25 ENSMUSG00000030483 Cyp2b10       1</div><div class='input'>
<span class='co'>## Grouping</span>
<span class='fu'><a href='AnnotationTrack-class.html'>group</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"
#&gt;  [4] "ENSMUST00000005477" "ENSMUST00000072438" "ENSMUST00000005477"
#&gt;  [7] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"
#&gt; [10] "ENSMUST00000005477" "ENSMUST00000072438" "ENSMUST00000005477"
#&gt; [13] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000144140"
#&gt; [16] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000072438"
#&gt; [19] "ENSMUST00000005477" "ENSMUST00000144140" "ENSMUST00000005477"
#&gt; [22] "ENSMUST00000072438" "ENSMUST00000005477" "ENSMUST00000005477"
#&gt; [25] "ENSMUST00000072438"</div><div class='input'><span class='fu'><a href='AnnotationTrack-class.html'>group</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"Group 1"</span>
<span class='fu'>transcript</span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "Group 1" "Group 1"</div><div class='input'><span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>subTrack</span><span class='op'>)</span>
</div><div class='img'><img src='GeneRegionTrack-class-4.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Collapsing transcripts</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>grTrack</span>, collapseTranscripts<span class='op'>=</span><span class='cn'>TRUE</span>, showId<span class='op'>=</span><span class='cn'>TRUE</span>,
extend.left<span class='op'>=</span><span class='fl'>10000</span>, shape<span class='op'>=</span><span class='st'>"arrow"</span><span class='op'>)</span>
</div><div class='img'><img src='GeneRegionTrack-class-5.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Stacking</span>
<span class='fu'><a href='StackedTrack-class.html'>stacking</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "squish"</div><div class='input'><span class='fu'><a href='StackedTrack-class.html'>stacking</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"dense"</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='img'><img src='GeneRegionTrack-class-6.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## coercion</span>
<span class='fu'>as</span><span class='op'>(</span><span class='va'>grTrack</span>, <span class='st'>"data.frame"</span><span class='op'>)</span>
</div><div class='output co'>#&gt;    X.seqnames  X.start    X.end X.width X.strand      X.feature    X.id
#&gt; 1        chr7 25897620 25897684      65        +           utr5 unknown
#&gt; 2        chr7 25897676 25897684       9        +           utr5 unknown
#&gt; 3        chr7 25897685 25897855     171        + protein_coding unknown
#&gt; 4        chr7 25897685 25897855     171        + protein_coding unknown
#&gt; 5        chr7 25911238 25911400     163        + protein_coding unknown
#&gt; 6        chr7 25911238 25911400     163        + protein_coding unknown
#&gt; 7        chr7 25911554 25911703     150        + protein_coding unknown
#&gt; 8        chr7 25911554 25911703     150        + protein_coding unknown
#&gt; 9        chr7 25913865 25914025     161        + protein_coding unknown
#&gt; 10       chr7 25913865 25914025     161        + protein_coding unknown
#&gt; 11       chr7 25914748 25914924     177        + protein_coding unknown
#&gt; 12       chr7 25914748 25914924     177        + protein_coding unknown
#&gt; 13       chr7 25915392 25915533     142        + protein_coding unknown
#&gt; 14       chr7 25915392 25915533     142        + protein_coding unknown
#&gt; 15       chr7 25916766 25917121     356        + protein_coding unknown
#&gt; 16       chr7 25916934 25917121     188        + protein_coding unknown
#&gt; 17       chr7 25916934 25917121     188        + protein_coding unknown
#&gt; 18       chr7 25917288 25917429     142        + protein_coding unknown
#&gt; 19       chr7 25917288 25917429     142        + protein_coding unknown
#&gt; 20       chr7 25917288 25917430     143        + protein_coding unknown
#&gt; 21       chr7 25925373 25925399      27        + protein_coding unknown
#&gt; 22       chr7 25926068 25926249     182        + protein_coding unknown
#&gt; 23       chr7 25926068 25926249     182        + protein_coding unknown
#&gt; 24       chr7 25926250 25926559     310        +           utr3 unknown
#&gt; 25       chr7 25926250 25926624     375        +           utr3 unknown
#&gt;                X.exon       X.transcript             X.gene X.symbol X.density
#&gt; 1  ENSMUSE00000742021 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1
#&gt; 2  ENSMUSE00000489385 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 3  ENSMUSE00000742021 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1
#&gt; 4  ENSMUSE00000489385 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 5  ENSMUSE00000973560 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1
#&gt; 6  ENSMUSE00000973560 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 7  ENSMUSE00001043622 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1
#&gt; 8  ENSMUSE00001043622 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 9  ENSMUSE00001013991 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1
#&gt; 10 ENSMUSE00001013991 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 11 ENSMUSE00000996531 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1
#&gt; 12 ENSMUSE00000996531 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 13 ENSMUSE00000999790 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1
#&gt; 14 ENSMUSE00000999790 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 15 ENSMUSE00000736922 ENSMUST00000144140 ENSMUSG00000030483  Cyp2b10         1
#&gt; 16 ENSMUSE00000963884 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1
#&gt; 17 ENSMUSE00000963884 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 18 ENSMUSE00001063063 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1
#&gt; 19 ENSMUSE00001063063 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 20 ENSMUSE00000748299 ENSMUST00000144140 ENSMUSG00000030483  Cyp2b10         1
#&gt; 21 ENSMUSE00000496705 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 22 ENSMUSE00000498616 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1
#&gt; 23 ENSMUSE00000750625 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 24 ENSMUSE00000750625 ENSMUST00000005477 ENSMUSG00000030483  Cyp2b10         1
#&gt; 25 ENSMUSE00000498616 ENSMUST00000072438 ENSMUSG00000030483  Cyp2b10         1
#&gt;           feature      id               exon         transcript
#&gt; 1            utr5 unknown ENSMUSE00000742021 ENSMUST00000072438
#&gt; 2            utr5 unknown ENSMUSE00000489385 ENSMUST00000005477
#&gt; 3  protein_coding unknown ENSMUSE00000742021 ENSMUST00000072438
#&gt; 4  protein_coding unknown ENSMUSE00000489385 ENSMUST00000005477
#&gt; 5  protein_coding unknown ENSMUSE00000973560 ENSMUST00000072438
#&gt; 6  protein_coding unknown ENSMUSE00000973560 ENSMUST00000005477
#&gt; 7  protein_coding unknown ENSMUSE00001043622 ENSMUST00000072438
#&gt; 8  protein_coding unknown ENSMUSE00001043622 ENSMUST00000005477
#&gt; 9  protein_coding unknown ENSMUSE00001013991 ENSMUST00000072438
#&gt; 10 protein_coding unknown ENSMUSE00001013991 ENSMUST00000005477
#&gt; 11 protein_coding unknown ENSMUSE00000996531 ENSMUST00000072438
#&gt; 12 protein_coding unknown ENSMUSE00000996531 ENSMUST00000005477
#&gt; 13 protein_coding unknown ENSMUSE00000999790 ENSMUST00000072438
#&gt; 14 protein_coding unknown ENSMUSE00000999790 ENSMUST00000005477
#&gt; 15 protein_coding unknown ENSMUSE00000736922 ENSMUST00000144140
#&gt; 16 protein_coding unknown ENSMUSE00000963884 ENSMUST00000072438
#&gt; 17 protein_coding unknown ENSMUSE00000963884 ENSMUST00000005477
#&gt; 18 protein_coding unknown ENSMUSE00001063063 ENSMUST00000072438
#&gt; 19 protein_coding unknown ENSMUSE00001063063 ENSMUST00000005477
#&gt; 20 protein_coding unknown ENSMUSE00000748299 ENSMUST00000144140
#&gt; 21 protein_coding unknown ENSMUSE00000496705 ENSMUST00000005477
#&gt; 22 protein_coding unknown ENSMUSE00000498616 ENSMUST00000072438
#&gt; 23 protein_coding unknown ENSMUSE00000750625 ENSMUST00000005477
#&gt; 24           utr3 unknown ENSMUSE00000750625 ENSMUST00000005477
#&gt; 25           utr3 unknown ENSMUSE00000498616 ENSMUST00000072438
#&gt;                  gene  symbol density
#&gt; 1  ENSMUSG00000030483 Cyp2b10       1
#&gt; 2  ENSMUSG00000030483 Cyp2b10       1
#&gt; 3  ENSMUSG00000030483 Cyp2b10       1
#&gt; 4  ENSMUSG00000030483 Cyp2b10       1
#&gt; 5  ENSMUSG00000030483 Cyp2b10       1
#&gt; 6  ENSMUSG00000030483 Cyp2b10       1
#&gt; 7  ENSMUSG00000030483 Cyp2b10       1
#&gt; 8  ENSMUSG00000030483 Cyp2b10       1
#&gt; 9  ENSMUSG00000030483 Cyp2b10       1
#&gt; 10 ENSMUSG00000030483 Cyp2b10       1
#&gt; 11 ENSMUSG00000030483 Cyp2b10       1
#&gt; 12 ENSMUSG00000030483 Cyp2b10       1
#&gt; 13 ENSMUSG00000030483 Cyp2b10       1
#&gt; 14 ENSMUSG00000030483 Cyp2b10       1
#&gt; 15 ENSMUSG00000030483 Cyp2b10       1
#&gt; 16 ENSMUSG00000030483 Cyp2b10       1
#&gt; 17 ENSMUSG00000030483 Cyp2b10       1
#&gt; 18 ENSMUSG00000030483 Cyp2b10       1
#&gt; 19 ENSMUSG00000030483 Cyp2b10       1
#&gt; 20 ENSMUSG00000030483 Cyp2b10       1
#&gt; 21 ENSMUSG00000030483 Cyp2b10       1
#&gt; 22 ENSMUSG00000030483 Cyp2b10       1
#&gt; 23 ENSMUSG00000030483 Cyp2b10       1
#&gt; 24 ENSMUSG00000030483 Cyp2b10       1
#&gt; 25 ENSMUSG00000030483 Cyp2b10       1</div><div class='input'><span class='fu'>as</span><span class='op'>(</span><span class='va'>grTrack</span>, <span class='st'>"UCSCData"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; UCSC track 'foo'
#&gt; UCSCData object with 3 ranges and 6 metadata columns:
#&gt;       seqnames            ranges strand |                 id        name
#&gt;          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt; &lt;character&gt;
#&gt;   [1]     chr7 25897676-25926559      + | ENSMUST00000005477     Cyp2b10
#&gt;   [2]     chr7 25897620-25926624      + | ENSMUST00000072438     Cyp2b10
#&gt;   [3]     chr7 25916766-25917430      + | ENSMUST00000144140     Cyp2b10
#&gt;           itemRgb blockCount             blockSizes            blockStarts
#&gt;       &lt;character&gt;  &lt;numeric&gt;            &lt;character&gt;            &lt;character&gt;
#&gt;   [1]     #FFD58A         12 9,171,163,150,161,17.. 0,9,13562,13878,1618..
#&gt;   [2]     #FFD58A         11 65,171,163,150,161,1.. 0,65,13618,13934,162..
#&gt;   [3]     #FFD58A          2                356,143                  0,522
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## HTML image map</span>
<span class='fu'><a href='GdObject-class.html'>coords</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; NULL</div><div class='input'><span class='fu'><a href='GdObject-class.html'>tags</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; NULL</div><div class='input'><span class='va'>grTrack</span> <span class='op'>&lt;-</span> <span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span><span class='op'>$</span><span class='va'>foo</span>
</div><div class='img'><img src='GeneRegionTrack-class-7.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'><a href='GdObject-class.html'>coords</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                             x1     y1        x2     y2
#&gt; ENSMUSE00000973560   239.54551  64.50 241.93921 415.50
#&gt; ENSMUSE00001043622   244.21469  64.50 246.41630 415.50
#&gt; ENSMUSE00001013991   278.36174  64.50 280.72588 415.50
#&gt; ENSMUSE00000996531   291.40884  64.50 294.00939 415.50
#&gt; ENSMUSE00000999790   300.92450  64.50 303.00790 415.50
#&gt; ENSMUSE00000963884   323.70890  64.50 326.47199 415.50
#&gt; ENSMUSE00001063063   328.93956  64.50 331.02296 415.50
#&gt; ENSMUSE00000498616   458.67175  64.50 461.34619 415.50
#&gt; ENSMUSE00000498616.1 461.36096 152.25 466.88714 327.75
#&gt; ENSMUSE00000973560.1 239.54551  64.50 241.93921 415.50
#&gt; ENSMUSE00001043622.1 244.21469  64.50 246.41630 415.50
#&gt; ENSMUSE00001013991.1 278.36174  64.50 280.72588 415.50
#&gt; ENSMUSE00000996531.1 291.40884  64.50 294.00939 415.50
#&gt; ENSMUSE00000999790.1 300.92450  64.50 303.00790 415.50
#&gt; ENSMUSE00000963884.1 323.70890  64.50 326.47199 415.50
#&gt; ENSMUSE00001063063.1 328.93956  64.50 331.02296 415.50
#&gt; ENSMUSE00000496705   448.09222  64.50 449.08221 415.50
#&gt; ENSMUSE00000750625   458.67175  64.50 461.34619 415.50
#&gt; ENSMUSE00000750625.1 461.36096 152.25 465.92671 327.75
#&gt; ENSMUSE00000736922   321.22656  64.50 326.47199 415.50
#&gt; ENSMUSE00000748299   328.93956  64.50 331.03774 415.50
#&gt; ENSMUSE00000742021    38.29809  64.50  41.79997 415.50
#&gt; ENSMUSE00000489385    38.71181  64.50  41.79997 415.50</div><div class='input'><span class='fu'><a href='GdObject-class.html'>tags</a></span><span class='op'>(</span><span class='va'>grTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; $fill
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt;            "#FFD58A"            "#FFD58A"            "#FFD58A" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt;            "#FFD58A"            "#FFD58A"            "#FFD58A" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt;            "#FFD58A"            "#FFD58A"            "#FFD58A" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt;            "#FFD58A"            "#FFD58A"            "#FFD58A" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt;            "#FFD58A"            "#FFD58A"            "#FFD58A" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt;            "#FFD58A"            "#FFD58A"            "#FFD58A" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;            "#FFD58A"            "#FFD58A"            "#FFD58A" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt;            "#FFD58A"            "#FFD58A" 
#&gt; 
#&gt; $strand
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt;                  "+"                  "+"                  "+" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt;                  "+"                  "+"                  "+" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt;                  "+"                  "+"                  "+" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt;                  "+"                  "+"                  "+" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt;                  "+"                  "+"                  "+" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt;                  "+"                  "+"                  "+" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;                  "+"                  "+"                  "+" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt;                  "+"                  "+" 
#&gt; 
#&gt; $text
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt; "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt; "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt; "ENSMUSE00001063063" "ENSMUSE00000498616" "ENSMUSE00000498616" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt; "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt; "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt; "ENSMUSE00001063063" "ENSMUSE00000496705" "ENSMUSE00000750625" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt; "ENSMUSE00000742021" "ENSMUSE00000489385" 
#&gt; 
#&gt; $start
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt;           "25911238"           "25911554"           "25913865" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt;           "25914748"           "25915392"           "25916934" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt;           "25917288"           "25926068"           "25926250" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt;           "25911238"           "25911554"           "25913865" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt;           "25914748"           "25915392"           "25916934" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt;           "25917288"           "25925352"           "25926068" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;           "25926250"           "25916766"           "25917288" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt;           "25897618"           "25897646" 
#&gt; 
#&gt; $end
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt;           "25911400"           "25911703"           "25914025" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt;           "25914924"           "25915533"           "25917121" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt;           "25917429"           "25926249"           "25926624" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt;           "25911400"           "25911703"           "25914025" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt;           "25914924"           "25915533"           "25917121" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt;           "25917429"           "25925419"           "25926249" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;           "25926559"           "25917121"           "25917430" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt;           "25897855"           "25897855" 
#&gt; 
#&gt; $feature
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt;     "protein_coding"     "protein_coding"               "utr3" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt;     "protein_coding"     "protein_coding"     "protein_coding" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;               "utr3"     "protein_coding"     "protein_coding" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt;          "composite"          "composite" 
#&gt; 
#&gt; $id
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt;            "unknown"            "unknown"            "unknown" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt;            "unknown"            "unknown"            "unknown" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt;            "unknown"            "unknown"            "unknown" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt;            "unknown"            "unknown"            "unknown" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt;            "unknown"            "unknown"            "unknown" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt;            "unknown"            "unknown"            "unknown" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;            "unknown"            "unknown"            "unknown" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt;            "unknown"            "unknown" 
#&gt; 
#&gt; $exon
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt; "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt; "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt; "ENSMUSE00001063063" "ENSMUSE00000498616" "ENSMUSE00000498616" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt; "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt; "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt; "ENSMUSE00001063063" "ENSMUSE00000496705" "ENSMUSE00000750625" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt; "ENSMUSE00000742021" "ENSMUSE00000489385" 
#&gt; 
#&gt; $transcript
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt; "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt; "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt; "ENSMUST00000072438" "ENSMUST00000072438" "ENSMUST00000072438" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt; "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt; "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt; "ENSMUST00000005477" "ENSMUST00000005477" "ENSMUST00000005477" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUST00000005477" "ENSMUST00000144140" "ENSMUST00000144140" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt; "ENSMUST00000072438" "ENSMUST00000005477" 
#&gt; 
#&gt; $gene
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt; "ENSMUSG00000030483" "ENSMUSG00000030483" 
#&gt; 
#&gt; $symbol
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;            "Cyp2b10"            "Cyp2b10"            "Cyp2b10" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt;            "Cyp2b10"            "Cyp2b10" 
#&gt; 
#&gt; $density
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt;                  "1"                  "1"                  "1" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt;                  "1"                  "1"                  "1" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt;                  "1"                  "1"                  "1" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt;                  "1"                  "1"                  "1" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt;                  "1"                  "1"                  "1" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt;                  "1"                  "1"                  "1" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;                  "1"                  "1"                  "1" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt;                  "2"                  "2" 
#&gt; 
#&gt; $gdensity
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt;                   NA                   NA                   NA 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt;                   NA                   NA                   NA 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt;                   NA                   NA                   NA 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt;                   NA                   NA                   NA 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt;                   NA                   NA                   NA 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt;                   NA                   NA                   NA 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;                   NA                   NA                   NA 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt;                  "1"                  "1" 
#&gt; 
#&gt; $exonId
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt; "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt; "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt; "ENSMUSE00001063063" "ENSMUSE00000498616" "ENSMUSE00000498616" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt; "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt; "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt; "ENSMUSE00001063063" "ENSMUSE00000496705" "ENSMUSE00000750625" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt; "ENSMUSE00000742021" "ENSMUSE00000489385" 
#&gt; 
#&gt; $origExonId
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt; "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt; "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt; "ENSMUSE00001063063" "ENSMUSE00000498616" "ENSMUSE00000498616" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt; "ENSMUSE00000973560" "ENSMUSE00001043622" "ENSMUSE00001013991" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt; "ENSMUSE00000996531" "ENSMUSE00000999790" "ENSMUSE00000963884" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt; "ENSMUSE00001063063" "ENSMUSE00000496705" "ENSMUSE00000750625" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt; "ENSMUSE00000750625" "ENSMUSE00000736922" "ENSMUSE00000748299" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt; "ENSMUSE00000742021" "ENSMUSE00000489385" 
#&gt; 
#&gt; $col
#&gt;   ENSMUSE00000973560   ENSMUSE00001043622   ENSMUSE00001013991 
#&gt;           "darkgray"           "darkgray"           "darkgray" 
#&gt;   ENSMUSE00000996531   ENSMUSE00000999790   ENSMUSE00000963884 
#&gt;           "darkgray"           "darkgray"           "darkgray" 
#&gt;   ENSMUSE00001063063   ENSMUSE00000498616 ENSMUSE00000498616.1 
#&gt;           "darkgray"           "darkgray"           "darkgray" 
#&gt; ENSMUSE00000973560.1 ENSMUSE00001043622.1 ENSMUSE00001013991.1 
#&gt;           "darkgray"           "darkgray"           "darkgray" 
#&gt; ENSMUSE00000996531.1 ENSMUSE00000999790.1 ENSMUSE00000963884.1 
#&gt;           "darkgray"           "darkgray"           "darkgray" 
#&gt; ENSMUSE00001063063.1   ENSMUSE00000496705   ENSMUSE00000750625 
#&gt;           "darkgray"           "darkgray"           "darkgray" 
#&gt; ENSMUSE00000750625.1   ENSMUSE00000736922   ENSMUSE00000748299 
#&gt;           "darkgray"           "darkgray"           "darkgray" 
#&gt;   ENSMUSE00000742021   ENSMUSE00000489385 
#&gt;           "darkgray"           "darkgray" 
#&gt; 
#&gt; $title
#&gt;  [1] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt;  [8] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt; [15] "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10" "Cyp2b10"
#&gt; [22] "Cyp2b10" "Cyp2b10"
#&gt; </div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Florian Hahne, Steffen Durinck, Robert Ivanek, Arne Mueller, Steve Lianoglou, Ge Tan, Lance Parsons, Shraddha Pai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


