<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>GenomeAxisTrack class and methods — GenomeAxisTrack-class • Gviz</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="GenomeAxisTrack class and methods — GenomeAxisTrack-class" />
<meta property="og:description" content="A class representing a customizable genomic axis." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Gviz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.33.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Gviz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ivanek/Gviz/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>GenomeAxisTrack class and methods</h1>
    
    <div class="hidden name"><code>GenomeAxisTrack-class.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A class representing a customizable genomic axis.</p>
    </div>

    <pre class="usage"><span class='fu'>GenomeAxisTrack</span>(<span class='kw'>range</span><span class='kw'>=</span><span class='kw'>NULL</span>, <span class='kw'>name</span><span class='kw'>=</span><span class='st'>"Axis"</span>, <span class='no'>id</span>, <span class='no'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>range</th>
      <td><p>Optional <code>GRanges</code> or
    <code>IRanges</code> object to highlight certain regions on
    the axis.</p></td>
    </tr>
    <tr>
      <th>name</th>
      <td><p>Character scalar of the track's name used in the title
    panel when plotting.</p></td>
    </tr>
    <tr>
      <th>id</th>
      <td><p>A character vector of the same length as <code>range</code>
   containing identifiers for the ranges. If missing, the constructor
   will try to extract the ids from <code><a href='https://rdrr.io/r/base/names.html'>names(range)</a></code>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional items which will all be interpreted as further
    display parameters. See <code><a href='settings.html'>settings</a></code> and the "Display
    Parameters" section below for details.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="objects-from-the-class"><a class="anchor" href="#objects-from-the-class"></a>Objects from the class</h2>

    

<p>Objects can be created using the constructor function
  <code>GenomeAxisTrack</code>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    

<p>A <code>GenomeAxisTrack</code> can be customized using the familiar display
  parameters. By providing a <code>GRanges</code> or <code>IRanges</code> object to
  the constructor, ranges on the axis can be further highlighted.\</p>
<p>With the <code>scale</code> display parameter, a small scale indicator can
  be shown instead of the entire genomic axis. The scale can either be
  provided as a fraction of the plotting region (it will be rounded to
  the nearest human readable absolute value) or as an absolute value and
  is always displayed in bp, kb, mb or gb units. Note that most display
  parameters for the <code>GenomeAxisTrack</code> are ignored when a scale is
  used insterad of the full axis. In particular, only the parameters
  <code>exponent</code>, <code>alpha</code>, <code>lwd</code>, <code>col</code>, <code>cex</code>,
  <code>distFromAxis</code> and <code>labelPos</code> are used.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    

<p>The return value of the constructor function is a new object of class
  <code>GenomeAxisTrack</code>.</p>
    <h2 class="hasAnchor" id="slots"><a class="anchor" href="#slots"></a>Slots</h2>

    <p></p><dl'>

    <dt><code>range</code>:</dt><dd><p>Object of class <code>GRanges</code>,
      highlighted on the axis.</p></dd>

    <dt><code>dp</code>:</dt><dd><p>Object of class
      <code><a href='DisplayPars-class.html'>DisplayPars</a></code>, inherited from class
      <code><a href='GdObject-class.html'>GdObject</a></code></p></dd>

    <dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code>, inherited
      from class <code><a href='GdObject-class.html'>GdObject</a></code></p></dd>

     <dt><code>imageMap</code>:</dt><dd><p>Object of class
      <code><a href='ImageMap-class.html'>ImageMap</a></code>, inherited from class
      <code><a href='GdObject-class.html'>GdObject</a></code></p></dd>

  
</dl>

    <h2 class="hasAnchor" id="extends"><a class="anchor" href="#extends"></a>Extends</h2>

    <p>Class <code>"<a href='GdObject-class.html'>GdObject</a>"</code>, directly.</p>
    <h2 class="hasAnchor" id="methods"><a class="anchor" href="#methods"></a>Methods</h2>

    

<p>In the following code chunks, <code>obj</code> is considered to be an object of class <code>GenomeAxisTrack</code>.</p>
<p><b><em>Exported in the name space:</em></b></p>
<p></p><dl'>

    <dt>[</dt><dd><p><code>signature(x="GenomeAxisTrack")</code>: subset the
      <code>GRanges</code> object in the <code>range</code> slot. For most
      applications, the <code>subset</code> method may be more appropriate.      <em>Additional Arguments:</em>      <dl'>    <dt></dt><dd><p><code>i</code>: subsetting incides.</p></dd></dl>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code>obj[1:5]</code></p></dd></dl></p></dd>

    <dt>start, end, width</dt><dd><p><code>signature(x="GenomeAxisTrack")</code>: the
      start or end coordinates of the track items, or their width in
      genomic coordinates.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/stats/start.html'>start(x)</a></code>      <code><a href='https://rdrr.io/r/stats/start.html'>end(x)</a></code>      <code>width(x)</code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/stats/start.html'>start(obj)</a></code></p></dd>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/stats/start.html'>end(obj)</a></code></p></dd>    <dt></dt><dd><p><code>width(obj)</code></p></dd></dl></p></dd>

    <dt>range</dt><dd><p><code>signature(x="GenomeAxisTrack")</code>: return the genomic
      coordinates for the track as an object of class
      <code>IRanges</code>.      <em>Usage:</em>      <code><a href='RangeTrack-class.html'>range(x)</a></code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='RangeTrack-class.html'>range(obj)</a></code></p></dd></dl></p></dd>

    <dt>ranges</dt><dd><p><code>signature(x="GenomeAxisTrack")</code>: return the genomic
      coordinates for the track along with all additional annotation
      information as an object of class <code>GRanges</code>.      <em>Usage:</em>      <code>ranges(x)</code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code>ranges(obj)</code></p></dd></dl></p></dd>

    <dt>strand</dt><dd><p><code>signature(x="GenomeAxisTrack")</code>: return a vector of
      strand specifiers for all track items, in the form '+' for the
      Watson strand, '-' for the Crick strand or '*' for either of the
      two.      <em>Usage:</em>      <code>strand(x)</code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code>strand(obj)</code></p></dd></dl></p></dd>

    <dt>values</dt><dd><p><code>signature(x="GenomeAxisTrack")</code>: return all
      additional annotation information except for the genomic coordinates
      for the track items.      <em>Usage:</em>      <code>values(x)</code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code>values(obj)</code></p></dd></dl></p></dd>

    <dt>subset</dt><dd><p><code>signature(x="GenomeAxisTrack")</code>: subset a
      <code>GenomeAxisTrack</code> by coordinates and sort if necessary.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/subset.html'>subset(x, from, to, sort=FALSE, ...)</a></code>      <em>Additional Arguments:</em>      <dl'>    <dt></dt><dd><p><code>from</code>, <code>to</code>: the coordinates range to subset
      to.</p></dd>    <dt></dt><dd><p><code>sort</code>: sort the object after subsetting. Usually
      not necessary.</p></dd>    <dt></dt><dd><p><code>...</code>: additional arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/subset.html'>subset(obj, from=10, to=20, sort=TRUE)</a></code></p></dd></dl></p></dd>

    <dt>length</dt><dd><p><code>signature(x="GenomeAxisTrack")</code>: return the
      number of items stored in the <code>ranges</code> slot.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/length.html'>length(x)</a></code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/length.html'>length(obj)</a></code></p></dd></dl></p></dd>
  
</dl>

<p><em>Internal methods:</em></p>
<p></p><dl'>

    <dt>drawGD</dt><dd><p><code>signature(GdObject="GenomeAxisTrack")</code>: the
      workhorse function to plot the object.      <em>Usage:</em>      <code><a href='GdObject-class.html'>drawGD(GdObject, minBase, maxBase, prepare=FALSE,
    subset=TRUE, ...)</a></code>      <em>Additional Arguments:</em>      <dl'>    <dt></dt><dd><p><code>minBase</code>, <code>maxBase</code>: the coordinate range to
      plot.</p></dd>    <dt></dt><dd><p><code>prepare</code>: run method in preparation or in
      production mode.</p></dd>    <dt></dt><dd><p><code>subset</code>: subset the object to the visible region
      or skip the potentially expensive subsetting operation.</p></dd>    <dt></dt><dd><p><code>...</code>: all further arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code>Gviz:::drawGD(obj)</code></p></dd>    <dt></dt><dd><p><code>Gviz:::drawGD(obj, minBase=1, maxBase=100)</code></p></dd>    <dt></dt><dd><p><code>Gviz:::drawGD(obj, prepare=TRUE,
        subset=FALSE)</code></p></dd></dl></p></dd>

    <dt>collapseTrack</dt><dd><p><code>signature(GdObject="GenomeAxisTrack")</code>:
      preprocess the track before plotting. This will collapse
      overlapping track items based on the available resolution and
      increase the width and height of all track objects to a minimum
      value to avoid rendering issues. See <code><a href='collapsing.html'>collapsing</a></code> for
      details.      <em>Usage:</em>      <code>collapseTrack(GdObject, diff=.pxResolution(coord="x"))</code>      <em>Additional Arguments:</em>      <dl'>    <dt></dt><dd><p><code>diff</code>: the minimum pixel width to display,
      everything below that will be inflated to a width of
      <code>diff</code>.</p></dd></dl>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code>Gviz:::collapseTrack(obj)</code></p></dd></dl></p></dd>

    <dt>initialize</dt><dd><p><code>signature(.Object="GenomeAxisTrack")</code>:
      initialize the object</p></dd>

    <dt>show</dt><dd><p><code>signature(object="GenomeAxisTrack")</code>: show a
      human-readable summary of the object</p></dd>

  
</dl>

<p><b><em>Inherited:</em></b></p>
<p></p><dl'>

    <dt>displayPars</dt><dd><p><code>signature(x="GenomeAxisTrack", name="character")</code>:
      list the value of the display parameter <code>name</code>. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>displayPars(x, name)</a></code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>displayPars(obj, "col")</a></code></p></dd></dl></p></dd>


    <dt>displayPars</dt><dd><p><code>signature(x="GenomeAxisTrack", name="missing")</code>:
      list the value of all available display parameters. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>displayPars(obj)</a></code></p></dd></dl></p></dd>

    <dt>getPar</dt><dd><p><code>signature(x="GenomeAxisTrack", name="character")</code>:
      alias for the <code>displayPars</code> method. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>getPar(x, name)</a></code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>getPar(obj, "col")</a></code></p></dd></dl></p></dd>

    <dt>getPar</dt><dd><p><code>signature(x="GenomeAxisTrack", name="missing")</code>: alias
      for the <code>displayPars</code> method. See <code><a href='settings.html'>settings</a></code> for
      details on display parameters and customization.      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>getPar(obj)</a></code></p></dd></dl></p></dd>

    <dt>displayPars&lt;-</dt><dd><p><code>signature(x="GenomeAxisTrack", value="list")</code>:
      set display parameters using the values of the named list in
      <code>value</code>. See <code><a href='settings.html'>settings</a></code> for details on display
      parameters and customization.      <em>Usage:</em>      <code>displayPars&lt;-(x, value)</code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code>displayPars(obj) &lt;- list(col="red", lwd=2)</code></p></dd></dl></p></dd>

    <dt>setPar</dt><dd><p><code>signature(x="GenomeAxisTrack", value="character")</code>: set
      the single display parameter <code>name</code> to <code>value</code>. Note
      that display parameters in the <code>GenomeAxisTrack</code> class are
      pass-by-reference, so no re-assignmnet to the symbol <code>obj</code> is
      necessary. See <code><a href='settings.html'>settings</a></code> for details on display
      parameters and customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>setPar(x, name, value)</a></code>      <em>Additional Arguments:</em>      <dl'>    <dt></dt><dd><p><code>name</code>: the name of the display parameter to set.</p></dd></dl>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>setPar(obj, "col", "red")</a></code></p></dd></dl></p></dd>

    <dt>setPar</dt><dd><p><code>signature(x="GenomeAxisTrack", value="list")</code>: set
      display parameters by the values of the named list in
      <code>value</code>. Note that display parameters in the <code>GenomeAxisTrack</code>
      class are pass-by-reference, so no re-assignmnet to the symbol
      <code>obj</code> is necessary. See <code><a href='settings.html'>settings</a></code> for details on
      display parameters and customization.      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>setPar(obj, list(col="red", lwd=2))</a></code></p></dd></dl></p></dd>

    <dt>group</dt><dd><p><code>signature(GdObject="GenomeAxisTrack")</code>: return
      grouping information for the individual items in the track. Unless
      overwritten in one of the sub-classes, this usualy returns
      <code>NULL</code>.      <em>Usage:</em>      <code><a href='AnnotationTrack-class.html'>group(GdObject)</a></code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='AnnotationTrack-class.html'>group(obj)</a></code></p></dd></dl></p></dd>

    <dt>names</dt><dd><p><code>signature(x="GenomeAxisTrack")</code>: return the value of
      the <code>name</code> slot.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/names.html'>names(x)</a></code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/names.html'>names(obj)</a></code></p></dd></dl></p></dd>

    <dt>names&lt;-</dt><dd><p><code>signature(x="GenomeAxisTrack", value="character")</code>:
      set the value of the <code>name</code> slot.      <em>Usage:</em>      <code>names&lt;-(x, value)</code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code>names(obj) &lt;- "foo"</code></p></dd></dl></p></dd>

    <dt>coords</dt><dd><p><code>signature(ImageMap="GenomeAxisTrack")</code>: return the
      coordinates from the internal image map.      <em>Usage:</em>      <code><a href='GdObject-class.html'>coords(ImageMap)</a></code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='GdObject-class.html'>coords(obj)</a></code></p></dd></dl></p></dd>

    <dt>tags</dt><dd><p><code>signature(x="GenomeAxisTrack")</code>: return the tags from the
      internal image map.      <em>Usage:</em>      <code><a href='GdObject-class.html'>tags(x)</a></code>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code><a href='GdObject-class.html'>tags(obj)</a></code></p></dd></dl></p></dd>

    <dt>drawAxis</dt><dd><p><code>signature(GdObject="GenomeAxisTrack")</code>: add a
      y-axis to the title panel of a track if necessary. Unless
      overwritten in one of the sub-classes this usualy does not plot
      anything and returns <code>NULL</code>.      <em>Usage:</em>      <code><a href='GdObject-class.html'>drawAxis(x, ...)</a></code>      <em>Additional Arguments:</em>      <dl'>    <dt></dt><dd><p><code>...</code>: all further arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code>Gviz:::drawAxis(obj)</code></p></dd></dl></p></dd>

    <dt>drawGrid</dt><dd><p><code>signature(GdObject="GenomeAxisTrack")</code>: superpose a
      grid on top of a track if necessary. Unless overwritten in one of
      the sub-classes this usualy does not plot anything and returns
      <code>NULL</code>.      <em>Usage:</em>      <code>drawGrid(GdObject, ...)</code>      <em>Additional Arguments:</em>      <dl'>    <dt></dt><dd><p><code>...</code>: additional arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl'>    <dt></dt><dd><p><code>Gviz:::drawGrid(obj)</code></p></dd></dl></p></dd>
  
</dl>

    <h2 class="hasAnchor" id="display-parameters"><a class="anchor" href="#display-parameters"></a>Display Parameters</h2>

     

<p>The following display parameters are set for objects of class
  <code>GenomeAxisTrack</code> upon instantiation, unless one or more of them
  have already been set by one of the optional sub-class initializers,
  which always get precedence over these global defaults. See
  <code><a href='settings.html'>settings</a></code> for details on setting graphical parameters
  for tracks.</p><dl'>

    <dt></dt><dd><p><code>add35=FALSE</code>: Logical scalar. Add 3' to 5' direction
      indicators.</p></dd> 

    <dt></dt><dd><p><code>add53=FALSE</code>: Logical scalar. Add 5' to 3' direction
      indicators.</p></dd> 

    <dt></dt><dd><p><code>background.title="transparent"</code>: Character scalar.
      The background color for the title panel. Defaults to omit the
      background.</p></dd> 

    <dt></dt><dd><p><code>cex=0.8</code>: Numeric scalar. The overall font expansion
      factor for the axis annotation text.</p></dd> 

    <dt></dt><dd><p><code>cex.id=0.7</code>: Numeric scalar. The text size for the
      optional range annotation.</p></dd> 

    <dt></dt><dd><p><code>col="darkgray"</code>: Character scalar. The color for the
      axis lines and tickmarks.</p></dd> 

    <dt></dt><dd><p><code>col.border.title="transparent"</code>: Integer or character
      scalar. The border color for the title panels.</p></dd> 

    <dt></dt><dd><p><code>col.id="white"</code>: Character scalar. The text color
      for the optional range annotation.</p></dd> 

    <dt></dt><dd><p><code>col.range="cornsilk4"</code>: Character scalar. The border
      color for highlighted regions on the axis.</p></dd> 

    <dt></dt><dd><p><code>distFromAxis=1</code>: Numeric scalar. Control the distance
      of the axis annotation from the tick marks.</p></dd> 

    <dt></dt><dd><p><code>exponent=NULL</code>: Numeric scalar. The exponent for the
      axis coordinates, e.g., 3 means mb, 6 means gb, etc. The default
      is to automatically determine the optimal exponent.</p></dd> 

    <dt></dt><dd><p><code>fill.range="cornsilk3"</code>: Character scalar. The fill
      color for highlighted regions on the axis.</p></dd> 

    <dt></dt><dd><p><code>fontcolor="#808080"</code>: Character scalar. The font
      color for the axis annotation text.</p></dd> 

    <dt></dt><dd><p><code>fontsize=10</code>: Numeric scalar. Font size for the axis
      annotation text in points.</p></dd> 

    <dt></dt><dd><p><code>labelPos="alternating"</code>: Character vector, one in
      "alternating", "revAlternating", "above" or "below". The vertical
      positioning of the axis labels. If <code>scale</code> is not <code>NULL</code>,
      the possible values are "above", "below" and "beside".</p></dd> 

    <dt></dt><dd><p><code>littleTicks=FALSE</code>: Logical scalar. Add more fine-grained
      tick marks.</p></dd> 

    <dt></dt><dd><p><code>lwd=2</code>: Numeric scalar. The line width for the axis
      elementes.</p></dd> 

    <dt></dt><dd><p><code>lwd.border.title=1</code>: Integer scalar. The border width
      for the title panels.</p></dd> 

    <dt></dt><dd><p><code>scale=NULL</code>: Numeric scalar. If not <code>NULL</code> a
      small scale is drawn instead of the full axis, if the value is
      between 0 and 1 it is interpreted as a fraction of the current
      plotting region, otherwise as an absolute length value in genomic
      coordinates.</p></dd> 

    <dt></dt><dd><p><code>showId=FALSE</code>: Logical scalar. Show the optional
      range highlighting annotation.</p></dd> 

    <dt></dt><dd><p><code>showTitle=FALSE</code>: Logical scalar. Plot a title panel.
      Defaults to omit the title panel.</p></dd> 

    <dt></dt><dd><p><code>size=NULL</code>: Numeric scalar. The relative size of the
      track. Can be overridden in the <code><a href='plotTracks.html'>plotTracks</a></code> function.
      Defaults to the ideal size based on the other track settings.</p></dd> 

    <dt></dt><dd><p><code>ticksAt=NULL</code>: Numeric scalar. The exact x-position
      for tickmarks (in base-pairs).</p></dd> 

  
</dl> 

<p>Additional display parameters are being inherited from the respective
  parent classes. Note that not all of them may have an effect on the
  plotting of <code>GenomeAxisTrack</code> objects.</p>
<p></p><dl'>

    <dt></dt><dd><p><code><a href='GdObject-class.html'>GdObject</a></code>:       <dl'>        <dt></dt><dd><p><code>alpha=1</code>: Numeric scalar. The transparency for
          all track items.</p></dd>         <dt></dt><dd><p><code>alpha.title=NULL</code>: Numeric scalar. The transparency
          for the title panel.</p></dd>         <dt></dt><dd><p><code>background.legend="transparent"</code>: Integer or
          character scalar. The background color for the legend.</p></dd>         <dt></dt><dd><p><code>background.panel="transparent"</code>: Integer or
          character scalar. The background color of the content panel.</p></dd>         <dt></dt><dd><p><code>cex.axis=NULL</code>: Numeric scalar. The expansion
          factor for the axis annotation. Defaults to <code>NULL</code>, in
          which case it is automatically determined based on the
          available space.</p></dd>         <dt></dt><dd><p><code>cex.title=NULL</code>: Numeric scalar. The expansion
          factor for the title panel. This effects the fontsize of both
          the title and the axis, if any. Defaults to <code>NULL</code>,
          which means that the text size is automatically adjusted to
          the available space.</p></dd>         <dt></dt><dd><p><code>col.axis="white"</code>: Integer or character scalar.
          The font and line color for the y axis, if any.</p></dd>         <dt></dt><dd><p><code>col.frame="lightgray"</code>: Integer or character
          scalar. The line color used for the panel frame, if
          <code>frame==TRUE</code></p></dd>         <dt></dt><dd><p><code>col.grid="#808080"</code>: Integer or character scalar.
          Default line color for grid lines, both when <code>type=="g"</code>
          in <code><a href='DataTrack-class.html'>DataTrack</a></code>s and when display parameter
          <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>col.line=NULL</code>: Integer or character scalar.
          Default colors for plot lines. Usually the same as the global
          <code>col</code> parameter.</p></dd>         <dt></dt><dd><p><code>col.symbol=NULL</code>: Integer or character scalar.
          Default colors for plot symbols. Usually the same as the
          global <code>col</code> parameter.</p></dd>         <dt></dt><dd><p><code>col.title="white"</code> <code>(Aliases: fontcolor.title)</code>:
          Integer or character scalar. The border color for the title panels</p></dd>         <dt></dt><dd><p><code>collapse=TRUE</code>: Boolean controlling whether to
          collapse the content of the track to accomodate the minimum
          current device resolution. See <code><a href='collapsing.html'>collapsing</a></code> for
          details.</p></dd>         <dt></dt><dd><p><code>fill="lightgray"</code>: Integer or character scalar.
          Default fill color setting for all plotting elements, unless
          there is a more specific control defined elsewhere.</p></dd>         <dt></dt><dd><p><code>fontface=1</code>: Integer or character scalar. The
          font face for all text, unless a more specific definition exists.</p></dd>         <dt></dt><dd><p><code>fontface.title=2</code>: Integer or character scalar.
          The font face for the title panels.</p></dd>         <dt></dt><dd><p><code>fontfamily="sans"</code>: Integer or character scalar.
          The font family for all text, unless a more specific definition
          exists.</p></dd>         <dt></dt><dd><p><code>fontfamily.title="sans"</code>: Integer or character
          scalar. The font family for the title panels.</p></dd>         <dt></dt><dd><p><code>frame=FALSE</code>: Boolean. Draw a frame around the
          track when plotting.</p></dd>         <dt></dt><dd><p><code>grid=FALSE</code>: Boolean, switching on/off the plotting
          of a grid.</p></dd>         <dt></dt><dd><p><code>h=-1</code>: Integer scalar. Parameter controlling the
          number of horizontal grid lines, see <code>panel.grid</code>
          for details.</p></dd>         <dt></dt><dd><p><code>lineheight=1</code>: Numeric scalar. The font line
          height for all text, unless a more specific definition exists.</p></dd>         <dt></dt><dd><p><code>lty="solid"</code>: Numeric scalar. Default line type
          setting for all plotting elements, unless there is a more
          specific control defined elsewhere.</p></dd>         <dt></dt><dd><p><code>lty.grid="solid"</code>: Integer or character scalar.
          Default line type for grid lines, both when <code>type=="g"</code>
          in <code><a href='DataTrack-class.html'>DataTrack</a></code>s and when display parameter
          <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>lwd.grid=1</code>: Numeric scalar. Default line width
          for grid lines, both when <code>type=="g"</code> in <code><a href='DataTrack-class.html'>DataTrack</a></code>s
          and when display parameter <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>lwd.title=1</code>: Integer scalar. The border width
          for the title panels</p></dd>         <dt></dt><dd><p><code>min.distance=1</code>: Numeric scalar. The minimum
          pixel distance before collapsing range items, only if
          <code>collapse==TRUE</code>. See <code><a href='collapsing.html'>collapsing</a></code> for details.</p></dd>         <dt></dt><dd><p><code>min.height=3</code>: Numeric scalar. The minimum range
          height in pixels to display. All ranges are expanded to this
          size in order to avoid rendering issues. See <code><a href='collapsing.html'>collapsing</a></code>
          for details.</p></dd>         <dt></dt><dd><p><code>min.width=1</code>: Numeric scalar. The minimum range
          width in pixels to display. All ranges are expanded to this
          size in order to avoid rendering issues. See <code><a href='collapsing.html'>collapsing</a></code>
          for details.</p></dd>         <dt></dt><dd><p><code>reverseStrand=FALSE</code>: Logical scalar. Set up the
          plotting coordinates in 3' -&gt; 5' direction if <code>TRUE</code>.
          This will effectively mirror the plot on the vertical axis.</p></dd>         <dt></dt><dd><p><code>rotation=0</code>: The rotation angle for all text
          unless a more specific definiton exists.</p></dd>         <dt></dt><dd><p><code>rotation.title=90</code> <code>(Aliases: rotation.title)</code>:
          The rotation angle for the text in the title panel. Even
          though this can be adjusted, the automatic resizing of the
          title panel will currently not work, so use at own risk.</p></dd>         <dt></dt><dd><p><code>showAxis=TRUE</code>: Boolean controlling whether to
          plot a y axis (only applies to track types where axes are
          implemented).</p></dd>         <dt></dt><dd><p><code>v=-1</code>: Integer scalar. Parameter controlling the
          number of vertical grid lines, see <code>panel.grid</code>
          for details.</p></dd> </dl></p></dd> 

  
</dl> 


    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'>

<p><code><a href='AnnotationTrack-class.html'>AnnotationTrack</a></code></p>
<p><code><a href='DisplayPars-class.html'>DisplayPars</a></code></p>
<p><code><a href='GdObject-class.html'>GdObject</a></code></p>
<p><code>GRanges</code></p>
<p><code><a href='ImageMap-class.html'>ImageMap</a></code></p>
<p><code>IRanges</code></p>
<p><code><a href='RangeTrack-class.html'>RangeTrack</a></code></p>
<p><code><a href='StackedTrack-class.html'>StackedTrack</a></code></p>
<p><code><a href='collapsing.html'>collapsing</a></code></p>
<p><code><a href='DataTrack-class.html'>DataTrack</a></code></p>
<p><code><a href='grouping.html'>grouping</a></code></p>
<p><code>panel.grid</code></p>
<p><code><a href='plotTracks.html'>plotTracks</a></code></p>
<p><code><a href='settings.html'>settings</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>## Construct object</span>
<span class='no'>axTrack</span> <span class='kw'>&lt;-</span> <span class='fu'>GenomeAxisTrack</span>(<span class='kw'>name</span><span class='kw'>=</span><span class='st'>"Axis"</span>,
<span class='no'>range</span> <span class='kw'>&lt;-</span> <span class='fu'>IRanges</span>(<span class='kw'>start</span><span class='kw'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>100</span>, <span class='fl'>300</span>, <span class='fl'>800</span>), <span class='kw'>end</span><span class='kw'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>150</span>, <span class='fl'>400</span>, <span class='fl'>1000</span>)))

<span class='co'># \dontshow{</span>
<span class='co'>## For some annoying reason the postscript device does not know about</span>
<span class='co'>## the sans font</span>
<span class='kw'>if</span>(!<span class='fu'><a href='https://rdrr.io/r/base/interactive.html'>interactive</a></span>())
{
<span class='no'>font</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/ps.options.html'>ps.options</a></span>()$<span class='no'>family</span>
<span class='fu'><a href='DisplayPars-class.html'>displayPars</a></span>(<span class='no'>axTrack</span>) <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='kw'>fontfamily</span><span class='kw'>=</span><span class='no'>font</span>, <span class='kw'>fontfamily.title</span><span class='kw'>=</span><span class='no'>font</span>)
}
<span class='co'># }</span>


<span class='co'>## Plotting</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span>(<span class='no'>axTrack</span>, <span class='kw'>from</span><span class='kw'>=</span><span class='fl'>0</span>, <span class='kw'>to</span><span class='kw'>=</span><span class='fl'>1100</span>)</div><div class='img'><img src='GenomeAxisTrack-class-1.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Track names</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; [1] "Axis"</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>axTrack</span>) <span class='kw'>&lt;-</span> <span class='st'>"foo"</span>

<span class='co'>## Subsetting and splitting</span>
<span class='no'>subTrack</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/subset.html'>subset</a></span>(<span class='no'>axTrack</span>, <span class='kw'>from</span><span class='kw'>=</span><span class='fl'>0</span>, <span class='kw'>to</span><span class='kw'>=</span><span class='fl'>500</span>)
<span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span>(<span class='no'>subTrack</span>)</div><div class='output co'>#&gt; [1] 2</div><div class='input'><span class='no'>subTrack</span>[<span class='fl'>1</span>]</div><div class='output co'>#&gt; Genome axis 'foo'
#&gt; There are annotated axis regions:
#&gt; GRanges object with 1 range and 0 metadata columns:
#&gt;       seqnames    ranges strand
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   [1]    dummy   100-150      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/split.html'>split</a></span>(<span class='no'>axTrack</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>1</span>,<span class='fl'>1</span>,<span class='fl'>2</span>))</div><div class='output co'>#&gt; $`1`
#&gt; Genome axis 'foo'
#&gt; There are annotated axis regions:
#&gt; GRanges object with 2 ranges and 0 metadata columns:
#&gt;       seqnames    ranges strand
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   [1]    dummy   100-150      *
#&gt;   [2]    dummy   300-400      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths
#&gt; 
#&gt; $`2`
#&gt; Genome axis 'foo'
#&gt; There are annotated axis regions:
#&gt; GRanges object with 1 range and 0 metadata columns:
#&gt;       seqnames    ranges strand
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   [1]    dummy  800-1000      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths
#&gt; </div><div class='input'>
<span class='co'>## Accessors</span>
<span class='fu'><a href='https://rdrr.io/r/stats/start.html'>start</a></span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; [1] 100 300 800</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/stats/start.html'>end</a></span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; [1]  150  400 1000</div><div class='input'><span class='fu'>width</span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; [1]  51 101 201</div><div class='input'>
<span class='fu'>strand</span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; [1] "*" "*" "*"</div><div class='input'>
<span class='fu'><a href='RangeTrack-class.html'>range</a></span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; IRanges object with 3 ranges and 0 metadata columns:
#&gt;           start       end     width
#&gt;       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
#&gt;   [1]       100       150        51
#&gt;   [2]       300       400       101
#&gt;   [3]       800      1000       201</div><div class='input'><span class='fu'>ranges</span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; GRanges object with 3 ranges and 0 metadata columns:
#&gt;       seqnames    ranges strand
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
#&gt;   [1]    dummy   100-150      *
#&gt;   [2]    dummy   300-400      *
#&gt;   [3]    dummy  800-1000      *
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## Annotation</span>
<span class='fu'>values</span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; data frame with 0 columns and 3 rows</div><div class='input'>
<span class='co'>## Grouping</span>
<span class='fu'><a href='AnnotationTrack-class.html'>group</a></span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; NULL</div><div class='input'>
<span class='co'>## HTML image map</span>
<span class='fu'><a href='GdObject-class.html'>coords</a></span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; NULL</div><div class='input'><span class='fu'><a href='GdObject-class.html'>tags</a></span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; NULL</div><div class='input'><span class='no'>axTrack</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='plotTracks.html'>plotTracks</a></span>(<span class='no'>axTrack</span>)$<span class='no'>foo</span></div><div class='img'><img src='GenomeAxisTrack-class-2.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'><a href='GdObject-class.html'>coords</a></span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt;          x1    y1        x2    y2
#&gt; 1  13.75372 227.4  40.22553 276.6
#&gt; 2 119.64095 227.4 172.58457 276.6
#&gt; 3 384.35905 227.4 489.71685 276.6</div><div class='input'><span class='fu'><a href='GdObject-class.html'>tags</a></span>(<span class='no'>axTrack</span>)</div><div class='output co'>#&gt; $title
#&gt;   1   2   3 
#&gt; "1" "2" "3" 
#&gt; 
#&gt; $start
#&gt;     1     2     3 
#&gt; "100" "300" "800" 
#&gt; 
#&gt; $end
#&gt;     1     2     3 
#&gt; "150" "400" "999" 
#&gt; </div><div class='input'>
<span class='co'>## adding an axis to another track</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>cyp2b10</span>)
<span class='no'>grTrack</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='GeneRegionTrack-class.html'>GeneRegionTrack</a></span>(<span class='kw'>start</span><span class='kw'>=</span><span class='fl'>26682683</span>, <span class='kw'>end</span><span class='kw'>=</span><span class='fl'>26711643</span>,
        <span class='kw'>rstart</span><span class='kw'>=</span><span class='no'>cyp2b10</span>$<span class='no'>start</span>, <span class='kw'>rends</span><span class='kw'>=</span><span class='no'>cyp2b10</span>$<span class='no'>end</span>, <span class='kw'>chromosome</span><span class='kw'>=</span><span class='fl'>7</span>, <span class='kw'>genome</span><span class='kw'>=</span><span class='st'>"mm9"</span>,
        <span class='kw'>transcript</span><span class='kw'>=</span><span class='no'>cyp2b10</span>$<span class='no'>transcript</span>, <span class='kw'>gene</span><span class='kw'>=</span><span class='no'>cyp2b10</span>$<span class='no'>gene</span>, <span class='kw'>symbol</span><span class='kw'>=</span><span class='no'>cyp2b10</span>$<span class='no'>symbol</span>,
        <span class='kw'>name</span><span class='kw'>=</span><span class='st'>"Cyp2b10"</span>, <span class='kw'>strand</span><span class='kw'>=</span><span class='no'>cyp2b10</span>$<span class='no'>strand</span>)

<span class='fu'><a href='plotTracks.html'>plotTracks</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='no'>grTrack</span>, <span class='fu'>GenomeAxisTrack</span>()))</div><div class='img'><img src='GenomeAxisTrack-class-3.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'><a href='plotTracks.html'>plotTracks</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='no'>grTrack</span>, <span class='fu'>GenomeAxisTrack</span>(<span class='kw'>scale</span><span class='kw'>=</span><span class='fl'>0.1</span>)))</div><div class='img'><img src='GenomeAxisTrack-class-4.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'><a href='plotTracks.html'>plotTracks</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='no'>grTrack</span>, <span class='fu'>GenomeAxisTrack</span>(<span class='kw'>scale</span><span class='kw'>=</span><span class='fl'>5000</span>)))</div><div class='img'><img src='GenomeAxisTrack-class-5.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'><a href='plotTracks.html'>plotTracks</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span>(<span class='no'>grTrack</span>, <span class='fu'>GenomeAxisTrack</span>(<span class='kw'>scale</span><span class='kw'>=</span><span class='fl'>0.5</span>, <span class='kw'>labelPos</span><span class='kw'>=</span><span class='st'>"below"</span>)))</div><div class='img'><img src='GenomeAxisTrack-class-6.png' alt='' width='700' height='433' /></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Florian Hahne, Steffen Durinck, Robert Ivanek, Arne Mueller, Steve Lianoglou, Ge Tan, Lance Parsons, Shraddha Pai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


