<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>GenomeAxisTrack class and methods — GenomeAxisTrack-class • Gviz</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="GenomeAxisTrack class and methods — GenomeAxisTrack-class"><meta property="og:description" content="A class representing a customizable genomic axis."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Gviz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.45.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/Gviz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ivanek/Gviz/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>GenomeAxisTrack class and methods</h1>
    <small class="dont-index">Source: <a href="https://github.com/ivanek/Gviz/blob/HEAD/R/GenomeAxisTrack-class.R" class="external-link"><code>R/GenomeAxisTrack-class.R</code></a></small>
    <div class="hidden name"><code>GenomeAxisTrack-class.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A class representing a customizable genomic axis.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre><code># S4 method for GenomeAxisTrack
initialize(.Object, range, ids, ...)

GenomeAxisTrack(range = NULL, name = "Axis", id, ...)

# S4 method for GenomeAxisTrack
ranges(x)

# S4 method for GenomeAxisTrack
range(x)

# S4 method for GenomeAxisTrack
start(x)

# S4 method for GenomeAxisTrack
start(x) &lt;- value

# S4 method for GenomeAxisTrack
end(x)

# S4 method for GenomeAxisTrack
end(x) &lt;- value

# S4 method for GenomeAxisTrack
width(x)

# S4 method for GenomeAxisTrack
length(x)

# S4 method for GenomeAxisTrack
values(x)

# S4 method for GenomeAxisTrack
strand(x)

# S4 method for GenomeAxisTrack
collapseTrack(
  GdObject,
  min.width = 1,
  min.distance = 0,
  collapse = TRUE,
  diff = .pxResolution(coord = "x"),
  xrange
)

# S4 method for GenomeAxisTrack,ANY,ANY,ANY
[(x, i, j, ..., drop = TRUE)

# S4 method for GenomeAxisTrack
subset(x, from = NULL, to = NULL, sort = FALSE, ...)

# S4 method for GenomeAxisTrack
drawGD(GdObject, minBase, maxBase, prepare = FALSE, subset = TRUE, ...)

# S4 method for GenomeAxisTrack
show(object)</code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>range</dt>
<dd><p>Optional object of class <code>GRanges</code> or <code>IRanges</code> containing
regions to be highlighted on the axis by coloured boxes.</p></dd>


<dt>...</dt>
<dd><p>Additional items which will all be interpreted as further
display parameters. See <code>settings</code> and the "Display Parameters"
section below for details.</p></dd>


<dt>name</dt>
<dd><p>Character scalar of the track's name used in the title panel
when plotting.</p></dd>


<dt>id</dt>
<dd><p>A character vector of the same length as <code>range</code> containing
identifiers for the ranges. If missing, the constructor will try to extract
the ids from <code>names(range)</code>.</p></dd>


<dt>x</dt>
<dd><p>A valid track object class name, or the object itself, in which
case the class is derived directly from it.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>The return value of the constructor function is a new object of class
<code>GenomeAxisTrack</code>.</p>


<p>Objects can be created using the constructor function <code>GenomeAxisTrack</code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>A <code>GenomeAxisTrack</code> can be customized using the familiar display
parameters. By providing a <code>GRanges</code> or <code>IRanges</code> object to the
constructor, ranges on the axis can be further highlighted.</p>
<p>With the <code>scale</code> display parameter, a small scale indicator can be
shown instead of the entire genomic axis. The scale can either be provided
as a fraction of the plotting region (it will be rounded to the nearest
human readable absolute value) or as an absolute value and is always
displayed in bp, kb, mb or gb units. Note that most display parameters for
the <code>GenomeAxisTrack</code> are ignored when a scale is used instead of the
full axis. In particular, only the parameters <code>exponent</code>, <code>alpha</code>,
<code>lwd</code>, <code>col</code>, <code>cex</code>, <code>distFromAxis</code> and <code>labelPos</code>
are used.</p>
    </div>
    <div id="functions">
    <h2>Functions</h2>
    
<ul><li><p><code>initialize(GenomeAxisTrack)</code>: Intialize.</p></li>
<li><p><code>GenomeAxisTrack()</code>: Constructor</p></li>
<li><p><code>ranges(GenomeAxisTrack)</code>: return the genomic coordinates for the
track along with all additional annotation information as an object of
class <code>GRanges</code>.</p></li>
<li><p><code>range(GenomeAxisTrack)</code>: return the genomic coordinates for the
track as an object of class <code>IRanges</code>.
@export</p></li>
<li><p><code>start(GenomeAxisTrack)</code>: return the start coordinates of the track
items.</p></li>
<li><p><code>start(GenomeAxisTrack) &lt;- value</code>: replace the start coordinates of the track
items.</p></li>
<li><p><code>end(GenomeAxisTrack)</code>: return the end coordinates of the track
items.</p></li>
<li><p><code>end(GenomeAxisTrack) &lt;- value</code>: replace the end coordinates of the track
items.</p></li>
<li><p><code>width(GenomeAxisTrack)</code>: return the with of the track items in
genomic coordinates.</p></li>
<li><p><code>length(GenomeAxisTrack)</code>: return the number of items stored in the ranges slot.</p></li>
<li><p><code>values(GenomeAxisTrack)</code>: return all additional annotation information
except for the genomic coordinates for the track items.</p></li>
<li><p><code>strand(GenomeAxisTrack)</code>: return a vector of strand specifiers for
all track items, in the form '+' for the Watson strand, '-' for the Crick
strand or '*' for either of the two.</p></li>
<li><p><code>collapseTrack(GenomeAxisTrack)</code>: preprocess the track before plotting.
This will collapse overlapping track items based on the available resolution
and increase the width and height of all track objects to a minimum value
to avoid rendering issues. See collapsing for details.</p></li>
<li><p><code>x[i</code>: subset the items in the <code>GenomeAxisTrack</code> object.
This is essentially similar to subsetting of the <code>GRanges</code> object in the
<code>range</code> slot. For most applications, the subset method may be more appropriate.</p></li>
<li><p><code>subset(GenomeAxisTrack)</code>: plot subset all the contained tracks in an
<code>GenomeAxisTrack</code> by coordinates and sort if necessary.</p></li>
<li><p><code>drawGD(GenomeAxisTrack)</code>: plot the object to a graphics device.
The return value of this method is the input object, potentially updated
during the plotting operation. Internally, there are two modes in which the
method can be called. Either in 'prepare' mode, in which case no plotting is
done but the object is preprocessed based on the available space, or in
'plotting' mode, in which case the actual graphical output is created.
Since subsetting of the object can be potentially costly, this can be
switched off in case subsetting has already been performed before or
is not necessary.</p></li>
<li><p><code>show(GenomeAxisTrack)</code>: Show method.</p></li>
</ul></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="DisplayPars-class.html">DisplayPars</a></code></p>
<p><code><a href="GdObject-class.html">GdObject</a></code></p>
<p><code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></code></p>
<p><code><a href="HighlightTrack-class.html">HighlightTrack</a></code></p>
<p><code><a href="ImageMap-class.html">ImageMap</a></code></p>
<p><code><a href="https://rdrr.io/pkg/IRanges/man/IRanges-class.html" class="external-link">IRanges</a></code></p>
<p><code><a href="RangeTrack-class.html">RangeTrack</a></code></p>
<p><code><a href="DataTrack-class.html">DataTrack</a></code></p>
<p><code><a href="collapsing.html">collapsing</a></code></p>
<p><code><a href="grouping.html">grouping</a></code></p>
<p><code><a href="https://rdrr.io/pkg/lattice/man/panel.functions.html" class="external-link">panel.grid</a></code></p>
<p><code><a href="plotTracks.html">plotTracks</a></code></p>
<p><code><a href="settings.html">settings</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Florian Hahne</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Construct object</span></span></span>
<span class="r-in"><span><span class="va">axTrack</span> <span class="op">&lt;-</span> <span class="fu">GenomeAxisTrack</span><span class="op">(</span></span></span>
<span class="r-in"><span>    name <span class="op">=</span> <span class="st">"Axis"</span>,</span></span>
<span class="r-in"><span>    <span class="va">range</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">300</span>, <span class="fl">800</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">150</span>, <span class="fl">400</span>, <span class="fl">1000</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># \dontshow{</span></span></span>
<span class="r-in"><span><span class="co">## For some annoying reason the postscript device does not know about</span></span></span>
<span class="r-in"><span><span class="co">## the sans font</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">font</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/ps.options.html" class="external-link">ps.options</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">family</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="DisplayPars-class.html">displayPars</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>fontfamily <span class="op">=</span> <span class="va">font</span>, fontfamily.title <span class="op">=</span> <span class="va">font</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Plotting</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">axTrack</span>, from <span class="op">=</span> <span class="fl">0</span>, to <span class="op">=</span> <span class="fl">1100</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="GenomeAxisTrack-class-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Track names</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Axis"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"foo"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Subsetting and splitting</span></span></span>
<span class="r-in"><span><span class="va">subTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">axTrack</span>, from <span class="op">=</span> <span class="fl">0</span>, to <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">subTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 2</span>
<span class="r-in"><span><span class="va">subTrack</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Genome axis 'foo'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are annotated axis regions:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 1 range and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]    dummy   100-150      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">split</a></span><span class="op">(</span><span class="va">axTrack</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $`1`</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Genome axis 'foo'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are annotated axis regions:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]    dummy   100-150      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]    dummy   300-400      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $`2`</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Genome axis 'foo'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are annotated axis regions:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 1 range and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]    dummy  800-1000      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Accessors</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 100 300 800</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]  150  400 1000</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]  51 101 201</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/strand.html" class="external-link">strand</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "*" "*" "*"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> IRanges object with 3 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           start       end     width</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]       100       150        51</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]       300       400       101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]       800      1000       201</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/range-squeezers.html" class="external-link">ranges</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 3 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]    dummy   100-150      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]    dummy   300-400      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]    dummy  800-1000      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Annotation</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">values</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> data frame with 0 columns and 3 rows</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Grouping</span></span></span>
<span class="r-in"><span><span class="fu"><a href="GdObject-class.html">group</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## HTML image map</span></span></span>
<span class="r-in"><span><span class="fu"><a href="ImageMap-class.html">coords</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span><span class="fu"><a href="ImageMap-class.html">tags</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span><span class="va">axTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span><span class="op">$</span><span class="va">foo</span></span></span>
<span class="r-plt img"><img src="GenomeAxisTrack-class-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="ImageMap-class.html">coords</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          x1    y1        x2    y2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  13.51842 216.6  38.68304 263.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 114.17690 216.6 164.50614 263.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 365.82310 216.6 465.97828 263.4</span>
<span class="r-in"><span><span class="fu"><a href="ImageMap-class.html">tags</a></span><span class="op">(</span><span class="va">axTrack</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $title</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   1   2   3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "1" "2" "3" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $start</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     1     2     3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "100" "300" "800" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $end</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     1     2     3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "150" "400" "999" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## adding an axis to another track</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cyp2b10</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">grTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="GeneRegionTrack-class.html">GeneRegionTrack</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    start <span class="op">=</span> <span class="fl">26682683</span>, end <span class="op">=</span> <span class="fl">26711643</span>,</span></span>
<span class="r-in"><span>    rstart <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">start</span>, rends <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">end</span>, chromosome <span class="op">=</span> <span class="fl">7</span>, genome <span class="op">=</span> <span class="st">"mm9"</span>,</span></span>
<span class="r-in"><span>    transcript <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">transcript</span>, gene <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">gene</span>, symbol <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">symbol</span>,</span></span>
<span class="r-in"><span>    name <span class="op">=</span> <span class="st">"Cyp2b10"</span>, strand <span class="op">=</span> <span class="va">cyp2b10</span><span class="op">$</span><span class="va">strand</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">grTrack</span>, <span class="fu">GenomeAxisTrack</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="GenomeAxisTrack-class-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">grTrack</span>, <span class="fu">GenomeAxisTrack</span><span class="op">(</span>scale <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="GenomeAxisTrack-class-4.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">grTrack</span>, <span class="fu">GenomeAxisTrack</span><span class="op">(</span>scale <span class="op">=</span> <span class="fl">5000</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="GenomeAxisTrack-class-5.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">grTrack</span>, <span class="fu">GenomeAxisTrack</span><span class="op">(</span>scale <span class="op">=</span> <span class="fl">0.5</span>, labelPos <span class="op">=</span> <span class="st">"below"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="GenomeAxisTrack-class-6.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Florian Hahne, Steffen Durinck, Robert Ivanek, Arne Mueller, Steve Lianoglou, Ge Tan, Lance Parsons, Shraddha Pai.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

