<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>SequenceTrack class and methods — SequenceTrack-class • Gviz</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="SequenceTrack class and methods — SequenceTrack-class" />
<meta property="og:description" content="

A track class to represent genomic sequences. The three child classes
  SequenceDNAStringSetTrack, SequenceRNAStringSetTrack 
  and SequenceBSgenomeTrack do most of the work, however in 
  practise they are of no particular relevance to the user." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Gviz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.37.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Gviz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ivanek/Gviz/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>SequenceTrack class and methods</h1>
    
    <div class="hidden name"><code>SequenceTrack-class.Rd</code></div>
    </div>

    <div class="ref-description">
    

<p>A track class to represent genomic sequences. The three child classes
  <code>SequenceDNAStringSetTrack</code>, <code>SequenceRNAStringSetTrack</code> 
  and <code>SequenceBSgenomeTrack</code> do most of the work, however in 
  practise they are of no particular relevance to the user.</p>
    </div>

    <pre class="usage"><span class='fu'>SequenceTrack</span><span class='op'>(</span><span class='va'>sequence</span>, <span class='va'>chromosome</span>, <span class='va'>genome</span>, name<span class='op'>=</span><span class='st'>"SequenceTrack"</span>,
<span class='va'>importFunction</span>, stream<span class='op'>=</span><span class='cn'>FALSE</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='fu'>RNASequenceTrack</span><span class='op'>(</span><span class='va'>sequence</span>, <span class='va'>chromosome</span>, <span class='va'>genome</span>, name<span class='op'>=</span><span class='st'>"SequenceTrack"</span>,
<span class='va'>importFunction</span>, stream<span class='op'>=</span><span class='cn'>FALSE</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>sequence</th>
      <td>

<p>A meta argument to handle the different input types, making the
    construction of a <code>SequenceTrack</code> as flexible as possible.</p>    
<p>The different input options for <code>sequence</code> are:</p>    
<p></p><dl>
      
      <dt></dt><dd><p>An object of class <code>DNAStringSet</code>. The
    individual <code>DNAString</code>s are considered to be
    the different chromosome sequences.</p></dd>
      
      <dt></dt><dd><p>An object of class <code>BSgenome</code>. The
    <code>Gviz</code> package tries to follow the <code>BSgenome</code>
    philosophy in that the respective chromosome sequences are only
    realized once they are first accessed.</p></dd>
      
      <dt></dt><dd><p>A <code>character</code> scalar: in this case the value of the
    <code>sequence</code> argument is considered to be a file path to an
    annotation file on disk. A range of file types are supported by
    the <code>Gviz</code> package as identified by the file extension. See
    the <code>importFunction</code> documentation below for further
    details.</p></dd>

    
</dl><p></p></td>
    </tr>
    <tr>
      <th>chromosome</th>
      <td><p>The currently active chromosome of the track. A
    valid UCSC chromosome identifier if
    <code><a href='https://rdrr.io/r/base/options.html'>options(ucscChromosomeNames=TRUE)</a></code>. Please note that in this
    case only syntactic checking takes place, i.e., the argument value
    needs to be an integer, numeric character or a character of the form
    <code>chrx</code>, where <code>x</code> may be any possible string. The user has
    to make sure that sequences for the respective chromosomes are
    indeed part of the object. If not provided here, the constructor
    will set it to the first available sequence. Please note that by
    definition all objects in the <code>Gviz</code> package can only have a
    single active chromosome at a time (although internally the
    information for more than one chromosome may be present), and the
    user has to call the <code>chromosome&lt;-</code> replacement method in order
    to change to a different active chromosome.</p></td>
    </tr>
    <tr>
      <th>genome</th>
      <td><p>The genome on which the track's ranges are
    defined. Usually this is a valid UCSC genome identifier, however
    this is not being formally checked at this point. For a
    <code>SequenceBSgenomeTrack</code> object, the genome information is
    extracted from the input <code>BSgenome</code> package. For a
    <code>DNAStringSet</code> it has too be provided or the constructor will
    fall back to the default value of <code>NA</code>.</p></td>
    </tr>
    <tr>
      <th>name</th>
      <td><p>Character scalar of the track's name used in the title
    panel when plotting.</p></td>
    </tr>
    <tr>
      <th>importFunction</th>
      <td><p>A user-defined function to be used to import the
    sequence data from a file. This only applies when the <code>sequence</code> argument
    is a character string with the path to the input data file. The
    function needs to accept an argument <code>file</code> containing the file
    path and has to return a proper <code>DNAStringSet</code> object with the
    sequence information per chromosome. A set of default import functions is
    already implemented in the package for a number of different file
    types, and one of these defaults will be picked automatically based
    on the extension of the input file name. If the extension can not be
    mapped to any of the existing import function, an error is raised
    asking for a user-defined import function. Currently the following
    file types can be imported with the default functions:
    <code>fa/fasta</code> and <code>2bit</code>.</p>
<p>Both file types support indexing by genomic coordinates, and it
    makes sense to only load the part of the file that is needed for
    plotting. To this end, the <code>Gviz</code> package defines the derived
    <code>ReferenceSequenceTrack</code> class, which supports streaming data
    from the file system. The user typically does not have to deal with
    this distinction but may rely on the constructor function to make
    the right choice as long as the default import functions are
    used. However, once a user-defined import function has been provided
    and if this function adds support for indexed files, you will have
    to make the constructor aware of this fact by setting the
    <code>stream</code> argument to <code>TRUE</code>. Please note that in this case
    the import function needs to accept a second mandatory argument
    <code>selection</code> which is a <code>GRanges</code> object containing the
    dimensions of the plotted genomic range. As before, the function has
    to return an appropriate <code>DNAStringSet</code> object.</p></td>
    </tr>
    <tr>
      <th>stream</th>
      <td><p>A logical flag indicating that the user-provided import
    function can deal with indexed files and knows how to process the
    additional <code>selection</code> argument when accessing the data on
    disk. This causes the constructor to return a
    <code>ReferenceSequenceTrack</code> object which will grab the necessary
    data on the fly during each plotting operation.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional items which will all be interpreted as further
    display parameters. See <code><a href='settings.html'>settings</a></code> and the "Display
    Parameters" section below for details.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="objects-from-the-class"><a class="anchor" href="#objects-from-the-class"></a>Objects from the class</h2>

    
  
<p>Objects can be created using the constructor function
  <code>SequenceTrack</code>.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    

<p>The return value of the constructor function is a new object of class
  <code>SequenceDNAStringSetTrack</code>, <code>SequenceBSgenomeTrack</code> ore
  <code>ReferenceSequenceTrack</code>, depending on the constructor
  arguments. Typically the user will not have to be troubled with this
  distinction and can rely on the constructor to make the right choice.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>details</h2>

    

<p>Depending on the available space the class will use different options
  to plot a sequence. If single letters can be accomodated without
  overplotting those will be show. Otherwise, colored boxes will be used
  to indicate letters, and if there is not enough horizontal room to
  show those, a simple line will indicate presence of a sequence. The
  <code>min.width</code> and <code>fontsize</code> display parameters directly
  control this behaviour. Each of the five possible nucleotides (G, A, T,
  C, and N) will be endoded in a separate color. As default we use the
  colors suggested in the <code>biovizBase</code> package, but a user is
  free to set their own color scheme by providing a named character
  vector with color as display parameter <code>fontcolor</code>, with names
  equal to the five possible bases.</p>
    <h2 class="hasAnchor" id="slots"><a class="anchor" href="#slots"></a>Slots</h2>

    

<p></p><dl>
        
    <dt><code>chromosome</code>:</dt><dd><p>Object of class <code>"character"</code>, the
      chromosome on which the track is defined. There can only be a
      single chromosome for one track. Throughout the
      package, chromosome name have to be entered either as a single
      integer scalar or as a character scalar of the form
      <code>chrXYZ</code>, where <em>XYZ</em> may be an arbitrary
      character string.</p></dd>
    
    <dt><code>genome</code>:</dt><dd><p>Object of class <code>"character"</code>, the genome
      for which the track is defined. This should be a valid UCSC genome
      identifier, however this may not always be formally checked upon
      object instantiation.</p></dd>
    
    <dt><code>dp</code>:</dt><dd><p>Object of class
      <code><a href='DisplayPars-class.html'>DisplayPars</a></code>, inherited from class
      <code><a href='GdObject-class.html'>GdObject</a></code>.</p></dd>
    
    <dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code>, inherited
      from class <code><a href='GdObject-class.html'>GdObject</a></code></p></dd>

    <dt><code>imageMap</code>:</dt><dd><p>Object of class
      <code><a href='ImageMap-class.html'>ImageMap</a></code>, inherited from class
      <code><a href='GdObject-class.html'>GdObject</a></code></p></dd>
  
  
</dl>
  

    <h2 class="hasAnchor" id="extends"><a class="anchor" href="#extends"></a>Extends</h2>

    
  
<p>Class <code>"<a href='GdObject-class.html'>GdObject</a>"</code>, directly.</p>
    <h2 class="hasAnchor" id="methods"><a class="anchor" href="#methods"></a>Methods</h2>

    
    
<p>In the following code chunks, <code>obj</code> is considered to be an object
  inheriting from class <code>SequenceTrack</code>.</p>  
<p><b><em>Exported in the name space:</em></b></p>
<p></p><dl>
      
      <dt>chromosome</dt><dd><p><code>signature(GdObject="SequenceTrack")</code>:
    return the chromosome for which the track is defined.    <em>Usage:</em>    <code><a href='RangeTrack-class.html'>chromosome(GdObject)</a></code>        <em>Examples:</em>    <dl>      <dt></dt><dd><p><code><a href='RangeTrack-class.html'>chromosome(obj)</a></code></p></dd></dl></p></dd>
    
      <dt>chromosome&lt;-</dt><dd><p><code>signature(GdObject="SequenceTrack")</code>:
    replace the value of the track's chromosome. This has to be a
    valid UCSC chromosome identifier or an integer or character
    scalar that can be reasonably coerced into one.    <em>Usage:</em>    <code>chromosome&lt;-(GdObject, value)</code>    <em>Additional Arguments:</em>    <dl>      <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>        <em>Examples:</em>    <dl>      <dt></dt><dd><p><code>chromosome(obj) &lt;- "chr12"</code></p></dd></dl></p></dd>
    
    <dt>genome</dt><dd><p><code>signature(x="SequenceTrack")</code>: return the track's genome.       <em>Usage:</em>      <code>genome(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>genome(obj)</code></p></dd></dl></p></dd>

    <dt>genome&lt;-</dt><dd><p><code>signature(x="SequenceTrack")</code>: set the track's
      genome. Usually this has to be a valid UCSC identifier, however
      this is not formally enforced here.      <em>Usage:</em>      <code>genome&lt;-(x, value)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>value</code>: replacement value.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>genome(obj) &lt;- "mm9"</code></p></dd></dl></p></dd>

    <dt>length</dt><dd><p><code>signature(x="SequenceTrack")</code>: return the number
      of nucleotides in the track's sequence.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/length.html'>length(x)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/length.html'>length(obj)</a></code></p></dd></dl></p></dd>
   
    <dt>seqnames</dt><dd><p><code>signature(x="SequenceTrack")</code>: return the
    names (i.e., the chromosome) of the sequences contained in the object.      <em>Usage:</em>      <code>values(x)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>seqnames(obj)</code></p></dd></dl></p></dd>

    <dt>subseq</dt><dd><p><code>signature(x="SequenceTrack")</code>: Extract a
      sub-sequence from the track.      <em>Usage:</em>      <code>subseq(x, start=NA, end=NA, width=NA)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>start</code>: the start coordinate for the sub-sequence.</p></dd>    <dt></dt><dd><p><code>end</code>: the end coordinate for the sub-sequence.</p></dd>    <dt></dt><dd><p><code>width</code>: the width of the sub-sequence.</p></dd></dl>       <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>subseq(obj, 1, 10)</code></p></dd></dl></p></dd>
  
</dl>
    
<p><b><em>Internal methods:</em></b></p>  
<p></p><dl>
    
    <dt>initialize</dt><dd><p><code>signature(.Object="SequenceTrack")</code>:
      initialize the object.</p></dd>
  
</dl>

<p><b><em>Inherited methods:</em></b></p>  
<p></p><dl>

     <dt>displayPars</dt><dd><p><code>signature(x="SequenceTrack",
      name="character")</code>: list the value of the display parameter
      <code>name</code>. See <code><a href='settings.html'>settings</a></code> for details on display
      parameters and customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>displayPars(x, name)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>displayPars(obj, "col")</a></code></p></dd></dl></p></dd>
      
    
    <dt>displayPars</dt><dd><p><code>signature(x="SequenceTrack", name="missing")</code>:
      list the value of all available display parameters. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>displayPars(obj)</a></code></p></dd></dl></p></dd>
    
    <dt>getPar</dt><dd><p><code>signature(x="SequenceTrack", name="character")</code>:
      alias for the <code>displayPars</code> method. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>getPar(x, name)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>getPar(obj, "col")</a></code></p></dd></dl></p></dd>
    
    <dt>getPar</dt><dd><p><code>signature(x="SequenceTrack", name="missing")</code>:
      alias for the <code>displayPars</code> method. See
      <code><a href='settings.html'>settings</a></code> for details on display parameters and
      customization.      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>getPar(obj)</a></code></p></dd></dl></p></dd>
    
    <dt>displayPars&lt;-</dt><dd><p><code>signature(x="SequenceTrack", value="list")</code>:
      set display parameters using the values of the named list in
      <code>value</code>. See <code><a href='settings.html'>settings</a></code> for details on display
      parameters and customization.      <em>Usage:</em>      <code>displayPars&lt;-(x, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>displayPars(obj) &lt;- list(col="red", lwd=2)</code></p></dd></dl></p></dd>
    
    <dt>setPar</dt><dd><p><code>signature(x="SequenceTrack", value="character")</code>:
      set the single display parameter <code>name</code> to <code>value</code>. Note
      that display parameters in the <code>SequenceTrack</code> class are
      pass-by-reference, so no re-assignmnet to the symbol <code>obj</code> is
      necessary. See <code><a href='settings.html'>settings</a></code> for details on display
      parameters and customization.      <em>Usage:</em>      <code><a href='DisplayPars-class.html'>setPar(x, name, value)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>name</code>: the name of the display parameter to set.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>setPar(obj, "col", "red")</a></code></p></dd></dl></p></dd>
    
    <dt>setPar</dt><dd><p><code>signature(x="SequenceTrack", value="list")</code>: set
      display parameters by the values of the named list in
      <code>value</code>.  Note that display parameters in the
      <code>SequenceTrack</code> class are pass-by-reference, so no re-assignmnet
      to the symbol <code>obj</code> is necessary. See <code><a href='settings.html'>settings</a></code>
      for details on display parameters and customization.      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='DisplayPars-class.html'>setPar(obj, list(col="red", lwd=2))</a></code></p></dd></dl></p></dd>
    
    <dt>names</dt><dd><p><code>signature(x="SequenceTrack")</code>: return the value of
      the <code>name</code> slot.      <em>Usage:</em>      <code><a href='https://rdrr.io/r/base/names.html'>names(x)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='https://rdrr.io/r/base/names.html'>names(obj)</a></code></p></dd></dl></p></dd>
    
    <dt>names&lt;-</dt><dd><p><code>signature(x="SequenceTrack", value="character")</code>:
      set the value of the <code>name</code> slot.      <em>Usage:</em>      <code>names&lt;-(x, value)</code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>names(obj) &lt;- "foo"</code></p></dd></dl></p></dd>

    <dt>coords</dt><dd><p><code>signature(ImageMap="SequenceTrack")</code>: return the
      coordinates from the internal image map.      <em>Usage:</em>      <code><a href='GdObject-class.html'>coords(ImageMap)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='GdObject-class.html'>coords(obj)</a></code></p></dd></dl></p></dd>

    <dt>tags</dt><dd><p><code>signature(x="SequenceTrack")</code>: return the tags from the
      internal image map.      <em>Usage:</em>      <code><a href='GdObject-class.html'>tags(x)</a></code>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code><a href='GdObject-class.html'>tags(obj)</a></code></p></dd></dl></p></dd>
    
    <dt>drawAxis</dt><dd><p><code>signature(GdObject="SequenceTrack")</code>: add a
      y-axis to the title panel of a track if necessary. Unless
      overwritten in one of the sub-classes this usualy does not plot
      anything and returns <code>NULL</code>.      <em>Usage:</em>      <code><a href='GdObject-class.html'>drawAxis(x, ...)</a></code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>...</code>: all further arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>Gviz:::drawAxis(obj)</code></p></dd></dl></p></dd>
    
    <dt>drawGrid</dt><dd><p><code>signature(GdObject="SequenceTrack")</code>: superpose a
      grid on top of a track if necessary. Unless overwritten in one of
      the sub-classes this usualy does not plot anything and returns
      <code>NULL</code>.      <em>Usage:</em>      <code>drawGrid(GdObject, ...)</code>      <em>Additional Arguments:</em>      <dl>    <dt></dt><dd><p><code>...</code>: additional arguments are ignored.</p></dd></dl>      <em>Examples:</em>      <dl>    <dt></dt><dd><p><code>Gviz:::drawGrid(obj)</code></p></dd></dl></p></dd>
  
</dl>

    <h2 class="hasAnchor" id="display-parameters"><a class="anchor" href="#display-parameters"></a>Display Parameters</h2>

     

<p>The following display parameters are set for objects of class
  <code>SequenceTrack</code> upon instantiation, unless one or more of them
  have already been set by one of the optional sub-class initializers,
  which always get precedence over these global defaults. See
  <code><a href='settings.html'>settings</a></code> for details on setting graphical parameters
  for tracks.</p><dl>

    <dt></dt><dd><p><code>add53=FALSE</code>: Logical scalar. Add a direction
      indicator.</p></dd> 

    <dt></dt><dd><p><code>background.title="transparent"</code>: Character scalar.
      Make the title panel background transparent by default.</p></dd> 

    <dt></dt><dd><p><code>col="darkgray"</code>: Character scalar. The color of the
      line when no indiviual letters can be plotted due to size
      limitations.</p></dd> 

    <dt></dt><dd><p><code>complement=FALSE</code>: Logical scalar. Plot the sequence
      complement.</p></dd> 

    <dt></dt><dd><p><code>fontface=2</code>: Numeric scalar. The face of the font.</p></dd> 

    <dt></dt><dd><p><code>fontsize=10</code>: Numeric scalar. Controls the size of
      the sequence letters and thus also the level of plotable details.</p></dd> 

    <dt></dt><dd><p><code>lwd=2</code>: Numeric scalar. The width of the line when
      no indiviual letters can be plotted due to size limitations.</p></dd> 

    <dt></dt><dd><p><code>min.width=2</code>: Numeric scalar. The minimum width in
      pixels of the colored boxes that are drawn when no indiviual
      letters can be plotted due to size limitations. If the horizontal
      space that a single base occupies is smaller than this value,
      only a horizontal line is drawn to indicate the presence of a
      sequence.</p></dd> 

    <dt></dt><dd><p><code>noLetters=FALSE</code>: Logical scalar. Always plot colored
      boxes (or a line) regardles of the available space.</p></dd> 

    <dt></dt><dd><p><code>showTitle=FALSE</code>: Logical scalar. Do not show a title
      panel by default.</p></dd> 

    <dt></dt><dd><p><code>size=NULL</code>: Numeric scalar. The size of the track
      item. Defaults to auto-detect the size based on the other parameter
      settings.</p></dd> 

  
</dl> 

<p>Additional display parameters are being inherited from the respective
  parent classes. Note that not all of them may have an effect on the
  plotting of <code>SequenceTrack</code> objects.</p>
<p></p><dl>

    <dt></dt><dd><p><code><a href='GdObject-class.html'>GdObject</a></code>:       <dl>        <dt></dt><dd><p><code>alpha=1</code>: Numeric scalar. The transparency for
          all track items.</p></dd>         <dt></dt><dd><p><code>alpha.title=NULL</code>: Numeric scalar. The transparency
          for the title panel.</p></dd>         <dt></dt><dd><p><code>background.legend="transparent"</code>: Integer or
          character scalar. The background color for the legend.</p></dd>         <dt></dt><dd><p><code>background.panel="transparent"</code>: Integer or
          character scalar. The background color of the content panel.</p></dd>         <dt></dt><dd><p><code>cex=1</code>: Numeric scalar. The overall font expansion
          factor for all text and glyphs, unless a more specific
          definition exists.</p></dd>         <dt></dt><dd><p><code>cex.axis=NULL</code>: Numeric scalar. The expansion
          factor for the axis annotation. Defaults to <code>NULL</code>, in
          which case it is automatically determined based on the
          available space.</p></dd>         <dt></dt><dd><p><code>cex.title=NULL</code>: Numeric scalar. The expansion
          factor for the title panel. This effects the fontsize of both
          the title and the axis, if any. Defaults to <code>NULL</code>,
          which means that the text size is automatically adjusted to
          the available space.</p></dd>         <dt></dt><dd><p><code>col.axis="white"</code>: Integer or character scalar.
          The font and line color for the y axis, if any.</p></dd>         <dt></dt><dd><p><code>col.border.title="white"</code>: Integer or character
          scalar. The border color for the title panels.</p></dd>         <dt></dt><dd><p><code>col.frame="lightgray"</code>: Integer or character
          scalar. The line color used for the panel frame, if
          <code>frame==TRUE</code></p></dd>         <dt></dt><dd><p><code>col.grid="#808080"</code>: Integer or character scalar.
          Default line color for grid lines, both when <code>type=="g"</code>
          in <code><a href='DataTrack-class.html'>DataTrack</a></code>s and when display parameter
          <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>col.line=NULL</code>: Integer or character scalar.
          Default colors for plot lines. Usually the same as the global
          <code>col</code> parameter.</p></dd>         <dt></dt><dd><p><code>col.symbol=NULL</code>: Integer or character scalar.
          Default colors for plot symbols. Usually the same as the
          global <code>col</code> parameter.</p></dd>         <dt></dt><dd><p><code>col.title="white"</code> <code>(Aliases: fontcolor.title)</code>:
          Integer or character scalar. The border color for the title panels</p></dd>         <dt></dt><dd><p><code>collapse=TRUE</code>: Boolean controlling whether to
          collapse the content of the track to accomodate the minimum
          current device resolution. See <code><a href='collapsing.html'>collapsing</a></code> for
          details.</p></dd>         <dt></dt><dd><p><code>fill="lightgray"</code>: Integer or character scalar.
          Default fill color setting for all plotting elements, unless
          there is a more specific control defined elsewhere.</p></dd>         <dt></dt><dd><p><code>fontcolor="black"</code>: Integer or character scalar.
          The font color for all text, unless a more specific definition
          exists.</p></dd>         <dt></dt><dd><p><code>fontface.title=2</code>: Integer or character scalar.
          The font face for the title panels.</p></dd>         <dt></dt><dd><p><code>fontfamily="sans"</code>: Integer or character scalar.
          The font family for all text, unless a more specific definition
          exists.</p></dd>         <dt></dt><dd><p><code>fontfamily.title="sans"</code>: Integer or character
          scalar. The font family for the title panels.</p></dd>         <dt></dt><dd><p><code>frame=FALSE</code>: Boolean. Draw a frame around the
          track when plotting.</p></dd>         <dt></dt><dd><p><code>grid=FALSE</code>: Boolean, switching on/off the plotting
          of a grid.</p></dd>         <dt></dt><dd><p><code>h=-1</code>: Integer scalar. Parameter controlling the
          number of horizontal grid lines, see <code><a href='https://rdrr.io/pkg/lattice/man/panel.functions.html'>panel.grid</a></code>
          for details.</p></dd>         <dt></dt><dd><p><code>lineheight=1</code>: Numeric scalar. The font line
          height for all text, unless a more specific definition exists.</p></dd>         <dt></dt><dd><p><code>lty="solid"</code>: Numeric scalar. Default line type
          setting for all plotting elements, unless there is a more
          specific control defined elsewhere.</p></dd>         <dt></dt><dd><p><code>lty.grid="solid"</code>: Integer or character scalar.
          Default line type for grid lines, both when <code>type=="g"</code>
          in <code><a href='DataTrack-class.html'>DataTrack</a></code>s and when display parameter
          <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>lwd.border.title=1</code>: Integer scalar. The border
          width for the title panels.</p></dd>         <dt></dt><dd><p><code>lwd.grid=1</code>: Numeric scalar. Default line width
          for grid lines, both when <code>type=="g"</code> in <code><a href='DataTrack-class.html'>DataTrack</a></code>s
          and when display parameter <code>grid==TRUE</code>.</p></dd>         <dt></dt><dd><p><code>lwd.title=1</code>: Integer scalar. The border width
          for the title panels</p></dd>         <dt></dt><dd><p><code>min.distance=1</code>: Numeric scalar. The minimum
          pixel distance before collapsing range items, only if
          <code>collapse==TRUE</code>. See <code><a href='collapsing.html'>collapsing</a></code> for details.</p></dd>         <dt></dt><dd><p><code>min.height=3</code>: Numeric scalar. The minimum range
          height in pixels to display. All ranges are expanded to this
          size in order to avoid rendering issues. See <code><a href='collapsing.html'>collapsing</a></code>
          for details.</p></dd>         <dt></dt><dd><p><code>reverseStrand=FALSE</code>: Logical scalar. Set up the
          plotting coordinates in 3' -&gt; 5' direction if <code>TRUE</code>.
          This will effectively mirror the plot on the vertical axis.</p></dd>         <dt></dt><dd><p><code>rotation=0</code>: The rotation angle for all text
          unless a more specific definiton exists.</p></dd>         <dt></dt><dd><p><code>rotation.title=90</code> <code>(Aliases: rotation.title)</code>:
          The rotation angle for the text in the title panel. Even
          though this can be adjusted, the automatic resizing of the
          title panel will currently not work, so use at own risk.</p></dd>         <dt></dt><dd><p><code>showAxis=TRUE</code>: Boolean controlling whether to
          plot a y axis (only applies to track types where axes are
          implemented).</p></dd>         <dt></dt><dd><p><code>v=-1</code>: Integer scalar. Parameter controlling the
          number of vertical grid lines, see <code><a href='https://rdrr.io/pkg/lattice/man/panel.functions.html'>panel.grid</a></code>
          for details.</p></dd> </dl></p></dd> 

  
</dl> 


    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Florian Hahne</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'>

<p><code><a href='AnnotationTrack-class.html'>AnnotationTrack</a></code></p>
<p><code>BSgenome</code></p>
<p><code><a href='DataTrack-class.html'>DataTrack</a></code></p>
<p><code><a href='DisplayPars-class.html'>DisplayPars</a></code></p>
<p><code><a href='https://rdrr.io/pkg/Biostrings/man/DNAString-class.html'>DNAString</a></code></p>
<p><code><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html'>DNAStringSet</a></code></p>
<p><code><a href='GdObject-class.html'>GdObject</a></code></p>
<p><code><a href='GeneRegionTrack-class.html'>GeneRegionTrack</a></code></p>
<p><code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></code></p>
<p><code><a href='ImageMap-class.html'>ImageMap</a></code></p>
<p><code>IRanges</code></p>
<p><code><a href='collapsing.html'>collapsing</a></code></p>
<p><code><a href='https://rdrr.io/pkg/lattice/man/panel.functions.html'>panel.grid</a></code></p>
<p><code><a href='plotTracks.html'>plotTracks</a></code></p>
<p><code><a href='settings.html'>settings</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## An empty object</span>
<span class='fu'>SequenceTrack</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; Sequence track 'SequenceTrack':
#&gt; | genome: NA
#&gt; | chromosomes: 0
#&gt; | active chromosome: chrNA (0 nucleotides)</div><div class='input'>
<span class='co'>## Construct from DNAStringSet</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://bioconductor.org/packages/Biostrings'>Biostrings</a></span><span class='op'>)</span>
<span class='va'>letters</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"A"</span>, <span class='st'>"C"</span>, <span class='st'>"T"</span>, <span class='st'>"G"</span>, <span class='st'>"N"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>999</span><span class='op'>)</span>
<span class='va'>seqs</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html'>DNAStringSet</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>chr1<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/sample.html'>sample</a></span><span class='op'>(</span><span class='va'>letters</span>, <span class='fl'>100000</span>, <span class='cn'>TRUE</span><span class='op'>)</span>,
collapse<span class='op'>=</span><span class='st'>""</span><span class='op'>)</span>, chr2<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/sample.html'>sample</a></span><span class='op'>(</span><span class='va'>letters</span>, <span class='fl'>200000</span>, <span class='cn'>TRUE</span><span class='op'>)</span>, collapse<span class='op'>=</span><span class='st'>""</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>sTrack</span> <span class='op'>&lt;-</span> <span class='fu'>SequenceTrack</span><span class='op'>(</span><span class='va'>seqs</span>, genome<span class='op'>=</span><span class='st'>"hg19"</span><span class='op'>)</span>
<span class='va'>sTrack</span>
</div><div class='output co'>#&gt; Sequence track 'SequenceTrack':
#&gt; | genome: hg19
#&gt; | chromosomes: 2
#&gt; | active chromosome: chr1 (100000 nucleotides)
#&gt; Call seqnames() to list all available chromosomes
#&gt; Call chromosome()&lt;- to change the active chromosome</div><div class='input'>
<span class='co'>## Construct from BSGenome object</span>
<span class='kw'>if</span><span class='op'>(</span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>require</a></span><span class='op'>(</span><span class='va'>BSgenome.Hsapiens.UCSC.hg19</span><span class='op'>)</span><span class='op'>)</span><span class='op'>{</span>
<span class='va'>sTrack</span> <span class='op'>&lt;-</span> <span class='fu'>SequenceTrack</span><span class='op'>(</span><span class='va'>Hsapiens</span><span class='op'>)</span>
<span class='va'>sTrack</span>
<span class='op'>}</span>
</div><div class='output co'>#&gt; Sequence track 'SequenceTrack':
#&gt; | genome: hg19
#&gt; | chromosomes: 298
#&gt; | active chromosome: chr1 (249250621 nucleotides)
#&gt; Call seqnames() to list all available chromosomes
#&gt; Call chromosome()&lt;- to change the active chromosome
#&gt; Parent BSgenome object:
#&gt; | organism: Homo sapiens
#&gt; | provider: UCSC
#&gt; | provider version: hg19
#&gt; | release date: June 2013
#&gt; | package name: BSgenome.Hsapiens.UCSC.hg19</div><div class='input'>

<span class='co'>## Set active chromosome</span>
<span class='fu'><a href='RangeTrack-class.html'>chromosome</a></span><span class='op'>(</span><span class='va'>sTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "chr1"</div><div class='input'><span class='fu'><a href='RangeTrack-class.html'>chromosome</a></span><span class='op'>(</span><span class='va'>sTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"chr2"</span>
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'>seqnames</span><span class='op'>(</span><span class='va'>sTrack</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "chr1" "chr2" "chr3" "chr4" "chr5" "chr6"</div><div class='input'>


<span class='co'># \dontshow{</span>
<span class='co'>## For some annoying reason the postscript device does not know about</span>
<span class='co'>## the sans font</span>
<span class='kw'>if</span><span class='op'>(</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/interactive.html'>interactive</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>
<span class='op'>{</span>
<span class='va'>font</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/grDevices/ps.options.html'>ps.options</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>$</span><span class='va'>family</span>
<span class='fu'><a href='DisplayPars-class.html'>displayPars</a></span><span class='op'>(</span><span class='va'>sTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>fontfamily<span class='op'>=</span><span class='va'>font</span>, fontfamily.title<span class='op'>=</span><span class='va'>font</span><span class='op'>)</span>
<span class='op'>}</span>
<span class='co'># }</span>

<span class='co'>## Plotting</span>
<span class='co'>## Sequences</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>sTrack</span>, from<span class='op'>=</span><span class='fl'>199970</span>, to<span class='op'>=</span><span class='fl'>200000</span><span class='op'>)</span>
</div><div class='img'><img src='SequenceTrack-class-1.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'>## Boxes</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>sTrack</span>, from<span class='op'>=</span><span class='fl'>199800</span>, to<span class='op'>=</span><span class='fl'>200000</span><span class='op'>)</span>
</div><div class='img'><img src='SequenceTrack-class-2.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'>## Line</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>sTrack</span>, from<span class='op'>=</span><span class='fl'>1</span>, to<span class='op'>=</span><span class='fl'>200000</span><span class='op'>)</span>
</div><div class='img'><img src='SequenceTrack-class-3.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'>## Force boxes</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>sTrack</span>, from<span class='op'>=</span><span class='fl'>199970</span>, to<span class='op'>=</span><span class='fl'>200000</span>, noLetters<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='img'><img src='SequenceTrack-class-4.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'>## Direction indicator</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>sTrack</span>, from<span class='op'>=</span><span class='fl'>199970</span>, to<span class='op'>=</span><span class='fl'>200000</span>, add53<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='img'><img src='SequenceTrack-class-5.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'>## Sequence complement</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>sTrack</span>, from<span class='op'>=</span><span class='fl'>199970</span>, to<span class='op'>=</span><span class='fl'>200000</span>, add53<span class='op'>=</span><span class='cn'>TRUE</span>, complement<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='img'><img src='SequenceTrack-class-6.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'>## Colors</span>
<span class='fu'><a href='plotTracks.html'>plotTracks</a></span><span class='op'>(</span><span class='va'>sTrack</span>, from<span class='op'>=</span><span class='fl'>199970</span>, to<span class='op'>=</span><span class='fl'>200000</span>, add53<span class='op'>=</span><span class='cn'>TRUE</span>, fontcolor<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>A<span class='op'>=</span><span class='fl'>1</span>,
C<span class='op'>=</span><span class='fl'>1</span>, G<span class='op'>=</span><span class='fl'>1</span>, T<span class='op'>=</span><span class='fl'>1</span>, N<span class='op'>=</span><span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='img'><img src='SequenceTrack-class-7.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Track names</span>
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>sTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "SequenceTrack"</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>sTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"foo"</span>

<span class='co'>## Accessors</span>
<span class='fu'>genome</span><span class='op'>(</span><span class='va'>sTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "hg19"</div><div class='input'><span class='fu'>genome</span><span class='op'>(</span><span class='va'>sTrack</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='st'>"mm9"</span>
<span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>sTrack</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] 243199373</div><div class='input'>
<span class='co'>## Sequence extraction</span>
<span class='fu'>subseq</span><span class='op'>(</span><span class='va'>sTrack</span>, start<span class='op'>=</span><span class='fl'>100000</span>, width<span class='op'>=</span><span class='fl'>20</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 20-letter DNAString object
#&gt; seq: TGTCCAAATATGTCTGGTGA</div><div class='input'><span class='co'>## beyond the stored sequence range</span>
<span class='fu'>subseq</span><span class='op'>(</span><span class='va'>sTrack</span>, start<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>sTrack</span><span class='op'>)</span>, width<span class='op'>=</span><span class='fl'>20</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 20-letter DNAString object
#&gt; seq: N-------------------</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Florian Hahne, Steffen Durinck, Robert Ivanek, Arne Mueller, Steve Lianoglou, Ge Tan, Lance Parsons, Shraddha Pai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


