<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Meta-constructor for Gviz tracks fetched directly from the various
UCSC data sources. — UcscTrack • Gviz</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Meta-constructor for Gviz tracks fetched directly from the various
UCSC data sources. — UcscTrack"><meta property="og:description" content="The UCSC data base provides a wealth of annotation information. This
function can be used to access UCSC, to retrieve the data available there
and to return it as an annotation track object amenable to plotting with
plotTracks."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Gviz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.45.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/Gviz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ivanek/Gviz/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Meta-constructor for Gviz tracks fetched directly from the various
UCSC data sources.</h1>
    <small class="dont-index">Source: <a href="https://github.com/ivanek/Gviz/blob/HEAD/R/UcscTrack.R" class="external-link"><code>R/UcscTrack.R</code></a></small>
    <div class="hidden name"><code>UcscTrack.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The UCSC data base provides a wealth of annotation information. This
function can be used to access UCSC, to retrieve the data available there
and to return it as an annotation track object amenable to plotting with
<code><a href="plotTracks.html">plotTracks</a></code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">UcscTrack</span><span class="op">(</span></span>
<span>  <span class="va">track</span>,</span>
<span>  table <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  trackType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AnnotationTrack"</span>, <span class="st">"GeneRegionTrack"</span>, <span class="st">"DataTrack"</span>, <span class="st">"GenomeAxisTrack"</span><span class="op">)</span>,</span>
<span>  <span class="va">genome</span>,</span>
<span>  <span class="va">chromosome</span>,</span>
<span>  name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">from</span>,</span>
<span>  <span class="va">to</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>track</dt>
<dd><p>Character, the name of the track to fetch from UCSC. To find
out about available tracks please consult the online table browser at
<a href="http://genome.ucsc.edu/cgi-bin/hgTables?command=start" class="external-link">http://genome.ucsc.edu/cgi-bin/hgTables?command=start</a>.</p></dd>


<dt>table</dt>
<dd><p>Character, the name of the table to fetch from UCSC, or
<code>NULL</code>, in which case the default selection of tables is used. To find
out about available tables for a given track please consult the online table
browser at <a href="http://genome.ucsc.edu/cgi-bin/hgTables?command=start" class="external-link">http://genome.ucsc.edu/cgi-bin/hgTables?command=start</a>.</p></dd>


<dt>trackType</dt>
<dd><p>Character, one in <code>c("AnnotationTrack",
"GeneRegionTrack", "DataTrack", "GenomeAxisTrack")</code>. The function will try
to coerce the downloaded data in an object of this class. See below for
details.</p></dd>


<dt>genome</dt>
<dd><p>Character, a valid USCS genome identifier for which to fetch
the data.</p></dd>


<dt>chromosome</dt>
<dd><p>Character, a valid USCS character identifier for which to
fetch the data.</p></dd>


<dt>name</dt>
<dd><p>Character, the name to use for the resulting track object.</p></dd>


<dt>from, to</dt>
<dd><p>A range of genomic locations for which to fetch data.</p></dd>


<dt>...</dt>
<dd><p>All additional named arguments are expected to be either
display parameters for the resulting objects, or character scalars of column
names in the downloaded UCSC data tables that are matched by name to
available arguments in the respective constructor functions as defined by
the <code>trackType</code> argument. See Details section for more information.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>An annotation track object as determined by <code>trackType</code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p><code>clearSessionCache</code> is can be called to remove all cached items from
the session which are generated when connecting with the UCSC data base.</p>
<p>The data stored at the UCSC data bases can be of different formats: gene or
transcript model data, simple annotation features like CpG Island locations
or SNPs, or numeric data like conservation or mapability. This function
presents a unified API to download all kinds of data and to map them back to
one of the annotation track objects defined in this package. The type of
object to hold the data has to be given in the <code>trackType</code> argument,
and subsequently the function passes all data on to the respective object
constructor. All additional named arguments are considered to be relevant
for the constructor of choice, and single character scalars are replaced by
the respective data columns in the downloaded UCSC tables if available. For
instance, assuming the table for track 'foo' contains the columns 'id',
'type', 'fromLoc' and 'toLoc', giving the feature identifier, type, start
end end location. In order to create an <code><a href="AnnotationTrack-class.html">AnnotationTrack</a></code>
object from that data, we have to pass the additional named arguments
<code>id="id"</code>, <code>feature="type"</code>, <code>start="fromLoc"</code> and
codeend="toLoc" to the <code>UcscTrack</code> function. The complete function call
could look like this:</p>
<p><code>UcscTrack(track="foo", genome="mm9", chromosome=3, from=1000,
to=10000, trackType="AnnotationTrack", id="id", feature="type",
start="from", end="to")</code></p>
<p>To reduce the bandwidth, some caching of the UCSC connection takes place. In
order to remove these cached session items, call <code>clearSessionCache</code>.</p>
<p>The <code>Gviz.ucscUrl</code> option controls which URL is being used to connect
to UCSC. For instance, one could switch to the European UCSC mirror by
calling <code>options(Gviz.ucscUrl="http://genome-euro.ucsc.edu/cgi-bin/"</code>.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="AnnotationTrack-class.html">AnnotationTrack</a></code></p>
<p><code><a href="DataTrack-class.html">DataTrack</a></code></p>
<p><code><a href="GeneRegionTrack-class.html">GeneRegionTrack</a></code></p>
<p><code><a href="GenomeAxisTrack-class.html">GenomeAxisTrack</a></code></p>
<p><code><a href="plotTracks.html">plotTracks</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Florian Hahne</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create UcscTrack for Known Genes from mm9 genome</span></span></span>
<span class="r-in"><span><span class="va">from</span> <span class="op">&lt;-</span> <span class="fl">65921878</span></span></span>
<span class="r-in"><span><span class="va">to</span> <span class="op">&lt;-</span> <span class="fl">65980988</span></span></span>
<span class="r-in"><span><span class="va">knownGenes</span> <span class="op">&lt;-</span> <span class="fu">UcscTrack</span><span class="op">(</span></span></span>
<span class="r-in"><span>    genome <span class="op">=</span> <span class="st">"mm9"</span>, chromosome <span class="op">=</span> <span class="st">"chrX"</span>, track <span class="op">=</span> <span class="st">"knownGene"</span>,</span></span>
<span class="r-in"><span>    from <span class="op">=</span> <span class="va">from</span>, to <span class="op">=</span> <span class="va">to</span>, trackType <span class="op">=</span> <span class="st">"GeneRegionTrack"</span>,</span></span>
<span class="r-in"><span>    rstarts <span class="op">=</span> <span class="st">"exonStarts"</span>, rends <span class="op">=</span> <span class="st">"exonEnds"</span>, gene <span class="op">=</span> <span class="st">"name"</span>,</span></span>
<span class="r-in"><span>    symbol <span class="op">=</span> <span class="st">"name"</span>, transcript <span class="op">=</span> <span class="st">"name"</span>, strand <span class="op">=</span> <span class="st">"strand"</span>,</span></span>
<span class="r-in"><span>    fill <span class="op">=</span> <span class="st">"#8282d2"</span>, name <span class="op">=</span> <span class="st">"UCSC Genes"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## if the UCSC is not accessible load prepared object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ucscItems</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## knownGenes is essentially GeneRegionTrack</span></span></span>
<span class="r-in"><span><span class="va">knownGenes</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GeneRegionTrack 'UCSC Genes'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | genome: mm9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | active chromosome: chrX</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> | annotation features: 76</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## plotting</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">knownGenes</span>, chromosome <span class="op">=</span> <span class="st">"chrX"</span>, from <span class="op">=</span> <span class="fl">65920688</span>, to <span class="op">=</span> <span class="fl">65960068</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="UcscTrack-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Florian Hahne, Steffen Durinck, Robert Ivanek, Arne Mueller, Steve Lianoglou, Ge Tan, Lance Parsons, Shraddha Pai.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

