<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>The Gviz User Guide • Gviz</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="The Gviz User Guide">
<meta property="og:description" content="Gviz">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Gviz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.35.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/Gviz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ivanek/Gviz/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Gviz_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>The Gviz User Guide</h1>
                        <h4 class="author">Florian Hahne</h4>
            <address class="author_afil">
      Novartis Institute for Biomedical Research, Basel, Switzerland<br><a class="author_email" href="mailto:#"></a><a href="mailto:florian.hahne@novartis.com" class="email">florian.hahne@novartis.com</a>
      </address>
                              <h4 class="author">Robert Ivánek</h4>
            <address class="author_afil">
      Department of Biomedicine, University of Basel, Basel, Switzerland<br><a class="author_email" href="mailto:#"></a><a href="mailto:robert.ivanek@unibas.ch" class="email">robert.ivanek@unibas.ch</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/ivanek/Gviz/blob/master/vignettes/Gviz.Rmd"><code>vignettes/Gviz.Rmd</code></a></small>
      <div class="hidden name"><code>Gviz.Rmd</code></div>

    </div>

    
    
<p><strong>Last edited</strong>: 19 May 2020</p>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>In order to make sense of genomic data one often aims to plot such data in a genome browser, along with a variety of genomic annotation features, such as gene or transcript models, CpG island, repeat regions, and so on. These features may either be extracted from public data bases like ENSEMBL or UCSC, or they may be generated or curated in-house. Many of the currently available genome browsers do a reasonable job in displaying genome annotation data, and there are options to connect to some of them from within <code>R</code> (e.g., using the <em><a href="https://bioconductor.org/packages/3.13/rtracklayer">rtracklayer</a></em> package). However, none of these solutions offer the flexibility of the full <code>R</code> graphics system to display large numeric data in a multitude of different ways. The <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package <span class="citation">(Hahne and Ivanek 2016)</span> aims to close this gap by providing a structured visualization framework to plot any type of data along genomic coordinates. It is loosely based on the <em><a href="https://bioconductor.org/packages/3.13/GenomeGraphs">GenomeGraphs</a></em> package by Steffen Durinck and James Bullard, however the complete class hierarchy as well as all the plotting methods have been restructured in order to increase performance and flexibility. All plotting is done using the grid graphics system, and several specialized annotation classes allow to integrate publicly available genomic annotation data from sources like <a href="https://genome.ucsc.edu/">UCSC</a> or <a href="https://www.ensembl.org/">ENSEMBL</a>.</p>
</div>
<div id="basic-features" class="section level1">
<h1 class="hasAnchor">
<a href="#basic-features" class="anchor"></a>Basic Features</h1>
<p>The fundamental concept behind the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package is similar to the approach taken by most genome browsers, in that individual types of genomic features or data are represented by separate tracks. Within the package, each track constitutes a single object inheriting from class <code>GdObject</code>, and there are constructor functions as well as a broad range of methods to interact with and to plot these tracks. When combining multiple objects, the individual tracks will always share the same genomic coordinate system, thus taking the burden of aligning the plot elements from the user.</p>
<p>It is worth mentioning that, at a given time, tracks in the sense of the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package are only defined for a single chromosome on a specific genome, at least for the duration of a given plotting operation. You will later see that a track may still contain information for multiple chromosomes, however most of this is hidden except for the currently active chromosome, and the user will have to explicitly switch the chromosome to access the inactive parts. While the package in principle imposes no fixed structure on the chromosome or on the genome names, it makes sense to stick to a standardized naming paradigm, in particular when fetching additional annotation information from online resources. By default this is enforced by a global option <code>ucscChromosomeNames</code>, which is set during package loading and which causes the package to check all supplied chromosome names for validity in the sense of the UCSC definition (chromosomes have to start with the <code>chr</code> string). You may decide to turn this feature off by calling <code><a href="https://rdrr.io/r/base/options.html">options(ucscChromosomeNames=FALSE)</a></code>. For the remainder of this vignette however, we will make use of the UCSC genome and chromosome identifiers, e.g., the <code>chr7</code> chromosome on the mouse <code>mm9</code> genome.</p>
<p>The different track classes will be described in more detail in the @ref(track-classes) section further below. For now, let’s just take a look at a typical <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> session to get an idea of what this is all about. We begin our presentation of the available functionality by loading the package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ivanek/Gviz">Gviz</a></span><span class="op">)</span></code></pre></div>
<p>The most simple genomic features consist of start and stop coordinates, possibly overlapping each other. CpG islands or microarray probes are real life examples for this class of features. In the Bioconductor world those are most often represented as run-length encoded vectors, for instance in the <code>IRanges</code> and <code>GRanges</code> classes. To seamlessly integrate with other Bioconductor packages, we can use the same data structures to generate our track objects. A sample set of CpG island coordinates has been saved in the <code>cpgIslands</code> object and we can use that for our first annotation track object. The constructor function <code>AnnotationTrack</code> is a convenient helper to create the object.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cpgIslands</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">cpgIslands</span><span class="op">)</span>
<span class="co">## [1] "GRanges"</span>
<span class="co">## attr(,"package")</span>
<span class="co">## [1] "GenomicRanges"</span>
<span class="va">chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">cpgIslands</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">gen</span> <span class="op">&lt;-</span> <span class="fu">genome</span><span class="op">(</span><span class="va">cpgIslands</span><span class="op">)</span>
<span class="va">atrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span><span class="va">cpgIslands</span>, name <span class="op">=</span> <span class="st">"CpG"</span><span class="op">)</span></code></pre></div>
<p>Please note that the <code>AnnotationTrack</code> constructor (as most constructors in this package) is fairly flexible and can accommodate many different types of inputs. For instance, the start and end coordinates of the annotation features could be passed in as individual arguments <code>start</code> and <code>end</code>, as a <code>data.frame</code> or even as an <code>IRanges</code> or <code>GRangesList</code> object. Furthermore, a whole bunch of coercion methods are available for those package users that prefer the more traditional R coding paradigm, and they should allow operations along the lines of <code>as(obj, 'AnnotationTrack')</code>. You may want to consult the class’ manual page for more information, or take a look at the @ref(bioconductor-integration) section for a listing of the most common data structures and their respective counterparts in the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package.</p>
<p>With our first track object being created we may now proceed to the plotting. There is a single function <code>plotTracks</code> that handles all of this. As we will learn in the remainder of this vignette, <code>plotTracks</code> is quite powerful and has a number of very useful additional arguments. For now we will keep things very simple and just plot the single CpG islands annotation track.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">atrack</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/plotAnnotationTrack-1.png"><!-- --></p>
<p>As you can see, the resulting graph is not particularly spectacular. There is a title region showing the track’s name on a gray background on the left side of the plot and a data region showing the seven individual CpG islands on the right. This structure is similar for all the available track objects classes and it somewhat mimics the layout of the popular UCSC Genome Browser. If you are not happy with the default settings, the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package offers a multitude of options to fine-tune the track appearance, which will be shown in the @ref(plotting-parameters) section.</p>
<p>Apart from the relative distance of the CpG islands, this visualization does not tell us much. One obvious next step would be to indicate the genomic coordinates we are currently looking at in order to provide some reference. For this purpose, the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package offers the <code>GenomeAxisTrack</code> class. Objects from the class can be created using the constructor function of the same name.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gtrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GenomeAxisTrack-class.html">GenomeAxisTrack</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Since a <code>GenomeAxisTrack</code> object is always relative to the other tracks that are plotted, there is little need for additional arguments. Essentially, the object just tells the <code>plotTracks</code> function to add a genomic axis to the plot. Nonetheless, it represent a separate annotation track just as the CpG island track does. We can pass this additional track on to <code>plotTracks</code> in the form of a list.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">gtrack</span>, <span class="va">atrack</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/plotGenomeAxisTrack-1.png"><!-- --></p>
<p>You may have realized that the genomic axis does not take up half of the available vertical plotting space, but only uses the space necessary to fit the axis and labels. Also the title region for this track is empty. In general, the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package tries to find reasonable defaults for all the parameters controlling the look and feel of a plots so that appealing visualizations can be created without too much tinkering. However, all features on the plot including the relative track sizes can also be adjusted manually.</p>
<p>As mentioned before in the beginning of this vignette, a plotted track is always defined for exactly one chromosome on a particular genome. We can include this information in our plot by means of a chromosome ideogram. An ideogram is a simplified visual representation of a chromosome, with the different chromosomal staining bands indicated by color, and the centromere (if present) indicated by the shape. The necessary information to produce this visualization is stored in online data repositories, for instance at UCSC. The <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package offers very convenient connections to some of these repositories, and the <code>IdeogramTrack</code> constructor function is one example for such a connection. With just the information about a valid UCSC genome and chromosome, we can directly fetch the chromosome ideogram information and construct a dedicated track object that can be visualized by <code>plotTracks</code>. Please not that you will need an established internet connection for this to work, and that fetching data from UCSC can take quite a long time, depending on the server load. The <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package tries to cache as much data as possible to reduce the bandwidth in future queries.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">itrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/IdeogramTrack-class.html">IdeogramTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="va">gen</span>, chromosome <span class="op">=</span> <span class="va">chr</span><span class="op">)</span></code></pre></div>
<p>Similar to the previous examples, we stick the additional track object into a list in order to plot it.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/plotIdeogramTrack-1.png"><!-- --></p>
<p>Ideogram tracks are the one exception in all of <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em>’s track objects in the sense that they are not really displayed on the same coordinate system like all the other tracks. Instead, the current genomic location is indicated on the chromosome by a red box (or, as in this case, a red line if the width is too small to fit a box).</p>
<p>So far we have only looked at very basic annotation features and how to give a point of reference to our plots. Naturally, we also want to be able to handle more complex genomic features, such as gene models. One potential use case would be to utilize gene model information from an existing local source. Alternatively, we could download such data from one of the available online resources like UCSC or ENSEBML, and there are constructor functions to handle these tasks. For this example we are going to load gene model data from a stored <code>data.frame</code>. The track class of choice here is a <code>GeneRegionTrack</code> object, which can be created via the constructor function of the same name. Similar to the <code>AnnotationTrack</code> constructor there are multiple possible ways to pass in the data.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneModels</span><span class="op">)</span>
<span class="va">grtrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GeneRegionTrack-class.html">GeneRegionTrack</a></span><span class="op">(</span><span class="va">geneModels</span>, genome <span class="op">=</span> <span class="va">gen</span>,
                           chromosome <span class="op">=</span> <span class="va">chr</span>, name <span class="op">=</span> <span class="st">"Gene Model"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span>, <span class="va">grtrack</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/GeneRegionTrack-1.png"><!-- --></p>
<p>In all those previous examples the plotted genomic range has been determined automatically from the input tracks. Unless told otherwise, the package will always display the region from the leftmost item to the rightmost item in any of the tracks. Of course such a static view on a chromosomal region is of rather limited use. We often want to zoom in or out on a particular plotting region to see more details or to get a broader overview. To that end, <code>plotTracks</code> supports the <code>from</code> and <code>to</code> arguments that let us choose an arbitrary genomic range to plot.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span>, <span class="va">grtrack</span><span class="op">)</span>,
           from <span class="op">=</span> <span class="fl">26700000</span>, to <span class="op">=</span> <span class="fl">26750000</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/zooming-1.png"><!-- --></p>
<p>Another pair of arguments that controls the zoom state are <code>extend.left</code> and <code>extend.right</code>. Rather than <code>from</code> and <code>to</code>, those arguments are relative to the currently displayed ranges, and can be used to quickly extend the view on one or both ends of the plot. In addition to positive or negative absolute integer values one can also provide a float value between -1 and 1 which will be interpreted as a zoom factor, i.e., a value of 0.5 will cause zooming in to half the currently displayed range.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span>, <span class="va">grtrack</span><span class="op">)</span>,
           extend.left <span class="op">=</span> <span class="fl">0.5</span>, extend.right <span class="op">=</span> <span class="fl">1000000</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/zooming2-1.png"><!-- --></p>
<p>You may have noticed that the layout of the gene model track has changed depending on the zoom level. This is a feature of the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package, which automatically tries to find the optimal visualization settings to make best use of the available space. At the same time, when features on a track are too close together to be plotted as separate items with the current device resolution, the package will try to reasonably merge them in order to avoid overplotting.</p>
<p>Often individual ranges on a plot tend to grow quite narrow, in particular when zooming far out, and a couple of tweaks become helpful in order to get nice plots, for instance to drop the bounding borders of the exons.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span>, <span class="va">grtrack</span><span class="op">)</span>, 
           extend.left <span class="op">=</span> <span class="fl">0.5</span>, extend.right <span class="op">=</span> <span class="fl">1000000</span>, col <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/zooming3-1.png"><!-- --></p>
<p>When zooming further in it may become interesting to take a look at the actual genomic sequence at a given position, and the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package provides the track class <code>SequenceTrack</code> that let’s you do just that. Among several other options it can draw the necessary sequence information from one of the <code>BSgenome</code> packages.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg19</span><span class="op">)</span>
<span class="va">strack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SequenceTrack-class.html">SequenceTrack</a></span><span class="op">(</span><span class="va">Hsapiens</span>, chromosome <span class="op">=</span> <span class="va">chr</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span>, <span class="va">grtrack</span>, <span class="va">strack</span><span class="op">)</span>, 
           from <span class="op">=</span> <span class="fl">26591822</span>, to <span class="op">=</span> <span class="fl">26591852</span>, cex <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/zooming4-1.png"><!-- --></p>
<p>So far we have replicated the features of a whole bunch of other genome browser tools out there. The real power of the package comes with a rather general track type, the <code>DataTrack</code>. <code>DataTrack</code> object are essentially run-length encoded numeric vectors or matrices, and we can use them to add all sorts of numeric data to our genomic coordinate plots. There are a whole bunch of different visualization options for these tracks, from dot plots to histograms to box-and-whisker plots. The individual rows in a numeric matrix are considered to be different data groups or samples, and the columns are the raster intervals in the genomic coordinates. Of course, the data points (or rather the data ranges) do not have to be evenly spaced; each column is associated with a particular genomic location. For demonstration purposes we can create a simple <code>DataTrack</code> object from randomly sampled data.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">255</span><span class="op">)</span>
<span class="va">lim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">26700000</span>, <span class="fl">26750000</span><span class="op">)</span>
<span class="va">coords</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">lim</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, 
                 <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span>from <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, to <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, <span class="fl">99</span><span class="op">)</span>, 
                 <span class="va">lim</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="fl">100</span>, min <span class="op">=</span> <span class="op">-</span><span class="fl">10</span>, max <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="va">dtrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DataTrack-class.html">DataTrack</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">dat</span>, start <span class="op">=</span> <span class="va">coords</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">coords</span><span class="op">)</span><span class="op">]</span>,
                    end <span class="op">=</span> <span class="va">coords</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>, chromosome <span class="op">=</span> <span class="va">chr</span>, genome <span class="op">=</span> <span class="va">gen</span>, 
                    name <span class="op">=</span> <span class="st">"Uniform"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span>, <span class="va">grtrack</span>, <span class="va">dtrack</span><span class="op">)</span>, 
           from <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, to <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/DataTrack-1.png"><!-- --></p>
<p>The first thing to notice is that the title panel to the right of the plot now contains a y-axis indicating the range of the displayed data. The default plotting type for numeric vectors is a simple dot plot. This is by far not the only visualization option, and in a sense it is wasting quite a lot of information because the run-length encoded ranges are not immediately apparent. We can change the plot type by supplying the <code>type</code> argument to <code>plotTracks</code>. A complete description of the available plotting options is given in section @ref(track-classes), and a more detailed treatment of the so-called ‘display parameters’ that control the look and feel of a track is given in the @ref(plotting-parameters) section.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span>, <span class="va">grtrack</span>, <span class="va">dtrack</span><span class="op">)</span>, 
           from <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, to <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">"histogram"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/DataTrackHist-1.png"><!-- --></p>
<p>As we can see, the data values in the numeric vector are indeed matched to the genomic coordinates of the <code>DataTrack</code> object. Such a visualization can be particularly helpful when displaying for instance the coverage of NGS reads along a chromosome, or to show the measurement values of mapped probes from a micro array experiment.</p>
<p>This concludes our first introduction into the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package. The remainder of this vignette will deal in much more depth with the different concepts and the various track classes and plotting options.</p>
</div>
<div id="plotting-parameters" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-parameters" class="anchor"></a>Plotting parameters</h1>
<div id="setting-parameters" class="section level2">
<h2 class="hasAnchor">
<a href="#setting-parameters" class="anchor"></a>Setting parameters</h2>
<p>Although not implicitly mentioned before, we have already made use of the plotting parameter facilities in the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package, or, as we will call them from now on, the ‘display parameters’. Display parameters are properties of individual track objects (i.e., of any object inheriting from the base <code>GdObject</code> class). They can either be set during object instantiation as additional arguments to the constructor functions or, for existing track objects, using the <code>displayPars</code> replacement method. In the former case, all named arguments that can not be matched to any of the constructor’s formal arguments are considered to be display parameters, regardless of their type or whether they are defined for a particular track class or not. The following code example rebuilds our <code>GeneRegionTrack</code> object with a bunch of display parameters and demonstrates the use of the <code>displayPars</code> accessor and replacement methods.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">grtrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GeneRegionTrack-class.html">GeneRegionTrack</a></span><span class="op">(</span><span class="va">geneModels</span>, genome <span class="op">=</span> <span class="va">gen</span>, chromosome <span class="op">=</span> <span class="va">chr</span>, 
                           name <span class="op">=</span> <span class="st">"Gene Model"</span>,
                           transcriptAnnotation <span class="op">=</span> <span class="st">"symbol"</span>,
                           background.title <span class="op">=</span> <span class="st">"brown"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/DisplayPars-class.html">displayPars</a></span><span class="op">(</span><span class="va">grtrack</span><span class="op">)</span><span class="op">)</span>
<span class="co">## $arrowHeadWidth</span>
<span class="co">## [1] 10</span>
<span class="co">## </span>
<span class="co">## $arrowHeadMaxWidth</span>
<span class="co">## [1] 20</span>
<span class="co">## </span>
<span class="co">## $col</span>
<span class="co">## [1] "darkgray"</span>
<span class="co">## </span>
<span class="co">## $collapseTranscripts</span>
<span class="co">## [1] FALSE</span>
<span class="co">## </span>
<span class="co">## $exonAnnotation</span>
<span class="co">## NULL</span>
<span class="co">## </span>
<span class="co">## $fill</span>
<span class="co">## [1] "#FFD58A"</span>
<span class="fu"><a href="../reference/DisplayPars-class.html">displayPars</a></span><span class="op">(</span><span class="va">grtrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>background.panel <span class="op">=</span> <span class="st">"#FFFEDB"</span>, col <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/DisplayPars-class.html">displayPars</a></span><span class="op">(</span><span class="va">grtrack</span><span class="op">)</span><span class="op">)</span>
<span class="co">## $arrowHeadWidth</span>
<span class="co">## [1] 10</span>
<span class="co">## </span>
<span class="co">## $arrowHeadMaxWidth</span>
<span class="co">## [1] 20</span>
<span class="co">## </span>
<span class="co">## $col</span>
<span class="co">## NULL</span>
<span class="co">## </span>
<span class="co">## $collapseTranscripts</span>
<span class="co">## [1] FALSE</span>
<span class="co">## </span>
<span class="co">## $exonAnnotation</span>
<span class="co">## NULL</span>
<span class="co">## </span>
<span class="co">## $fill</span>
<span class="co">## [1] "#FFD58A"</span>

<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span>, <span class="va">grtrack</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/displayPars1f-1.png"><!-- --></p>
<p>For our gene model track we have now added the gene symbols of the different transcripts to the plot, removed the gray border around the individual exons models, and changed the background color of both the title and the data panel to a yellowish hue. There is a third option to set display parameters for a single plotting operation (rather than the permanent setting in the track object) by passing in additional arguments to the <code>plotTracks</code> function. We have already made use of this feature, for instance in the previous data plotting type example. It is worth mentioning that all display parameters which are passed along with the <code>plotTracks</code> function apply to <em>all</em> track objects in the plot. For some objects classes a particular display parameter may not make any sense, and in that case it is simply ignored. Also, the settings only apply for one single plotting operation and will not be retained in the plotted track objects. They do however get precedence over the object-internal parameters. The following line of code exemplifies this behavior.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span>, <span class="va">grtrack</span><span class="op">)</span>, 
           background.panel <span class="op">=</span> <span class="st">"#FFFEDB"</span>, background.title <span class="op">=</span> <span class="st">"darkblue"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/displayPars2f-1.png"><!-- --></p>
<p>In order to make full use of the flexible parameter system we need to know which display parameters control which aspect of which track class. The obvious source for this information are the man pages of the respective track classes, which list all available parameters along with a short description of their effect and default values in the <code>Display Parameters</code> section. Alternatively, we can use the <code>availableDisplayPars</code> function, which prints out the available parameters for a class as well as their default values in a list-like structure. The single argument to the function is either a class name of a track object class, or the object itself, in which case its class is automatically detected.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DisplayPars-class.html">availableDisplayPars</a></span><span class="op">(</span><span class="va">grtrack</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="va">dp</span><span class="op">)</span>
<span class="co">## </span>
<span class="co">## The following display parameters are available for 'GeneRegionTrack' objects:</span>
<span class="co">## (see ? GeneRegionTrack for details on their usage)</span>
<span class="co">## </span>
<span class="co">## showAxis (inherited from class 'GdObject'): TRUE </span>
<span class="co">## showExonId: NULL</span>
<span class="co">## showFeatureId (inherited from class 'AnnotationTrack'): FALSE </span>
<span class="co">## showId (inherited from class 'AnnotationTrack'): FALSE </span>
<span class="co">## showOverplotting (inherited from class 'AnnotationTrack'): FALSE </span>
<span class="co">## showTitle (inherited from class 'GdObject'): TRUE </span>
<span class="co">## size (inherited from class 'GdObject'): 1 </span>
<span class="co">## stackHeight (inherited from class 'StackedTrack'): 0.75 </span>
<span class="co">## thinBoxFeature: utr ncRNA utr3 utr5 3UTR 5UTR miRNA lincRNA three_prime_UTR five_prime_UTR </span>
<span class="co">## transcriptAnnotation: NULL</span>
<span class="co">## v (inherited from class 'GdObject'): -1</span></code></pre></div>
<p>As we can see, display parameters can be inherited from parent classes. For the regular user this is not important at all, however it nicely exemplifies the structure of the class hierarchy in the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package.</p>
</div>
<div id="schemes" class="section level2">
<h2 class="hasAnchor">
<a href="#schemes" class="anchor"></a>Schemes</h2>
<p>Users might find themselves changing the same parameters over and over again, and it would make sense to register these modifications in a central location once and for all. To this end the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package supports display parameter schemes. A scheme is essentially just a bunch of nested named lists, where the names on the first level of nesting should correspond to track class names, and the names on the second level to the display parameters to set. The currently active scheme can be changed by setting the global option <code>Gviz.scheme</code>, and a new scheme can be registered by using the <code>addScheme</code> function, providing both the list and the name for the new scheme. The <code>getScheme</code> function is useful to get the current scheme as a list structure, for instance to use as a skeleton for your own custom scheme.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/options.html">getOption</a></span><span class="op">(</span><span class="st">"Gviz.scheme"</span><span class="op">)</span>
<span class="co">## [1] "default"</span>
<span class="va">scheme</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/settings.html">getScheme</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">scheme</span><span class="op">$</span><span class="va">GeneRegionTrack</span><span class="op">$</span><span class="va">fill</span> <span class="op">&lt;-</span> <span class="st">"salmon"</span>
<span class="va">scheme</span><span class="op">$</span><span class="va">GeneRegionTrack</span><span class="op">$</span><span class="va">col</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">scheme</span><span class="op">$</span><span class="va">GeneRegionTrack</span><span class="op">$</span><span class="va">transcriptAnnotation</span> <span class="op">&lt;-</span> <span class="st">"transcript"</span>
<span class="fu"><a href="../reference/settings.html">addScheme</a></span><span class="op">(</span><span class="va">scheme</span>, <span class="st">"myScheme"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>Gviz.scheme <span class="op">=</span> <span class="st">"myScheme"</span><span class="op">)</span>
<span class="va">grtrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GeneRegionTrack-class.html">GeneRegionTrack</a></span><span class="op">(</span><span class="va">geneModels</span>, genome <span class="op">=</span> <span class="va">gen</span>,
                           chromosome <span class="op">=</span> <span class="va">chr</span>, name <span class="op">=</span> <span class="st">"Gene Model"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grtrack</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/displayPars4-1.png"><!-- --></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>Gviz.scheme<span class="op">=</span><span class="st">"default"</span><span class="op">)</span>
<span class="va">grtrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GeneRegionTrack-class.html">GeneRegionTrack</a></span><span class="op">(</span><span class="va">geneModels</span>, genome <span class="op">=</span> <span class="va">gen</span>, chromosome <span class="op">=</span> <span class="va">chr</span>,
                           name <span class="op">=</span> <span class="st">"Gene Model"</span>,
                           transcriptAnnotation <span class="op">=</span> <span class="st">"symbol"</span><span class="op">)</span></code></pre></div>
<p>In order to make these settings persistent across R sessions one can create one or several schemes in the global environment in the special object <code>.GvizSchemes</code>, for instance by putting the necessary code in the <code>.Rprofile</code> file. This object needs to be a named list of schemes, and it will be collected when the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package loads. Its content is then automatically added to the collection of available schemes. The following pseudo-code exemplifies this and could go into an <code>.Rprofile</code> file.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">.GvizSchemes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>myScheme <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  GeneRegionTrack<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"salmon"</span>, col <span class="op">=</span> <span class="cn">NULL</span>, 
                       transcriptAnnotation <span class="op">=</span> <span class="st">"transcript"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Please note that because display parameters are stored with the track objects, a scheme change only has an effect on those objects that are created after the change has taken place.</p>
</div>
<div id="plotting-direction" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-direction" class="anchor"></a>Plotting direction</h2>
<p>By default all tracks will be plotted in a 5’ -&gt; 3’ direction. It sometimes can be useful to actually show the data relative to the opposite strand. To this end one can use the <code>reverseStrand</code> display parameter, which does just what its name suggests. Since the combination of forward and reverse stranded tracks on a single plot does not make too much sense, one should usually set this as a global display parameter in the <code>plotTracks</code> function. The function will however cast a warning if a mixture of forward and reverse strand tracks has been passed in for plotting.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span>, <span class="va">grtrack</span><span class="op">)</span>, reverseStrand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/plottingdirections-1.png"><!-- --></p>
<p>As you can see, the fact that the data has been plotted on the reverse strand is also reflected in the <code>GenomeAxis</code> track.</p>
</div>
</div>
<div id="track-classes" class="section level1">
<h1 class="hasAnchor">
<a href="#track-classes" class="anchor"></a>Track classes</h1>
<p>In this section we will highlight all of the available annotation track classes in the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package. For the complete reference of all the nuts and bolts, including all the available methods, please see the respective class man pages. We will try to keep this vignette up to date, but in cases of discrepancies between here and the man pages you should assume the latter to be correct.</p>
<div id="genomeaxistrack" class="section level2">
<h2 class="hasAnchor">
<a href="#genomeaxistrack" class="anchor"></a>GenomeAxisTrack</h2>
<p><code>GenomeAxisTrack</code> objects can be used to add some reference to the currently displayed genomic location to a <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> plot. In their most basic form they are really just a horizontal axis with genomic coordinate tick-marks. Using the <code>GenomeAxisTrack</code> constructor function is the recommended way to instantiate objects from the class. There is no need to know in advance about a particular genomic location when constructing the object. Instead, the displayed coordinates will be determined from the context, e.g., from the <code>from</code> and <code>to</code> arguments of the <code>plotTracks</code> function, or, when plotted together with other track objects, from their genomic locations.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">axisTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GenomeAxisTrack-class.html">GenomeAxisTrack</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">axisTrack</span>, from <span class="op">=</span> <span class="fl">1e6</span>, to <span class="op">=</span> <span class="fl">9e6</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/GenomeAxisTrackClass1-1.png"><!-- --></p>
<p>As an optional feature one can highlight particular regions on the axis, for instance to indicated stretches of N nucleotides or gaps in genomic alignments. Such regions have to be supplied to the optional <code>range</code> argument of the constructor function as either an <code>GRanges</code> or an <code>IRanges</code> object.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">axisTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GenomeAxisTrack-class.html">GenomeAxisTrack</a></span><span class="op">(</span>range<span class="op">=</span><span class="fu">IRanges</span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2e6</span>, <span class="fl">4e6</span><span class="op">)</span>,
                                           end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3e6</span>, <span class="fl">7e6</span><span class="op">)</span>,
                                           names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"N-stretch"</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">axisTrack</span>, from <span class="op">=</span> <span class="fl">1e6</span>, to <span class="op">=</span> <span class="fl">9e6</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/GenomeAxisTrackClass2-1.png"><!-- --></p>
<p>If names have been supplied with the <code>range</code> argument, those can also be added to the plot.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">axisTrack</span>, from <span class="op">=</span> <span class="fl">1e6</span>, to <span class="op">=</span> <span class="fl">9e6</span>, showId <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/GenomeAxisTrackClass2a-1.png"><!-- --></p>
<div id="display-parameters-for-genomeaxistrack-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#display-parameters-for-genomeaxistrack-objects" class="anchor"></a>Display parameters for GenomeAxisTrack objects</h3>
<p>There are a whole bunch of display parameters to alter the appearance of <code>GenomeAxisTrack</code> plots. For instance, one could add directional indicators to the axis using the <code>add53</code> and <code>add35</code> parameters.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">axisTrack</span>, from <span class="op">=</span> <span class="fl">1e6</span>, to <span class="op">=</span> <span class="fl">9e6</span>, add53 <span class="op">=</span> <span class="cn">TRUE</span>, add35 <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/GenomeAxisTrackClass3-1.png"><!-- --></p>
<p>Sometimes the resolution of the tick marks is not sufficient, in which case the <code>littleTicks</code> argument can be used to have a more fine-grained axis annotation.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">axisTrack</span>, from <span class="op">=</span> <span class="fl">1e6</span>, to <span class="op">=</span> <span class="fl">9e6</span>, add53 <span class="op">=</span> <span class="cn">TRUE</span>,
           add35 <span class="op">=</span> <span class="cn">TRUE</span>, littleTicks <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/GenomeAxisTrackClass4-1.png"><!-- --></p>
<p>The <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package tries to come up with reasonable defaults for the axis annotation. In our previous example, the genomic coordinates are indicated in megabases. We can control this via the <code>exponent</code> parameter, which takes an integer value greater then zero. The location of the tick marks are displayed as a fraction of <span class="math inline">\(10^{exponent}\)</span>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">axisTrack</span>, from <span class="op">=</span> <span class="fl">1e6</span>, to <span class="op">=</span> <span class="fl">9e6</span>, exponent <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/GenomeAxisTrackClass5-1.png"><!-- --></p>
<p>Another useful parameter, <code>labelPos</code> controls the arrangement of the tick marks. It takes one of the values <code>alternating</code>, <code>revAlternating</code>, <code>above</code> or <code>below</code>. For instance we could align all tick marks underneath the axis.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">axisTrack</span>, from <span class="op">=</span> <span class="fl">1e6</span>, to <span class="op">=</span> <span class="fl">9e6</span>, labelPos <span class="op">=</span> <span class="st">"below"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/GenomeAxisTrackClass6-1.png"><!-- --></p>
<p>Sometimes a full-blown axis is just too much, and all we really need in the plot is a small scale indicator. We can change the appearance of the <code>GenomeAxisTrack</code> object to such a limited representation by setting the <code>scale</code> display parameter. Typically, this will be a numeric value between 0 and 1, which is interpreted as the fraction of the plotting region used for the scale. The plotting method will apply some rounding to come up with reasonable and human-readable values. For even more control we can pass in a value larger than 1, which is considered to be an absolute range length. In this case, the user is responsible for the scale to actually fit in the current plotting range.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">axisTrack</span>, from <span class="op">=</span> <span class="fl">1e6</span>, to <span class="op">=</span> <span class="fl">9e6</span>, scale <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/GenomeAxisTrackClass7-1.png"><!-- --></p>
<p>We still have control over the placement of the scale label via the <code>labelPos</code>, parameter, which now takes the values <code>above</code>, <code>below</code> and <code>beside</code>.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">axisTrack</span>, from <span class="op">=</span> <span class="fl">1e6</span>, to <span class="op">=</span> <span class="fl">9e6</span>, scale <span class="op">=</span> <span class="fl">0.5</span>, 
           labelPos <span class="op">=</span> <span class="st">"below"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/GenomeAxisTrackClass8-1.png"><!-- --></p>
<p>For a complete listing of all the available display parameters please see the table below or the man page of the <code>GenomeAxisTrack</code> class by typing in <code><a href="../reference/GenomeAxisTrack-class.html">?GenomeAxisTrack</a></code> on the <code>R</code> command line.</p>
<table class="table">
<colgroup>
<col width="7%">
<col width="92%">
</colgroup>
<thead><tr class="header">
<th align="left">Display Parameter</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">add35</td>
<td align="left">Logical scalar. Add 3’ to 5’ direction indicators.</td>
</tr>
<tr class="even">
<td align="left">add53</td>
<td align="left">Logical scalar. Add 5’ to 3’ direction indicators.</td>
</tr>
<tr class="odd">
<td align="left">cex</td>
<td align="left">Numeric scalar. The overall font expansion factor for the axis annotation text.</td>
</tr>
<tr class="even">
<td align="left">cex.id</td>
<td align="left">Numeric scalar. The text size for the optional range annotation.</td>
</tr>
<tr class="odd">
<td align="left">col</td>
<td align="left">Character scalar. The color for the axis lines and tickmarks.</td>
</tr>
<tr class="even">
<td align="left">col.border.title</td>
<td align="left">Integer or character scalar. The border color for the title panels.</td>
</tr>
<tr class="odd">
<td align="left">col.id</td>
<td align="left">Character scalar. The text color for the optional range annotation.</td>
</tr>
<tr class="even">
<td align="left">col.range</td>
<td align="left">Character scalar. The border color for highlighted regions on the axis.</td>
</tr>
<tr class="odd">
<td align="left">distFromAxis</td>
<td align="left">Numeric scalar. Control the distance of the axis annotation from the tick marks.</td>
</tr>
<tr class="even">
<td align="left">exponent</td>
<td align="left">Numeric scalar. The exponent for the axis coordinates, e.g., 3 means mb, 6 means gb, etc. The default is to automatically determine the optimal exponent.</td>
</tr>
<tr class="odd">
<td align="left">fill.range</td>
<td align="left">Character scalar. The fill color for highlighted regions on the axis.</td>
</tr>
<tr class="even">
<td align="left">fontcolor</td>
<td align="left">Character scalar. The font color for the axis annotation text.</td>
</tr>
<tr class="odd">
<td align="left">fontsize</td>
<td align="left">Numeric scalar. Font size for the axis annotation text in points.</td>
</tr>
<tr class="even">
<td align="left">labelPos</td>
<td align="left">Character vector, one in “alternating”, “revAlternating”, “above” or “below”. The vertical positioning of the axis labels. If <code>scale</code> is not <code>NULL</code>, the possible values are “above”, “below” and “beside”.</td>
</tr>
<tr class="odd">
<td align="left">littleTicks</td>
<td align="left">Logical scalar. Add more fine-grained tick marks.</td>
</tr>
<tr class="even">
<td align="left">lwd</td>
<td align="left">Numeric scalar. The line width for the axis elementes.</td>
</tr>
<tr class="odd">
<td align="left">lwd.border.title</td>
<td align="left">Integer scalar. The border width for the title panels.</td>
</tr>
<tr class="even">
<td align="left">scale</td>
<td align="left">Numeric scalar. If not <code>NULL</code> a small scale is drawn instead of the full axis, if the value is between 0 and 1 it is interpreted as a fraction of the current plotting region, otherwise as an absolute length value in genomic coordinates.</td>
</tr>
<tr class="odd">
<td align="left">showId</td>
<td align="left">Logical scalar. Show the optional range highlighting annotation.</td>
</tr>
<tr class="even">
<td align="left">ticksAt</td>
<td align="left">Numeric scalar. The exact x-position for tickmarks (in base-pairs).</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="ideogramtrack" class="section level2">
<h2 class="hasAnchor">
<a href="#ideogramtrack" class="anchor"></a>IdeogramTrack</h2>
<p>While a genomic axis provides helpful points of reference to a plot, it is sometimes important to show the currently displayed region in the broader context of the whole chromosome. Are we looking at distal regions, or somewhere close to the centromere? And how much of the complete chromosome is covered in our plot. To that end the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package defines the <code>IdeogramTrack</code> class, which is an idealized representation of a single chromosome. When plotted, these track objects will always show the whole chromosome, regardless of the selected genomic region. However, the displayed coordinates are indicated by a box that sits on the ideogram image. The chromosomal data necessary to draw the ideogram is not part of the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package itself, instead it is downloaded from an online source (UCSC). Thus it is important to use both chromosome and genome names that are recognizable in the UCSC data base when dealing with <code>IdeogramTrack</code> objects. You might want to consult the <a href="http://genome.ucsc.edu/">UCSC webpage</a> or use the <code>ucscGenomes</code> function in the <code>rtracklayer</code> package for a listing of available genomes.</p>
<p>Assuming the chromosome data are available online, a simple call to the <code>IdeogramTrack</code> constructor function including the desired genome and chromosome name are enough to instantiate the object. Since the connection to UCSC can be slow, the package tries to cache data that has already been downloaded for the duration of the <code>R</code> session. If needed, the user can manually clear the cache by calling the <code>clearSessionCache</code> function. Of course it is also possible to construct <code>IdeogramTrack</code> objects from local data. Please see the class’ man page for details.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ideoTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/IdeogramTrack-class.html">IdeogramTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg19"</span>, chromosome <span class="op">=</span> <span class="st">"chrX"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">ideoTrack</span>, from <span class="op">=</span> <span class="fl">85e6</span>, to <span class="op">=</span> <span class="fl">129e6</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/IdeogramTrackClass1Do-1.png"><!-- --></p>
<p>We can turn off the explicit plotting of the chromosome name by setting the <code>showId</code> display parameter to <code>FALSE</code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">ideoTrack</span>, from <span class="op">=</span> <span class="fl">85e6</span>, to <span class="op">=</span> <span class="fl">129e6</span>, showId <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/IdeogramTrackClass2-1.png"><!-- --></p>
<p>The chromosome bands in the ideogram come with a unique identifier, and we can add this information to the plot, at least for those bands that are wide enough to accommodate the text.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">ideoTrack</span>, from <span class="op">=</span> <span class="fl">85e6</span>, to <span class="op">=</span> <span class="fl">129e6</span>, showId <span class="op">=</span> <span class="cn">FALSE</span>, 
           showBandId <span class="op">=</span> <span class="cn">TRUE</span>, cex.bands <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/IdeogramTrackClass3-1.png"><!-- --></p>
<p>We can also change the shape of the centromere to “circle” instead of default “triangle”.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">ideoTrack</span>, from <span class="op">=</span> <span class="fl">85e6</span>, to <span class="op">=</span> <span class="fl">129e6</span>, showId <span class="op">=</span> <span class="cn">FALSE</span>,
           centromereShape <span class="op">=</span> <span class="st">"circle"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/IdeogramTrackClass4-1.png"><!-- --></p>
<div id="display-parameters-for-ideogramtrack-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#display-parameters-for-ideogramtrack-objects" class="anchor"></a>Display parameters for IdeogramTrack objects</h3>
<p>For a complete listing of all the available display parameters please see the table below or the man page of the <code>IdeogramTrack</code> class by typing in <code><a href="../reference/IdeogramTrack-class.html">?IdeogramTrack</a></code> on the <code>R</code> command line.</p>
<table class="table">
<colgroup>
<col width="13%">
<col width="86%">
</colgroup>
<thead><tr class="header">
<th align="left">Display Parameter</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">bevel</td>
<td align="left">Numeric scalar, between 0 and 1. The level of smoothness for the two ends of the ideogram.</td>
</tr>
<tr class="even">
<td align="left">centromereShape</td>
<td align="left">Character scalar. The shape of the centromere. Only “triangle” or “circle” is accepted. Default to “triangle”</td>
</tr>
<tr class="odd">
<td align="left">cex</td>
<td align="left">Numeric scalar. The overall font expansion factor for the chromosome name text.</td>
</tr>
<tr class="even">
<td align="left">cex.bands</td>
<td align="left">Numeric scalar. The font expansion factor for the chromosome band identifier text.</td>
</tr>
<tr class="odd">
<td align="left">col</td>
<td align="left">Character scalar. The border color used for the highlighting of the currently displayed genomic region.</td>
</tr>
<tr class="even">
<td align="left">col.border.title</td>
<td align="left">Integer or character scalar. The border color for the title panels.</td>
</tr>
<tr class="odd">
<td align="left">fill</td>
<td align="left">Character scalar. The fill color used for the highlighting of the currently displayed genomic region.</td>
</tr>
<tr class="even">
<td align="left">fontcolor</td>
<td align="left">Character scalar. The font color for the chromosome name text.</td>
</tr>
<tr class="odd">
<td align="left">fontface</td>
<td align="left">Character scalar. The font face for the chromosome name text.</td>
</tr>
<tr class="even">
<td align="left">fontfamily</td>
<td align="left">Character scalar. The font family for the chromosome name text.</td>
</tr>
<tr class="odd">
<td align="left">fontsize</td>
<td align="left">Numeric scalar. The font size for the chromosome name text.</td>
</tr>
<tr class="even">
<td align="left">lty</td>
<td align="left">Character or integer scalar. The line type used for the highlighting of the currently displayed genomic region.</td>
</tr>
<tr class="odd">
<td align="left">lwd</td>
<td align="left">Numeric scalar. The line width used for the highlighting of the currently displayed genomic region.</td>
</tr>
<tr class="even">
<td align="left">lwd.border.title</td>
<td align="left">Integer scalar. The border width for the title panels.</td>
</tr>
<tr class="odd">
<td align="left">outline</td>
<td align="left">Logical scalar. Add borders to the individual chromosome staining bands.</td>
</tr>
<tr class="even">
<td align="left">showBandId</td>
<td align="left">Logical scalar. Show the identifier for the chromosome bands if there is space for it.</td>
</tr>
<tr class="odd">
<td align="left">showId</td>
<td align="left">Logical scalar. Indicate the chromosome name next to the ideogram.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="datatrack" class="section level2">
<h2 class="hasAnchor">
<a href="#datatrack" class="anchor"></a>DataTrack</h2>
<p>Probably the most powerful of all the track classes in the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package are <code>DataTracks</code>. Essentially they constitute run-length encoded numeric vectors or matrices, meaning that one or several numeric values are associated to a particular genomic coordinate range. These ranges may even be overlapping, for instance when looking at results from a running window operation. There can be multiple samples in a single data set, in which case the ranges are associated to the columns of a numeric matrix rather than a numeric vector, and the plotting method provides tools to incorporate sample group information. Thus the starting point for creating <code>DataTrack</code> objects will always be a set of ranges, either in the form of an <code>IRanges</code> or <code>GRanges</code> object, or individually as start and end coordinates or widths. The second ingredient is a numeric vector of the same length as the number of ranges, or a numeric matrix with the same number of columns. Those may even already be part of the input <code>GRanges</code> object as <code>elemenMetadata</code> values. For a complete description of all the possible inputs please see the class’ online documentation. We can pass all this information to the <code>DataTrack</code> constructor function to instantiate an object. We will load our sample data from an <code>GRanges</code> object that comes as part of the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">twoGroups</span><span class="op">)</span>
<span class="va">dTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DataTrack-class.html">DataTrack</a></span><span class="op">(</span><span class="va">twoGroups</span>, name <span class="op">=</span> <span class="st">"uniform"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/DataClass1-1.png"><!-- --></p>
<p>The default visualization for our very simplistic sample <code>DataTrack</code> is a rather uninspiring dot plot. The track comes with a scale to indicate the range of the numeric values on the y-axis, apart from that it looks very much like the previous examples. A whole battery of display parameters is to our disposal to control the track’s look and feel. The most important one is the <code>type</code> parameter. It determines the type of plot to use and takes one or several of the following values:</p>
<table class="table">
<colgroup>
<col width="10%">
<col width="89%">
</colgroup>
<thead><tr class="header">
<th align="left">Value</th>
<th align="left">Type</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">p</td>
<td align="left">dot plot</td>
</tr>
<tr class="even">
<td align="left">l</td>
<td align="left">lines plot</td>
</tr>
<tr class="odd">
<td align="left">b</td>
<td align="left">dot and lines plot</td>
</tr>
<tr class="even">
<td align="left">a</td>
<td align="left">lines plot of average (i.e., mean) values</td>
</tr>
<tr class="odd">
<td align="left">s</td>
<td align="left">stair steps (horizontal first)</td>
</tr>
<tr class="even">
<td align="left">S</td>
<td align="left">stair steps (vertical first)</td>
</tr>
<tr class="odd">
<td align="left">g</td>
<td align="left">add grid lines</td>
</tr>
<tr class="even">
<td align="left">r</td>
<td align="left">add linear regression line</td>
</tr>
<tr class="odd">
<td align="left">h</td>
<td align="left">histogram lines</td>
</tr>
<tr class="even">
<td align="left">confint</td>
<td align="left">confidence intervals for average values</td>
</tr>
<tr class="odd">
<td align="left">smooth</td>
<td align="left">add loess curve</td>
</tr>
<tr class="even">
<td align="left">histogram</td>
<td align="left">histogram (bar width equal to range with)</td>
</tr>
<tr class="odd">
<td align="left">mountain</td>
<td align="left">‘mountain-type’ plot relative to a baseline</td>
</tr>
<tr class="even">
<td align="left">polygon</td>
<td align="left">‘polygon-type’ plot relative to a baseline</td>
</tr>
<tr class="odd">
<td align="left">boxplot</td>
<td align="left">box and whisker plot</td>
</tr>
<tr class="even">
<td align="left">gradient</td>
<td align="left">false color image of the summarized values</td>
</tr>
<tr class="odd">
<td align="left">heatmap</td>
<td align="left">false color image of the individual values</td>
</tr>
<tr class="even">
<td align="left">horizon</td>
<td align="left">Horizon plot indicating magnitude and direction of a change relative to a baseline</td>
</tr>
</tbody>
</table>
<p>Displayed below are the same sample data as before but plotted with the different type settings:<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a></p>
<p><img src="Gviz_files/figure-html/typePlots-1.png"><!-- --></p>
<p>data while others are clearly designed for multivariate inputs. The <code>a</code> type for instance averages the values at each genomic location before plotting the derived values as a line. The decision for a particular plot type is totally up to the user, and one could even overlay multiple types by supplying a character vector rather than a character scalar as the <code>type</code> argument. For example, this will combine a boxplot with an average line and a data grid.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack</span>, type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"boxplot"</span>, <span class="st">"a"</span>, <span class="st">"g"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/mutitype-1.png"><!-- --></p>
<p>For the heatmap plotting type we arrange all the data in a well-structured two-dimensional matrix which gives us the opportunity to add a little extra information about the individual samples. Depending on how the <code>DataTrack</code> was created in the first place we can choose to display the sample names (which in our case correspond to the column names of the input <code>GRanges</code> object). The plot also highlights another feature of the heatmap type: the y-axis now shows a mapping of the numeric values into the color range.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">twoGroups</span><span class="op">)</span><span class="op">)</span>
<span class="co">## [1] "control"   "control.1" "control.2" "treated"   "treated.1" "treated.2"</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack</span>, type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"heatmap"</span><span class="op">)</span>, showSampleNames <span class="op">=</span> <span class="cn">TRUE</span>, 
           cex.sampleNames <span class="op">=</span> <span class="fl">0.6</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/sampNames-1.png"><!-- --></p>
<div id="data-grouping" class="section level3">
<h3 class="hasAnchor">
<a href="#data-grouping" class="anchor"></a>Data Grouping</h3>
<p>An additional layer of flexibility is added by making use of <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em>’s grouping functionality. The individual samples (i.e., rows in the data matrix) can be grouped together using a factor variable, and, if reasonable, this grouping is reflected in the layout of the respective track types. For instance our example data could be derived from two different sample groups with three replicates each, and we could easily integrate this information into our plot.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack</span>, groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"control"</span>, <span class="st">"treated"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>,
           type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>, <span class="st">"p"</span>, <span class="st">"confint"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/grouping-1.png"><!-- --></p>
<p>For the dot plot representation the individual group levels are indicated by color coding. For the <code>a</code> type, the averages are now computed for each group separately and also indicated by two lines with similar color coding. Grouping is not supported for all plotting types, for example the <code>mountain</code> and <code>polygon</code> type already use color coding to convey a different message and for the <code>gradient</code> type the data are already collapsed to a single variable. The following gives an overview over some of the other groupable <code>DataTrack</code> types. Please note that there are many more display parameters that control the layout of both grouped and of ungrouped <code>DataTracks</code>. You may want to check the class’ help page for details.<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a></p>
<p><img src="Gviz_files/figure-html/typeGroupedPlots-1.png"><!-- --></p>
<p>If we need to display some additional information about the individual group levels we can make use of the <code>legend</code> display parameter to add a simple legend to the plot. Depending on the plot type and on some of the other display parameters, the look of this legend may vary slightly.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack</span>, groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"control"</span>, <span class="st">"treated"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>,
           type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>, <span class="st">"p"</span><span class="op">)</span>, legend <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/groupingLegend-1.png"><!-- --></p>
<p>For a grouped horizon plot the group labels have to be shown in a similar fashion as for heatmaps, i.e., by setting the <code>showSampleNames</code> argument to <code>TRUE</code>.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">dtHoriz</span><span class="op">)</span>
<span class="va">dtHoriz</span> <span class="op">&lt;-</span> <span class="va">dtHoriz</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="op">]</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dtHoriz</span>, type <span class="op">=</span> <span class="st">"horiz"</span>, groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">values</span><span class="op">(</span><span class="va">dtHoriz</span><span class="op">)</span><span class="op">)</span>,
           showSampleNames <span class="op">=</span> <span class="cn">TRUE</span>, cex.sampleNames <span class="op">=</span> <span class="fl">0.6</span>, separator <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/horizLegend-1.png"><!-- --></p>
</div>
<div id="building-datatrack-objects-from-files" class="section level3">
<h3 class="hasAnchor">
<a href="#building-datatrack-objects-from-files" class="anchor"></a>Building DataTrack objects from files</h3>
<p>A number of standard file types exist that all store numeric data along genomic coordinates. We have tried to make such files accessible in the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package by providing additional options to the <code>DataTrack</code> constructor function. In the previous examples the <code>range</code> argument was a <code>GRanges</code> object. Instead, we can also pass in the path to a file on disk by means of a character scalar. The <code>DataTrack</code> class supports the most common file types like <code>wig</code>, <code>bigWig</code> or <code>bedGraph</code>, but also knows how to deal with <code>bam</code> files. You may have realized that some of these files are indexed, and we have taken the approach to stream the data from indexed files on the fly when it is needed for plotting.</p>
<p>However let’s first start with the simple example of a <code>bedGraph</code> file. These files support a single data sample, and thus are equivalent to a <code>GRanges</code> object with a single numeric metadata column. <code>bedGraph</code> files are not indexed, so we have to load the whole file content when instantiating the object.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bgFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/test.bedGraph"</span>, package <span class="op">=</span> <span class="st">"Gviz"</span><span class="op">)</span>
<span class="va">dTrack2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DataTrack-class.html">DataTrack</a></span><span class="op">(</span>range <span class="op">=</span> <span class="va">bgFile</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, type <span class="op">=</span> <span class="st">"l"</span>, 
                     chromosome <span class="op">=</span> <span class="st">"chr19"</span>, name <span class="op">=</span> <span class="st">"bedGraph"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">dTrack2</span><span class="op">)</span>
<span class="co">## [1] "DataTrack"</span>
<span class="co">## attr(,"package")</span>
<span class="co">## [1] "Gviz"</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack2</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/filedt1-1.png"><!-- --></p>
<p>As we can see the constructor has returned a regular <code>DataTrack</code> object. The function to be used in order to read the data off the file has been automatically chosen by the package based on the file extension of the input file. Of course the number of these supported standard file types is limited, and a user may want to import a non-standard file through the same mechanism. To this end, the <code>DataTrack</code> constructor defines an additional argument called <code>importFunction</code>. As the name suggests, the value of this argument is a function which needs to handle the mandatory <code>file</code> argument. Upon evaluation this argument will be filled in with the path to the data file, and the user-defined function needs to provide all logic necessary to parse that file into a valid <code>GRanges</code> object. From this point on everything will happen just as if the <code>range</code> argument had been this <code>GRanges</code> object. In other words, numeric metadata columns will be shown as individual samples and non-numeric columns will be silently ignored. We can exemplify this in the next code chunk. Note that the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package is using functionality from the <em><a href="https://bioconductor.org/packages/3.13/rtracklayer">rtracklayer</a></em> package for most of the file import operations, just as we do here in a more explicit way.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span>
<span class="va">dTrack3</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/DataTrack-class.html">DataTrack</a></span><span class="op">(</span>range <span class="op">=</span> <span class="va">bgFile</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, type <span class="op">=</span> <span class="st">"l"</span>, 
                      chromosome <span class="op">=</span> <span class="st">"chr19"</span>, name <span class="op">=</span> <span class="st">"bedGraph"</span>,
                      importFunction <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">file</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/export.html">import</a></span><span class="op">(</span>con<span class="op">=</span><span class="va">file</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span><span class="op">(</span><span class="va">dTrack2</span>, <span class="va">dTrack3</span><span class="op">)</span>
<span class="co">## [1] TRUE</span></code></pre></div>
<p>So far one could have easily done the whole process in two separate steps: first import the data from the file into a <code>GRanges</code> object and then provided this object to the constructor. The real power of the file support in the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package comes with streaming from indexed files. As mentioned before, only the relevant part of the data has to be loaded during the plotting operation, so the underlying data files may be quite large without decreasing the performance or causing too big of a memory footprint. We will exemplify this feature here using a small <code>bam</code> file that is provided with the package. <code>bam</code> files contain alignments of sequences (typically from a next generation sequencing experiment) to a common reference. The most natural representation of such data in a <code>DataTrack</code> is to look at the alignment coverage at a given position only and to encode this in a single metadata column.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bamFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/test.bam"</span>, package <span class="op">=</span> <span class="st">"Gviz"</span><span class="op">)</span>
<span class="va">dTrack4</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DataTrack-class.html">DataTrack</a></span><span class="op">(</span>range <span class="op">=</span> <span class="va">bamFile</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, type <span class="op">=</span> <span class="st">"l"</span>, 
                     name <span class="op">=</span> <span class="st">"Coverage"</span>, window <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, 
                     chromosome <span class="op">=</span> <span class="st">"chr1"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">dTrack4</span><span class="op">)</span>
<span class="co">## [1] "ReferenceDataTrack"</span>
<span class="co">## attr(,"package")</span>
<span class="co">## [1] "Gviz"</span>
<span class="va">dTrack4</span>
<span class="co">## ReferenceDataTrack 'Coverage'</span>
<span class="co">## | genome: hg19</span>
<span class="co">## | active chromosome: chr1</span>
<span class="co">## | referenced file: /__w/_temp/Library/Gviz/extdata/test.bam</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack4</span>, from <span class="op">=</span> <span class="fl">189990000</span>, to <span class="op">=</span> <span class="fl">190000000</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/filedt3-1.png"><!-- --></p>
<p>As seen in the previous code chunk, the <code>dTrack4</code> object is now of class <code>ReferenceDataTrack</code>. For the user this distinction is not particularly relevant with the exception that the <code>length</code> method for this class almost always returns <code>0</code> because the content of the object is only realized during the plotting operation. Obviously, streaming from the disk comes with a price in that file access is much slower than accessing RAM, however the file indexing allows for fairly rapid data retrieval, and other processes during the plotting operation tend to be much more costly, anyways. It is worth mentioning however that each plotting operation will cause reading off the file, and there are currently no caching mechanisms in place to avoid that. Nevertheless, plotting a larger chunk of the <code>bam</code> file still finishes in a reasonable time.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack4</span>, chromosome <span class="op">=</span> <span class="st">"chr1"</span>, from <span class="op">=</span> <span class="fl">189891483</span>, to <span class="op">=</span> <span class="fl">190087517</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/filedt4-1.png"><!-- --></p>
<p>Of course users can provided their own file parsing function just like we showed in the previous example. The import function now needs to be able to deal with a second mandatory argument <code>selection</code>, which is a <code>GRanges</code> object giving the genomic interval that has to be imported from the file. In addition one needs to tell the <code>DataTrack</code> constructor that data should be streamed off a file by setting the <code>stream</code> argument to <code>TRUE</code>.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">myImportFun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">file</span>, <span class="va">selection</span><span class="op">)</span><span class="op">{</span>
    <span class="co">## do something here</span>
<span class="op">}</span>
<span class="fu"><a href="../reference/DataTrack-class.html">DataTrack</a></span><span class="op">(</span>range <span class="op">=</span> <span class="va">bamFile</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, type <span class="op">=</span> <span class="st">"l"</span>,
          name <span class="op">=</span> <span class="st">"Coverage"</span>, window <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, chromosome <span class="op">=</span> <span class="st">"chr1"</span>,
          importFunction <span class="op">=</span> <span class="va">myImportFun</span>, stream <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## ReferenceDataTrack 'Coverage'</span>
<span class="co">## | genome: hg19</span>
<span class="co">## | active chromosome: chr1</span>
<span class="co">## | referenced file: /__w/_temp/Library/Gviz/extdata/test.bam</span></code></pre></div>
</div>
<div id="data-transformations" class="section level3">
<h3 class="hasAnchor">
<a href="#data-transformations" class="anchor"></a>Data transformations</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package offers quite some flexibility to transform data on the fly. This involves both rescaling operations (each data point is transformed on the track’s y-axis by a transformation function) as well as summarization and smoothing operations (the values for several genomic locations are summarized into one derived value on the track’s x-axis). To illustrate this let’s create a significantly bigger <code>DataTrack</code> than the one we used before, containing purely synthetic data for only a single sample.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Trig.html">sin</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="va">pi</span>, <span class="fl">10</span><span class="op">*</span><span class="va">pi</span>, len<span class="op">=</span><span class="fl">500</span><span class="op">)</span><span class="op">)</span>
<span class="va">dTrack.big</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DataTrack-class.html">DataTrack</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">100000</span>, len <span class="op">=</span> <span class="fl">500</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">15</span>, 
                        chromosome <span class="op">=</span> <span class="st">"chrX"</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, 
                        name <span class="op">=</span> <span class="st">"sinus"</span>,
                        data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Trig.html">sin</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="va">pi</span>, <span class="fl">5</span> <span class="op">*</span> <span class="va">pi</span>, len <span class="op">=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span> 
                          <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="fl">500</span>, <span class="fl">0.5</span>, <span class="fl">1.5</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack.big</span>, type<span class="op">=</span><span class="st">"hist"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/biggerdata-1.png"><!-- --></p>
<p>Since the available resolution on our screen is limited we can no longer distinguish between individual coordinate ranges. The <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package tries to avoid overplotting by collapsing overlapping ranges (assuming the <code>collapseTracks</code> parameter is set to <code>TRUE</code>). However, it is often desirable to summarize the data, for instance by binning values into a fixed number of windows followed by the calculation of a meaningful summary statistic. This can be archived by a combination of the <code>window</code> and <code>aggregation</code> display parameters. The former can be an integer value greater than zero giving the number of evenly-sized bins to aggregate the data in. The latter is supposed to be a user-supplied function that accepts a numeric vector as a single input parameter and returns a single aggregated numerical value. For simplicity, the most obvious aggregation functions can be selected by passing in a character scalar rather than a function. Possible values are <code>mean</code>, <code>median,</code>extreme<code>,</code>sum, <code>min</code> and <code>max</code>. These presets are also much faster because they have been optimized to operate on large numeric matrices. The default is to compute the mean value of all the binned data points.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack.big</span>, type <span class="op">=</span> <span class="st">"hist"</span>, window <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/aggregation-1.png"><!-- --></p>
<p>Instead of binning the data in fixed width bins one can also use the <code>window</code> parameter to perform more elaborate running window operations. For this to happen the parameter value has to be smaller than zero, and the additional display parameter <code>windowSize</code> can be used to control the size of the running window. This operation does not change the number of coordinate ranges on the plot, but instead the original value at a particular position is replaced by the respective sliding window value at the same position. A common use case for sliding windows on genomic ranges is to introduce a certain degree of smoothing to the data.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack.big</span>, type <span class="op">=</span> <span class="st">"hist"</span>, window <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, windowSize <span class="op">=</span> <span class="fl">2500</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/aggregation2-1.png"><!-- --></p>
<p>In addition to transforming the data on the x-axis we can also apply arbitrary transformation functions on the y-axis. One obvious use-case would be to log-transform the data prior to plotting. The framework is flexible enough however to allow for arbitrary transformation operations. The mechanism works by providing a function as the <code>transformation</code> display parameter, which takes as input a numeric vector and returns a transformed numeric vector of the same length. The following code for instance truncates the plotted data to values greater than zero.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack.big</span>, type <span class="op">=</span> <span class="st">"l"</span>, 
           transformation <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span> <span class="va">x</span><span class="op">[</span><span class="va">x</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span>; <span class="va">x</span> <span class="op">}</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/transformation-1.png"><!-- --></p>
<p>As seen before, the <code>a</code> type allows to plot average values for each of the separate groups. There is however an additional parameter <code>aggregateGroups</code> that generalizes group value aggregations. In the following example we display, for each group and at each position, the average values in the form of a dot-and-lines plot.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack</span>, groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"control"</span>, <span class="st">"treated"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>, 
           type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"b"</span><span class="op">)</span>, aggregateGroups <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/groupingAv1-1.png"><!-- --></p>
<p>This functionality again also relies on the setting of the <code>aggregation</code> parameter, and we can easily change it to display the maximum group values instead.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">dTrack</span>, groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"control"</span>, <span class="st">"treated"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>, 
           type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"b"</span><span class="op">)</span>, aggregateGroups <span class="op">=</span> <span class="cn">TRUE</span>, aggregation <span class="op">=</span> <span class="st">"max"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/groupingAv2-1.png"><!-- --></p>
</div>
<div id="display-parameters-for-datatrack-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#display-parameters-for-datatrack-objects" class="anchor"></a>Display parameters for DataTrack objects</h3>
<p>For a complete listing of all the available display parameters please see the table below or the man page of the <code>DataTrack</code> class by typing in <code><a href="../reference/DataTrack-class.html">?DataTrack</a></code> on the <code>R</code> command line.</p>
<table class="table">
<colgroup>
<col width="3%">
<col width="96%">
</colgroup>
<thead><tr class="header">
<th align="left">Display Parameter</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">aggregateGroups</td>
<td align="left">Logical scalar. Aggregate the values within a sample group using the aggregation funnction specified in the <code>aggregation</code> parameter.</td>
</tr>
<tr class="even">
<td align="left">aggregation</td>
<td align="left">Function or character scalar. Used to aggregate values in windows or for collapsing overlapping items. The function has to accept a numeric vector as a single input parameter and has to return a numeric scalar with the aggregated value. Alternatively, one of the predefined options <code>mean</code>, <code>median</code> <code>sum</code>, <code>min</code>, <code>max</code> or <code>extreme</code> can be supplied as a character scalar. Defaults to <code>mean</code>.</td>
</tr>
<tr class="odd">
<td align="left">alpha.confint</td>
<td align="left">Numeric scalar. The transparency for the confidence intervalls in confint-type plots.</td>
</tr>
<tr class="even">
<td align="left">amount</td>
<td align="left">Numeric scalar. Amount of jittering in xy-type plots. See <code>panel.xyplot</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">baseline</td>
<td align="left">Numeric scalar. Y-axis position of an optional baseline. This parameter has a special meaning for mountain-type and polygon-type plots, see the ‘Details’ section in <code>DataTrack</code> for more information.</td>
</tr>
<tr class="even">
<td align="left">box.legend</td>
<td align="left">Logical scalar. Draw a box around a legend.</td>
</tr>
<tr class="odd">
<td align="left">box.ratio</td>
<td align="left">Numeric scalar. Parameter controlling the boxplot appearance. See <code>panel.bwplot</code> for details.</td>
</tr>
<tr class="even">
<td align="left">box.width</td>
<td align="left">Numeric scalar. Parameter controlling the boxplot appearance. See <code>panel.bwplot</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">cex</td>
<td align="left">Numeric scalar. The default pixel size for plotting symbols.</td>
</tr>
<tr class="even">
<td align="left">cex.legend</td>
<td align="left">Numeric scalar. The size factor for the legend text.</td>
</tr>
<tr class="odd">
<td align="left">cex.sampleNames</td>
<td align="left">Numeric scalar. The size factor for the sample names text in heatmap or horizon plots. Defaults to an automatic setting.</td>
</tr>
<tr class="even">
<td align="left">coef</td>
<td align="left">Numeric scalar. Parameter controlling the boxplot appearance. See <code>panel.bwplot</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">col</td>
<td align="left">Character or integer vector. The color used for all line and symbol elements, unless there is a more specific control defined elsewhere. Unless <code>groups</code> are specified, only the first color in the vector is usually regarded.</td>
</tr>
<tr class="even">
<td align="left">col.baseline</td>
<td align="left">Character scalar. Color for the optional baseline, defaults to the setting of <code>col</code>.</td>
</tr>
<tr class="odd">
<td align="left">col.boxplotFrame</td>
<td align="left">Character scalar. Line color of the frame around grouped boxplots.</td>
</tr>
<tr class="even">
<td align="left">col.confint</td>
<td align="left">Character vector. Border colors for the confidence intervals for confint-type plots.</td>
</tr>
<tr class="odd">
<td align="left">col.histogram</td>
<td align="left">Character scalar. Line color in histogram-type plots.</td>
</tr>
<tr class="even">
<td align="left">col.horizon</td>
<td align="left">The line color for the segments in the <code>horizon</code>-type plot. See <code>horizonplot</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">col.mountain</td>
<td align="left">Character scalar. Line color in mountain-type and polygon-type plots, defaults to the setting of <code>col</code>.</td>
</tr>
<tr class="even">
<td align="left">col.sampleNames</td>
<td align="left">Character or integer scalar. The color used for the sample names in heatmap plots.</td>
</tr>
<tr class="odd">
<td align="left">collapse</td>
<td align="left">Logical scalar. Collapse overlapping ranges and aggregate the underlying data.</td>
</tr>
<tr class="even">
<td align="left">degree</td>
<td align="left">Numeric scalar. Parameter controlling the loess calculation for smooth and mountain-type plots. See <code>panel.loess</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">do.out</td>
<td align="left">Logical scalar. Parameter controlling the boxplot appearance. See <code>panel.bwplot</code> for details.</td>
</tr>
<tr class="even">
<td align="left">evaluation</td>
<td align="left">Numeric scalar. Parameter controlling the loess calculation for smooth and mountain-type plots. See <code>panel.loess</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">factor</td>
<td align="left">Numeric scalar. Factor to control amount of jittering in xy-type plots. See <code>panel.xyplot</code> for details.</td>
</tr>
<tr class="even">
<td align="left">family</td>
<td align="left">Character scalar. Parameter controlling the loess calculation for smooth and mountain-type plots. See <code>panel.loess</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">fill.confint</td>
<td align="left">Character vector. Fill colors for the confidence intervals for confint-type plots.</td>
</tr>
<tr class="even">
<td align="left">fill.histogram</td>
<td align="left">Character scalar. Fill color in histogram-type plots, defaults to the setting of <code>fill</code>.</td>
</tr>
<tr class="odd">
<td align="left">fill.horizon</td>
<td align="left">The fill colors for the segments in the <code>horizon</code>-type plot. This should be a vector of length six, where the first three entries are the colors for positive changes, and the latter three entries are the colors for negative changes. Defaults to a red-blue color scheme. See <code>horizonplot</code> for details.</td>
</tr>
<tr class="even">
<td align="left">fill.mountain</td>
<td align="left">Character vector of length 2. Fill color in mountain-type and polygon-type plots.</td>
</tr>
<tr class="odd">
<td align="left">fontcolor.legend</td>
<td align="left">Integer or character scalar. The font color for the legend text.</td>
</tr>
<tr class="even">
<td align="left">fontface.legend</td>
<td align="left">Integer or character scalar. The font face for the legend text.</td>
</tr>
<tr class="odd">
<td align="left">fontfamily.legend</td>
<td align="left">Integer or character scalar. The font family for the legend text.</td>
</tr>
<tr class="even">
<td align="left">fontsize.legend</td>
<td align="left">Numeric scalar. The pixel size for the legend text.</td>
</tr>
<tr class="odd">
<td align="left">gradient</td>
<td align="left">Character vector. The base colors for the <code>gradient</code> plotting type or the <code>heatmap</code> type with a single group. When plotting heatmaps with more than one group, the <code>col</code> parameter can be used to control the group color scheme, however the gradient will always be from white to ‘col’ and thus does not offer as much flexibility as this <code>gradient</code> parameter.</td>
</tr>
<tr class="even">
<td align="left">grid</td>
<td align="left">Logical vector. Draw a line grid under the track content.</td>
</tr>
<tr class="odd">
<td align="left">groups</td>
<td align="left">Vector coercable to a factor. Optional sample grouping. See ‘Details’ section in <code>DataTrack</code> for further information.</td>
</tr>
<tr class="even">
<td align="left">horizon.origin</td>
<td align="left">The baseline relative to which changes are indicated on the <code>horizon</code>-type plot. See <code>horizonplot</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">horizon.scale</td>
<td align="left">The scale for each of the segments in the <code>horizon</code>-type plot. Defaults to 1/3 of the absolute data range. See <code>horizonplot</code> for details.</td>
</tr>
<tr class="even">
<td align="left">jitter.x</td>
<td align="left">Logical scalar. Toggle on jittering on the x axis in xy-type plots. See <code>panel.xyplot</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">jitter.y</td>
<td align="left">Logical scalar. Toggle off jittering on the y axis in xy-type plots. See <code>panel.xyplot</code> for details.</td>
</tr>
<tr class="even">
<td align="left">legend</td>
<td align="left">Boolean triggering the addition of a legend to the track to indicate groups. This only has an effect if at least two groups are present.</td>
</tr>
<tr class="odd">
<td align="left">levels.fos</td>
<td align="left">Numeric scalar. Parameter controlling the boxplot appearance. See <code>panel.bwplot</code> for details.</td>
</tr>
<tr class="even">
<td align="left">lineheight.legend</td>
<td align="left">Numeric scalar. The line height for the legend text.</td>
</tr>
<tr class="odd">
<td align="left">lty.baseline</td>
<td align="left">Character or numeric scalar. Line type of the optional baseline, defaults to the setting of <code>lty</code>.</td>
</tr>
<tr class="even">
<td align="left">lty.mountain</td>
<td align="left">Character or numeric scalar. Line type in mountain-type and polygon-type plots, defaults to the setting of <code>lty</code>.</td>
</tr>
<tr class="odd">
<td align="left">lwd.baseline</td>
<td align="left">Numeric scalar. Line width of the optional baseline, defaults to the setting of <code>lwd</code>.</td>
</tr>
<tr class="even">
<td align="left">lwd.mountain</td>
<td align="left">Numeric scalar. Line width in mountain-type and polygon-type plots, defaults to the setting of <code>lwd</code>.</td>
</tr>
<tr class="odd">
<td align="left">min.distance</td>
<td align="left">Numeric scalar. The mimimum distance in pixel below which to collapse ranges.</td>
</tr>
<tr class="even">
<td align="left">missingAsZero</td>
<td align="left">Logical scalar. Defines how the missing values are treated in the aggregation procedure with running window. Setting it to <code>TRUE</code> fills empty positions with zeros, which is default. <code>FALSE</code> fills empty positions with <code>NA</code>.</td>
</tr>
<tr class="odd">
<td align="left">na.rm</td>
<td align="left">Boolean controlling whether to discard all NA values when plotting or to keep empty spaces for NAs</td>
</tr>
<tr class="even">
<td align="left">ncolor</td>
<td align="left">Integer scalar. The number of colors for the ‘gradient’ plotting type</td>
</tr>
<tr class="odd">
<td align="left">notch</td>
<td align="left">Logical scalar. Parameter controlling the boxplot appearance. See <code>panel.bwplot</code> for details.</td>
</tr>
<tr class="even">
<td align="left">notch.frac</td>
<td align="left">Numeric scalar. Parameter controlling the boxplot appearance. See <code>panel.bwplot</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">pch</td>
<td align="left">Integer scalar. The type of glyph used for plotting symbols.</td>
</tr>
<tr class="even">
<td align="left">separator</td>
<td align="left">Numeric scalar. Number of pixels used to separate individual samples in heatmap- and horizon-type plots.</td>
</tr>
<tr class="odd">
<td align="left">showColorBar</td>
<td align="left">Boolean. Indicate the data range color mapping in the axis for ‘heatmap’ or ‘gradient’ types.</td>
</tr>
<tr class="even">
<td align="left">showSampleNames</td>
<td align="left">Boolean. Display the names of the individual samples in a heatmap or a horizon plot.</td>
</tr>
<tr class="odd">
<td align="left">span</td>
<td align="left">Numeric scalar. Parameter controlling the loess calculation for smooth and mountain-type plots. See <code>panel.loess</code> for details.</td>
</tr>
<tr class="even">
<td align="left">stackedBars</td>
<td align="left">Logical scalar. When there are several data groups, draw the histogram-type plots as stacked barplots or grouped side by side.</td>
</tr>
<tr class="odd">
<td align="left">stats</td>
<td align="left">Function. Parameter controlling the boxplot appearance. See <code>panel.bwplot</code> for details.</td>
</tr>
<tr class="even">
<td align="left">transformation</td>
<td align="left">Function. Applied to the data matrix prior to plotting or when calling the <code>score</code> method. The function should accept exactly one input argument and its return value needs to be a numeric vector which can be coerced back into a data matrix of identical dimensionality as the input data.</td>
</tr>
<tr class="odd">
<td align="left">type</td>
<td align="left">Character vector. The plot type, one or several in <code>p</code>,<code>l</code>, <code>b</code>, <code>a</code>, <code>a_confint</code>, <code>s</code>, <code>g</code>, <code>r</code>, <code>S</code>, <code>confint</code>, <code>smooth</code>, <code>histogram</code>, <code>mountain</code>, <code>polygon</code>, <code>h</code>, <code>boxplot</code>, <code>gradient</code>, <code>heatmap</code>, <code>horizon</code>. See ‘Details’ section in <code>DataTrack</code> for more information on the individual plotting types.</td>
</tr>
<tr class="even">
<td align="left">varwidth</td>
<td align="left">Logical scalar. Parameter controlling the boxplot appearance. See <code>panel.bwplot</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">window</td>
<td align="left">Numeric or character scalar. Aggregate the rows values of the data matrix to <code>window</code> equally sized slices on the data range using the method defined in <code>aggregation</code>. If negative, apply a running window of size <code>windowSize</code> using the same aggregation method. Alternatively, the special value <code>auto</code> causes the function to determine the optimal window size to avoid overplotting, and <code>fixed</code> uses fixed-size windows of size <code>windowSize</code>.</td>
</tr>
<tr class="even">
<td align="left">windowSize</td>
<td align="left">Numeric scalar. The size of the running window when the value of <code>window</code> is negative.</td>
</tr>
<tr class="odd">
<td align="left">ylim</td>
<td align="left">Numeric vector of length 2. The range of the y-axis scale.</td>
</tr>
<tr class="even">
<td align="left">yTicksAt</td>
<td align="left">Numeric vector. The points at which y-axis tick-marks are to be drawn. By default, when <code>NULL</code>, tickmark locations are computed.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="annotationtrack" class="section level2">
<h2 class="hasAnchor">
<a href="#annotationtrack" class="anchor"></a>AnnotationTrack</h2>
<p><code>AnnotationTrack</code> objects are the multi-purpose tracks in the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package. Essentially they consist of one or several genomic ranges that can be grouped into composite annotation elements if needed. In principle this would be enough to represent everything from CpG islands to complex gene models, however for the latter the package defines the specialized <code>GeneRegionTrack</code> class, which will be highlighted in a separate section. Most of the features discussed here will also apply to <code>GeneRegionTrack</code> objects, though. As a matter of fact, the <code>GeneRegionTrack</code> class inherits directly from class <code>AnnotationTrack</code>.</p>
<p><code>AnnotationTrack</code> objects are easily instantiated using the constructor function of the same name. The necessary building blocks are the range coordinates, a chromosome and a genome identifier. Again we try to be flexible in the way this information can be passed to the function, either in the form of separate function arguments, as <code>IRanges</code>, <code>GRanges</code> or <code>data.frame</code> objects. Optionally, we can pass in the strand information for the annotation features and some useful identifiers. A somewhat special case is to build the object from a <code>GRangesList</code> object, which will automatically preserve the element grouping information contained in the list structure. For the full details on the constructor function and the accepted arguments see <code><a href="../reference/AnnotationTrack-class.html">?AnnotationTrack</a></code>. Let’s take a look at a very simple track:</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">40</span>, <span class="fl">120</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">15</span>, 
                          chromosome <span class="op">=</span> <span class="st">"chrX"</span>, 
                          strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"*"</span>, <span class="st">"-"</span><span class="op">)</span>,
                          id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Huey"</span>, <span class="st">"Dewey"</span>, <span class="st">"Louie"</span><span class="op">)</span>, 
                          genome <span class="op">=</span> <span class="st">"hg19"</span>, name <span class="op">=</span> <span class="st">"foo"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/anntrack1-1.png"><!-- --></p>
<p>The ranges are plotted as simple boxes if no strand information is available, or as arrows to indicate their direction. We can change the range item shapes by setting the <code>shape</code> display parameter. It can also be helpful to add the names for the individual features to the plot. This can be archived by setting the <code>featureAnnotation</code> parameter to <code>'id'</code>.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack</span>, shape <span class="op">=</span> <span class="st">"box"</span>, featureAnnotation <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/anntrack2-1.png"><!-- --></p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack</span>, shape <span class="op">=</span> <span class="st">"ellipse"</span>, featureAnnotation <span class="op">=</span> <span class="st">"id"</span>, 
           fontcolor.feature <span class="op">=</span> <span class="st">"darkblue"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/anntrack3-1.png"><!-- --></p>
<p>In this very simplistic example each annotation feature consisted of a single range. In real life the genomic annotation features that we encounter often consists of several sub-units. We can create such composite <code>AnnotationTrack</code> objects by providing a grouping factor to the constructor. It needs to be of similar length as the total number of atomic features in the track, i.e, the number of genomic ranges that are passed to the constructor. The levels of the this factor will be used as internal identifiers for the individual composite feature groups, and we can toggle on their printing by setting <code>groupAnnotation</code> to ‘group’.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aTrack.groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">180</span>, <span class="fl">260</span>, <span class="fl">460</span>, <span class="fl">860</span>, <span class="fl">1240</span><span class="op">)</span>, 
                                 width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">15</span>, <span class="fl">20</span>, <span class="fl">40</span>, <span class="fl">100</span>, <span class="fl">200</span>, <span class="fl">20</span><span class="op">)</span>,
                                 chromosome <span class="op">=</span> <span class="st">"chrX"</span>,
                                 strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"*"</span>, <span class="st">"-"</span><span class="op">)</span>, 
                                              <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>,
                                 group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Huey"</span>, <span class="st">"Dewey"</span>, <span class="st">"Louie"</span><span class="op">)</span>, 
                                             <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>,
                                 genome <span class="op">=</span> <span class="st">"hg19"</span>, name <span class="op">=</span> <span class="st">"foo"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack.groups</span>, groupAnnotation <span class="op">=</span> <span class="st">"group"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/anntrack4f-1.png"><!-- --></p>
<p>We can control the placement of the group labels through the <code>just.group</code> parameter.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack.groups</span>, groupAnnotation <span class="op">=</span> <span class="st">"group"</span>, 
           just.group <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/anntrack4af-1.png"><!-- --></p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack.groups</span>, groupAnnotation <span class="op">=</span> <span class="st">"group"</span>, 
           just.group <span class="op">=</span> <span class="st">"above"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/anntrack4bf-1.png"><!-- --></p>
<p>Arranging items on the plotting canvas is relatively straight forward as long as there are no overlaps between individual regions or groups of regions. Those inevitably cause overplotting which could seriously obfuscate the information on the plot. A logical solution to this problem is to stack overlapping items in separate horizontal lines to accommodate all of them. This involves some optimization, and the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package automatically tries to come up with the most compact arrangement. Let’s exemplify this feature with a slightly modified <code>AnnotationTrack</code> object.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aTrack.stacked</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">180</span>, <span class="fl">260</span>, <span class="fl">800</span>, <span class="fl">600</span>, <span class="fl">1240</span><span class="op">)</span>,
                                  width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">15</span>, <span class="fl">20</span>, <span class="fl">40</span>, <span class="fl">100</span>, <span class="fl">500</span>, <span class="fl">20</span><span class="op">)</span>,
                                  chromosome <span class="op">=</span> <span class="st">"chrX"</span>,
                                  strand <span class="op">=</span> <span class="st">"*"</span>,
                                  group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Huey"</span>, <span class="st">"Dewey"</span>, <span class="st">"Louie"</span><span class="op">)</span>, 
                                              <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>,
                                  genome <span class="op">=</span> <span class="st">"hg19"</span>, name <span class="op">=</span> <span class="st">"foo"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack.stacked</span>, groupAnnotation<span class="op">=</span><span class="st">"group"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/stacking1-1.png"><!-- --></p>
<p>We now have our three annotation feature groups distributed over two horizontal lines. One can control the stacking of overlapping items using the <code>stacking</code> display parameter. Currently the three values <code>squish</code>, <code>dense</code> and <code>hide</code> are supported. Horizontal stacking is enabled via the <code>squish</code> option, which also is the default. The <code>dense</code> setting forces overlapping items to be joined in one meta-item and <code>hide</code> all together disables the plotting of <code>AnnotationTrack</code> items. Please note that adding identifiers to the plot only works for the <code>squish</code> option.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack.stacked</span>, stacking <span class="op">=</span> <span class="st">"dense"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/stacking2-1.png"><!-- --></p>
<p>In addition to annotation groups there is also the notion of a feature type in the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package. Feature types are simply different types of annotation regions (e.g., mRNA transcripts, miRNAs, rRNAs, etc.) that are indicated by different colors. There is no limit on the number of different features, however each element in a grouped annotation item needs to be of the same feature type. We can query and set features using the <code>feature</code> and <code>feature&lt;-</code> methods.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/RangeTrack-class.html">feature</a></span><span class="op">(</span><span class="va">aTrack.stacked</span><span class="op">)</span>
<span class="co">## [1] "unknown" "unknown" "unknown" "unknown" "unknown" "unknown"</span>
<span class="fu"><a href="../reference/RangeTrack-class.html">feature</a></span><span class="op">(</span><span class="va">aTrack.stacked</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"foo"</span>, <span class="st">"bar"</span>, <span class="st">"bar"</span>, <span class="st">"bar"</span>, <span class="st">"no"</span>, <span class="st">"no"</span><span class="op">)</span></code></pre></div>
<p>Features types can also be shown as either the group or the item annotation on the plot.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack.stacked</span>, featureAnnotation <span class="op">=</span> <span class="st">"feature"</span>, 
           groupAnnotation <span class="op">=</span> <span class="st">"feature"</span>, fontcolor.feature <span class="op">=</span> <span class="fl">1</span>, 
           cex.feature <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/featuresIdPlot-1.png"><!-- --></p>
<p>Unless we tell the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package how to deal with the respective feature types they will all be treated in a similar fashion, i.e., they will be plotted using the default color as defined by the <code>fill</code> display parameter. To define colors for individual feature types we simply have to add them as additional display parameters, where the parameter name matches the feature type and its value is supposed to be a valid R color qualifier. Of course this implies that we can only use feature names that are not already occupied by other display parameters defined in the package.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack.stacked</span>, groupAnnotation <span class="op">=</span> <span class="st">"group"</span>, 
           foo <span class="op">=</span> <span class="st">"darkred"</span>, bar <span class="op">=</span> <span class="st">"darkgreen"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/featuresPlotf-1.png"><!-- --></p>
<p>Stacking of annotation items to avoid overplotting only works as long as there is enough real estate on the plotting canvas to separate all items, i.e., we need all items to be at least a single pixel wide to correctly display them. This limitation is automatically enforced by the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package, however it implies that unless neighboring items are more than one pixel apart we can not distinguish between them and will inevitably introduce a certain amount of overplotting. This means that on a common screen device we can only look at a very limited genomic region of a few kb in full resolution. Given that an average chromosome is in the order of a few gb, we still need a reasonable way to deal with the overplotting problem despite the item stacking functionality. As default, the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package will merge all overlapping items into one unified meta-item and only plot that (see ‘Collapse’ section below for details). In order to indicate the amount of overplotting that was introduced by this process we can use the <code>showOverplotting</code> display parameter. It uses a color scale (based on the original colors defined for the track), with lighter colors indicating areas of low or no overplotting, and more saturated colors indicating areas of high overplotting density. We exemplify this feature on an <code>AnnotationTrack</code> object that represents a good portion of a real human chromosome.</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"denseAnnTrack"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">denseAnnTrack</span>, showOverplotting <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/overplotting-1.png"><!-- --></p>
<div id="collapsing" class="section level3">
<h3 class="hasAnchor">
<a href="#collapsing" class="anchor"></a>Collapsing</h3>
<p>All track types that inherit from class <code>AnnotationTrack</code> support the collapsing of overlapping track items, either because they have initially been defined as overlapping coordinates, or because the current device resolution does not allow to sufficiently separate them. For instance, two elements of a feature group may be separated by 100 base pairs on the genomic scale, however when plotted to the screen, those 100 base pairs translate to a distance of less than one pixel. In this case we can no longer show the items as two separate entities. One solution to this problem would be to allow for arbitrary overplotting, in which case the last one of the overlapping items that is drawn on the device wins. This is not optimal in many ways, and it also poses a significant burden on the graphical engine because a lot of stuff has to be drawn which no one will ever see.</p>
<p>To this end the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package provides an infrastructure to reasonably collapse overlapping items, thereby adjusting the information content that can be shown to the available device resolution. By default this feature is turned on, and the user does not have to worry too much about it. However, one should be aware of the consequences this may have on a given visualization. If you absolutely do not want collapsing to take place, you may completely turn it off by setting the display parameter <code>collapse</code> to <code>FALSE</code>. Please note that by doing this the <code>showOverplotting</code> parameter will also stop working. If you opt in, there is some considerable amount of detailed control to fine tune the collapsing to your needs.</p>
<p>Let’s start with a small example track for which element collapsing has been turned off and no adjustments to the ranges have been made. We plot both the item identifiers and the group identifiers to exemplify what is going on.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">collapseTrack</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">ctrack</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/collapse1f-1.png"><!-- --></p>
<p>The first thing to notice is that the for item <code>d</code> we do see the item identifier but not the range itself. This is due to the fact that the width of the item is smaller than a single pixel, and hence the graphics system can not display it (Note that this is only true for certain devices. The quartz device on the Mac seems to be a little smarter about this). There are also the two items <code>e</code> and <code>f</code> which seem to overlay each other completely, and another two items which appear to be just a single fused item (<code>k</code> and <code>l</code>). Again, this is a resolution issue as their relative distance is smaller than a single pixel, so all we see is a single range and some ugly overplotted identifiers. We can control the first issue by setting the minimum pixel width of a plotted item to be one pixel using the <code>min.width</code> display parameter.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">ctrack</span>, min.width <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/collapse2f-1.png"><!-- --></p>
<p>Now the item <code>d</code> has a plotable size and can be drawn to the device. The overplotted items are still rather annoying, but the only way to get rid of those is to turn item collapsing back on.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">ctrack</span>, min.width <span class="op">=</span> <span class="fl">1</span>, collapse <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/collapse3f-1.png"><!-- --></p>
<p>Now all items that could not be separated by at least one pixel have been merged into a single meta-item, and the software has taken care of the identifiers for you, too. The merging operation is aware of the grouping information, so no two groups where joint together. Sometimes a single pixel width or a single pixel distance is not enough to get a good visualization. In these cases one could decide to enforce even larger values. We can do this not only for the minimum width, but also for the minimum distance by setting the <code>min.distance</code> parameter.</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">ctrack</span>, min.width <span class="op">=</span> <span class="fl">3</span>, min.distance <span class="op">=</span> <span class="fl">5</span>, collapse <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/collapse4f-1.png"><!-- --></p>
<p>This time also the two items <code>b</code> and <code>c</code> have been merged, and all ranges are now at least 3 pixels wide. Depending on the density of items on the plot even this reduction can be insufficient. Because we did not merge complete groups we might still end up with quite a lot of stacks to accommodate all the information. To this end the display parameter <code>mergeGroups</code> can be used to disable absolute group separation. Rather than blindly merging all groups (as it is done when <code>stacking='dense'</code>) however, the software will only join those overlapping group ranges for which all items are already merged into a single meta item.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">ctrack</span>, min.width <span class="op">=</span> <span class="fl">3</span>, min.distance <span class="op">=</span> <span class="fl">5</span>, collapse <span class="op">=</span> <span class="cn">TRUE</span>,
           mergeGroups <span class="op">=</span> <span class="cn">TRUE</span>, extend.left <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/collapse5f-1.png"><!-- --></p>
</div>
<div id="building-annotationtrack-objects-from-files" class="section level3">
<h3 class="hasAnchor">
<a href="#building-annotationtrack-objects-from-files" class="anchor"></a>Building AnnotationTrack objects from files</h3>
<p>Just like we showed before for <code>DataTrack</code> objects it is possible to stream the data for an <code>AnnotationTrack</code> object off the file system. We will again use the <code>bam</code> file as an example. This time the default import function reads the coordinates of all the sequence alignments from the file and even knows how to deal with grouped elements.</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aTrack2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span>range <span class="op">=</span> <span class="va">bamFile</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, 
                           name <span class="op">=</span> <span class="st">"Reads"</span>, chromosome <span class="op">=</span> <span class="st">"chr1"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">aTrack2</span><span class="op">)</span>
<span class="co">## [1] "ReferenceAnnotationTrack"</span>
<span class="co">## attr(,"package")</span>
<span class="co">## [1] "Gviz"</span>
<span class="va">aTrack2</span>
<span class="co">## ReferenceAnnotationTrack 'Reads'</span>
<span class="co">## | genome: hg19</span>
<span class="co">## | active chromosome: chr1</span>
<span class="co">## | referenced file: /__w/_temp/Library/Gviz/extdata/test.bam</span>
<span class="co">## | mapping: id=id, group=group</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack2</span>, from <span class="op">=</span> <span class="fl">189995000</span>, to <span class="op">=</span> <span class="fl">190000000</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/fileat1-1.png"><!-- --></p>
<p>Since <code>AnnotationTrack</code> objects are somewhat more complicated than <code>DataTrack</code> objects, the constructor provides another level of flexibility when reading the data from a file. More specifically, the user has the ability to map the provided metadata columns of the <code>GRanges</code> object that is returned by the import function to the respective columns in the final <code>AnnotationTrack</code> object. If no explicit mapping is provided, the package will try to find a default mapping based on the input file type and the desired track type. The current mapping of a track is displayed by its <code>show</code> method as can be seen above. The mechanism for changing this default mapping is quite straight forward: if the values in the relevant constructor arguments (in this case <code>group</code>, <code>id</code> and <code>feature</code>) are character scalars, the metadata columns of the <code>GRanges</code> object that is returned by the import function are mapped against their values. For instance we can map the <code>id</code> column to the <code>group</code> attribute and thus suppress the grouping of sequence alignments that originate from the same sequencing read.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aTrack3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span>range <span class="op">=</span> <span class="va">bamFile</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, 
                           name <span class="op">=</span> <span class="st">"Reads"</span>, chromosome <span class="op">=</span> <span class="st">"chr1"</span>, 
                           group <span class="op">=</span> <span class="st">"id"</span><span class="op">)</span>
<span class="va">aTrack3</span>
<span class="co">## ReferenceAnnotationTrack 'Reads'</span>
<span class="co">## | genome: hg19</span>
<span class="co">## | active chromosome: chr1</span>
<span class="co">## | referenced file: /__w/_temp/Library/Gviz/extdata/test.bam</span>
<span class="co">## | mapping: id=id, group=id</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">aTrack3</span>, from <span class="op">=</span> <span class="fl">189995000</span>, to <span class="op">=</span> <span class="fl">190000000</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/fileat2-1.png"><!-- --></p>
<p>Please note that if no metadata column can be mapped to the constructor arguments their values are assumed to be normal character scalars and they will be recycled accordingly to fill the track’s slots. One can ask for the package’s default mappings by using the <code>availableDefaultMapping</code> function.</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/ReferenceTrack-class.html">availableDefaultMapping</a></span><span class="op">(</span><span class="va">bamFile</span>, <span class="st">"AnnotationTrack"</span><span class="op">)</span>
<span class="co">## $id</span>
<span class="co">## [1] "id"</span>
<span class="co">## </span>
<span class="co">## $group</span>
<span class="co">## [1] "group"</span></code></pre></div>
<p>We can now plot both the <code>DataTrack</code> representation as well as the <code>AnnotationTrack</code> representation of the <code>bam</code> file together to prove that the underlying data are indeed identical.</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">dTrack4</span>, <span class="va">aTrack2</span><span class="op">)</span>, from <span class="op">=</span> <span class="fl">189990000</span>, to <span class="op">=</span> <span class="fl">190000000</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/fileat4-1.png"><!-- --></p>
</div>
<div id="display-parameters-for-annotationtrack-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#display-parameters-for-annotationtrack-objects" class="anchor"></a>Display parameters for AnnotationTrack objects</h3>
<p>For a complete listing of all the available display parameters please see the table below or the man page of the <code>AnnotationTrack</code> class by typing in <code><a href="../reference/AnnotationTrack-class.html">?AnnotationTrack</a></code> on the <code>R</code> command line.</p>
<table class="table">
<colgroup>
<col width="5%">
<col width="94%">
</colgroup>
<thead><tr class="header">
<th align="left">Display Parameter</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">arrowHeadMaxWidth</td>
<td align="left">Numeric scalar. The maximum width of the arrow head in pixels if <code>shape</code> is <code>arrow</code>.</td>
</tr>
<tr class="even">
<td align="left">arrowHeadWidth</td>
<td align="left">Numeric scalar. The width of the arrow head in pixels if <code>shape</code> is <code>fixedArrow</code>.</td>
</tr>
<tr class="odd">
<td align="left">cex</td>
<td align="left">Numeric scalar. The font expansion factor for item identifiers.</td>
</tr>
<tr class="even">
<td align="left">cex.group</td>
<td align="left">Numeric scalar. The font expansion factor for the group-level annotation.</td>
</tr>
<tr class="odd">
<td align="left">col</td>
<td align="left">Character or integer scalar. The border color for all track items.</td>
</tr>
<tr class="even">
<td align="left">col.line</td>
<td align="left">Character scalar. The color used for connecting lines between grouped items. Defaults to a light gray, but if set to <code>NULL</code> the same color as for the first item in the group is used.</td>
</tr>
<tr class="odd">
<td align="left">featureAnnotation</td>
<td align="left">Character scalar. Add annotation information to the individual track elements. This can be a value in <code>id</code>, <code>group</code> or <code>feature</code>. Defaults to <code>id</code>. Only works if <code>showFeatureId</code> is not <code>FALSE</code>.</td>
</tr>
<tr class="even">
<td align="left">fill</td>
<td align="left">Character or integer scalar. The fill color for untyped items. This is also used to connect grouped items. See <code>grouping</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">fontcolor.group</td>
<td align="left">Character or integer scalar. The font color for the group-level annotation.</td>
</tr>
<tr class="even">
<td align="left">fontcolor.item</td>
<td align="left">Character or integer scalar. The font color for item identifiers.</td>
</tr>
<tr class="odd">
<td align="left">fontface.group</td>
<td align="left">Numeric scalar. The font face for the group-level annotation.</td>
</tr>
<tr class="even">
<td align="left">fontfamily.group</td>
<td align="left">Character scalar. The font family for the group-level annotation.</td>
</tr>
<tr class="odd">
<td align="left">fontsize.group</td>
<td align="left">Numeric scalar. The font size for the group-level annotation.</td>
</tr>
<tr class="even">
<td align="left">groupAnnotation</td>
<td align="left">Character scalar. Add annotation information as group labels. This can be a value in <code>id</code>, <code>group</code> or <code>feature</code>. Defaults to <code>group</code>. Only works if <code>showId</code> is not <code>FALSE</code>.</td>
</tr>
<tr class="odd">
<td align="left">just.group</td>
<td align="left">Character scalar. the justification of group labels. Either <code>left</code>, <code>right</code>, <code>above</code> or <code>below</code>.</td>
</tr>
<tr class="even">
<td align="left">lex</td>
<td align="left">Numeric scalar. The line expansion factor for all track items. This is also used to connect grouped items. See <code>grouping</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">lineheight</td>
<td align="left">Numeric scalar. The font line height for item identifiers.</td>
</tr>
<tr class="even">
<td align="left">lty</td>
<td align="left">Character or integer scalar. The line type for all track items. This is also used to connect grouped items. See <code>grouping</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">lwd</td>
<td align="left">Integer scalar. The line width for all track items. This is also used to connect grouped items. See <code>grouping</code> for details.</td>
</tr>
<tr class="even">
<td align="left">mergeGroups</td>
<td align="left">Logical scalar. Merge fully overlapping groups if <code>collapse==TRUE</code>.</td>
</tr>
<tr class="odd">
<td align="left">min.height</td>
<td align="left">Numeric scalar. The minimum range height in pixels to display. All ranges are expanded to this size in order to avoid rendering issues. See <code>collapsing</code> for details. For feathered bars indicating the strandedness of grouped items this also controls the height of the arrow feathers.</td>
</tr>
<tr class="even">
<td align="left">min.width</td>
<td align="left">Numeric scalar. The minimum range width in pixels to display. All ranges are expanded to this size in order to avoid rendering issues. See <code>collapsing</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">rotation</td>
<td align="left">Numeric scalar. The degree of text rotation for item identifiers.</td>
</tr>
<tr class="even">
<td align="left">rotation.group</td>
<td align="left">Numeric scalar. The degree of text rotation for group labels.</td>
</tr>
<tr class="odd">
<td align="left">rotation.item</td>
<td align="left">Numeric scalar. The degree of text rotation for item identifiers.</td>
</tr>
<tr class="even">
<td align="left">shape</td>
<td align="left">Character scalar. The shape in which to display the track items. Currently only <code>box</code>, <code>arrow</code>, <code>fixedArrow</code>, <code>ellipse</code>, and <code>smallArrow</code> are implemented.</td>
</tr>
<tr class="odd">
<td align="left">showFeatureId</td>
<td align="left">Logical scalar. Control whether to plot the individual track item identifiers.</td>
</tr>
<tr class="even">
<td align="left">showId</td>
<td align="left">Logical scalar. Control whether to annotate individual groups.</td>
</tr>
<tr class="odd">
<td align="left">showOverplotting</td>
<td align="left">Logical scalar. Use a color gradient to show the amount of overplotting for collapsed items. This implies that <code>collapse==TRUE</code>
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="generegiontrack" class="section level2">
<h2 class="hasAnchor">
<a href="#generegiontrack" class="anchor"></a>GeneRegionTrack</h2>
<p><code>GeneRegionTrack</code> objects are in principle very similar to <code>AnnotationTrack</code> objects. The only difference is that they are a little more gene/transcript centric, both in terms of plotting layout and user interaction, and that they may define a global start and end position (a feature which is not particularly relevant for the normal user). The constructor function of the same name is a convenient tool to instantiate the object from a variety of different sources. In a nutshell, we need to pass start and end positions (or the width) of each annotation feature in the track and also supply the exon, transcript and gene identifiers for each item which will be used to create the transcript groupings. A somewhat special case is to build a <code>GeneRegionTrack</code> object directly from one of the popular <code>TxDb</code> objects, an option that is treated in more detail below. For more information about all the available options see the class’s manual page (<code><a href="../reference/GeneRegionTrack-class.html">?GeneRegionTrack</a></code>).</p>
<p>There are a number of accessor methods that make it easy to query and replace for instance exon, transcript or gene assignments. There is also some support for gene aliases or gene symbols which are often times more useful than cryptic data base gene identifiers. The following code that re-uses the <code>GeneRegionTrack</code> object from the first section exemplifies some of these features.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneModels</span><span class="op">)</span>
<span class="va">grtrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GeneRegionTrack-class.html">GeneRegionTrack</a></span><span class="op">(</span><span class="va">geneModels</span>, genome <span class="op">=</span> <span class="va">gen</span>, chromosome <span class="op">=</span> <span class="va">chr</span>, 
                           name <span class="op">=</span> <span class="st">"foo"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/GeneRegionTrack-class.html">gene</a></span><span class="op">(</span><span class="va">grtrack</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/GeneRegionTrack-class.html">transcript</a></span><span class="op">(</span><span class="va">grtrack</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/GeneRegionTrack-class.html">exon</a></span><span class="op">(</span><span class="va">grtrack</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/GeneRegionTrack-class.html">symbol</a></span><span class="op">(</span><span class="va">grtrack</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grtrack</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/generegtrackf-1.png"><!-- --></p>
<p>Both exon- and transcript-centric annotation can be added to the plot using the <code>transcriptAnnotation</code> and <code>exonAnnotation</code> display parameters.</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grtrack</span>, transcriptAnnotation <span class="op">=</span> <span class="st">"symbol"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/generegtrack2af-1.png"><!-- --></p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grtrack</span>, transcriptAnnotation <span class="op">=</span> <span class="st">"transcript"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/generegtrack2bf-1.png"><!-- --></p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grtrack</span>, exonAnnotation <span class="op">=</span> <span class="st">"exon"</span>, extend.left <span class="op">=</span> <span class="op">-</span><span class="fl">0.8</span>, 
           fontcolor.exon <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/generegtrack2cf-1.png"><!-- --></p>
<p>Since we have the gene and transcript level information as part of our <code>GeneRegionTrack</code> objects we can ask the package to collapse all of our gene models from individual exons and transcripts down to gene body locations by setting the <code>collapseTranscripts</code> display parameter to <code>TRUE</code>.</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grtrack</span>, collapseTranscripts <span class="op">=</span> <span class="cn">TRUE</span>, shape <span class="op">=</span> <span class="st">"arrow"</span>, 
           transcriptAnnotation <span class="op">=</span> <span class="st">"symbol"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/generegtrack3f-1.png"><!-- --></p>
<p>Collapsing down all the way to a gene is not always the desired operation, and the parameters offers a bit more control. For instance we could decide to plot just the longest transcript by setting it to <code>longest</code>.</p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grtrack</span>, collapseTranscripts <span class="op">=</span> <span class="st">"longest"</span>, shape <span class="op">=</span> <span class="st">"arrow"</span>, 
           transcriptAnnotation <span class="op">=</span> <span class="st">"symbol"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/generegtrack3g-1.png"><!-- --></p>
<p>Or we can plot the union of all the exons in some sort of meta-transcript.</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">grtrack</span>, collapseTranscripts <span class="op">=</span> <span class="st">"meta"</span>, shape <span class="op">=</span> <span class="st">"arrow"</span>, 
           transcriptAnnotation <span class="op">=</span> <span class="st">"symbol"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/generegtrack3h-1.png"><!-- --></p>
<div id="building-generegiontrack-objects-from-txdbs" class="section level3">
<h3 class="hasAnchor">
<a href="#building-generegiontrack-objects-from-txdbs" class="anchor"></a>Building GeneRegionTrack objects from TxDbs</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.13/GenomicFeatures">GenomicFeatures</a></em> packages provides an elegant framework to download gene model information from online sources and to store it locally in a SQLite data base. Because these so called <code>TxDb</code> objects have become the de-facto standard for genome annotation information in Bioconductor we tried to make it as simple as possible to convert them into <code>GeneRegionTracks</code>. Essentially one only has to call the constructor function with the <code>TxDb</code> object as a single argument. We exemplify this on a small sample data set that comes with the <em><a href="https://bioconductor.org/packages/3.13/GenomicFeatures">GenomicFeatures</a></em> package.</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicFeatures">GenomicFeatures</a></span><span class="op">)</span>
<span class="va">samplefile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"hg19_knownGene_sample.sqlite"</span>, 
                          package <span class="op">=</span> <span class="st">"GenomicFeatures"</span><span class="op">)</span>
<span class="va">txdb</span> <span class="op">&lt;-</span> <span class="fu">loadDb</span><span class="op">(</span><span class="va">samplefile</span><span class="op">)</span>
<span class="fu"><a href="../reference/GeneRegionTrack-class.html">GeneRegionTrack</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span>
<span class="co">## GeneRegionTrack 'GeneRegionTrack'</span>
<span class="co">## | genome: hg19</span>
<span class="co">## | active chromosome: chr1</span>
<span class="co">## | annotation features: 162</span>
<span class="co">## There are 1808 additional annotation features on 26 further chromosomes</span>
<span class="co">##   chr10: 17</span>
<span class="co">##   chr11: 123</span>
<span class="co">##   chr12: 25</span>
<span class="co">##   chr13: 37</span>
<span class="co">##   chr14: 55</span>
<span class="co">##   ...</span>
<span class="co">##   chr6_ssto_hap7: 28</span>
<span class="co">##   chr7: 46</span>
<span class="co">##   chr8: 40</span>
<span class="co">##   chr9: 52</span>
<span class="co">##   chrX: 250</span>
<span class="co">## Call seqlevels(obj) to list all available chromosomes or seqinfo(obj) for more detailed output</span>
<span class="co">## Call chromosome(obj) &lt;- 'chrId' to change the active chromosome</span></code></pre></div>
<p>In this context, the constructor’s <code>chromosome</code>, <code>start</code> and <code>end</code> argument take on a slightly different meaning in that they can be used to subset the data that is fetched from the <code>TxDb</code> object. Please note that while the <code>chromosome</code> alone can be supplied, providing <code>start</code> or <code>end</code> without the chromosome information will not work.</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">txTr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GeneRegionTrack-class.html">GeneRegionTrack</a></span><span class="op">(</span><span class="va">txdb</span>, chromosome <span class="op">=</span> <span class="st">"chr6"</span>, 
                        start <span class="op">=</span> <span class="fl">35000000</span>,  end <span class="op">=</span> <span class="fl">40000000</span><span class="op">)</span></code></pre></div>
<p>A nice bonus when building <code>GeneRegionTracks</code> from <code>TxDb</code> objects is that we get additional information about coding and non-coding regions of the transcripts, i.e., coordinates of the 5’ and 3’ UTRs and of the CDS regions. The class’ plotting method can use this information to distinguish between coding and non-coding regions based on the shape of the feature. All coding regions are plotted just as we have seen in the previous examples, whereas the non-coding regions are drawn as slightly thinner boxes. The distinction between coding and non-coding is made on the basis of the object’s <code>feature</code> values in combination with a special display parameter <code>thinBoxFeature</code> that enumerates all feature types that are to be treated as non-coding. Obviously this feature is available to all <code>GeneRegionTracks</code>, not only the ones that were build from <code>TxDb</code> objects. However, the coding information has to be added manually and the default value of the <code>thinBoxFeature</code> parameter may not be sufficient to cover all possible cases. It is up to the user to come up with a complete list of non-coding feature types depending on the source of the data.</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/RangeTrack-class.html">feature</a></span><span class="op">(</span><span class="va">txTr</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">txTr</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/generegtrack4f-1.png"><!-- --></p>
<!--
The `TxDb` do not provide gene symbols, however there is a possibility to set 
them using the *[AnnotationDbi](https://bioconductor.org/packages/3.13/AnnotationDbi)* framework and corresponding 
`org.eg...db` package. For example here we can use *[org.eg.Hs.db](https://bioconductor.org/packages/3.13/org.eg.Hs.db)*.


-->
</div>
<div id="building-generegiontrack-objects-from-ensdbs" class="section level3">
<h3 class="hasAnchor">
<a href="#building-generegiontrack-objects-from-ensdbs" class="anchor"></a>Building GeneRegionTrack objects from EnsDbs</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> is able to build <code>GeneRegionTrack</code> from <code>EnsDb</code> objects. They are similar to <code>TxDb</code> objects however their content is based on Ensembl Genome Browser. The <em><a href="https://bioconductor.org/packages/3.13/ensembldb">ensembldb</a></em> package provides a complete framework to download gene model information from Ensembl Genome Browser and to store it locally in a SQLite database as well. For commonly used organisms and recent Ensembl versions, those databases are available directly in the <em><a href="https://bioconductor.org/packages/3.13/AnnotationHub">AnnotationHub</a></em>. The <em><a href="https://bioconductor.org/packages/3.13/ensembldb">ensembldb</a></em> package itself provides <code>getGeneRegionTrackForGviz</code> function to build the <code>GeneRegionTrack</code> object from the <code>EnsDb</code> objects.</p>
<p>In recent versions of <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package, the <code>GeneRegionTrack</code>can be also build by calling the constructor function with the <code>EnsDb</code> object as a single argument. We exemplify this using the database provided by <em><a href="https://bioconductor.org/packages/3.13/EnsDb.Hsapiens.v75">EnsDb.Hsapiens.v75</a></em> package.</p>
<p>It is important to mention that naming convention for chromosomes is different between UCSC and Ensembl genome browsers. The <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> uses by default the convention from UCSC and therefore it is important to either: - set the chromosome names of <code>EnsDb</code> to UCSC style by calling <code>seqlevelsStyle(edb) &lt;- "UCSC"</code> - or set <code>Gviz</code> to accept non-UCSC style chromosome names by calling <code><a href="https://rdrr.io/r/base/options.html">options(ucscChromosomeNames = FALSE)</a></code></p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jorainer/ensembldb">ensembldb</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v75</span><span class="op">)</span>

<span class="va">edb</span> <span class="op">&lt;-</span> <span class="va">EnsDb.Hsapiens.v75</span>
<span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-seqlevels.html">seqlevelsStyle</a></span><span class="op">(</span><span class="va">edb</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span>

<span class="va">eTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GeneRegionTrack-class.html">GeneRegionTrack</a></span><span class="op">(</span><span class="va">edb</span>, chromosome <span class="op">=</span> <span class="st">"chr6"</span>, 
                          start <span class="op">=</span> <span class="fl">35000000</span>, end <span class="op">=</span> <span class="fl">40000000</span><span class="op">)</span>

<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">eTrack</span><span class="op">)</span></code></pre></div>
</div>
<div id="display-parameters-for-generegiontrack-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#display-parameters-for-generegiontrack-objects" class="anchor"></a>Display parameters for GeneRegionTrack objects</h3>
<p>For a complete listing of all the available display parameters please see the table below or the man page of the <code>GeneRegionTrack</code> class by typing in <code><a href="../reference/GeneRegionTrack-class.html">?GeneRegionTrack</a></code> on the <code>R</code> command line.</p>
<table class="table">
<colgroup>
<col width="3%">
<col width="96%">
</colgroup>
<thead><tr class="header">
<th align="left">Display Parameter</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">alpha</td>
<td align="left">Numeric scalar between 0 and 1. The opacity of the plotting elements, if supported by the device.</td>
</tr>
<tr class="even">
<td align="left">arrowHeadMaxWidth</td>
<td align="left">Numeric scalar. The maximum width of the arrow head in pixels if <code>shape</code> is <code>arrow</code>.</td>
</tr>
<tr class="odd">
<td align="left">arrowHeadWidth</td>
<td align="left">Numeric scalar. The width of the arrow head in pixels if <code>shape</code> is <code>fixedArrow</code>.</td>
</tr>
<tr class="even">
<td align="left">cex</td>
<td align="left">Numeric scalar. The font expansion factor for item identifiers.</td>
</tr>
<tr class="odd">
<td align="left">cex.group</td>
<td align="left">Numeric scalar. The font expansion factor for the group-level annotation.</td>
</tr>
<tr class="even">
<td align="left">col</td>
<td align="left">Character or integer scalar. The border color for all track items. Defaults to using the same color as in <code>fill</code>, also taking into account different track <code>features</code>.</td>
</tr>
<tr class="odd">
<td align="left">collapseTranscripts</td>
<td align="left">Logical or character scalar. Can be one in <code>gene</code>, <code>longest</code>, <code>shortest</code> or <code>meta</code>. Merge all transcripts of the same gene into one single gene model. In the case of <code>gene</code> (or <code>TRUE</code>), this will only keep the start location of the first exon and the end location of the last exon from all transcripts of the gene. For <code>shortest</code> and <code>longest</code>, only the longest or shortest transcript model is retained. For <code>meta</code>, a meta-transcript containing the union of all exons is formed (essentially identical to the operation <code><a href="https://rdrr.io/pkg/GenomicRanges/man/inter-range-methods.html">reduce(geneModel)</a></code>).</td>
</tr>
<tr class="even">
<td align="left">exonAnnotation</td>
<td align="left">Character scalar. Add annotation information to the individual exon models. This can be a value in <code>symbol</code>, <code>gene</code>, <code>transcript</code>, <code>exon</code> or <code>feature</code>. Defaults to <code>exon</code>. Only works if <code>showExonId</code> is not <code>FALSE</code>.</td>
</tr>
<tr class="odd">
<td align="left">fill</td>
<td align="left">Character or integer scalar. The fill color for untyped items. This is also used to connect grouped items. See <code>grouping</code> for details.</td>
</tr>
<tr class="even">
<td align="left">fontcolor</td>
<td align="left">Character or integer scalar. The font color for item identifiers.</td>
</tr>
<tr class="odd">
<td align="left">fontcolor.group</td>
<td align="left">Character or integer scalar. The font color for the group-level annotation.</td>
</tr>
<tr class="even">
<td align="left">fontface</td>
<td align="left">Integer scalar. The font face for item identifiers.</td>
</tr>
<tr class="odd">
<td align="left">fontface.group</td>
<td align="left">Numeric scalar. The font face for the group-level annotation.</td>
</tr>
<tr class="even">
<td align="left">fontfamily</td>
<td align="left">Character scalar. The font family for item identifiers.</td>
</tr>
<tr class="odd">
<td align="left">fontfamily.group</td>
<td align="left">Character scalar. The font family for the group-level annotation.</td>
</tr>
<tr class="even">
<td align="left">fontsize</td>
<td align="left">Numeric scalar. The font size for item identifiers.</td>
</tr>
<tr class="odd">
<td align="left">fontsize.group</td>
<td align="left">Numeric scalar. The font size for the group-level annotation.</td>
</tr>
<tr class="even">
<td align="left">geneSymbols</td>
<td align="left">Logical scalar. Use human-readable gene symbols or gene IDs for the transcript annotation.</td>
</tr>
<tr class="odd">
<td align="left">lex</td>
<td align="left">Numeric scalar. The line expansion factor for all track items. This is also used to connect grouped items. See <code>grouping</code> for details.</td>
</tr>
<tr class="even">
<td align="left">lineheight</td>
<td align="left">Numeric scalar. The font line height for item identifiers.</td>
</tr>
<tr class="odd">
<td align="left">lty</td>
<td align="left">Character or integer scalar. The line type for all track items. This is also used to connect grouped items. See <code>grouping</code> for details.</td>
</tr>
<tr class="even">
<td align="left">lwd</td>
<td align="left">Integer scalar. The line width for all track items. This is also used to connect grouped items. See <code>grouping</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">min.distance</td>
<td align="left">Numeric scalar. The minimum pixel distance before collapsing range items, only if <code>collapse==TRUE</code>. See <code>collapsing</code> for details. Note that a value larger than 0 may lead to UTR regions being merged to CDS regions, which in most cases is not particularly useful.</td>
</tr>
<tr class="even">
<td align="left">min.width</td>
<td align="left">Numeric scalar. The minimum range width in pixels to display. All ranges are expanded to this size in order to avoid rendering issues. See <code>collapsing</code> for details.</td>
</tr>
<tr class="odd">
<td align="left">rotation</td>
<td align="left">Numeric scalar. The degree of text rotation for item identifiers.</td>
</tr>
<tr class="even">
<td align="left">shape</td>
<td align="left">Character scalar. The shape in which to display the track items. Currently only <code>box</code>, <code>arrow</code>, <code>ellipse</code>, and <code>smallArrow</code> are implemented.</td>
</tr>
<tr class="odd">
<td align="left">showExonId</td>
<td align="left">Logical scalar. Control whether to plot the individual exon identifiers.</td>
</tr>
<tr class="even">
<td align="left">showId</td>
<td align="left">Logical scalar. Control whether to annotate individual groups.</td>
</tr>
<tr class="odd">
<td align="left">showOverplotting</td>
<td align="left">Logical scalar. Use a color gradient to show the amount of overplotting for collapsed items. This implies that <code>collapse==TRUE</code>
</td>
</tr>
<tr class="even">
<td align="left">thinBoxFeature</td>
<td align="left">Character vector. A listing of feature types that should be drawn with thin boxes. Typically those are non-coding elements.</td>
</tr>
<tr class="odd">
<td align="left">transcriptAnnotation</td>
<td align="left">Character scalar. Add annotation information as transcript labels. This can be a value in <code>symbol</code>, <code>gene</code>, <code>transcript</code>, <code>exon</code> or <code>feature</code>. Defaults to <code>symbol</code>. Only works if <code>showId</code> is not <code>FALSE</code>.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="biomartgeneregiontrack" class="section level2">
<h2 class="hasAnchor">
<a href="#biomartgeneregiontrack" class="anchor"></a>BiomartGeneRegionTrack</h2>
<p>As seen before it can be very useful to quickly download gene annotation information from an online repository rather than having to construct it each time from scratch. To this end, the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package also defines the <code>BiomartGeneRegionTrack</code> class, which directly extends <code>GeneRegionTrack</code> but provides a direct interface to the ENSEMBL Biomart service (yet another interface to the UCSC data base content is highlighted in one of the next sections). Rather than providing all the bits and pieces for the full gene model, we just enter a genome, chromosome and a start and end position on this chromosome, and the constructor function <code>BiomartGeneRegionTrack</code> will automatically contact ENSEMBL, fetch the necessary information and build the gene model on the fly. Please note that you will need an internet connection for this to work, and that contacting Biomart can take a significant amount of time depending on usage and network traffic. Hence the results are almost never going to be returned instantaneously.</p>
<p>NOTE: <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> tries hard to automatically set up the necessary connection to the Biomart service to retrieve the relevant data. Sometimes however it can be advantageous to fully control the Biomart connector, for instance when trying to retrieve data from an archived Ensembl version (by default, <em><a href="https://bioconductor.org/packages/3.13/biomaRt">biomaRt</a></em> will always link to the most recent version). To that end, the <code>biomart</code> parameter can be used to supply a <code>biomaRt</code> object. Please note that a data set has to be selected in that object. In the example code below, the older version of human genome assembly (hg19) and matching Biomart version are used.</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomaRt</span><span class="op">)</span>
<span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html">useMart</a></span><span class="op">(</span>host <span class="op">=</span> <span class="st">"grch37.ensembl.org"</span>, 
              biomart <span class="op">=</span> <span class="st">"ENSEMBL_MART_ENSEMBL"</span>, 
              dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span>
<span class="va">biomTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/BiomartGeneRegionTrack-class.html">BiomartGeneRegionTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg19"</span>, chromosome <span class="op">=</span> <span class="va">chr</span>, 
                                    start <span class="op">=</span> <span class="fl">20e6</span>, end <span class="op">=</span> <span class="fl">21e6</span>,
                                    name <span class="op">=</span> <span class="st">"ENSEMBL"</span>, biomart <span class="op">=</span> <span class="va">bm</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">biomTrack</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/BiomartGeneRegionTrackDo-1.png"><!-- --></p>
<p>You may have noticed in the above plot that the track includes <code>feature</code> information which is displayed by the different feature colors and box sizes. This information has been automatically extracted from Biomart and the respective color coding is part of the class’ definition. We can highlight the feature classes even more by using similarly colored bars to connect the grouped elements and by dropping the bounding boxes around the individual exon models.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">biomTrack</span>, col.line <span class="op">=</span> <span class="cn">NULL</span>, col <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/BiomartGeneRegionTrackCol-1.png"><!-- --></p>
<p>For dense stacked plots like this it can sometimes be helpful to control the vertical spacing between the stacking rows. A simple way to do this is available for each object inheriting form the <code>StackedTrack</code> class by means of the <code>stackHeight</code> display parameter. It takes a value between 0 and 1 that controls which fraction of the available vertical space on a stacking line should be used to draw the feature glyphs.</p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">biomTrack</span>, col.line <span class="op">=</span> <span class="cn">NULL</span>, col <span class="op">=</span> <span class="cn">NULL</span>, stackHeight <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/BiomartGeneRegionTrackHeight-1.png"><!-- --></p>
<p>The <code>filter</code> argument to the <code>BiomartGeneRegionTrack</code> constructor can be used to pass on arbitrary additional filters to the Biomart query. We could for instance limit the returned gene models to RefSeq models only. Please see the documentation in the <em><a href="https://bioconductor.org/packages/3.13/biomaRt">biomaRt</a></em> package for details.</p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html">useMart</a></span><span class="op">(</span>host <span class="op">=</span> <span class="st">"grch37.ensembl.org"</span>, 
              biomart <span class="op">=</span> <span class="st">"ENSEMBL_MART_ENSEMBL"</span>, 
              dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span>
<span class="va">biomTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/BiomartGeneRegionTrack-class.html">BiomartGeneRegionTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg19"</span>, chromosome <span class="op">=</span> <span class="va">chr</span>, 
                                    start <span class="op">=</span> <span class="fl">20e6</span>, end <span class="op">=</span> <span class="fl">21e6</span>,
                                    name <span class="op">=</span> <span class="st">"ENSEMBL"</span>, 
                                    filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>with_refseq_mrna<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>, 
                                    biomart <span class="op">=</span> <span class="va">bm</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">biomTrack</span>, col.line <span class="op">=</span> <span class="cn">NULL</span>, col <span class="op">=</span> <span class="cn">NULL</span>, stackHeight <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/BiomartGeneRegionTrackFilterDo-1.png"><!-- --></p>
<p>Sometimes it can be more convenient to get a model for a particular gene rather than defining a genomic range. One could use the filter mechanisms as described before for this purpose, however the <code>BiomartGeneRegionTrack</code> implements a more generic way. By providing the gene symbol as the <code>symbol</code> argument, the track will be initialized around the locus for this gene, if it can be found. Alternatively, one can provide the Ensembl transcript or gene id via the <code>transcript</code> or <code>gene</code> arguments, or an Entrez id via the <code>entrez</code> argument. In all of these cases the constructor will also fetch models for other genes that overlap the same region. If that is not desired one needs to provide an explicit Biomart filter.</p>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html">useMart</a></span><span class="op">(</span>host <span class="op">=</span> <span class="st">"grch37.ensembl.org"</span>, 
              biomart <span class="op">=</span> <span class="st">"ENSEMBL_MART_ENSEMBL"</span>,
              dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span>
<span class="va">biomTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/BiomartGeneRegionTrack-class.html">BiomartGeneRegionTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg19"</span>, name <span class="op">=</span> <span class="st">"ENSEMBL"</span>, 
                                    symbol <span class="op">=</span> <span class="st">"ABCB5"</span>, biomart <span class="op">=</span> <span class="va">bm</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">biomTrack</span>, transcriptAnnotation <span class="op">=</span> <span class="st">"symbol"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/BiomartGeneRegionTrackSymbolDo-1.png"><!-- --></p>
<!--
### Using custom biomaRt objects for BiomartGeneRegionTracks
*[Gviz](https://bioconductor.org/packages/3.13/Gviz)* tries hard to automatically set up the necessary connection to
the Biomart service to retrieve the relevant data. Sometimes however
it can be advantageous to fully control the Biomart connector, for
instance when trying to retrieve data from an archived Ensembl version
(by default, *[biomaRt](https://bioconductor.org/packages/3.13/biomaRt)* will always link to the most recent
version). To that end, the `biomart` parameter can be used to
supply a `biomaRt` object. Please note that a data set has to
be selected in that object. A second prerequisite for this to work is
that *[Gviz](https://bioconductor.org/packages/3.13/Gviz)* is able to guess which of the values that are queried
from the Ensembl data base represent which track feature. If there
were no changes in the schema of Ensembl, or if the schema for all
marts was similar this would not be an issue, but that is not the
case. In the following example we show how to fetch data for the same
region from the older version Ensembl version 47. The internal
`.getBMFeatureMap` is helpful to set up the feature mapping
which is necessary because Ensembl used to call the gene symbol field
`external_gene_id` in that release.


-->
<div id="display-parameters-for-biomartgeneregiontrack-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#display-parameters-for-biomartgeneregiontrack-objects" class="anchor"></a>Display parameters for BiomartGeneRegionTrack objects</h3>
<p>For a complete listing of all the available display parameters please see the table above in the previous <code>GeneRegionTrack</code> section or the man page of the <code>BiomartGeneRegionTrack</code> class by typing in <code><a href="../reference/BiomartGeneRegionTrack-class.html">?BiomartGeneRegionTrack</a></code> on the <code>R</code> command line.</p>
<p>As mentioned above, one additional benefit when fetching the data through Biomart is that we also receive some information about the annotation feature types, which is automatically used for the color coding of the track. The following table shows the available feature types.</p>
<table class="table">
<colgroup>
<col width="15%">
<col width="72%">
<col width="12%">
</colgroup>
<thead><tr class="header">
<th align="left">Display Parameter</th>
<th align="left">Description</th>
<th align="left">Color</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">C_segment</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘C_segment’.</td>
<td align="left">burlywood4</td>
</tr>
<tr class="even">
<td align="left">D_segment</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘C_segment’.</td>
<td align="left">lightblue</td>
</tr>
<tr class="odd">
<td align="left">J_segment</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘C_segment’.</td>
<td align="left">dodgerblue2</td>
</tr>
<tr class="even">
<td align="left">miRNA</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘L_segment’.</td>
<td align="left">cornflowerblue</td>
</tr>
<tr class="odd">
<td align="left">miRNA_pseudogene</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘miRNA_pseudogene’.</td>
<td align="left">cornsilk</td>
</tr>
<tr class="even">
<td align="left">misc_RNA</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘misc_RNA’.</td>
<td align="left">cornsilk3</td>
</tr>
<tr class="odd">
<td align="left">misc_RNA_pseudogene</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘misc_RNA_pseudogene’.</td>
<td align="left">cornsilk4</td>
</tr>
<tr class="even">
<td align="left">Mt_rRNA</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘Mt_rRNA’.</td>
<td align="left">yellow</td>
</tr>
<tr class="odd">
<td align="left">Mt_tRNA</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘Mt_tRNA’.</td>
<td align="left">darkgoldenrod</td>
</tr>
<tr class="even">
<td align="left">Mt_tRNA_pseudogene</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘Mt_tRNA_pseudogene’.</td>
<td align="left">darkgoldenrod1</td>
</tr>
<tr class="odd">
<td align="left">protein_coding</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘protein_coding’.</td>
<td align="left">orange</td>
</tr>
<tr class="even">
<td align="left">pseudogene</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘pseudogene’.</td>
<td align="left">brown1</td>
</tr>
<tr class="odd">
<td align="left">retrotransposed</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘retrotransposed’.</td>
<td align="left">blueviolet</td>
</tr>
<tr class="even">
<td align="left">rRNA</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘rRNA’.</td>
<td align="left">darkolivegreen1</td>
</tr>
<tr class="odd">
<td align="left">rRNA_pseudogene</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘rRNA_pseudogene’.</td>
<td align="left">darkolivegreen</td>
</tr>
<tr class="even">
<td align="left">scRNA</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘scRNA’.</td>
<td align="left">gold4</td>
</tr>
<tr class="odd">
<td align="left">scRNA_pseudogene</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘scRNA_pseudogene’.</td>
<td align="left">darkorange2</td>
</tr>
<tr class="even">
<td align="left">snoRNA</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘snoRNA’.</td>
<td align="left">cyan</td>
</tr>
<tr class="odd">
<td align="left">snoRNA_pseudogene</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘snoRNA_pseudogene’.</td>
<td align="left">cyan2</td>
</tr>
<tr class="even">
<td align="left">snRNA</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘snRNA’.</td>
<td align="left">coral</td>
</tr>
<tr class="odd">
<td align="left">snRNA_pseudogene</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘snRNA_pseudogene’.</td>
<td align="left">coral3</td>
</tr>
<tr class="even">
<td align="left">tRNA_pseudogene</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘tRNA_pseudogene’.</td>
<td align="left">antiquewhite3</td>
</tr>
<tr class="odd">
<td align="left">utr3</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘utr3’.</td>
<td align="left">orange</td>
</tr>
<tr class="even">
<td align="left">utr5</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘utr5’.</td>
<td align="left">orange</td>
</tr>
<tr class="odd">
<td align="left">V_segment</td>
<td align="left">Character or integer scalar. Fill color for annotation objects of type ‘V_segment’.</td>
<td align="left">aquamarine</td>
</tr>
<tr class="even">
<td align="left">verbose</td>
<td align="left">Logical scalar. Report data loading events from Bioamart or retrieval from cache.</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="detailsannotationtrack" class="section level2">
<h2 class="hasAnchor">
<a href="#detailsannotationtrack" class="anchor"></a>DetailsAnnotationTrack</h2>
<p>It is sometimes desirable to add more detailed information to particular ranges in an <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> plot for which the notion of genomic coordinates no longer makes sense. For instance, the ranges in an <code>AnnotationTrack</code> may represent probe locations on a genome, and for each of these probes a number of measurements from multiple samples and from different sample groups are available. To this end, the <code>DetailsAnnotationTrack</code> provides a flexible interface to further annotate genomic regions with arbitrary additional information. This is archived by splitting the <code>AnnotationTrack</code> plotting region into two horizontal sections: the lower section containing the range data in genomic coordinates, and the upper one containing the additional data for each of the displayed ranges in vertically tiled panels of equal size. The connection between a range item and its details panel is indicated by connecting lines.</p>
<p>The content of the individual details panels has to be filled in by a user-defined plotting function that uses grid (or lattice) plotting commands. This function has to accept a number of mandatory parameters, notably the start, end, strand, chromosome and identifier information for the genomic range, as well as an integer counter indicating the index of the currently plotted details tile. This information can be used to fetch arbitrary details, e.g. from a list, and environment or even from a <code>GRanges</code> object which will then be processed and visualized within the plotting function. This may sound rather abstract, and for more details please refer to the class’ help page. For now we just want to demonstrate the functionality in a simple little example. We begin by defining a <code>GRanges</code> object containing 4 genomic locations. In our example those are considered to be probe locations from a methylation array.</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></span><span class="op">)</span>
<span class="va">probes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">"chr7"</span>, ranges <span class="op">=</span> <span class="fu">IRanges</span><span class="op">(</span>
      start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2000000</span>, <span class="fl">2070000</span>, <span class="fl">2100000</span>, <span class="fl">2160000</span><span class="op">)</span>,
      end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2050000</span>, <span class="fl">2130000</span>, <span class="fl">2150000</span>, <span class="fl">2170000</span><span class="op">)</span><span class="op">)</span>, 
  strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"-"</span>, <span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"-"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>For each of these probes we have methylation measurements from a large number of different samples in a numeric matrix, and within the samples there are two treatment groups. The aim is to compare the distribution of measurement values between these two groups at each probe locus.</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">methylation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/GammaDist.html">rgamma</a></span><span class="op">(</span><span class="fl">400</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span>,
                  ncol <span class="op">=</span> <span class="fl">100</span>,
                  dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"probe"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, <span class="cn">NULL</span><span class="op">)</span><span class="op">)</span>
<span class="va">methylation</span><span class="op">[</span>, <span class="fl">51</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">methylation</span><span class="op">[</span>, <span class="fl">51</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span> <span class="op">+</span> <span class="fl">0</span><span class="op">:</span><span class="fl">3</span>
<span class="va">sgroups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grp1"</span>, <span class="st">"grp2"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></code></pre></div>
<p>Of course we could use a <code>DataTrack</code> with the box-plot representation for this task, however we do have strand-specific data here and some of the probes can be overlapping, so all this information would be lost. We are also interested in the particular shape of the data distribution, so a density plot representation is what we really need. Luckily, the <em>lattice</em> package gives us a nice <code>densityplot</code> function that supports grouping, so all that’s left to do now is to write a little wrapper that handles the extraction of the relevant data from the matrix. This is easily achieved by using the range identifiers, which conveniently map to the row names of the data matrix.</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://lattice.r-forge.r-project.org/">lattice</a></span><span class="op">)</span>
<span class="va">details</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">identifier</span>, <span class="va">...</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>signal <span class="op">=</span> <span class="va">methylation</span><span class="op">[</span><span class="va">identifier</span>, <span class="op">]</span>, group <span class="op">=</span> <span class="va">sgroups</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/lattice/man/histogram.html">densityplot</a></span><span class="op">(</span><span class="op">~</span><span class="va">signal</span>, group <span class="op">=</span> <span class="va">group</span>, data <span class="op">=</span> <span class="va">d</span>, 
                    main <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">identifier</span>, cex <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span>,
                    scales <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>draw <span class="op">=</span> <span class="cn">FALSE</span>, x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>draw <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>, 
                    ylab <span class="op">=</span> <span class="st">""</span>, xlab <span class="op">=</span> <span class="st">""</span><span class="op">)</span>, 
        newpage <span class="op">=</span> <span class="cn">FALSE</span>, prefix <span class="op">=</span> <span class="st">"plot"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>Finally, it is as simple as calling the <code>AnnotationTrack</code> constructor, passing along the wrapper function and calling <code>plotTracks</code>.</p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">deTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span>range <span class="op">=</span> <span class="va">probes</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, 
                           chromosome <span class="op">=</span> <span class="fl">7</span>, id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">methylation</span><span class="op">)</span>,
                           name <span class="op">=</span> <span class="st">"probe details"</span>, stacking <span class="op">=</span> <span class="st">"squish"</span>,
                           fun <span class="op">=</span> <span class="va">details</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">deTrack</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/DetailsAnnotationTrack4-1.png"><!-- --></p>
<p>It should be noted here that in our little example we rely on the methylation data matrix and the grouping vector to be present in the working environment. This is not necessarily the cleanest solution and one should consider storing additional data in an environment, passing it along using the <code>detailFunArgs</code> parameter, or making it part of the details function in form of a closure. The class’ help page provides further instructions.</p>
<p>Another use case for the <code>DetailsAnnotationTrack</code> class is to deal with the problem of very different feature sizes within a single track. For instance, we may be looking at a rather large genomic region containing one big transcript with many widely spaced exons and a bunch of smaller, more compact transcripts. In this case it would be helpful to provide a zoomed in version of those smaller transcripts. In order to achieve this we can make use of the class’ <code>groupDetails</code> display parameter, which applies the detail plotting function over each range group rather than over individual range items. First we define a function that selects those groups with a plotted size smaller than 10 pixels. We make use of the unexported function <code>.pxResolution</code> here to come up with the mapping between pixel coordinates and genomic coordinates.</p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">selFun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">identifier</span>, <span class="va">start</span>, <span class="va">end</span>, <span class="va">track</span>, <span class="va">GdObject</span>, <span class="va">...</span><span class="op">)</span><span class="op">{</span>
    <span class="va">gcount</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="../reference/AnnotationTrack-class.html">group</a></span><span class="op">(</span><span class="va">GdObject</span><span class="op">)</span><span class="op">)</span>
    <span class="co">## This computes the width of 2 pixels in genomic coordinates</span>
    <span class="va">pxRange</span> <span class="op">&lt;-</span> <span class="fu">Gviz</span><span class="fu">:::</span><span class="fu">.pxResolution</span><span class="op">(</span>min.width <span class="op">=</span> <span class="fl">20</span>, coord <span class="op">=</span> <span class="st">"x"</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="op">(</span><span class="va">end</span> <span class="op">-</span> <span class="va">start</span><span class="op">)</span> <span class="op">&lt;</span> <span class="va">pxRange</span> <span class="op">&amp;&amp;</span> <span class="va">gcount</span><span class="op">[</span><span class="va">identifier</span><span class="op">]</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>The actual detail plotting function is fairly trivial. In the details viewport we simply call <code>plotTracks</code> function on the subset of items from each group that has been selected before without plotting the track titles, and also add a scale indicator for some reference. It is worth mentioning however that we need to extract the ranges of the zoomed in track items from the unmodified track object (the <code>Gdobject.original</code> argument in the details function) because the object that is plotted in the lower panel contains only the collapsed ranges.</p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">detFun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">identifier</span>, <span class="va">GdObject.original</span>, <span class="va">...</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="../reference/GenomeAxisTrack-class.html">GenomeAxisTrack</a></span><span class="op">(</span>scale <span class="op">=</span> <span class="fl">0.3</span>, size <span class="op">=</span> <span class="fl">0.2</span>, cex <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span>, 
        <span class="va">GdObject.original</span><span class="op">[</span><span class="fu"><a href="../reference/AnnotationTrack-class.html">group</a></span><span class="op">(</span><span class="va">GdObject.original</span><span class="op">)</span> <span class="op">==</span> <span class="va">identifier</span><span class="op">]</span><span class="op">)</span>,
        add <span class="op">=</span> <span class="cn">TRUE</span>, showTitle <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>Finally, we load some sample data, turn it into a <code>DetailsAnnotationTrack</code> object and plot it.</p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneDetails</span><span class="op">)</span>
<span class="va">deTrack2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span><span class="va">geneDetails</span>, fun <span class="op">=</span> <span class="va">detFun</span>, 
                            selectFun <span class="op">=</span> <span class="va">selFun</span>,
                            groupDetails <span class="op">=</span> <span class="cn">TRUE</span>, details.size <span class="op">=</span> <span class="fl">0.5</span>, 
                            detailsConnector.cex <span class="op">=</span> <span class="fl">0.5</span>, 
                            detailsConnector.lty <span class="op">=</span> <span class="st">"dotted"</span>,
                            shape <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"smallArrow"</span>, <span class="st">"arrow"</span><span class="op">)</span>, 
                            groupAnnotation <span class="op">=</span> <span class="st">"group"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">deTrack2</span>, extend.left <span class="op">=</span> <span class="fl">90000</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/DetailsAnnotationTrack7f-1.png"><!-- --></p>
<div id="display-parameters-for-detailsannotationtrack-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#display-parameters-for-detailsannotationtrack-objects" class="anchor"></a>Display parameters for DetailsAnnotationTrack objects</h3>
<p>In addition to the display parameters for the <code>AnnotationTrack</code> class, some additional parameters can be used to control the look and feel of the details sections. For a complete listing of all the available display parameters please see the tables below and the one above in the <code>AnnotationTrack</code> section or the man page of the <code>DetailsAnnotationTrack</code> class by typing in <code><a href="../reference/AnnotationTrack-class.html">?DetailsAnnotationTrack</a></code> on the <code>R</code> command line.</p>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">deTrack</span>, details.size <span class="op">=</span> <span class="fl">0.75</span>, detailsConnector.pch <span class="op">=</span> <span class="cn">NA</span>, 
           detailsConnector.col <span class="op">=</span> <span class="st">"darkred"</span>, 
           detailsBorder.fill <span class="op">=</span> <span class="st">"#FFE3BF"</span>, 
           detailsBorder.col <span class="op">=</span> <span class="st">"darkred"</span>, shape <span class="op">=</span><span class="st">"box"</span>, 
           detailsConnector.lty <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/DetailsAnnotationTrack8-1.png"><!-- --></p>
<table class="table">
<colgroup>
<col width="3%">
<col width="96%">
</colgroup>
<thead><tr class="header">
<th align="left">Display Parameter</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">details.minWidth</td>
<td align="left">Numeric scalar. The minium width in pixels for a details panel, if less space is available no details are plotted.</td>
</tr>
<tr class="even">
<td align="left">details.ratio</td>
<td align="left">Numeric scalar. By default, the plotting method tries to fill all available space of the details panel tiles. Depending on the dimensions of your plot and the number of tiles this may lead to fairly stretched plots. Restricting the ration of width over height can help to fine tune for somewhat more sane graphics in these cases. Essentially this adds some white space in between individual tiles to force the desired ratio. Together with the <code>size</code> and <code>details.size</code> arguments, which control the vertical extension of the whole track and of the details section, this allows for some fairly generic resizing of the tiles.</td>
</tr>
<tr class="odd">
<td align="left">details.size</td>
<td align="left">Numeric scalar. The fraction of vertical space of the track used for the details section.</td>
</tr>
<tr class="even">
<td align="left">detailsBorder.col</td>
<td align="left">Character or integer scalar. Line color of the border.</td>
</tr>
<tr class="odd">
<td align="left">detailsBorder.fill</td>
<td align="left">Character or integer scalar. Background color of the border.</td>
</tr>
<tr class="even">
<td align="left">detailsBorder.lty</td>
<td align="left">Character or integer scalar. Line type of the border around each details panel.</td>
</tr>
<tr class="odd">
<td align="left">detailsBorder.lwd</td>
<td align="left">Integer scalar. Line width of the border.</td>
</tr>
<tr class="even">
<td align="left">detailsConnector.cex</td>
<td align="left">Numeric scalar. Relative size of the connector’s end points.</td>
</tr>
<tr class="odd">
<td align="left">detailsConnector.col</td>
<td align="left">Character or integer scalar. Color of the line connecting the <code>AnnotstionTrack</code> item with its details panel.</td>
</tr>
<tr class="even">
<td align="left">detailsConnector.lty</td>
<td align="left">Character or integer scalar. Type of connecting line.</td>
</tr>
<tr class="odd">
<td align="left">detailsConnector.lwd</td>
<td align="left">Integer scalar. Line width of the connector.</td>
</tr>
<tr class="even">
<td align="left">detailsConnector.pch</td>
<td align="left">Integer scalar. Type of the connector’s ends.</td>
</tr>
<tr class="odd">
<td align="left">detailsFunArgs</td>
<td align="left">List.Additional arguments that get passed on the the details plotting function.</td>
</tr>
<tr class="even">
<td align="left">groupDetails</td>
<td align="left">Logial scalar. Plot details for feature groups rather than for individual features.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="sequencetrack" class="section level2">
<h2 class="hasAnchor">
<a href="#sequencetrack" class="anchor"></a>SequenceTrack</h2>
<p>So far we have displayed all kinds of ranges on a genome, but never really looked at the underlying genomic sequence. To that end, the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package defines the <code>SequenceTrack</code> class which can hold genomic sequence information derived from either a <code>DNAStringSet</code> object of, even more convenient, from one of the <code>BSgenome</code> packages. There is not formal checking of the track’s genome, so it is up to the user to provide reasonable inputs to the class’ constructor. As with all the other track types, the constructor has the same name as the class. In this example we build a <code>SequenceTrack</code> from the human hg19 UCSC genome.</p>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg19</span><span class="op">)</span>
<span class="va">sTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SequenceTrack-class.html">SequenceTrack</a></span><span class="op">(</span><span class="va">Hsapiens</span><span class="op">)</span>
<span class="va">sTrack</span>
<span class="co">## Sequence track 'SequenceTrack':</span>
<span class="co">## | genome: hg19</span>
<span class="co">## | chromosomes: 298</span>
<span class="co">## | active chromosome: chr1 (249250621 nucleotides)</span>
<span class="co">## Call seqnames() to list all available chromosomes</span>
<span class="co">## Call chromosome()&lt;- to change the active chromosome</span>
<span class="co">## Parent BSgenome object:</span>
<span class="co">## | organism: Homo sapiens</span>
<span class="co">## | provider: UCSC</span>
<span class="co">## | provider version: hg19</span>
<span class="co">## | release date: June 2013</span>
<span class="co">## | package name: BSgenome.Hsapiens.UCSC.hg19</span></code></pre></div>
<p>Here we retain the same benefits as for regular <code>BSgenome</code> objects in that the actual sequence is only loaded into memory when a particular chromosome is first accessed. Plotting the track yields the expected result: a character representation of the genomic sequence in the current plot window. It may be worth noting that the color scheme that is used to encode the individual nucleotides is defined in the <em><a href="https://bioconductor.org/packages/3.13/biovizBase">biovizBase</a></em> package and the user is referred to its documentation for more details.</p>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">sTrack</span>, chromosome <span class="op">=</span> <span class="fl">1</span>, from <span class="op">=</span> <span class="fl">20000</span>, to <span class="op">=</span> <span class="fl">20050</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/SequenceTrack2-1.png"><!-- --></p>
<p>Sometimes it may be too busy to plot both the sequence letters and the color coding, and we can assign arbitrary colors by setting the <code>fontcolor</code> display parameter. The convention here is that we need to supply a named vector, with one entry for each of the five possible nucleotides (A, C, T, G, and N). If any of the entries is missing, the respective letter will not be drawn at all.</p>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fcol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"darkgray"</span>, C<span class="op">=</span><span class="st">"darkgray"</span>, T<span class="op">=</span><span class="st">"darkgray"</span>, G<span class="op">=</span><span class="st">"darkgray"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">sTrack</span>, chromosome <span class="op">=</span> <span class="fl">1</span>, from <span class="op">=</span> <span class="fl">20000</span>, to <span class="op">=</span> <span class="fl">20050</span>, 
           fontcolor <span class="op">=</span> <span class="va">fcol</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/SequenceTrack3-1.png"><!-- --></p>
<p>In this case we are plotting the sequence on the forward strand and we may want to make this clear by adding direction indicators.</p>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">sTrack</span>, chromosome <span class="op">=</span> <span class="fl">1</span>, from <span class="op">=</span> <span class="fl">20000</span>, to <span class="op">=</span> <span class="fl">20050</span>,
           add53 <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/SequenceTrack4-1.png"><!-- --></p>
<p>If instead we plot the complement sequence on the reverse strand, the indicators are automatically adjusted.</p>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">sTrack</span>, chromosome <span class="op">=</span> <span class="fl">1</span>, from <span class="op">=</span> <span class="fl">20000</span>, to <span class="op">=</span> <span class="fl">20050</span>, 
           add53 <span class="op">=</span> <span class="cn">TRUE</span>, complement <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/SequenceTrack5-1.png"><!-- --></p>
<p>So far we have been able to fit the sequence onto our plotting device without overplotting. I.e., all the letters nicely fit in to their respective position in the genomic coordinate system. However this restricts us to very small windows which we can reasonably display. A much more compact version of the same information is to use colored boxes rather than individual letters. Those boxes can be stacked much closer together, which increases the possible window size quite a bit. The user does not really need to worry about this as the plotting method will automatically make a reasonable decision based on the available space.</p>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">sTrack</span>, chromosome <span class="op">=</span> <span class="fl">1</span>, from <span class="op">=</span> <span class="fl">20000</span>, to <span class="op">=</span> <span class="fl">20100</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/SequenceTrack6-1.png"><!-- --></p>
<p>For added flexibility one can set the <code>noLetters</code> display parameter to <code>TRUE</code> to always force the drawing of boxes. Of course also the colored box representation of a sequence has its limits, and if we cross the threshold when individual boxes can not be separated anymore, the plotting method falls back to drawing a single line indicating the presence of a sequence at the given position.</p>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">sTrack</span>, chromosome <span class="op">=</span> <span class="fl">1</span>, from <span class="op">=</span> <span class="fl">20000</span>, to <span class="op">=</span> <span class="fl">201000</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/SequenceTrack7-1.png"><!-- --></p>
<p>Finally, the selected font size is also contributing to the available space, and we can cram a little more sequence into a given window by decreasing it.</p>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">sTrack</span>, chromosome <span class="op">=</span> <span class="fl">1</span>, from <span class="op">=</span> <span class="fl">20000</span>, to <span class="op">=</span> <span class="fl">20100</span>, cex <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/SequenceTrack8-1.png"><!-- --></p>
<div id="display-parameters-for-sequencetrack-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#display-parameters-for-sequencetrack-objects" class="anchor"></a>Display parameters for SequenceTrack objects</h3>
<p>For a complete listing of all the available display parameters please see the table below or the man page of the <code>SequenceTrack</code> class by typing in <code><a href="../reference/SequenceTrack-class.html">?SequenceTrack</a></code> on the <code>R</code> command line.</p>
<table class="table">
<colgroup>
<col width="5%">
<col width="94%">
</colgroup>
<thead><tr class="header">
<th align="left">Display Parameter</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">add53</td>
<td align="left">Logical scalar. Add a direction indicator.</td>
</tr>
<tr class="even">
<td align="left">cex</td>
<td align="left">The character expansion factor for the size of the sequence letters. Together with <code>fontsize</code> this determines the final font size and thus the level of plotable details.</td>
</tr>
<tr class="odd">
<td align="left">col</td>
<td align="left">Character scalar. The color of the line when no indiviual letters can be plotted due to size limitations.</td>
</tr>
<tr class="even">
<td align="left">col.border.title</td>
<td align="left">Integer or character scalar. The border color for the title panels.</td>
</tr>
<tr class="odd">
<td align="left">complement</td>
<td align="left">Logical scalar. Plot the sequence complement.</td>
</tr>
<tr class="even">
<td align="left">fontcolor</td>
<td align="left">Character vector. The colors used for the 5 possible nucleotides (G, A, T, C, N). Defaults to use colors as defined in the <code>biovizBase</code> package.</td>
</tr>
<tr class="odd">
<td align="left">fontface</td>
<td align="left">Numeric scalar. The face of the font.</td>
</tr>
<tr class="even">
<td align="left">fontsize</td>
<td align="left">Numeric scalar. Controls the size of the sequence letters and thus also the level of plotable details.</td>
</tr>
<tr class="odd">
<td align="left">lwd</td>
<td align="left">Numeric scalar. The width of the line when no indiviual letters can be plotted due to size limitations.</td>
</tr>
<tr class="even">
<td align="left">lwd.border.title</td>
<td align="left">Integer scalar. The border width for the title panels.</td>
</tr>
<tr class="odd">
<td align="left">min.width</td>
<td align="left">Numeric scalar. The minimum width in pixels of the colored boxes that are drawn when no indiviual letters can be plotted due to size limitations. If the horizontal space that a single base occupies is smaller than this value, only a horizontal line is drawn to indicate the presence of a sequence.</td>
</tr>
<tr class="even">
<td align="left">noLetters</td>
<td align="left">Logical scalar. Always plot colored boxes (or a line) regardles of the available space.</td>
</tr>
<tr class="odd">
<td align="left">rotation</td>
<td align="left">Numeric scalar. The rotation angle for each individual letter in the sequence.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="alignmentstrack" class="section level2">
<h2 class="hasAnchor">
<a href="#alignmentstrack" class="anchor"></a>AlignmentsTrack</h2>
<p>Plots of aligned sequences, typically from next generation sequencing experiments can be quite helpful, for instance when visually inspecting the validity of a called SNP. Those alignments are usually stored in <code>BAM</code> files. As shown in some of the previous section we can deal with these files in a fairly crude way by coupling them to either an <code>AnnotationTrack</code> or a <code>DataTrack</code>. The utility of the generated plots however is fairly limited, and we want in particular support for gapped and paired alignments, and show differences of the individual read sequences to a reference.</p>
<p>The <code>AlignmentsTrack</code> class is designed to help in exactly these tasks. In its typical manifestation it is also a member of the <code>ReferenceTrack</code> class, enabling memory efficient streaming directly off a file. Even though this file does not necessarily have to be a <code>BAM</code> file, the automatic import will only work for those. For all other file representations of alignments a user will have to provide their own import function, and this topic is further discussed in the class’ documentation. For this demonstration let’s use a small <code>BAM</code> file for which paired NGS reads have been mapped to an extract of the human hg19 genome. The data originate from an RNASeq experiment, and the alignments have been performed using the <code>STAR</code> aligner allowing for gaps. We also download some gene annotation data for that region from Biomart.</p>
<p>Again, creating the <code>AlignmentsTrack</code> object is as simple as calling the constructor function of the same name. We set the <code>isPaired</code> argument to <code>TRUE</code> in order to tell the constructor that it should expect paired reads. Please note that <code>TRUE</code> is the default value for this parameter and usually things will still work even if the reads do not originate from a paired end library. In other words, the default setting could be used as a poor-man’s auto-detection, however it make more sense to supply the argument explicitly if you know what type of data to expect.</p>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">afrom</span> <span class="op">&lt;-</span> <span class="fl">2960000</span>
<span class="va">ato</span> <span class="op">&lt;-</span> <span class="fl">3160000</span>
<span class="va">alTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AlignmentsTrack-class.html">AlignmentsTrack</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"Gviz"</span>, <span class="st">"extdata"</span>, <span class="st">"gapped.bam"</span><span class="op">)</span>,
  isPaired <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">bmt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/BiomartGeneRegionTrack-class.html">BiomartGeneRegionTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg19"</span>, chromosome <span class="op">=</span> <span class="st">"chr12"</span>,
                              start <span class="op">=</span> <span class="va">afrom</span>, end <span class="op">=</span> <span class="va">ato</span>,
                              filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>with_refseq_mrna <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                              stacking <span class="op">=</span> <span class="st">"dense"</span><span class="op">)</span></code></pre></div>
<p>We can create a first plot of our read alignments data from a birds eye perspective, say a range of 20,000 nucleotides.</p>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">bmt</span>, <span class="va">alTrack</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">afrom</span>, to <span class="op">=</span> <span class="va">ato</span>, 
           chromosome <span class="op">=</span> <span class="st">"chr12"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_2-1.png"><!-- --></p>
<p>Now this already shows us the general layout of the track: on top we have a panel with the read coverage information in the form of a histogram, and below that a pile-up view of the individual reads. There is only a certain amount of vertical space available for the plotting, and not the whole depth of the pile-up can be displayed here. This fact is indicated by the white downward-pointing arrows in the title panel. We could address this issue by playing around with the <code>max.height</code>, <code>min.height</code> or <code>stackHeight</code> display parameters which all control the height or the vertical spacing of the stacked reads. Or we could reduce the size of the coverage section by setting the <code>coverageHeight</code> or the <code>minCoverageHeight</code> parameters. Please see the class documentation for more details.</p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">bmt</span>, <span class="va">alTrack</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">afrom</span>, to <span class="op">=</span> <span class="va">ato</span>,
           chromosome <span class="op">=</span> <span class="st">"chr12"</span>, min.height <span class="op">=</span> <span class="fl">0</span>, 
           coverageHeight <span class="op">=</span> <span class="fl">0.08</span>, minCoverageHeight <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_3-1.png"><!-- --></p>
<p>From that far out the pile-ups are not particularly useful, and we can turn those off by setting the <code>type</code> display parameter accordingly.</p>
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">alTrack</span>, <span class="va">bmt</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">afrom</span>, to <span class="op">=</span> <span class="va">ato</span>, 
           chromosome <span class="op">=</span> <span class="st">"chr12"</span>, type <span class="op">=</span> <span class="st">"coverage"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_4-1.png"><!-- --></p>
<p>Let’s zoom in a bit further to check out the details of the pile-ups section.</p>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">bmt</span>, <span class="va">alTrack</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">12700</span>, 
           to <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">15200</span>, chromosome <span class="op">=</span> <span class="st">"chr12"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_5-1.png"><!-- --></p>
<p>The direction of the individual reads is indicated by the arrow head, and read pairs are connected by a bright gray line. Gaps in the alignments are show by the connecting dark gray lines. Numbers of reads supporting each gap can be visualized as “sashimi plot” <span class="citation">(Katz et al. 2015)</span> by adding <code>'sashimi'</code> as an additional argument to the <code>type</code> display parameter.</p>
<p>The axis in the title panel shows the relationship between the line width the the number of reads spanning a given junction. The height of each arc is not informative, it is optimized to fit all overlapping junctions in a small window.</p>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">bmt</span>, <span class="va">alTrack</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">12700</span>, 
           to <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">15200</span>, chromosome <span class="op">=</span> <span class="st">"chr12"</span>,
           type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"coverage"</span>, <span class="st">"sashimi"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_5_1-1.png"><!-- --></p>
<p>It is also possible to filter the junctions by including the coordinates as an additional argument <code>sashimiFilter</code>. In this case only junction which equally overlaps with specified <code>GRanges</code> object are shown.</p>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">introns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span><span class="st">"chr12"</span>, <span class="fu">IRanges</span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2973662</span>, <span class="fl">2973919</span><span class="op">)</span>,
                                    end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2973848</span>, <span class="fl">2974520</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">bmt</span>, <span class="va">alTrack</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">12700</span>, to <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">15200</span>, 
           chromosome <span class="op">=</span> <span class="st">"chr12"</span>, type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"coverage"</span>, <span class="st">"sashimi"</span><span class="op">)</span>, 
           sashimiFilter <span class="op">=</span> <span class="va">introns</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_5_2-1.png"><!-- --></p>
<p>However sometimes the aligners do not place the junction reads precisely on the exon-intron boundaries and it is therefore useful to include also junctions which are shifted by few bases. This can be done by setting the additional argument <code>sashimiFilterTolerance</code> to small positive integer value.</p>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">bmt</span>, <span class="va">alTrack</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">12700</span>, to <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">15200</span>, 
           chromosome <span class="op">=</span> <span class="st">"chr12"</span>, type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"coverage"</span>, <span class="st">"sashimi"</span><span class="op">)</span>, 
           sashimiFilter <span class="op">=</span> <span class="va">introns</span>, sashimiFilterTolerance <span class="op">=</span> <span class="fl">5L</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_5_3-1.png"><!-- --></p>
<p>On devices that support transparency we can also see that some of the read pairs are actually overlapping. Like before, we can not display the full depth of the pile-up, and the clipping that takes place hides a lot of the reads from the view. Another way of getting to see at least some of them is by reversing the stacking order. We can also play around with some of the other display parameters that control the look and feel of the reads.</p>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">bmt</span>, <span class="va">alTrack</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">12700</span>, to <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">15200</span>,
           chromosome <span class="op">=</span> <span class="st">"chr12"</span>, reverseStacking <span class="op">=</span> <span class="cn">TRUE</span>,
           col.mates <span class="op">=</span> <span class="st">"purple"</span>, col.gap <span class="op">=</span> <span class="st">"orange"</span>, type <span class="op">=</span> <span class="st">"pileup"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_6-1.png"><!-- --></p>
<p>As mentioned before we can control whether the data should be treated as paired end or single end data by setting the <code>isPaired</code> argument in the constructor. Here is how we could take a look at the data in the same file, but in single end mode.</p>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AlignmentsTrack-class.html">AlignmentsTrack</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"Gviz"</span>, <span class="st">"extdata"</span>, <span class="st">"gapped.bam"</span><span class="op">)</span>, 
  isPaired <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">bmt</span>, <span class="va">alTrack</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">12700</span>, to <span class="op">=</span> <span class="va">afrom</span> <span class="op">+</span> <span class="fl">15200</span>, 
           chromosome <span class="op">=</span> <span class="st">"chr12"</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_6_1-1.png"><!-- --></p>
<p>So far we did not look at the individual read sequences at all. Even though they are contained in the <code>BAM</code> file, their value is fairly limited without the reference sequence against which the reads have initially been aligned. Typically the mismatch positions and the variant bases are not included. (Please note that this is not entirely true for all <code>BAM</code> files. One can store base mismatch positions in the CIGAR string, or even the full variant description in the non-standard <code>MD</code> field. However very few aligners currently make use of these features. We may implement support for those later if found necessary.)</p>
<p>To better show the features of the <code>AlignmentsTrack</code> for sequence variants we will load a different data set, this time from a whole genome DNASeq SNP calling experiment. Again the reference genome is hg19 and the alignments have been performed using <code>Bowtie2</code>.</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">afrom</span> <span class="op">&lt;-</span> <span class="fl">44945200</span>
<span class="va">ato</span> <span class="op">&lt;-</span> <span class="fl">44947200</span>
<span class="va">alTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AlignmentsTrack-class.html">AlignmentsTrack</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"Gviz"</span>, <span class="st">"extdata"</span>, <span class="st">"snps.bam"</span><span class="op">)</span>, isPaired <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="va">alTrack</span>, chromosome <span class="op">=</span> <span class="st">"chr21"</span>, from <span class="op">=</span> <span class="va">afrom</span>, to <span class="op">=</span> <span class="va">ato</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_7-1.png"><!-- --></p>
<p>Somehow we need to tell the <code>AlignmentsTrack</code> about the reference genome, and there are two ways to archive this. Either we provide a <code>SequenceTrack</code> object with the reference to the constructor function via the <code>referenceSequence</code> argument, or we stick it into the track list that is provided to the <code>plotTrack</code> which is smart enough to detect it there. For this example we chose the second option and re-use the <code>SequenceTrack</code> object from the previous section.</p>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">alTrack</span>, <span class="va">sTrack</span><span class="op">)</span>, chromosome <span class="op">=</span> <span class="st">"chr21"</span>, 
           from <span class="op">=</span> <span class="va">afrom</span>, to <span class="op">=</span> <span class="va">ato</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_8-1.png"><!-- --></p>
<p>The mismatched bases are now indicated on both the individual reads in the pileup section and also in the coverage plot in the form of a stacked histogram. When zooming in to one of the obvious heterozygous SNP positions we can reveal even more details.</p>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">alTrack</span>, <span class="va">sTrack</span><span class="op">)</span>, chromosome <span class="op">=</span> <span class="st">"chr21"</span>,
           from <span class="op">=</span> <span class="fl">44946590</span>, to <span class="op">=</span> <span class="fl">44946660</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_9-1.png"><!-- --></p>
<p>Just like shown before in the <code>SequenceTrack</code> we can even show individual letters here as long as there is enough space to fit them. This can be controlled using the display parameters <code>min.height</code>, <code>max.height</code> and <code>cex</code>.</p>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">alTrack</span>, <span class="va">sTrack</span><span class="op">)</span>, chromosome <span class="op">=</span> <span class="st">"chr21"</span>, from <span class="op">=</span> <span class="fl">44946590</span>, 
           to <span class="op">=</span> <span class="fl">44946660</span>, cex <span class="op">=</span> <span class="fl">0.5</span>, min.height <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_10-1.png"><!-- --></p>
<p>It is also possible to visualize the “indels” present in the alignments. This can be achieved by setting the <code>showIndels</code> parameter to <code>TRUE</code>. The coverage plot will no longer count the part of the reads corresponding to deletions. The appearance of insertions and deletions in pile-up track can be defined by setting the color, line width and line type (<code>col.deletion</code>, <code>col.insertion</code>, <code>lty.deletion</code>, <code>lty.insertion</code>, <code>lwd.deletion</code>, <code>lwd.insertion</code>). By default the insertions are indicated with violet vertical bars and the deletions are shown as black connecting lines.</p>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">indelTrack1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AlignmentsTrack-class.html">AlignmentsTrack</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"Gviz"</span>, <span class="st">"extdata"</span>, <span class="st">"indels.bam"</span><span class="op">)</span>, 
  name <span class="op">=</span> <span class="st">"Standard"</span><span class="op">)</span>
<span class="va">indelTrack2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AlignmentsTrack-class.html">AlignmentsTrack</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"Gviz"</span>, <span class="st">"extdata"</span>, <span class="st">"indels.bam"</span><span class="op">)</span>, 
  showIndels<span class="op">=</span><span class="cn">TRUE</span>, name<span class="op">=</span><span class="st">"Indels"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">indelTrack1</span>, <span class="va">indelTrack2</span><span class="op">)</span>, 
           chromosome <span class="op">=</span> <span class="st">"chr2"</span>, from <span class="op">=</span> <span class="fl">126442000</span>, to <span class="op">=</span> <span class="fl">126453000</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/alignmentstrack_11-1.png"><!-- --></p>
<div id="display-parameters-for-alignmentstrack-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#display-parameters-for-alignmentstrack-objects" class="anchor"></a>Display parameters for AlignmentsTrack objects</h3>
<p>For a complete listing of all the available display parameters please see the table below or the man page of the <code>AlignmentsTrack</code> class by typing in <code><a href="../reference/AlignmentsTrack-class.html">?AlignmentsTrack</a></code> on the <code>R</code> command line.</p>
<table class="table">
<colgroup>
<col width="7%">
<col width="92%">
</colgroup>
<thead><tr class="header">
<th align="left">Display Parameter</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">alpha.mismatch</td>
<td align="left">Numeric scalar between 0 and 1. The transparency of the mismatch base information.</td>
</tr>
<tr class="even">
<td align="left">alpha.reads</td>
<td align="left">Numeric scalar between 0 and 1. The transparency of the individual read icons. Can be used to indicate overlapping regions in read pairs. Only on supported devices.</td>
</tr>
<tr class="odd">
<td align="left">cex</td>
<td align="left">Numeric Scalar. The global character expansion factor.</td>
</tr>
<tr class="even">
<td align="left">cex.mismatch</td>
<td align="left">Numeric Scalar. The character expansion factor for the mismatch base letters.</td>
</tr>
<tr class="odd">
<td align="left">col</td>
<td align="left">Integer or character scalar. The default color of all line elements.</td>
</tr>
<tr class="even">
<td align="left">col.coverage</td>
<td align="left">Integer or character scalar. The line color for the coverage profile.</td>
</tr>
<tr class="odd">
<td align="left">col.deletion</td>
<td align="left">Integer or character scalar. The color of the line that is bridging the deleted regions in alignments.</td>
</tr>
<tr class="even">
<td align="left">col.gap</td>
<td align="left">Integer or character scalar. The color of the line that is bridging the gap regions in gapped alignments.</td>
</tr>
<tr class="odd">
<td align="left">col.insertion</td>
<td align="left">Integer or character scalar. The color of the line that highlighting insertions in alignments.</td>
</tr>
<tr class="even">
<td align="left">col.mates</td>
<td align="left">Integer or character scalar. The color of the line that is connecting two paired reads.</td>
</tr>
<tr class="odd">
<td align="left">col.mismatch</td>
<td align="left">Integer or character scalar. The box color around mismatch bases.</td>
</tr>
<tr class="even">
<td align="left">col.reads</td>
<td align="left">Integer or character scalar. The box color around reads.</td>
</tr>
<tr class="odd">
<td align="left">col.sashimi</td>
<td align="left">Integer or character scalar. The line color for sashimi plots.</td>
</tr>
<tr class="even">
<td align="left">collapse</td>
<td align="left">Logical scalar. Do not perform any collapsing of overlapping elements. Currently not supported.</td>
</tr>
<tr class="odd">
<td align="left">coverageHeight</td>
<td align="left">Numeric scalar. The height of the coverage region of the track. Can either be a value between 0 and 1 in which case it is taken as a relative height, or a positive value greater 1 in which case it is interpreted as pixels.</td>
</tr>
<tr class="even">
<td align="left">fill</td>
<td align="left">Integer or character scalar. The default fill color of all plot elements.</td>
</tr>
<tr class="odd">
<td align="left">fill.coverage</td>
<td align="left">Integer or character scalar. The fill color for the coverage profile.</td>
</tr>
<tr class="even">
<td align="left">fill.reads</td>
<td align="left">Integer or character scalar. The fill color for the read icons.</td>
</tr>
<tr class="odd">
<td align="left">fontface.mismatch</td>
<td align="left">Integer scalar. The font face for mismatch bases.</td>
</tr>
<tr class="even">
<td align="left">lty</td>
<td align="left">Integer or character scalar. The default type of all line elements.</td>
</tr>
<tr class="odd">
<td align="left">lty.coverage</td>
<td align="left">Integer or character scalar. The line type of the coverage profile.</td>
</tr>
<tr class="even">
<td align="left">lty.deletion</td>
<td align="left">Integer or character scalar. The type of the line that is bridging the deleted regions in alignments.</td>
</tr>
<tr class="odd">
<td align="left">lty.gap</td>
<td align="left">Integer or character scalar. The type of the line that is bridging the gap regions in gapped alignments.</td>
</tr>
<tr class="even">
<td align="left">lty.insertion</td>
<td align="left">Integer or character scalar. The type of the line that highlighting insertions in alignments.</td>
</tr>
<tr class="odd">
<td align="left">lty.mates</td>
<td align="left">Integer or character scalar. The type of the line that is connecting two paired reads.</td>
</tr>
<tr class="even">
<td align="left">lty.mismatch</td>
<td align="left">Integer or character scalar. The box line type around mismatch bases.</td>
</tr>
<tr class="odd">
<td align="left">lty.reads</td>
<td align="left">Integer or character scalar. The box line type around mismatch reads.</td>
</tr>
<tr class="even">
<td align="left">lwd</td>
<td align="left">Integer scalar. The default width of all line elements.</td>
</tr>
<tr class="odd">
<td align="left">lwd.coverage</td>
<td align="left">Integer or character scalar. The line width of the coverage profile.</td>
</tr>
<tr class="even">
<td align="left">lwd.deletion</td>
<td align="left">Integer scalar. The width of the line that is bridging the deleted regions in alignments.</td>
</tr>
<tr class="odd">
<td align="left">lwd.gap</td>
<td align="left">Integer scalar. The width of the line that is bridging the gap regions in gapped alignments.</td>
</tr>
<tr class="even">
<td align="left">lwd.insertion</td>
<td align="left">Integer scalar. The width of the line that highlighting insertions in alignments.</td>
</tr>
<tr class="odd">
<td align="left">lwd.mates</td>
<td align="left">Integer scalar. The width of the line that is connecting two paired reads.</td>
</tr>
<tr class="even">
<td align="left">lwd.mismatch</td>
<td align="left">Integer scalar. The box line width around mismatch bases.</td>
</tr>
<tr class="odd">
<td align="left">lwd.reads</td>
<td align="left">Integer scalar. The box line width around reads.</td>
</tr>
<tr class="even">
<td align="left">lwd.sashimiMax</td>
<td align="left">Integer scalar. The maximal width of the line in sashimi plots.</td>
</tr>
<tr class="odd">
<td align="left">max.height</td>
<td align="left">Integer scalar. The maximum height of an individual read in pixels. Can be used in combination with <code>min.height</code> to control the read and stacking appearance.</td>
</tr>
<tr class="even">
<td align="left">min.height</td>
<td align="left">Integer scalar. The minimum height of an individual read in pixels. Can be used in combination with <code>max.height</code> to control the read and stacking appearance.</td>
</tr>
<tr class="odd">
<td align="left">minCoverageHeight</td>
<td align="left">Integer scalar. The minimum height of the coverage section. Uselful in combination with a relative setting of <code>coverageHeight</code>.</td>
</tr>
<tr class="even">
<td align="left">minSashimiHeight</td>
<td align="left">Integer scalar. The minimum height of the sashimi section. Uselful in combination with a relative setting of <code>sashimiHeight</code>.</td>
</tr>
<tr class="odd">
<td align="left">noLetters</td>
<td align="left">Logical scalar. Always plot colored boxes for mismatch bases regardles of the available space.</td>
</tr>
<tr class="even">
<td align="left">sashimiFilter</td>
<td align="left">GRanges object. Only junctions which overlap equally with <code>sashimiFilter</code> GRanges are shown. Default <code>NULL</code>, no filtering.</td>
</tr>
<tr class="odd">
<td align="left">sashimiFilterTolerance</td>
<td align="left">Integer scalar. Only used in combination with <code>sashimiFilter</code>. It allows to include junctions whose starts/ends are within specified distance from <code>sashimiFilter</code> GRanges. This is useful for cases where the aligner did not place the junction reads precisely. Default <code>0L</code> , no tolerance.</td>
</tr>
<tr class="even">
<td align="left">sashimiHeight</td>
<td align="left">Integer scalar. The height of the sashimi part of the track. Can either be a value between 0 and 1 in which case it is taken as a relative height, or a positive value greater 1 in which case it is interpreted as pixels.</td>
</tr>
<tr class="odd">
<td align="left">sashimiScore</td>
<td align="left">Integer scalar. The minimum number of reads supporting the junction.</td>
</tr>
<tr class="even">
<td align="left">sashimiStrand</td>
<td align="left">Integer scalar. Only reads which have the specified strand are considered to count the junctions.</td>
</tr>
<tr class="odd">
<td align="left">sashimiTransformation</td>
<td align="left">Function. Applied to the junction score vector prior to plotting. The function should accept exactly one input argument and its return value needs to be a numeric vector of identical length as the input data.</td>
</tr>
<tr class="even">
<td align="left">showIndels</td>
<td align="left">Logical scalar. Consider insertions and deletions in coverage and pile-up. Default is <code>FALSE</code>. If set to <code>TRUE</code> the deletions defined in CIGAR string are not considered in coverage plot. The deletions are displayed as bridging lines in pile-up track. Insertions are shown as vertical bars.</td>
</tr>
<tr class="odd">
<td align="left">showMismatches</td>
<td align="left">Logical scalar. Add mismatch information, either as individual base letters or using color coded bars. This implies that the reference sequence has been provided, either to the class constructor or as part of the track list.</td>
</tr>
<tr class="even">
<td align="left">transformation</td>
<td align="left">Function. Applied to the coverage vector prior to plotting. The function should accept exactly one input argument and its return value needs to be a numeric Rle of identical length as the input data.</td>
</tr>
<tr class="odd">
<td align="left">type</td>
<td align="left">Character vactor. The type of information to plot. For <code>coverage</code> a coverage plot, potentially augmented by base mismatch information, for <code>sashimi</code> a sashimi plot, showing the juctions, and for <code>pileup</code> the pileups of the individual reads. These three can be combined.</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="creating-tracks-from-ucsc-data" class="section level1">
<h1 class="hasAnchor">
<a href="#creating-tracks-from-ucsc-data" class="anchor"></a>Creating tracks from UCSC data</h1>
<p>The UCSC data bases contain a multitude of genome annotation data for dozens of different organisms. Some of those data are very simple annotations like CpG island locations or SNP locations. Others are more complicated gene models, or even numeric annotations like conservation information. In order to provide a unified interface to all this information, the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package defines a meta-constructor function <code>UcscTrack</code>. The idea here is that we can express all of the available UCSC data in one of the package’s track types. We use the functionality provided in the <em><a href="https://bioconductor.org/packages/3.13/rtracklayer">rtracklayer</a></em> package to connect to UCSC and to download the relevant information. As a little illustrative example, let’s reproduce a view from the famous UCSC genome browser using the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package. As a final result we want to show something similar to Figure @ref(fig:UCSC1).</p>
<div class="figure">
<img src="ucsc1.png" alt="A screen shot of a UCSC genome browser view around the FMR1 locus on the mouse chromosome." width="1387"><p class="caption">
A screen shot of a UCSC genome browser view around the FMR1 locus on the mouse chromosome.
</p>
</div>
<p>To start we first need to know about the available data in the UCSC data base and about their structure. A good way to do this is to use the table browser on the <a href="http://genome.ucsc.edu/cgi-bin/hgTables?command=start">UCSC web site</a>. Figure @ref(fig:UCSC2) shows the table structure for the first gene model track, the known UCSC genes, in the table browser. We can see that there are multiple fields, some with genomic locations, other with additional data like labels or identifiers. If we go back to the section about the <code>GeneRegionTrack</code> class we remember that we need exactly this type of information for the constructor function. So in order to take the UCSC data and build an object of class <code>GeneRegionTrack</code> we need a way to map them to the individual constructor arguments. This is exactly what the <code>UcscTrack</code> meta-constructor is supposed to do for us.</p>
<div class="figure">
<img src="ucsc2.png" alt="A screen shot of a UCSC table browser view on the UCSC Known Genes track." width="1720"><p class="caption">
A screen shot of a UCSC table browser view on the UCSC Known Genes track.
</p>
</div>
<p>It needs to know about the track for which to extract the data (and optionally one or several of the tables that make up the collective track data, see <code><a href="../reference/UcscTrack.html">?UcscTrack</a></code> for details), about the genomic range including the chromosome for which to extract data, about the type of <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> track that we want to translate this data into, and about the individual track columns and their counterparts in the respective track class constructor. In our example, the track is called <code>knownGene</code>, the track type to construct is <code>GeneRegionTrack</code>, and the relevant columns are <code>exonStarts</code>, <code>exonEnds</code>, <code>name</code> and <code>strand</code>, which we will use as the start and end coordinates of the ranges and for all the exon, transcript and gene identifiers. Here we make use of the high flexibility of the <code>GeneRegionTrack</code> constructor in the sense that the exon coordinates actually come in the form of a comma-separated list, combining all the information for one transcript in one row of the table. The function is smart enough to detect this and to split the annotation regions accordingly. The full function call to create the <code>GeneRegionTrack</code> from the UCSC data looks like this:</p>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">from</span> <span class="op">&lt;-</span> <span class="fl">65921878</span>
<span class="va">to</span> <span class="op">&lt;-</span> <span class="fl">65980988</span>
<span class="va">knownGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/UcscTrack.html">UcscTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"mm9"</span>, chromosome <span class="op">=</span> <span class="st">"chrX"</span>, 
                        track <span class="op">=</span> <span class="st">"knownGene"</span>, from <span class="op">=</span> <span class="va">from</span>, to <span class="op">=</span> <span class="va">to</span>,
                        trackType <span class="op">=</span> <span class="st">"GeneRegionTrack"</span>, 
                        rstarts <span class="op">=</span> <span class="st">"exonStarts"</span>, rends <span class="op">=</span> <span class="st">"exonEnds"</span>, 
                        gene <span class="op">=</span> <span class="st">"name"</span>, symbol <span class="op">=</span> <span class="st">"name"</span>, 
                        transcript <span class="op">=</span> <span class="st">"name"</span>, strand <span class="op">=</span> <span class="st">"strand"</span>, 
                        fill <span class="op">=</span> <span class="st">"#8282d2"</span>, name <span class="op">=</span> <span class="st">"UCSC Genes"</span><span class="op">)</span></code></pre></div>
<p>With a similar approach we can construct the next two gene model tracks based on the <code>xenoRefGene</code> and <code>ensGene</code> data tables.</p>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">refGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/UcscTrack.html">UcscTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"mm9"</span>, chromosome <span class="op">=</span> <span class="st">"chrX"</span>,
                      track <span class="op">=</span> <span class="st">"xenoRefGene"</span>, from <span class="op">=</span> <span class="va">from</span>, to <span class="op">=</span> <span class="va">to</span>,
                      trackType <span class="op">=</span> <span class="st">"GeneRegionTrack"</span>, 
                      rstarts <span class="op">=</span> <span class="st">"exonStarts"</span>, rends <span class="op">=</span> <span class="st">"exonEnds"</span>, 
                      gene <span class="op">=</span> <span class="st">"name"</span>,  symbol <span class="op">=</span> <span class="st">"name2"</span>, 
                      transcript <span class="op">=</span> <span class="st">"name"</span>, strand <span class="op">=</span> <span class="st">"strand"</span>,
                      fill <span class="op">=</span> <span class="st">"#8282d2"</span>, stacking <span class="op">=</span> <span class="st">"dense"</span>, 
                      name <span class="op">=</span> <span class="st">"Other RefSeq"</span><span class="op">)</span>

<span class="va">ensGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/UcscTrack.html">UcscTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"mm9"</span>, chromosome <span class="op">=</span> <span class="st">"chrX"</span>, 
                      track <span class="op">=</span> <span class="st">"ensGene"</span>, from <span class="op">=</span> <span class="va">from</span>, to <span class="op">=</span> <span class="va">to</span>,
                      trackType <span class="op">=</span> <span class="st">"GeneRegionTrack"</span>, 
                      rstarts <span class="op">=</span> <span class="st">"exonStarts"</span>, rends <span class="op">=</span> <span class="st">"exonEnds"</span>,
                      gene <span class="op">=</span> <span class="st">"name"</span>, symbol <span class="op">=</span> <span class="st">"name2"</span>, 
                      transcript <span class="op">=</span> <span class="st">"name"</span>, strand <span class="op">=</span> <span class="st">"strand"</span>, 
                      fill <span class="op">=</span> <span class="st">"#960000"</span>, name <span class="op">=</span> <span class="st">"Ensembl Genes"</span><span class="op">)</span></code></pre></div>
<p>The CpG and SNP tracks are slightly different since a <code>GeneRegionTrack</code> representation would not be particularly useful. Instead, we can use <code>AnnotationTrack</code> objects as containers. The overall process using the <code>UcscTrack</code> meta-constructor remains the same.</p>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cpgIslands</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/UcscTrack.html">UcscTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"mm9"</span>, chromosome <span class="op">=</span> <span class="st">"chrX"</span>, 
                        track <span class="op">=</span> <span class="st">"cpgIslandExt"</span>, from <span class="op">=</span> <span class="va">from</span>, to <span class="op">=</span> <span class="va">to</span>,
                        trackType <span class="op">=</span> <span class="st">"AnnotationTrack"</span>, 
                        start <span class="op">=</span> <span class="st">"chromStart"</span>, end <span class="op">=</span> <span class="st">"chromEnd"</span>, 
                        id <span class="op">=</span> <span class="st">"name"</span>, shape <span class="op">=</span> <span class="st">"box"</span>, fill <span class="op">=</span> <span class="st">"#006400"</span>, 
                        name <span class="op">=</span> <span class="st">"CpG Islands"</span><span class="op">)</span>

<span class="va">snpLocations</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/UcscTrack.html">UcscTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"mm9"</span>, chromosome <span class="op">=</span> <span class="st">"chrX"</span>, 
                           track <span class="op">=</span> <span class="st">"snp128"</span>, from <span class="op">=</span> <span class="va">from</span>, to <span class="op">=</span> <span class="va">to</span>,
                           trackType <span class="op">=</span> <span class="st">"AnnotationTrack"</span>, 
                           start <span class="op">=</span> <span class="st">"chromStart"</span>, end <span class="op">=</span> <span class="st">"chromEnd"</span>, 
                           id <span class="op">=</span> <span class="st">"name"</span>, feature <span class="op">=</span> <span class="st">"func"</span>, 
                           strand <span class="op">=</span> <span class="st">"strand"</span>, shape <span class="op">=</span> <span class="st">"box"</span>, 
                           stacking <span class="op">=</span> <span class="st">"dense"</span>, fill <span class="op">=</span> <span class="st">"black"</span>,
                           name <span class="op">=</span> <span class="st">"SNPs"</span><span class="op">)</span></code></pre></div>
<p>Most of UCSC’s <code>DataTrack</code>-like tracks are a little more complex and represent a collection of several sub-tracks, with data originating from multiple tables. To make sure that we get the correct information we have to be a little bit more specific here and also define the particular table on the UCSC data base to use.</p>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conservation</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/UcscTrack.html">UcscTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"mm9"</span>, chromosome <span class="op">=</span> <span class="st">"chrX"</span>, 
                          track <span class="op">=</span> <span class="st">"Conservation"</span>, 
                          table <span class="op">=</span> <span class="st">"phyloP30wayPlacental"</span>,
                          from <span class="op">=</span> <span class="va">from</span>, to <span class="op">=</span> <span class="va">to</span>, trackType <span class="op">=</span> <span class="st">"DataTrack"</span>, 
                          start <span class="op">=</span> <span class="st">"start"</span>, end <span class="op">=</span> <span class="st">"end"</span>, data <span class="op">=</span> <span class="st">"score"</span>,
                          type <span class="op">=</span> <span class="st">"hist"</span>, window <span class="op">=</span> <span class="st">"auto"</span>, 
                          col.histogram <span class="op">=</span> <span class="st">"darkblue"</span>, 
                          fill.histogram <span class="op">=</span> <span class="st">"darkblue"</span>,
                          ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3.7</span>, <span class="fl">4</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">"Conservation"</span><span class="op">)</span>

<span class="va">gcContent</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/UcscTrack.html">UcscTrack</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"mm9"</span>, chromosome <span class="op">=</span> <span class="st">"chrX"</span>, 
                       track <span class="op">=</span> <span class="st">"GC Percent"</span>, table <span class="op">=</span> <span class="st">"gc5Base"</span>,
                       from <span class="op">=</span> <span class="va">from</span>, to <span class="op">=</span> <span class="va">to</span>, trackType <span class="op">=</span> <span class="st">"DataTrack"</span>, 
                       start <span class="op">=</span> <span class="st">"start"</span>, end <span class="op">=</span> <span class="st">"end"</span>, data <span class="op">=</span> <span class="st">"score"</span>,
                       type <span class="op">=</span> <span class="st">"hist"</span>, window <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, windowSize <span class="op">=</span> <span class="fl">1500</span>, 
                       fill.histogram <span class="op">=</span> <span class="st">"black"</span>, col.histogram <span class="op">=</span> <span class="st">"black"</span>,
                       ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">30</span>, <span class="fl">70</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">"GC Percent"</span><span class="op">)</span></code></pre></div>
<p>To add some reference points we also need a genome axis and an <code>IdeogramTrack</code> of the x chromosome.</p>
<div class="sourceCode" id="cb124"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">axTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GenomeAxisTrack-class.html">GenomeAxisTrack</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">idxTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/IdeogramTrack-class.html">IdeogramTrack</a></span><span class="op">(</span>genome<span class="op">=</span><span class="st">"mm9"</span>, chromosome<span class="op">=</span><span class="st">"chrX"</span><span class="op">)</span></code></pre></div>
<p>And finally we can plot all of our tracks.</p>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">idxTrack</span>, <span class="va">axTrack</span>, <span class="va">knownGenes</span>, <span class="va">refGenes</span>, <span class="va">ensGenes</span>, 
                <span class="va">cpgIslands</span>, <span class="va">gcContent</span>, <span class="va">conservation</span>, <span class="va">snpLocations</span><span class="op">)</span>, 
           from <span class="op">=</span> <span class="va">from</span>, to <span class="op">=</span> <span class="va">to</span>, showTitle <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/ucscTrack6-1.png"><!-- --></p>
</div>
<div id="track-highlighting-and-overlays" class="section level1">
<h1 class="hasAnchor">
<a href="#track-highlighting-and-overlays" class="anchor"></a>Track highlighting and overlays</h1>
<div id="highlighting" class="section level2">
<h2 class="hasAnchor">
<a href="#highlighting" class="anchor"></a>Highlighting</h2>
<p>Some features of a typical genome browser plot are not specific for just a single track, but instead span multiple tracks. One prominent example are highlighting regions. Assume you want to draw the attention on the plot to a particular genomic stretch where the visual clues across several or maybe even all tracks indicate a particular interesting behavior. Because the individual tracks in the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package are all represented by individual objects, there is no straight forward way to define such a region. Here is where the <code>HighlightingTrack</code> class comes into play. Objects from this class are essentially just containers for one or several of the previously described track objects, together with a very simple range definition that is to be superimposed over the individual tracks once they have been rendered. Going back to our final example from the second chapter we could highlight two areas on the plot on the <code>AnnotationTrack</code>, the <code>GeneRegionTrack</code> and the <code>DataTrack</code> using the <code>HighlightTrack</code> constructor in the following call:</p>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ht</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HighlightTrack-class.html">HighlightTrack</a></span><span class="op">(</span>trackList <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">atrack</span>, <span class="va">grtrack</span>, <span class="va">dtrack</span><span class="op">)</span>,
                     start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">26705000</span>, <span class="fl">26720000</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">7000</span>,
                     chromosome <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">ht</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, to <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/HighlightTrack-1.png"><!-- --></p>
<p>One can also build multiple <code>HighlightTrack</code> collections to create gaps in the highlighting regions. For <code>IdeogramTracks</code> we never plot highlighting ranges because they do not live on the same x-axis coordinate system like the other tracks.</p>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ht1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HighlightTrack-class.html">HighlightTrack</a></span><span class="op">(</span>trackList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">atrack</span><span class="op">)</span>, 
                      start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">26705000</span>, <span class="fl">26720000</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">7000</span>, 
                      chromosome <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>
<span class="va">ht2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HighlightTrack-class.html">HighlightTrack</a></span><span class="op">(</span>trackList <span class="op">=</span> <span class="va">dtrack</span>, start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">26705000</span>, <span class="fl">26720000</span><span class="op">)</span>,
                      width <span class="op">=</span> <span class="fl">7000</span>, chromosome <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">ht1</span>, <span class="va">grtrack</span>, <span class="va">ht2</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, to <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/HighlightTrack2-1.png"><!-- --></p>
<div id="display-parameters-for-highlighttrack-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#display-parameters-for-highlighttrack-objects" class="anchor"></a>Display parameters for HighlightTrack objects</h3>
<p>For a complete listing of all the available display parameters please see the table below or the man page of the <code>HighlightTrack</code> class by typing in <code><a href="../reference/HighlightTrack-class.html">?HighlightTrack</a></code> on the <code>R</code> command line.</p>
<table class="table">
<colgroup>
<col width="19%">
<col width="80%">
</colgroup>
<thead><tr class="header">
<th align="left">Display Parameter</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">col</td>
<td align="left">Integer or character vector. The border color for the highlighting regions.</td>
</tr>
<tr class="even">
<td align="left">fill</td>
<td align="left">Integer or character vector. The fill color for the highlighting regions.</td>
</tr>
<tr class="odd">
<td align="left">inBackground</td>
<td align="left">Logical scalar. Place the box in the background or in the foreground.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="overlays" class="section level2">
<h2 class="hasAnchor">
<a href="#overlays" class="anchor"></a>Overlays</h2>
<p>For certain applications it can make sense to overlay multiple tracks on the same area of the plot. For instance when dealing with multiple data sets that do not necessarily share the same genomic coordinates for their data points the design of the <code>DataTrack</code> class is not optimal. To contrast these two data sets an overlay of two separate <code>DataTrack</code> objects however can be quite powerful.The <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package allows for such overlay operations by means of the <code>OverlayTrack</code> class. Similar to the <code>HighlightTrack</code> class from the previous section this is merely a container for an arbitrary number of other <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> track objects. For the purpose of an instructive example we will generate a second <code>DataTrack</code> object and combine it with the existing one from the second chapter.</p>
<div class="sourceCode" id="cb128"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="fl">100</span>, min <span class="op">=</span> <span class="op">-</span><span class="fl">2</span>, max <span class="op">=</span> <span class="fl">22</span><span class="op">)</span>
<span class="va">dtrack2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DataTrack-class.html">DataTrack</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">dat</span>, start <span class="op">=</span> <span class="va">coords</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">coords</span><span class="op">)</span><span class="op">]</span>, 
                end <span class="op">=</span> <span class="va">coords</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>, chromosome <span class="op">=</span> <span class="va">chr</span>, genome <span class="op">=</span> <span class="va">gen</span>, 
                name <span class="op">=</span> <span class="st">"Uniform2"</span>,  groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="st">"sample 2"</span>, 
                  levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample 1"</span>, <span class="st">"sample 2"</span><span class="op">)</span><span class="op">)</span>, legend <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="../reference/DisplayPars-class.html">displayPars</a></span><span class="op">(</span><span class="va">dtrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="st">"sample 1"</span>, 
              levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample 1"</span>, <span class="st">"sample 2"</span><span class="op">)</span><span class="op">)</span>, legend <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">ot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OverlayTrack-class.html">OverlayTrack</a></span><span class="op">(</span>trackList<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">dtrack2</span>, <span class="va">dtrack</span><span class="op">)</span><span class="op">)</span>
<span class="va">ylims</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/extendrange.html">extendrange</a></span><span class="op">(</span><span class="fu"><a href="../reference/RangeTrack-class.html">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">values</span><span class="op">(</span><span class="va">dtrack</span><span class="op">)</span>, <span class="fu">values</span><span class="op">(</span><span class="va">dtrack2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">ot</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, to <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, 
           ylim <span class="op">=</span> <span class="va">ylims</span>, type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"smooth"</span>, <span class="st">"p"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/OverlayTrack-1.png"><!-- --></p>
<p>It is important to know that <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> is fully unaware of the intention and any specifics of the track merging operation. Any track type can be merged with any other track type, and there are no limitations on the number of overlays. It is totally up to the user to design reasonable merges and to control critical aspects of the plot, like the y-axis scaling or the grouping in the above example. During rendering, the software will only use the first track object in the <code>trackList</code> container of <code>OverlayTracks</code> object to set up the track title. Thus, the ranges on the y-axis are those that have been automatically determined for the first of our two <code>DataTracks</code>. Only by forcing a common y-range for both objects do we actually plot the data from both tracks on a common scale. Had we added another track object that is not a <code>DataTrack</code> is the first element of our <code>trackList</code> there wouldn’t even be an axis. In a similar fashion, by nicely setting up the group levels for the two <code>DataTracks</code> we can even ask for a legend to be added.</p>
<p>On devices that support it, alpha blending can be a useful tool to tease out even more information out of track overlays, at least when comparing just a small number of samples. The resulting transparency effectively eliminates the problem of overplotting. The following example will only work if this vignette has been built on a system with alpha blending support.</p>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/DisplayPars-class.html">displayPars</a></span><span class="op">(</span><span class="va">dtrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>alpha.title <span class="op">=</span> <span class="fl">1</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="fu"><a href="../reference/DisplayPars-class.html">displayPars</a></span><span class="op">(</span><span class="va">dtrack2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>alpha.title <span class="op">=</span> <span class="fl">1</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="va">ot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OverlayTrack-class.html">OverlayTrack</a></span><span class="op">(</span>trackList <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">dtrack</span>, <span class="va">dtrack2</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">gtrack</span>, <span class="va">ot</span><span class="op">)</span>, from <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, to <span class="op">=</span> <span class="va">lim</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, 
           ylim <span class="op">=</span> <span class="va">ylims</span>, type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"hist"</span><span class="op">)</span>, window <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/OverlayTrack2-1.png"><!-- --></p>
</div>
</div>
<div id="composite-plots-for-multiple-chromosomes" class="section level1">
<h1 class="hasAnchor">
<a href="#composite-plots-for-multiple-chromosomes" class="anchor"></a>Composite plots for multiple chromosomes</h1>
<p>As mentioned in the introduction section, a set of <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> tracks has to share the same chromosome when plotted, i.e., only a single chromosome can be active during a given plotting operation. Consequently, we can not directly create plots for multiple chromosomes in a single call to the <code>plotTracks</code> function. However, since the underlying graphical infrastructure of the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package uses grid graphics, we can build our own composite plot using multiple consecutive <code>plotTracks</code> calls. All we need to take care of is an adequate layout structure to plot into, and we also need to tell <code>plotTracks</code> not to clear the graphics device before plotting, which can be achieved by setting the function’s <code>add</code> argument to <code>FALSE</code>. For details on how to create a layout structure in the grid graphics system, please see the help page at <code>?grid.layout)</code>.</p>
<p>We start by creating an <code>AnnotationTrack</code> objects and a <code>DataTrack</code> object which both contain data for several chromosomes.</p>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">chroms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr3"</span>, <span class="st">"chr4"</span><span class="op">)</span>
<span class="va">maTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AnnotationTrack-class.html">AnnotationTrack</a></span><span class="op">(</span>range<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">chroms</span>, 
    ranges <span class="op">=</span> <span class="fu">IRanges</span><span class="op">(</span>start <span class="op">=</span> <span class="fl">1</span>,  width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">400</span>, <span class="fl">200</span>,<span class="fl">1000</span><span class="op">)</span><span class="op">)</span>,
    strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span><span class="op">)</span><span class="op">)</span>, genome <span class="op">=</span> <span class="st">"mm9"</span>, 
    chromosome <span class="op">=</span> <span class="st">"chr1"</span>, name <span class="op">=</span> <span class="st">"foo"</span><span class="op">)</span>

<span class="va">mdTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DataTrack-class.html">DataTrack</a></span><span class="op">(</span>
  range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">chroms</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">40</span>, <span class="fl">20</span>, <span class="fl">100</span><span class="op">)</span><span class="op">)</span>,
                  ranges <span class="op">=</span> <span class="fu">IRanges</span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">100</span>, len <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
                                             <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">400</span>, len <span class="op">=</span> <span class="fl">40</span><span class="op">)</span>, 
                                             <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">200</span>, len <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,
                                             <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1000</span>, len <span class="op">=</span> <span class="fl">100</span><span class="op">)</span><span class="op">)</span>, 
                                   width <span class="op">=</span> <span class="fl">9</span><span class="op">)</span>, values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="fl">170</span><span class="op">)</span><span class="op">)</span>,
  data <span class="op">=</span> <span class="st">"values"</span>, chromosome <span class="op">=</span> <span class="st">"chr1"</span>, genome <span class="op">=</span> <span class="st">"mm9"</span>, name <span class="op">=</span> <span class="st">"bar"</span><span class="op">)</span></code></pre></div>
<p>Now we also want a genome axis and an <code>IdeogramTrack</code> object to indicate the genomic context.</p>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mgTrack</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GenomeAxisTrack-class.html">GenomeAxisTrack</a></span><span class="op">(</span>scale <span class="op">=</span> <span class="fl">50</span>, labelPos <span class="op">=</span> <span class="st">"below"</span>, exponent <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="../reference/RangeTrack-class.html">chromosome</a></span><span class="op">(</span><span class="va">itrack</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"chr1"</span></code></pre></div>
<p>Finally, we build a layout in which the plots for each chromosome are placed in a rectangular grid and repeatedly call <code>plotTracks</code> for each chromosome.</p>
<div class="sourceCode" id="cb132"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ncols</span> <span class="op">&lt;-</span> <span class="fl">2</span>
<span class="va">nrows</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">chroms</span><span class="op">)</span> <span class="op">%/%</span> <span class="va">ncols</span>
<span class="fu"><a href="https://rdrr.io/r/grid/grid.newpage.html">grid.newpage</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grid/viewports.html">pushViewport</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grid/viewport.html">viewport</a></span><span class="op">(</span>layout <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/grid.layout.html">grid.layout</a></span><span class="op">(</span><span class="va">nrows</span>, <span class="va">ncols</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_along</a></span><span class="op">(</span><span class="va">chroms</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/grid/viewports.html">pushViewport</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grid/viewport.html">viewport</a></span><span class="op">(</span>layout.pos.col <span class="op">=</span> <span class="op">(</span><span class="op">(</span><span class="va">i</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%%</span> <span class="va">ncols</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>,
                        layout.pos.row <span class="op">=</span> <span class="op">(</span><span class="op">(</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%/%</span> <span class="va">ncols</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span>, <span class="va">maTrack</span>, <span class="va">mdTrack</span>, <span class="va">mgTrack</span><span class="op">)</span>, 
               chromosome <span class="op">=</span> <span class="va">chroms</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, add <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/grid/viewports.html">popViewport</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p><img src="Gviz_files/figure-html/multPlot3-1.png"><!-- --></p>
<p>Maybe an even more compact version of this would be to use the lattice package for building the actual trellis, with <code>plotTracks</code> as the panel function.</p>
<div class="sourceCode" id="cb133"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://lattice.r-forge.r-project.org/">lattice</a></span><span class="op">)</span>
<span class="va">chroms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>chromosome <span class="op">=</span> <span class="va">chroms</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/lattice/man/xyplot.html">xyplot</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">~</span> <span class="va">chromosome</span> <span class="op">|</span> <span class="va">chromosome</span>, data <span class="op">=</span> <span class="va">chroms</span>, panel <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="../reference/plotTracks.html">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">itrack</span> , <span class="va">maTrack</span>, <span class="va">mdTrack</span>, <span class="va">mgTrack</span><span class="op">)</span>, 
             chromosome <span class="op">=</span> <span class="va">x</span>, add <span class="op">=</span> <span class="cn">TRUE</span>, showId <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">}</span>,
  scales <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>draw <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>, xlab <span class="op">=</span> <span class="cn">NULL</span>, ylab <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<p><img src="Gviz_files/figure-html/multPlot4-1.png"><!-- --></p>
</div>
<div id="bioconductor-integration" class="section level1">
<h1 class="hasAnchor">
<a href="#bioconductor-integration" class="anchor"></a>Bioconductor integration and file support</h1>
<p>This short section is supposed to give a very brief overview over the different track classes in the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package and how those can be constructed from the typical Bioconductor classes that deal with genomic data. The list is by no means complete, and a closer look at a track class’ documentation should provide all the possible options.</p>
<table class="table">
<colgroup>
<col width="16%">
<col width="19%">
<col width="65%">
</colgroup>
<thead><tr class="header">
<th align="left">Gviz class</th>
<th align="left">Bioconductor class</th>
<th align="left">Method</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">AnnotationTrack</td>
<td align="left">data.frame</td>
<td align="left">Constructor</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">IRanges</td>
<td align="left">Constructor + additional arguments</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">GRanges</td>
<td align="left">Constructor or setAs method, additional data in metadata columns</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">GRangesList</td>
<td align="left">Constructor or setAs method</td>
</tr>
<tr class="odd">
<td align="left">GeneRegionTrack</td>
<td align="left">data.frame</td>
<td align="left">Constructor</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">IRanges</td>
<td align="left">Constructor + additional arguments</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">GRanges</td>
<td align="left">Constructor or setAs method, additional data in metadata columns</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">GRangesList</td>
<td align="left">Constructor or setAs method, additional data in metadata columns</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">TxDb</td>
<td align="left">Constructor or setAs method</td>
</tr>
<tr class="even">
<td align="left">DataTrack</td>
<td align="left">data.frame</td>
<td align="left">Constructor</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">IRanges</td>
<td align="left">Constructor + additional data matrix</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">GRanges</td>
<td align="left">Constructor or setAs method, numeric data in metadata columns</td>
</tr>
<tr class="odd">
<td align="left">SequenceTrack</td>
<td align="left">DNAStringSet</td>
<td align="left">DNAStringSet &amp; Constructor</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">BSgenome</td>
<td align="left">Constructor</td>
</tr>
</tbody>
</table>
<p>We have seen in previous examples that the <em><a href="https://bioconductor.org/packages/3.13/Gviz">Gviz</a></em> package supports quite a wide range of standard file types as inputs for the class’ constructor functions. In this section we summarize the file support for the different track classes in one place.</p>
<table class="table">
<colgroup>
<col width="4%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="85%">
</colgroup>
<thead><tr class="header">
<th align="left">Gviz class</th>
<th align="left">File type</th>
<th align="left">Extension</th>
<th align="left">Streaming</th>
<th align="left">Details</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">AnnotationTrack</td>
<td align="left">BED</td>
<td align="left">.bed</td>
<td align="left">no</td>
<td align="left">Genomic locations from the mandatory <code>chrom</code>, <code>chromStart</code> and <code>chromEnd</code> fields, and optionally the strand from the strand field. If present, the information in the field is mapped to track item ids, and is mapped to track item feature type. All other fields are currently ignored.</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">GFF</td>
<td align="left">.gff, .gff1,</td>
<td align="left">no</td>
<td align="left">Only the following basic GFF fields are recognized: <code>seqname</code>, <code>start</code>, <code>end</code>, <code>strand</code>, (mapped to track item feature type) and <code>group</code> (to allow for track item grouping).</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">GFF2</td>
<td align="left">.gff2</td>
<td align="left">no</td>
<td align="left">Same as above, but feature grouping information may be provided either as <code>Group</code> or <code>Parent</code> attribute. Feature ids are mapped to one of the ID, Name or Alias attributes.</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">GFF3</td>
<td align="left">.gff3</td>
<td align="left">no</td>
<td align="left">Same as above, but feature grouping information has to be provided as the <code>Parent</code> attribute.</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">BAM</td>
<td align="left">.bam</td>
<td align="left">YES</td>
<td align="left">Only start and end locations as well as the strand information for the reads are used. Read identifiers are used for track item grouping.</td>
</tr>
<tr class="even">
<td align="left">GeneRegionTrack</td>
<td align="left">GTF</td>
<td align="left">.gtf</td>
<td align="left">no</td>
<td align="left">A somewhat looser format definition for <code>gtf</code> files is applied here where gene, transcript and exon identifiers and names can be parsed from the <code>gene_id</code>, <code>gene_name</code>, <code>transcript_id</code>, <code>transcript_name</code>, <code>exon_id</code>, or <code>exon_id attributes</code>
</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">GFF</td>
<td align="left">.gff, .gff1</td>
<td align="left">no</td>
<td align="left">This only supports very limited item grouping and thus complete gene models can not be properly encoded.</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">GFF2</td>
<td align="left">.gff2</td>
<td align="left">no</td>
<td align="left">In most instances this is identical to the <code>GTF</code> standard and it could make sense to rename the file accordingly.</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">GFF3</td>
<td align="left">.gff3</td>
<td align="left">no</td>
<td align="left">The gene-to-transcript and transcript-to- exon relationships are encoded in the parent and <code>type</code> attributes and the parser tries to accommodate most of the existing <code>GFF3</code> variants.</td>
</tr>
<tr class="even">
<td align="left">DataTrack</td>
<td align="left">BedGraph</td>
<td align="left">.bedGraph</td>
<td align="left">no</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">WIG</td>
<td align="left">.wig</td>
<td align="left">no</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">BigWig</td>
<td align="left">.bigWig</td>
<td align="left">YES</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">BAM</td>
<td align="left">.bam</td>
<td align="left">YES</td>
<td align="left">Read coverage only is extracted from the bam file.</td>
</tr>
<tr class="even">
<td align="left">SequenceTrack</td>
<td align="left">FASTA</td>
<td align="left">.fa, .fasta</td>
<td align="left">YES</td>
<td align="left">Streaming only possible if an index file is found in the same directory as the original fasta file.</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">2Bit</td>
<td align="left">.2bit</td>
<td align="left">YES</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">AlignmentsTrack</td>
<td align="left">BAM</td>
<td align="left">.bam</td>
<td align="left">YES</td>
<td align="left">Always needs an index file is found in the same directory as the original <code>BAM</code> file.</td>
</tr>
</tbody>
</table>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>SessionInfo</h1>
<p>The following is the session info that generated this vignette:</p>
<div class="sourceCode" id="cb134"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## R Under development (unstable) (2020-11-05 r79401)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.1 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    </span>
<span class="co">##  [8] datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] lattice_0.20-41                   biomaRt_2.47.1                   </span>
<span class="co">##  [3] GenomicFeatures_1.43.2            AnnotationDbi_1.53.0             </span>
<span class="co">##  [5] Biobase_2.51.0                    BSgenome.Hsapiens.UCSC.hg19_1.4.3</span>
<span class="co">##  [7] BSgenome_1.59.0                   rtracklayer_1.51.0               </span>
<span class="co">##  [9] Biostrings_2.59.0                 XVector_0.31.0                   </span>
<span class="co">## [11] Gviz_1.35.0                       GenomicRanges_1.43.0             </span>
<span class="co">## [13] GenomeInfoDb_1.27.0               IRanges_2.25.2                   </span>
<span class="co">## [15] S4Vectors_0.29.3                  BiocGenerics_0.37.0              </span>
<span class="co">## [17] knitr_1.30                        BiocStyle_2.19.0                 </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] colorspace_2.0-0            ellipsis_0.3.1             </span>
<span class="co">##   [3] rprojroot_1.3-2             biovizBase_1.39.0          </span>
<span class="co">##   [5] htmlTable_2.1.0             base64enc_0.1-3            </span>
<span class="co">##   [7] fs_1.5.0                    dichromat_2.0-0            </span>
<span class="co">##   [9] rstudioapi_0.12             bit64_4.0.5                </span>
<span class="co">##  [11] xml2_1.3.2                  splines_4.1.0              </span>
<span class="co">##  [13] Formula_1.2-4               Rsamtools_2.7.0            </span>
<span class="co">##  [15] cluster_2.1.0               dbplyr_2.0.0               </span>
<span class="co">##  [17] png_0.1-7                   BiocManager_1.30.10        </span>
<span class="co">##  [19] compiler_4.1.0              httr_1.4.2                 </span>
<span class="co">##  [21] backports_1.2.0             assertthat_0.2.1           </span>
<span class="co">##  [23] Matrix_1.2-18               lazyeval_0.2.2             </span>
<span class="co">##  [25] htmltools_0.5.0             prettyunits_1.1.1          </span>
<span class="co">##  [27] tools_4.1.0                 gtable_0.3.0               </span>
<span class="co">##  [29] glue_1.4.2                  GenomeInfoDbData_1.2.4     </span>
<span class="co">##  [31] dplyr_1.0.2                 rappdirs_0.3.1             </span>
<span class="co">##  [33] Rcpp_1.0.5                  pkgdown_1.6.1.9000         </span>
<span class="co">##  [35] vctrs_0.3.4                 xfun_0.19                  </span>
<span class="co">##  [37] stringr_1.4.0               lifecycle_0.2.0            </span>
<span class="co">##  [39] ensembldb_2.15.0            XML_3.99-0.5               </span>
<span class="co">##  [41] zlibbioc_1.37.0             scales_1.1.1               </span>
<span class="co">##  [43] VariantAnnotation_1.37.0    ProtGenerics_1.23.1        </span>
<span class="co">##  [45] ragg_0.4.0                  hms_0.5.3                  </span>
<span class="co">##  [47] MatrixGenerics_1.3.0        SummarizedExperiment_1.21.0</span>
<span class="co">##  [49] AnnotationFilter_1.15.0     RColorBrewer_1.1-2         </span>
<span class="co">##  [51] yaml_2.2.1                  curl_4.3                   </span>
<span class="co">##  [53] memoise_1.1.0               gridExtra_2.3              </span>
<span class="co">##  [55] ggplot2_3.3.2               rpart_4.1-15               </span>
<span class="co">##  [57] latticeExtra_0.6-29         stringi_1.5.3              </span>
<span class="co">##  [59] RSQLite_2.2.1               highr_0.8                  </span>
<span class="co">##  [61] desc_1.2.0                  checkmate_2.0.0            </span>
<span class="co">##  [63] filelock_1.0.2              BiocParallel_1.25.1        </span>
<span class="co">##  [65] rlang_0.4.8                 pkgconfig_2.0.3            </span>
<span class="co">##  [67] systemfonts_0.3.2           matrixStats_0.57.0         </span>
<span class="co">##  [69] bitops_1.0-6                evaluate_0.14              </span>
<span class="co">##  [71] purrr_0.3.4                 GenomicAlignments_1.27.1   </span>
<span class="co">##  [73] htmlwidgets_1.5.2           bit_4.0.4                  </span>
<span class="co">##  [75] tidyselect_1.1.0            magrittr_1.5               </span>
<span class="co">##  [77] bookdown_0.21               R6_2.5.0                   </span>
<span class="co">##  [79] generics_0.1.0              Hmisc_4.4-1                </span>
<span class="co">##  [81] DelayedArray_0.17.0         DBI_1.1.0                  </span>
<span class="co">##  [83] withr_2.3.0                 pillar_1.4.6               </span>
<span class="co">##  [85] foreign_0.8-80              survival_3.2-7             </span>
<span class="co">##  [87] RCurl_1.98-1.2              nnet_7.3-14                </span>
<span class="co">##  [89] tibble_3.0.4                crayon_1.3.4               </span>
<span class="co">##  [91] BiocFileCache_1.15.1        rmarkdown_2.5              </span>
<span class="co">##  [93] jpeg_0.1-8.1                progress_1.2.2             </span>
<span class="co">##  [95] data.table_1.13.2           blob_1.2.1                 </span>
<span class="co">##  [97] digest_0.6.27               textshaping_0.1.2          </span>
<span class="co">##  [99] openssl_1.4.3               munsell_0.5.0              </span>
<span class="co">## [101] askpass_1.1</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Hahne2016">
<p>Hahne, Florian, and Robert Ivanek. 2016. “Statistical Genomics: Methods and Protocols.” In, edited by Ewy Mathé and Sean Davis, 335–51. New York, NY: Springer New York. <a href="https://doi.org/10.1007/978-1-4939-3578-9_16">https://doi.org/10.1007/978-1-4939-3578-9_16</a>.</p>
</div>
<div id="ref-Katz22012015">
<p>Katz, Yarden, Eric T. Wang, Jacob Silterra, Schraga Schwartz, Bang Wong, Helga Thorvaldsdóttir, James T. Robinson, Jill P. Mesirov, Edoardo M. Airoldi, and Christopher B. Burge. 2015. “Quantitative Visualization of Alternative Exon Expression from Rna-Seq Data.” <em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btv034">https://doi.org/10.1093/bioinformatics/btv034</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>A different data set is plotted for the horizon type for the sake of clarity.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>Please note, that in the example, a different data set is plotted for the horizon type for the sake of clarity.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Florian Hahne, Steffen Durinck, Robert Ivanek, Arne Mueller, Steve Lianoglou, Ge Tan, Lance Parsons, Shraddha Pai.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
